<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>About</title>

    <!-- javascript imports -->
    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
    <script src="js/bootstrap.bundle.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.8.1/dist/igv.min.js"></script>
    <script src="js/tabulator.min.js" type="text/javascript"></script>

    <!-- CSS imports -->
    <link rel="stylesheet" href="css/bootstrap.css" type="text/css">
    <link rel="stylesheet" href="css/common.css" type="text/css">
    <link rel="stylesheet" href="css/tf_dashboard.css" type="text/css">
    <link rel="stylesheet" href="css/about-page.css" type="text/css">
    <link rel="stylesheet" href="css/tabulator_bootstrap4.min.css" rel="stylesheet">

    <!-- font -->
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&family=Sulphur+Point:wght@300;400;700&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <!-- CSS -->
    <style>
        * {
            font-family: 'Lato', 'Roboto', 'Helvetica', 'Segoe UI', sans-serif;
        }
    </style>

</head>

<body>
<!-- Page Content -->
<div class="container-fluid page-container">
    <!-- Top Navigation Bar -->
    <nav class="navbar navbar-expand-md sticky-top header">
        <a class="navbar-brand" href="index.html">
             <img src="images/ChIPdb_logo.png" height=35 class="d-inline-block align-top"
            alt="" loading="lazy">
            proChIPdb 1.0.0 </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav mr-auto">
                <li class="nav-item dropdown">
                    <button class="btn nav-color dropdown-toggle" type="button" id="dropdownMenu2"
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        Organism Dataset
                    </button>
                    <div class="dropdown-menu" aria-labelledby="dropdownMenu2">
                        <a class="dropdown-item" href="dataset_page.html?organism=e_coli"><i>E. coli</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=all_other">All Other Organisms</a>
                    </div>
                </li>
            </ul>
            <ul class="navbar-nav">
                <li class="nav-item">
                    <a class="mr-4" href="about.html"><span><i class="material-icons">info</i></span>
                        <span>About</span>
                    </a>
                </li>
                <li class="nav-item">
                    <a class="mr-4" href="search.html"><span><i class="material-icons">search</i></span>
                        <span>Search</span>
                    </a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="row row-no-margin">
        <!-- Left Navigation -->
        <div id="left_col" class="col-sm-12 col-md-2">
            <div id="left_col_content">
                <ul class="nav flex-column">
                    <li class="nav-item">
                        <a class="nav-link" href="#use_site">How to Use this Site</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#tf_dash">Transcription Factor Dashboards</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#search">Search</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#ref">References</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#contact">Contact Us</a>
                    </li>
                </ul>
            </div>
        </div>
        <!-- page content -->
        <div class="col-sm-10">
            <!-- Title -->
            <h1 class="text-center py-3">About proChIPdb</h1>

            <p>
                Welcome to <b>proChIPdb</b>, the <b>PRO</b>karyotic <b>Ch</b>romatin <b>I</b>mmuno<b>P</b>recipitation <b>D</b>ata<b>B</b>ase! This tool enables microbiologists to easily browse 271 chip-seq and chip-exo profiles for various transcription factors (TFs) across 13 organisms. Currently, most of our profiles are for <i>Escherichia coli</i>. We provide curated tables of binding sites, interactive plots and genome viewers, as well as comparisons against literature binding sites.
            </p>

            <!-- Using the Site -->
            <h3 id="use_site" class="text-center py-3">
                How to Use this Site
            </h3>

            <p>
                Begin by selecting your organism of interest from the splash page. Next, you can browse the table of transcription factors from our dataset page. proChIPdb covers well-studied transcription factors, as well as a wide scope of <i>E. coli</i> y-TFs (relatively poorly characterized transcription factors). 
            </p><p>
                You can also <a href="#search">search</a> for transcription factors or genes using the search button in the upper left corner, or <a href="#download">download all site data</a> to do your own custom analysis.
            </p><p>
                Once you’ve reached a transcription factor dashboard, you can browse its binding sites and target genes in the curated table. Interested in a specific binding site? Click on it to see the peak in our genome viewer. You can also click through the tabs in the upper right panel to see global characterizations of the transcription factor’s binding, such as its binding motif, peak width, peak location relative to its target genes, and the concordance between the data in proChIPdb and other databases of transcription factor binding. For more details on the dashboard components, see the sections below.
            </p><p>
                It is our hope that proChIPdb will enrich your research by allowing easy access to this compendium of binding intensity data, curated binding sites, and summary statistics. If you are trying to understand how genes are transcriptionally regulated (e.g. differential expression result interpretation, gene module identification, relative binding strength comparison, binding motif analysis), then proChIPdb can provide detailed information relevant to you.
            </p>

            <!-- TF Dashboards -->
            <h3 id="tf_dash" class="text-center py-3">
                Transcription Factor Dashboards
            </h3>

            <p>
                Each dashboard features a specific transcription factor (for a specific organism and strain) and its ChIP results for at least one condition. Most transcription factors were characterized under M9 media conditions, and some were also characterized under other relevant conditions. In the following section, we include examples of each page element from <a href="tf_dashboard.html?organism=e_coli&tf=Fur&genome=NC_000913_3&i=3" target="_blank">the <i>E. coli</i> K-12 Fur dashboard</a>.
            </p>

            <div class="container-fluid p-0 m-0">
                <div class="row">
                    <div class="col-sm-3">
                        <div class="card p-2" id="meta_text"></div>
                    </div>
                    <div class="col-sm-9">
                        <h4 class="text-center py-3">
                            Metadata Panel
                        </h4>
                        <p>
                            Located in the left-side column of the page, this panel contains basic details about the organism, strain, media used, and supplement(s), if applicable. Hovering over the<button class="meta-info-btn" id="unused-media-info" type="button"><i class="material-icons meta-icons">info</i></button>button adjacent to the media name will reveal additional details about the composition of the media. Supplements are included in detailed, nested dictionaries which describe the concentrations in the units used by their original publications. The accession number from <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank">GEO</a> or <a href="https://www.ncbi.nlm.nih.gov/sra" target="_blank">SRA</a> is also displayed, along with the PMID and DOI of the associated publication.
                        </p><p>
                            Links to the transcription factor's page on external databases are also included here, featuring <a href="ecocyc.org" target="_blank">EcoCyc</a>, <a href="http://regulondb.ccg.unam.mx/" target="_blank">RegulonDB</a>, <a href="https://www.uniprot.org/" target="_blank">UniProt</a>, <a href="https://www.rcsb.org/" target="_blank">the Protein Data Bank</a>, <a href="https://www.pseudomonas.com/" target="_blank">Pseudomonas Genome DB</a>, and <a href="https://aureowiki.med.uni-greifswald.de/Main_Page" target="_blank"><i>Aureo</i>Wiki</a>. Use these to access up-to-date information about the transcription factor, including relevant journal articles and protein structures.
                        </p><p>
                            Below the main links, there may be additional links under the heading "iModulons". iModulons are machine-learning derived gene groupings from analysis of transcriptomes, which can be associated with transcription factors. If the page's transcription factor is predicted to regulate any iModulons, then those iModulons will be listed as links here. Since iModulons capture independent signals in a dataset, they may combine the effects of several transcription factors or approximate nonlinear responses as multiple iModulons, which means that some cases (including Fur) will have more than one associated iModulon. Links go to <a href="imodulondb.org" target="_blank">iModulonDB</a>, which has an <a href="imodulondb.org/about" target="_blank">about page</a> at which you can learn more about this approach. It may be interesting to compare the genes from the iModulon with the binding sites from proChIPdb, as well as to use iModulonDB to learn more about the transcription factor's activity over a wide range of conditions.
                        </p><p>
                            The example on the left shows the metadata for Fur. Fur is the <b>f</b>erric <b>u</b>ptake <b>r</b>egulator, meaning it controls genes relevant to iron transport. It was tested with two conditions: iron-replete (Fe) and iron-starvation (DPD) conditions. In the presence of iron, it is expected that Fur will bind to DNA and repress its target genes. In the presence of DPD, DPD will bind any free iron to create iron starvation conditions, leading to decreased Fur binding (more details available <a href="https://www.nature.com/articles/ncomms5910" target="_blank">here</a>). More information about Fur is available at the provided links. Fur is a major cellular regulator, so it takes part in the regulation of several iModulons.
                    </div>
                </div>
            </div>

            <h4 class="text-center py-3">
                Binding Site Table
            </h4>

            <p>
                This table represents <a href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">MACE</a> results from processing the sequence read data presented in the Genome Viewer. Tabs across the top correspond to each binding condition. Each row represents a curated binding peak. Clicking on a row will update the Genome Viewer to display a zoomed in plot of the corresponding peak. The rows are initially sorted by genome location, and the tallest peaks can easily be found by sorting by descending "Peak Intensity" (clicking the column header twice). The columns are as follows:
            </p>

            <ul>
                <li><b>TF-Condition-#</b>: A unique identifier for the peak, ordered by genome location.</li>
                <li><b>Start Position & End Position</b>: The genome locations of the peak in base pairs are determined by MACE. For each peak, a single-nucleotide resolution border peak is detected by using the Chebyshev Inequality and the border is matched by using the Gale–Shapley stable matching algorithm. For E. coli K-12 MG1655, these locations refer to the <a href="https://www.ncbi.nlm.nih.gov/nuccore/556503834" target="_blank">NC_000913.3 genome</a>.</li>
                <li><b>Peak Intensity</b>: Peak signal intensity is determined with two replicates using MACE peak calling algorithm, which employs Shannon’s relative entropy (H) to compute signal to noise values across the genome for each replicate. Note that the noise value varies for each condition because it captures variation due to inexact cross-linking, exonuclease digestion, dynamic protein conformations, and PCR amplification.</li>
                <li><b>Closest gene</b>: Each peak is assigned to its closest gene, according to genomic positions. This column will contain a gene name regardless of the distance to the gene.</li>
                <li><b>TU Target genes</b>: If a transcriptional unit exists downstream of the binding peak and is within 500 base pairs of the peak, its genes will be listed here. If that TU also has a page in an online database, such as <a href="http://regulondb.ccg.unam.mx/" target="_blank">RegulonDB</a> for <i>E. coli</i>, then the TU will appear as a link to that page.</li>
            </ul>
            <div class="container">
                <div class ="row">
                    <div class="col-md-0 col-lg-1"></div>
                    <div class="col-md-12 col-lg-10">
                        <div class="card">
                            <div class="section-header"> Binding Site Table
                                <button class="btn btn-sm btn-mod" id="download-table">Download CSV</button>
                            </div>
                            <ul class="nav nav-tabs tab-strip" id="TableTab" role="tablist"></ul>
                            <div id="binding_site_table" style="width: 100%; height:300px;"></div>
                        </div>
                    </div>
                    <div class="col-md-0 col-lg-1"></div>
                </div>
            </div>
            <p>
                In the above table, we can learn a great deal about Fur binding. The DPD tab shows relatively low peak intensities, because DPD induces iron starvation that suppresses Fur binding. Switching to the Fe tab, we can sort by peak intensity to see the strongest binding events when Fur binding is stimulated. For example, the binding site Fur-11 has a very high peak intensity and corresponds to the <i>entCEBAH</i> operon, which produces the iron chelator enterobactin. Note that searching on our search page for any of the target genes in this table would return this page as a result.
            </p>

            <h4 class="text-center py-3">
                Genome Viewer
            </h4>

            <p>
                proChIPdb’s genome viewer provides access to a complete, genome-wide view of the transcription factor’s activity under the given conditions. The genome viewer was made using <a href="https://doi.org/10.1101/2020.05.03.075499" target="_blank">igv.js</a>, and it visualizes bigWig ChIP read files generated from bam files using <a href="https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html" target="_blank">deeptools</a>. From top to bottom, the features and tracks of the viewer are:
            </p>
            <ul>
                <li><b>Toolbar</b>: The toolbar contains a search bar, which can be used to search for specific loci by gene location or gene name. It also has useful tools for adding cursor guides, a center line, toggling track labels, downloading the current view, and adjusting the zoom level.</li>
                <li><b>Genome Location</b>: The white bar with red outlines may help orient you to which region of the genome is currently displayed. Underneath that, an axis is labeled with specific base pair numbers.</li>
                <li><b>Nucleotides</b>: If the viewer is zoomed close enough, color-coded nucleotide labels will appear.</li>
                <li><b>Genes</b>: Gene annotations as provided in the genome. Click on a track to see additional details, such as the gene name, locus tag, and product description.</li>
                <li><b>Published TUs</b>: Transcriptional unit annotations, as provided by <a href="https://academic.oup.com/nar/article/48/18/10157/5911745" target="_blank">the Bitome</a>.</li>
                <li><b>Published TSS</b>: Transcription start sites, as provided by the Bitome.</li>
                <li><b>Published TFBS</b>: Transcription factor binding sites, as provided by the Bitome.</li>
                <li><b>Data Tracks</b>: The y-axis of these plots is the number of ChIP reads mapped to the x-axis nucleotide. Replicate numbers and unique conditions (if applicable) are indicated by the track name. To download this data for custom analysis, click “Download bigWig Files” in the upper left corner of the pane.</li>
            </ul>

            <div class="card mt-3 pb-2">
                <div class="section-header"> Genome Viewer
                    <a class="btn btn-sm btn-mod" id="download-bw">Download bigWig files</a>
                </div>
                <div id="igv-div"></div>
            </div>

            <p>
                In the above genome viewer, the bottom four rows show the ChIP data for Fur binding (2 conditions with 2 replicates each). Note that in the DPD conditions, the y axis does not reach a very high value because none of the binding events are particularly strong; this means that noise dominates our view in those conditions. On the other hand, the Fe condition creates several strong peaks. If you'd like, you can scroll back up to the table for the Fe condition and select a binding site like Fur-11. This will zoom the genome viewer into that binding site to see its specific shape. The two peaks for Fur-11 align with the annotated "Fur" binding sites in the "Published TFBS" row. The target genes (starting with <i>entC</i>) can be seen in the "Genes" row. In our less studied transcription factors, this view represents a powerful opportunity for discovery.
            </p>


            <h4 class="text-center py-3">
                Feature Visualization Panel
            </h4>
            <p>
                The tabs within this panel provide additional characterization of the data. In the upper right hand corner of each tab, the menu button (<i class="material-icons">menu</i>) enables PNG, SVG, and data download.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Width
            </h5>
            <p>
                This tab shows a histogram of the binding peak widths from the binding site table. Hover over each bar to see a count of peaks that fall within its corresponding bin.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Position
            </h5>
            <p>
                In this tab, proChIPdb compares the binding locations relative to each target gene. For each gene (as listed in the Target Genes column of the Binding Site Table), the distance from the gene start site to the binding site is measured in base pairs and normalized to the length of the gene. Points are then plotted with this value on the x axis and the peak intensity (S/N) on the y axis. Genes from the same transcriptional unit will have identical y axis values. Clustering on the x axis indicates the distance at which the transcription factor usually exerts its influence on gene expression. Hover over a point to view more details, such as the gene name.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Motif
            </h5>
            <p>
                If this tab exists, then it contains a sequence logo of a significantly enriched motif and its corresponding E-value (E < 0.001). Motifs provide valuable insight about which sequences will bind the TF. Sub-tabs across the bottom of the panel allow you to select from each of the conditions in the dashboard. The menu button in the upper right hand corner of each tab allows download of both the image itself and the position weight matrix (PWM) of the motif. 
            </p><p>
                These motifs were generated using <a href="https://meme-suite.org/meme/doc/meme-chip.html" target="_blank">MEME-ChIP</a> (parameters: meme-minw=5bp, meme-maxw=45bp, -meme-nmotifs=4, filter-thresh=0.001), run on the binding peaks extended with a 20bp margin.
            </p><p>
                In addition, a final tab named using a PMID may be available; this contains the sequence logo that as it appears in the publication from which the data is from. You are encouraged to refer to the original publications for more details about those sequence logos. 
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Venn
            </h5>
            <p>
                The venn diagram compares the target genes obtained in proChIPdb to other target gene sets in the literature. Hover over a section of the venn diagram to see the genes it contains. Areas of agreement indicate strong evidence of direct regulation, and areas of disagreement represent opportunities to improve literature annotation or elucidate condition-specific differences in binding. The specific literature sources are mentioned below each diagram. For <i>E. coli</i>, <a href="https://ecocyc.org/" target="_blank">EcoCyc</a>’s transcriptional regulatory network was used.
            </p>
            <div class="container">
                <div class ="row">
                    <div class="col-md-2 col-lg-3"></div>
                    <div class="col-md-8 col-lg-5">
                        <div class="card">
                            <div class="section-header"> Feature Visualization</div>

                            <ul class="nav nav-tabs tab-strip" id="myTab" role="tablist">

                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab active" id="header-tab-1" data-bs-toggle="tab"
                                            data-bs-target="#tab-1" data-toggle="tab" data-target="#tab-1"
                                            type="button" role="tab" aria-controls="tab-1"
                                            aria-selected="true">Width
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab" id="header-tab-4" data-bs-toggle="tab"
                                            data-bs-target="#tab-4" data-toggle="tab" data-target="#tab-4"
                                            type="button" role="tab" aria-controls="tab-4" aria-selected="false">Position
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab disabled" id="header-tab-3" data-bs-toggle="tab"
                                            data-bs-target="#tab-3" data-toggle="tab" data-target="#tab-3"
                                            type="button" role="tab" aria-controls="tab-3" aria-selected="false">Motif
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab" id="header-tab-2" data-bs-toggle="tab"
                                            data-bs-target="#tab-2" data-toggle="tab" data-target="#tab-2"
                                            type="button" role="tab" aria-controls="tab-2" aria-selected="false">Venn
                                    </button>
                                </li>
                            </ul>
                            <!-- Binding width histogram-->
                            <div class="tab-content" id="myTabContent">
                                <div class="tab-pane fade show active" id="tab-1" role="tabpanel"
                                     aria-labelledby="hist-tab">
                                    <div id="widthHist" style="width:100%; height:300px;"></div>
                                </div>

                                <!--Venn Diagram-->
                                <div class="tab-pane fade" id="tab-2" role="tabpanel" aria-labelledby="venn-tab">
                                    <div id="venn" style="width:100%; height:300px;"></div>
                                </div>

                                <!-- Motif Sequence-->
                                <div class="tab-pane fade" id="tab-3" role="tabpanel" aria-labelledby="contact-tab">
                                    <!--Hamburger Dropdown-->
                                    <div class="dropdown menu-right">
                                        <!--Trigger-->
                                        <button class="btn menu-hamburger-height" type="button" id="dropdownMenuMotif"
                                                data-toggle="dropdown"
                                                aria-haspopup="true" aria-expanded="false">
                                            <i class="material-icons">menu</i>
                                        </button>
                                        <!--Menu-->
                                        <div class="dropdown-menu dropdown-primary">
                                            <a id="motif-download-png" class="dropdown-item" download>Download PNG image</a>
                                            <a id="motif-download-svg" class="dropdown-item" download>Download SVG vector
                                                image</a>
                                            <a id="motif-download-csv" class="dropdown-item" download>Download PWM CSV</a>
                                        </div>
                                    </div>
                                    <div id="motif-banner-title" class="motif-banner-title"></div>

                                    <!--Motif Logo and Selector Buttons-->
                                    <div class="fill-parent">
                                        <div class="fill">
                                            <img style="max-width:95%; max-height:95%" id="motif_sequence" src="">
                                        </div>
                                        <div class="motif-selector-buttons" id="motif_selector_buttons"></div>
                                    </div>

                                </div>

                                <!-- Position Scatterplot-->
                                <div class="tab-pane fade" id="tab-4" role="tabpanel" aria-labelledby="hist-tab">
                                    <div id="position_scatter" style="width:100%; height:300px;"></div>
                                </div>
                            </div>

                        </div>
                    </div>
                    <div class="col-md-2 col-lg-3"></div>
                </div>
            </div>

            <!-- Search -->
            <h3 id="search" class="text-center py-3">
                Search
            </h3>
            <p>
                The <a href="search.html">search page</a> can be reached from any page by clicking "Search" in the upper right hand corner. You have the option to search transcription factors by name (e.g. "AtoC"), by PMID (e.g. "25222563") or by accession number (e.g. "GSE54901"). You can also search genes by name (e.g. "thrA"; includes common synonyms) or locus tag (e.g. "b0002"). Leave both options selected to return all relevant results. Search terms are case-insensitive. Each result that appears below the search bar will be a link to a transcription factor dashboard. The portion of the result that matches your search term will appear bolded and underlined.
            </p><p>
                Transcription factor results will simply list the name, organism, strain, PMID, and accession number of the matching page. Gene results are associated with specific binding peaks upstream of the gene of interest, so they include additional details:
            </p>
            <ul>
                <li>Gene name and locus tag (in parentheses)</li>
                <li>Binding site identifier in the form "Transcription factor name - #"</li>
                <li>Organism and strain name</li>
                <li>Condition name in the form "transcription factor name + condition". The condition may be the media type or the supplement used, for example.</li>
                <li>Peak strength as a signal to noise (S/N) ratio. Note that these values cannot be directly mathematically compared because noise varies from condition to condition, but the peak strength will still provide a valuable indicator of how important the peak is in the ChIP profile.</li>
            </ul>

            <!-- Download -->
            <h3 id="download" class="text-center py-3">
                Download
            </h3>
            <p>
                As described throughout this page, any specific content on a transcription factor dashboard may be downloaded using the buttons in the panels. If you would prefer to download all of the proChIPdb data, you can do so by following the link in the lower left corner of the splash page <a href="https://zenodo.org/record/5168081" target="_blank">(or here)</a>. The folder is organized as follows:
            </p>
            <ul>
                <li><b>Organisms</b>: The first level allows you to select an organism, named "g_species" (e.g. "e_coli"). JSON files used for searching the database are also stored in the top level folder. These JSON files may be useful as full lists of all TFs or target genes.</li>
                <li><b>Strains</b>: After selecting an organism, choose a strain in the second level. Strains are usually named after the RefSeq accession number of the strain's genome. This level also contains a "TF_list" file that lists each TF with some additional metadata, and is used to generate the dataset page and metadata panels. Use the TF_lists as your guide for matching each of the files to their appropriate TFs. If the strain you selected does not have a TF_list available, the TF_list should be in the "all_other" organism folder.</li>
                <li><b>Site Content</b>: Each subfolder for a given strain contains the individual files for the TFs, organized by category.
                    <ul>
                        <li><b>annotation</b>: Genome annotations, including GFF files for the four annotation tracks in the genome viewer.</li>
                        <li><b>binding_widths</b>: Input files for the binding width histograms in the feature visualization panel.</li>
                        <li><b>bw</b>: BigWig files representing the ChIP data, which are displayed in the genome viewer.</li>
                        <li><b>curated_input</b>: GFF files containing curated binding peak locations.</li>
                        <li><b>positions</b>: Input files for the position scatter plot in the feature visualization panel.</li>
                        <li><b>sequence</b>: FASTA and other sequence files for the strain, used in the genome viewer.</li>
                        <li><b>table</b>: Binding site tables.</li>
                        <li><b>venn</b>: Input files for the venn diagrams in the feature visualization panel.</li>
                    </ul>
                </li>
            </ul>

            <!-- References -->
            <h3 id="ref" class="text-center py-3">
                References
            </h3>
            <p class="text-center" style="font-size:16pt">
                proChIPdb manuscript coming soon!
            </p>
            <h4 class="text-center py-3">
                Methods
            </h4>
            <ul>
                <li><b>Experimental</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gky752" target="_blank">Ye Gao, James T Yurkovich, Sang Woo Seo, Ilyas Kabimoldayev, Andreas Dräger, Ke Chen, Anand V Sastry, Xin Fang, Nathan Mih, Laurence Yang, Johannes Eichner, Byung-Kwan Cho, Donghyuk Kim, Bernhard O Palsson, Systematic discovery of uncharacterized transcription factors in Escherichia coli K-12 MG1655, Nucleic Acids Research, 46(20), 2018, 10682–10696.</a></li>
                <li><b>deepTools</b>: <a class="ref_link" href="http://doi.org/10.1093/nar/gkw257" target="_blank">Ramírez, Fidel, Devon P. Ryan, Björn Grüning, Vivek Bhardwaj, Fabian Kilpert, Andreas S. Richter, Steffen Heyne, Friederike Dündar, and Thomas Manke. deepTools2: A next Generation Web Server for Deep-Sequencing Data Analysis. Nucleic Acids Research (2016), Volume 44, Issue W1, 8 July 2016, Pages W160–W165.</a></li>
                <li><b>MACE</b>: <a class="ref_link" href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">Liguo Wang, Junsheng Chen, Chen Wang, Liis Uusküla-Reimand, Kaifu Chen, Alejandra Medina-Rivera, Edwin J. Young, Michael T. Zimmermann, Huihuang Yan, Zhifu Sun, Yuji Zhang, Stephen T. Wu, Haojie Huang, Michael D. Wilson, Jean-Pierre A. Kocher, Wei Li, MACE: model based analysis of ChIP-exo, Nucleic Acids Research, Volume 42, Issue 20, 10 November 2014, Page e156.</a></li>
                <li><b>IGV</b>: <a class="ref_link" href="http://www.nature.com/nbt/journal/v29/n1/abs/nbt.1754.html" target="_blank">James T. Robinson, Helga Thorvaldsdóttir, Wendy Winckler, Mitchell Guttman, Eric S. Lander, Gad Getz, Jill P. Mesirov. Integrative Genomics Viewer. Nature Biotechnology 29, 24–26 (2011).</a></li>
                <li><b>IGV.js</b>: <a class="ref_link" href="https://www.biorxiv.org/content/10.1101/2020.05.03.075499v1" target="_blank">James T. Robinson, Helga Thorvaldsdóttir, Douglass Turner, Jill P. Mesirov. igv.js: an embeddable JavaScript implementation of the Integrative Genomics Viewer (IGV). bioRxiv 2020.05.03075499.</a></li>
                <li><b>MEME</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkp335" target="_blank">Timothy L. Bailey, Mikael Bodén, Fabian A. Buske, Martin Frith, Charles E. Grant, Luca Clementi, Jingyuan Ren, Wilfred W. Li, William S. Noble, "MEME SUITE: tools for motif discovery and searching", Nucleic Acids Research, 37:W202-W208, 2009.</a></li>
            </ul>
            <h4 class="text-center py-3">
                Annotation Sources
            </h4>
            <ul>
                <li><b>The Bitome</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkaa774" target="_blank">Cameron R Lamoureux, Kumari Sonal Choudhary, Zachary A King, Troy E Sandberg, Ye Gao, Anand V Sastry, Patrick V Phaneuf, Donghui Choe, Byung-Kwan Cho, Bernhard O Palsson, The Bitome: digitized genomic features reveal fundamental genome organization, Nucleic Acids Research, Volume 48, Issue 18, 09 October 2020, Pages 10157–10163.</a></li>
                <li><b>EcoCyc</b>: <a class="ref_link" href="https://doi.org/10.1128/ecosalplus.esp-0006-2018" target="_blank">Karp PD, Ong WK, Paley S, Billington R, Caspi R, Fulcher C, Kothari A, Krummenacker M, Latendresse M, Midford PE, Subhraveti P, Gama-Castro S, Muñiz-Rascado L, Bonavides-Martinez C, Santos-Zavaleta A, Mackie A, Collado-Vides J, Keseler IM, Paulsen I. The EcoCyc Database. EcoSal Plus. 2018 Nov.</a></li>
                <li><b>RefSeq</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkv1189" target="_blank">O'Leary NA, et al. Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. Nucleic Acids Res. 2016 Jan.</a></li>
                <li><b>RegulonDB</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gky1077" target="_blank">Alberto SZ, et al. RegulonDB v 10.5: tackling challenges to unify classic and high throughput knowledge of gene regulation in E. coli K-12. Nucleic Acids Research, Volume 47, Issue D1, 08 January 2019, Pages D212–D220.</a></li>
                <li><b>iModulonDB</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkaa810" target="_blank">Rychel, K., Decker, K., Sastry, A.V. et al. iModulonDB: a knowledgebase of microbial transcriptional regulation derived from machine learning. Nucleic Acids Research, Volume 49, Issue D1, 8 January 2021, Pages D112–D120.</a></li>
            </ul>
            <h4 class="text-center py-3">
                Additional References
            </h4>
            <ul>
                <li><a class="ref_link" href="https://doi.org/10.1371/journal.pone.0197272" target="_blank">Latif, H., Federowicz, S., Ebrahim, A., Tarasova, J., Szubin, R., Utrilla, J., Zengler, K., & Palsson, B. O. (2018). ChIP-exo interrogation of Crp, DNA, and RNAP holoenzyme interactions. PloS One, 13(5), e0197272.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1093/nar/gky069" target="_blank">Kim, D., Seo, S. W., Gao, Y., Nam, H., Guzman, G. I., Cho, B.-K., & Palsson, B. O. (2018). Systems assessment of transcriptional regulation on central carbon metabolism by Cra and CRP. Nucleic Acids Research, 46(6), 2901–2917.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/ncomms5910" target="_blank">Seo, S. W., Kim, D., Latif, H., O’Brien, E. J., Szubin, R., & Palsson, B. O. (2014). Deciphering Fur transcriptional regulatory network highlights its complex role beyond iron metabolism in Escherichia coli. Nature Communications, 5, 4910.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/ncomms8970" target="_blank">Seo, S. W., Kim, D., O’Brien, E. J., Szubin, R., & Palsson, B. O. (2015). Decoding genome-wide GadEWX-transcriptional regulatory networks reveals multifaceted cellular responses to acid stress in Escherichia coli. Nature Communications, 6, 7970.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1016/j.celrep.2015.07.043" target="_blank">Seo, S. W., Kim, D., Szubin, R., & Palsson, B. O. (2015). Genome-wide Reconstruction of OxyR and SoxRS Transcriptional Regulatory Networks under Oxidative Stress in Escherichia coli K-12 MG1655. Cell Reports, 12(8), 1289–1299.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/s41598-017-02110-7" target="_blank">Seo, S. W., Gao, Y., Kim, D., Szubin, R., Yang, J., Cho, B.-K., & Palsson, B. O. (2017). Revealing genome-scale transcriptional regulatory landscape of OmpR highlights its expanded regulatory roles under osmotic stress in Escherichia coli K-12 MG1655. Scientific Reports, 7(1), 2181.</a></li>
                <li><a class="ref_link" href="https://www.nature.com/articles/s41467-019-13483-w" target="_blank">Sastry, A. V., Gao, Y., Szubin, R., Hefner, Y., Xu, S., Kim, D., Choudhary, K. S., Yang, L., King, Z. A., & Palsson, B. O. (2019). The Escherichia coli transcriptome mostly consists of independently regulated modules. Nature Communications, 10(1), 5536.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1101/2020.05.21.109694" target="_blank">Gao, Y., Bang, I., Seif, Y., Nam, G., Sastry, A. V., Chen, K., Monk, J. M., Choudhary, K. S., Seo, S. W., Lee, E.-Y., Kim, D., & Palsson, B. O. Reconstruction of Fur pan-regulon uncovers the complexity and diversity of transcriptional regulation in E. coli. </a></li>
            </ul>

            <!-- Contact -->
            <h3 id="contact" class="text-center py-3">
                Contact Us
            </h3>
            <p class="text-center" style="font-size:16pt">
                To ask questions, provide feedback, report an issue, or collaborate with us, please email us at <a href="mailto:be-chip-pro@eng.ucsd.edu">be-chip-pro@eng.ucsd.edu</a>.
            </p>
            <!-- Footer -->
            <footer>
                <hr>
                <div class="container">
                    <div class="row">
                        <div class="col-md-3">
                            <a href="http://systemsbiology.ucsd.edu/">
                                <img class="img-fluid" src="images/sbrg-logo.png">
                            </a>
                        </div>
                        <div class="col-sm-9 m-auto">
                            <p class="copyright text-muted small">
                                proChIPdb is maintained by the Systems Biology Research Group at the University of
                                California, San Diego. <br/>
                            </p>
                        </div>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>

<!-- Load Outside Libraries -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.1.0/papaparse.min.js"></script>
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/venn.js"></script>
<script src="https://code.highcharts.com/modules/histogram-bellcurve.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
<script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>

<!-- Load scripts -->
<script src="js/igv_and_table.js"></script>
<script src="js/peak_gene_positions.js"></script>
<script src="js/width_histogram.js"></script>
<script src="js/venn2.js"></script>

<!-- Determine which TF -->
<script>
    var organism = 'e_coli';
    var TF_name = 'Fur';
    var genome = 'NC_000913.3';
    var index = 3;
    $('#TF_header').append(TF_name);

    $.get('/data/' + organism + '/TF_list.csv', function (file_data) {
        var data = Papa.parse(file_data, {dynamicTyping: true}).data;
        const row = data[parseInt(index) + 1];

        // TF details
        $('#meta_text').append('<p class="meta_cat">ORGANISM</p>');
        $('#meta_text').append('<p class="meta_val"><i>' + row[2] + '</i></p>');
        $('#meta_text').append('<p class="meta_cat">STRAIN</p>');
        $('#meta_text').append('<p class="meta_val">' + row[3] + '</p>');
        $('#meta_text').append('<p class="meta_cat">GENOME ID</p>');
        $('#meta_text').append('<p class="meta_val">' + row[6] + '</p>');
        $('#meta_text').append('<p class="meta_cat">MEDIA</p>');
        $('#meta_text').append('<p class="meta_val">' + row[20] + '<button class="meta-info-btn" id="media-info" type="button" ><i class="material-icons meta-icons">info</i></button></p>');
        $('#meta_text').append('<p class="meta_cat">SUPPLEMENT</p>');
        $('#meta_text').append('<p class="meta_val">' + row[21] + '</p>');


        $('#media-info').popover({
            container: "body",
            title: "additional media information",
            html: true,
            trigger: "hover focus",
            content: function () {
                return `<div class="popover-message">${row[22]}</div>`;
            }
        });


        // out links
        if (organism == 'e_coli') {
            var link_link = '/data/e_coli/outlinks/' + TF_name + '_links.csv';
            var db_num = 4
        } else {
            var link_link = '/data/all_other/outlinks/' + organism + '_' + TF_name + '_links.csv';
            var db_num = 4 //may change as addl data is added
        }
        $.get(link_link, function (link_data) {
            link_table = Papa.parse(link_data, {dynamicTyping: true, skipEmptyLines: true}).data;
            
            var link_text = '<p class="meta_cat">LINKS</p>';
            var include_links = false;
            for (let i = 0; i < db_num; i++) {
                if (link_table[i][1]) {

                    link_text += '<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>';
                    include_links = true;
                }
            }
            if (include_links) {
                $('#meta_text').append(link_text);
            }
            if (link_table.length > db_num) {
                $('#meta_text').append('<p class="meta_cat mt-3">iMODULONS</p>');
                for (let i = db_num; i < link_table.length; i++) {
                    $('#meta_text').append('<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>');
                }
            }

        })
        $('#meta_text').append('<p class="meta_cat">RAW DATA ACCESSION</p>');
        $('#meta_text').append('<p class="meta_val">' + row[14] + '</p>');
        if (row[15]|row[19]) {
            $('#meta_text').append('<p class="meta_cat">REFERENCE</p>');
            $('#meta_text').append('<p class="meta_val mb-0">PMID: ' + row[19] + '</p>');
            $('#meta_text').append('<p class="meta_val">DOI: <a href="' + row[15] + '">' + row[15].replace("https://doi.org/", "") + '</a></p>');

        }
        if (row[26]) {
            $('#meta_text').append('<p class="meta_cat">EXTRA NOTES</p>');
            $('#meta_text').append('<p class="meta_val">' + row[26] + '</p>');
        }

        // IGV and table
        const [title, fileName] = row[12].split(";")[0].split("|");
        generateIGVandTable(TF_name, row, fileName);
        // First, get row[12] to get list of filenames to create the tabs for x number of files
        const conditions = row[12].split(";");
        // parse the table row of Title A|xyz.json;Title B|def.json;...
        // access the tabs parent component by ID
        $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab active" id="table-tab-1" data-bs-toggle="tab"
                    data-bs-target="#table-tab-1" data-toggle="tab" data-target="#table-tab-1"
                    type="button" role="tab" aria-controls="table-tab-1"
                    aria-selected="true">${title}
            </button>
        </li>
        `);
        $('#table-tab-1').click(() => {
            generateIGVandTable(TF_name, row, fileName);
        });
        // loop through the loaded values of `Title A|xyz.json` and create a tab for each
        conditions.slice(1, conditions.length).forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab" id="table-tab-${i + 2}" data-bs-toggle="tab"
                    data-bs-target="#table-tab-${i + 2}" data-toggle="tab" data-target="#table-tab-${i + 2}"
                    type="button" role="tab" aria-controls="table-tab-${i + 2}"
                    aria-selected="true">${conditionTitle}
            </button>
        </li>
        `);
            $(`#table-tab-${i + 2}`).click(() => {
                generateIGVandTable(TF_name, row, conditionFile);
            });
        });

        // Loop through it all again and create buttons to choose the motif graphic
        let hasFoundFirstImageToRender = false;
        const clearAllActiveMotifButtonFunctions = [];

        conditions.forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            const motifImageAndCSVBase = conditionFile.replace("_binding_table.json", "");
            const imageFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.png`;
            const csvFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.csv`;
            const spacelessConditionTitle = conditionTitle.replace(/ /g, "_");

            $('#motif_selector_buttons').append(`
                <button class="btn btn-outline-dark btn-sm" id="motif-button-${spacelessConditionTitle}">${conditionTitle}</button>
            `);

            $.get(imageFilePath, function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = imageFilePath;
                    document.getElementById("motif-download-svg").classList.add('d-none');
                    document.getElementById("motif-download-png").href = imageFilePath;
                    document.getElementById("motif-download-csv").href = csvFilePath;
                    document.getElementById("motif-download-csv").classList.remove('d-none');

                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.add('active');
                    document.getElementById('motif-banner-title').innerText = 'Logo identified by MEME-ChIP';
                };

                clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.remove('active'));
                $(`#motif-button-${spacelessConditionTitle}`).click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }
            }).catch(() => {
                //     $('#motif_selector_buttons').append(`
                //     <button disabled title="No motif logo found using MEME-ChIP">${conditionTitle}</button>
                // `);
                $(`#motif-button-${spacelessConditionTitle}`).addClass('disabled');
                $(`#motif-button-${spacelessConditionTitle}`).addClass('gray-btn');
                $(`#motif-button-${spacelessConditionTitle}`).attr('disabled', true);
                $(`#motif-button-${spacelessConditionTitle}`).attr("title", "No motif logo found using MEME-ChIP");
            });
        });

        // Add literature motif information, if available
        const motifSVG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.svg';
        const motifPNG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.png';

        $('#motif_selector_buttons').append(`
            <button class="btn btn-outline-dark btn-sm" id="motif-button-lit">PMID: ${row[19]}</button>
        `);


        $.get(motifSVG)
            .done(function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = motifSVG;
                    document.getElementById("motif-download-svg").href = motifSVG;
                    document.getElementById("motif-download-svg").classList.remove('d-none');
                    document.getElementById("motif-download-png").href = motifPNG;
                    document.getElementById("motif-download-csv").classList.add('d-none');


                    clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-lit`).classList.remove('active'));
                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-lit`).classList.add('active');
                    document.getElementById('motif-banner-title').innerText = `Logo from PMID ${row[19]}`;
                };

                $('#motif-button-lit').click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }

                // document.getElementById("motif_sequence").src = motifSVG;
                // document.getElementById("motif-download-svg").href = motifSVG;
            }).fail(function () {
            $.get(motifPNG)
                .done(function () {
                    // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                    document.getElementById("header-tab-3").classList.remove('disabled');

                    const changeImageHandler = function () {
                        document.getElementById("motif_sequence").src = motifPNG;
                        document.getElementById("motif-download-svg").classList.add('d-none');
                        document.getElementById("motif-download-png").href = motifPNG;
                        document.getElementById("motif-download-csv").classList.add('d-none');

                        clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-lit`).classList.remove('active'));
                        clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                        document.getElementById(`motif-button-lit`).classList.add('active');
                        document.getElementById('motif-banner-title').innerText = `Logo from PMID ${row[19]}`;
                    };

                    $('#motif-button-lit').click(changeImageHandler);

                    if (!hasFoundFirstImageToRender) {
                        changeImageHandler();
                        hasFoundFirstImageToRender = true;
                    }
                    // document.getElementById("motif_sequence").src = motifPNG;
                    // document.getElementById("motif-download-svg").remove();
                }).fail(function () {
                // document.getElementById("header-tab-3").classList.add('disabled')
                // document.getElementById("motif-button-lit").classList.add('disabled');
                // document.getElementById("motif-button-lit").classList.add('gray-btn');
                $('#motif-button-lit').remove();
            })
        });

        // if (!hasFoundFirstImageToRender) {
        //     document.getElementById("header-tab-3").classList.add('disabled');
        // }

        // deal with disabling header tab here

        // Venn
        $.get('/data/' + organism + '/' + genome + '/venn/' + TF_name.substr(0, 4) + '_venn.csv', function (data) {
            generateVenn2(data, 'venn');
        }).catch(() => {
            document.getElementById("header-tab-2").classList.add('disabled')
        });

        // Binding width histogram
        $.get('/data/' + organism + '/' + genome + '/binding_widths/' + TF_name + "_widths.csv", function (width_data) {
            generateWidthHistogram(width_data, 'widthHist')
        }).catch(() => {
            document.getElementById("widthHist").innerText = 'Binding width histogram not available.'
            document.getElementById("widthHist").classList.add('NA_text')
        });

        // Peak/Gene Position Scatter
        $.get('/data/' + organism + '/' + genome + '/positions/' + TF_name + "_positions.csv", function (positions_data) {
            generatePeakScatter(positions_data, row, "position_scatter");
        }).catch(() => {
            document.getElementById("header-tab-4").classList.add('disabled')
        });

    });

</script>
</body>
</html>
