<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>About</title>

    <!-- CSS imports -->
    <link rel="stylesheet" href="css/bootstrap.css" type="text/css">
    <link rel="stylesheet" href="css/common.css" type="text/css">
    <link rel="stylesheet" href="css/tf_dashboard.css" type="text/css">
    <link rel="stylesheet" href="css/about-page.css" type="text/css">
    <link rel="stylesheet" href="css/tabulator_bootstrap4.min.css" rel="stylesheet">

    <!-- javascript imports -->
    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
    <script src="js/bootstrap.bundle.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.8.1/dist/igv.min.js"></script>
    <script src="js/tabulator.min.js" type="text/javascript"></script>

    <!-- font -->
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&family=Sulphur+Point:wght@300;400;700&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <!-- CSS -->
    <style>
        * {
            font-family: 'Lato', 'Roboto', 'Helvetica', 'Segoe UI', sans-serif;
        }
    </style>

</head>

<body>
<!-- Page Content -->
<div class="container-fluid page-container">
    <!-- Top Navigation Bar -->
    <nav class="navbar navbar-expand-md sticky-top header">
        <a class="navbar-brand" href="index.html">
             <img src="images/ChIPdb_logo.png" height=35 class="d-inline-block align-top"
            alt="" loading="lazy">
            ChIP-pro 1.0.0 </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav mr-auto">
                <li class="nav-item dropdown">
                    <button class="btn nav-color dropdown-toggle" type="button" id="dropdownMenu2"
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        Organism Dataset
                    </button>
                    <div class="dropdown-menu" aria-labelledby="dropdownMenu2">
                        <a class="dropdown-item" href="dataset_page.html?organism=e_coli"><i>E. coli</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=all_other">All Other Organisms</a>
                    </div>
                </li>
            </ul>
            <ul class="navbar-nav">
                <li class="nav-item">
                    <a class="mr-4" href="about.html"><span><i class="material-icons">info</i></span>
                        <span>About</span>
                    </a>
                </li>
                <li class="nav-item">
                    <a class="mr-4" href="search.html"><span><i class="material-icons">search</i></span>
                        <span>Search</span>
                    </a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="row row-no-margin">
        <!-- Left Navigation -->
        <div id="left_col" class="col-sm-12 col-md-2">
            <div id="left_col_content">
                <ul class="nav flex-column">
                    <li class="nav-item">
                        <a class="nav-link" href="#use_site">How to Use this Site</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#tf_dash">Transcription Factor Dashboards</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#search">Search</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#ref">References</a>
                    </li>
                    <li class="nav-item">
                        <a class="nav-link" href="#contact">Contact Us</a>
                    </li>
                </ul>
            </div>
        </div>
        <!-- page content -->
        <div class="col-sm-10">
            <!-- Title -->
            <h1 class="text-center py-3">About ChIP-pro</h1>

            <p>
                Welcome to <b>ChIP-pro</b>, the <b>C</b>hromatin <b>I</b>mmuno<b>P</b>recipitation database for <b>PRO</b>karyotic organisms! This tool enables microbiologists to easily browse 140 chip-seq and chip-exo profiles for various transcription factors (TFs) across 7 organisms. Currently, most of our profiles are for <i>Escherichia coli</i>. We provide curated tables of binding sites, interactive plots and genome viewers, as well as comparisons against literature binding sites. 
            </p>

            <!-- Using the Site -->
            <h3 id="use_site" class="text-center py-3">
                How to Use this Site
            </h3>

            <p>
                Begin by selecting your organism of interest from the splash page. Next, you can browse the table of transcription factors from our dataset page. ChIP-pro covers well-studied transcription factors, as well as a wide scope of <i>E. coli</i> y-TFs (relatively poorly characterized transcription factors). 
            </p><p>
                You can also <a href="#search">search</a> for transcription factors or genes using the search button in the upper left corner.
            </p><p>
                Once you’ve reached a transcription factor dashboard, you can browse its binding sites and target genes in the curated table. Interested in a specific binding site? Click on it to see the peak in our genome viewer. You can also click through the tabs in the upper right panel to see global characterizations of the transcription factor’s binding, such as its binding motif, peak width, peak location relative to its target genes, and the concordance between the data in ChIP-pro and other databases of transcription factor binding. For more details on the dashboard components, see the sections below.
            </p><p>
                It is our hope that ChIP-pro will enrich your research by allowing easy access to this compendium of binding intensity data, curated binding sites, and summary statistics. If you are trying to understand how genes are transcriptionally regulated (e.g. differential expression result interpretation, gene module identification, relative binding strength comparison, binding motif analysis), then ChIP-pro can provide detailed information relevant to you.
            </p>

            <!-- TF Dashboards -->
            <h3 id="tf_dash" class="text-center py-3">
                Transcription Factor Dashboards
            </h3>

            <p>
                Each dashboard features a specific transcription factor (for a specific organism and strain) and its ChIP results for at least one condition. Most transcription factors were characterized under M9 media conditions, and some were also characterized under other relevant conditions. In the following section, we include examples of each page element from <a href="tf_dashboard.html?organism=e_coli&tf=YdcI&genome=NC_000913_3&i=16" target="_blank">the <i>E. coli</i> YddM dashboard</a>.
            </p>

            <div class="container-fluid p-0 m-0">
                <div class="row">
                    <div class="col-sm-2">
                        <div class="card p-2" id="meta_text"></div>
                    </div>
                    <div class="col-sm-10">
                        <h4 class="text-center py-3">
                            Metadata Panel
                        </h4>
                        <p>
                            Located in the left-side column of the page, this panel contains basic details about the organism, strain, media used, and supplement(s), if applicable.
                        </p>
                        <p>
                            The example on the left shows the metadata for YddM, in which a simple M9 media condition was used.
                    </div>
                </div>
            </div>

            <h4 class="text-center py-3">
                Binding Site Table
            </h4>

            <p>
                This table represents <a href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">MACE</a> results from processing the sequence read data presented in the Genome Viewer. Each row represents a curated binding peak. Clicking on a row will update the Genome Viewer to display a zoomed in plot of the corresponding peak. The rows are initially sorted by genome location, and the tallest peaks can easily be found by sorting by descending "Peak Intensity" (clicking the column header twice). The columns are as follows:
            </p>

            <ul>
                <li><b>TF-#</b>: A unique identifier for the peak, ordered by genome location.</li>
                <li><b>Binding Condtion</b>: Takes the form “TF + condition.”</li>
                <li><b>Start Position & End Position</b>: The genome locations of the peak in base pairs are determined by MACE. For each peak, a single-nucleotide resolution border peak is detected by using the Chebyshev Inequality and the border is matched by using the Gale–Shapley stable matching algorithm. For E. coli K-12 MG1655, these locations refer to the <a href="https://www.ncbi.nlm.nih.gov/nuccore/556503834" target="_blank">NC_000913.3 genome</a>.</li>
                <li><b>Peak Intensity</b>: Peak signal intensity is determined with two replicates using MACE peak calling algorithm, which employs Shannon’s relative entropy (H)to compute signal to noise values across the genome for each replicate. Note that the noise value varies for each condition because it captures variation due to inexact cross-linking, exonuclease digestion, dynamic protein conformations, and PCR amplification.</li>
                <li><b>Target genes</b>: Each peak is assigned to the target gene, according to genomic positions.</li>
            </ul>
            <div class="container">
                <div class ="row">
                    <div class="col-md-0 col-lg-1"></div>
                    <div class="col-md-12 col-lg-10">
                        <div class="card">
                            <div class="section-header"> Binding Site Table
                                <button class="btn btn-sm btn-mod" id="download-table">Download CSV</button>
                            </div>
                            <div id="binding_site_table" style="width: 100%; height:342px;"></div>
                        </div>
                    </div>
                    <div class="col-md-0 col-lg-1"></div>
                </div>
            </div>
            <p>
                In the above table, we see that the strongest binding by YddM is YddM-5. The target gene is <i>yddM</i>, so this transcripion factor strongly self-regulates. 

            <h4 class="text-center py-3">
                Genome Viewer
            </h4>

            <p>
                ChIP-pro’s genome viewer provides access to a complete, genome-wide view of the transcription factor’s activity under the given conditions. The genome viewer was made using <a href="https://doi.org/10.1101/2020.05.03.075499" target="_blank">igv.js</a>, and it visualizes bigWig ChIP read files generated from bam files using <a href="https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html" target="_blank">deeptools</a>. From top to bottom, the features and tracks of the viewer are:
            </p>
            <ul>
                <li><b>Toolbar</b>: The toolbar contains a search bar, which can be used to search for specific loci by gene location or gene name. It also has useful tools for adding cursor guides, a center line, toggling track labels, downloading the current view, and adjusting the zoom level.</li>
                <li><b>Genome Location</b>: The white bar with red outlines may help orient you to which region of the genome is currently displayed. Underneath that, an axis is labeled with specific base pair numbers.</li>
                <li><b>Nucleotides</b>: If the viewer is zoomed close enough, color-coded nucleotide labels will appear.</li>
                <li><b>Genes</b>: Gene annotations as provided in the genome. Click on a track to see additional details, such as the gene name, locus tag, and product description.</li>
                <li><b>Published TUs</b>: Transcriptional unit annotations, as provided by <a href="https://academic.oup.com/nar/article/48/18/10157/5911745" target="_blank">the Bitome</a>.</li>
                <li><b>Published TSS</b>: Transcription start sites, as provided by the Bitome.</li>
                <li><b>Published TFBS</b>: Transcription factor binding sites, as provided by the Bitome.</li>
                <li><b>Data Tracks</b>: The y-axis of these plots is the number of ChIP reads mapped to the x-axis nucleotide. Replicate numbers and unique conditions (if applicable) are indicated by the track name. To download this data for custom analysis, click “Download bigWig Files” in the upper left corner of the pane.</li>
            </ul>

            <div class="card mt-3 pb-2">
                <div class="section-header"> Genome Viewer
                    <a class="btn btn-sm btn-mod" id="download-bw">Download bigWig files</a>
                </div>
                <div id="igv-div"></div>
            </div>


            <h4 class="text-center py-3">
                Feature Visualization Panel
            </h4>
            <p>
                The tabs within this panel provide additional characterization of the data. In the upper right hand corner of each tab, the menu button (<i class="material-icons">menu</i>) enables PNG, SVG, and data download.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Width
            </h5>
            <p>
                This tab shows a histogram of the binding peak widths from the binding site table. Hover over each bar to see a count of peaks that fall within its corresponding bin.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Position
            </h5>
            <p>
                In this tab, ChIP-pro compares the binding locations relative to each target gene. For each gene (as listed in the Target Genes column of the Binding Site Table), the distance from the gene start site to the binding site is measured in base pairs and normalized to the length of the gene. Points are then plotted with this value on the x axis and the peak intensity (S/N) on the y axis. Genes from the same transcriptional unit will have identical y axis values. Clustering on the x axis indicates the distance at which the transcription factor usually exerts its influence on gene expression. Hover over a point to view more details, such as the gene name.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Motif
            </h5>
            <p>
                If this tab exists, then it contains a sequence logo of a significantly enriched motif for these binding sites and its corresponding E-value. The motif and logo are generated by <a href="https://meme-suite.org/meme/index.html" target="_blank">the MEME Suite</a> (parameters: width = 20bp, E-value cutoff = 0.001). The dot with arrows represents the center of the motif, which assists in identifying palindromic sequences. 
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Venn
            </h5>
            <p>
                The venn diagram compares the target genes obtained in ChIP-pro to other target gene sets in the literature. Hover over a section of the venn diagram to see the genes it contains. Areas of agreement indicate strong evidence of direct regulation, and areas of disagreement represent opportunities to improve literature annotation or elucidate condition-specific differences in binding. The specific literature sources are mentioned below each diagram. For <i>E. coli</i>, <a href="https://ecocyc.org/" target="_blank">EcoCyc</a>’s transcriptional regulatory network was used.
            </p>
            <div class="container">
                <div class ="row">
                    <div class="col-md-0 col-lg-4"></div>
                    <div class="col-md-12 col-lg-4">
                        <div class="card">
                            <div class="section-header"> Feature Visualization</div>
                            <ul class="nav nav-tabs tab-strip" id="myTab" role="tablist">

                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab active" id="header-tab-1" data-bs-toggle="tab"
                                            data-bs-target="#tab-1" data-toggle="tab" data-target="#tab-1"
                                            type="button" role="tab" aria-controls="tab-1"
                                            aria-selected="true">Width
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab" id="header-tab-4" data-bs-toggle="tab"
                                            data-bs-target="#tab-4" data-toggle="tab" data-target="#tab-4"
                                            type="button" role="tab" aria-controls="tab-4" aria-selected="false">Position
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab" id="header-tab-3" data-bs-toggle="tab"
                                            data-bs-target="#tab-3" data-toggle="tab" data-target="#tab-3"
                                            type="button" role="tab" aria-controls="tab-3" aria-selected="false">Motif
                                    </button>
                                </li>
                                <li class="nav-item" role="presentation">
                                    <button class="nav-link round-tab" id="header-tab-2" data-bs-toggle="tab"
                                            data-bs-target="#tab-2" data-toggle="tab" data-target="#tab-2"
                                            type="button" role="tab" aria-controls="tab-2" aria-selected="false">Venn
                                    </button>
                                </li>
                            </ul>
                            <!-- Binding width histogram-->
                            <div class="tab-content" id="myTabContent">
                                <div class="tab-pane fade show active" id="tab-1" role="tabpanel"
                                     aria-labelledby="hist-tab">
                                    <div id="widthHist" style="width:100%; height:300px;"></div>
                                </div>

                                <!--Venn Diagram-->
                                <div class="tab-pane fade" id="tab-2" role="tabpanel" aria-labelledby="venn-tab">
                                    <div id="venn" style="width:100%; height:300px;"></div>
                                </div>

                                <!-- Motif Sequence-->
                                <div class="tab-pane fade" id="tab-3" role="tabpanel" aria-labelledby="contact-tab">

                                    <!--Hamburger Dropdown-->
                                    <div class="dropdown menu-right">
                                        <!--Trigger-->
                                        <button class="btn menu-hamburger-height" type="button" id="dropdownMenuMotif"
                                                data-toggle="dropdown"
                                                aria-haspopup="true" aria-expanded="false">
                                            <i class="material-icons">menu</i>
                                        </button>
                                        <!--Menu-->
                                        <div class="dropdown-menu dropdown-primary">
                                            <a class="dropdown-item" href="#">Download PNG image</a>
                                            <a id="motif-download" class="dropdown-item" download>Download SVG vector image</a>
                                        </div>
                                    </div>
                                    <!--/Hamburger Dropdown-->

                                    <div style="width:100%; height:300px" class="fill">
                                        <img id="motif_sequence" src=""/>
                                    </div>
                                </div>

                                <!-- Position Scatterplot-->
                                <div class="tab-pane fade" id="tab-4" role="tabpanel" aria-labelledby="hist-tab">
                                    <div id="position_scatter" style="width:100%; height:300px;"></div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="col-md-0 col-lg-4"></div>
                </div>
            </div>

            <!-- Search -->
            <h3 id="search" class="text-center py-3">
                Search
            </h3>
            <p>
                The <a href="search.html">search page</a> can be reached from any page by clicking "Search" in the upper right hand corner. You have the option to search transcription factors by name (e.g. "AtoC") or genes by name (e.g. "thrA") or locus tag (e.g. "b0002"); you can also leave both options selected to return all relevant results. Search terms are case-insensitive. Each result that appears below the search bar will be a link to a transcription factor dashboard.
            </p><p>
                Transcription factor results will simply list the name, organism, and strain of the matching page. Gene results are associated with specific binding peaks upstream of the gene of interest, so they include additional details:
            </p>
            <ul>
                <li>Gene name and locus tag (in parentheses)</li>
                <li>Binding site identifier in the form "Transcription factor name - #"</li>
                <li>Organism and strain name</li>
                <li>Condition name in the form "transcription factor name + condition". The condition may be the media type or the supplement used, for example.</li>
                <li>Peak strength as a signal to noise (S/N) ratio. Note that these values cannot be directly mathematically compared because noise varies from condition to condition, but the peak strength will still provide a valuable indicator of how important the peak is in the ChIP profile.</li>
            </ul>

            <!-- References -->
            <h3 id="ref" class="text-center py-3">
                References
            </h3>
            <p class="text-center" style="font-size:16pt">
                ChIP-pro manuscript coming soon!
            </p>
            <h4 class="text-center py-3">
                Methods
            </h4>
            <ul>
                <li><b>Experimental</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gky752" target="_blank">Ye Gao, James T Yurkovich, Sang Woo Seo, Ilyas Kabimoldayev, Andreas Dräger, Ke Chen, Anand V Sastry, Xin Fang, Nathan Mih, Laurence Yang, Johannes Eichner, Byung-Kwan Cho, Donghyuk Kim, Bernhard O Palsson, Systematic discovery of uncharacterized transcription factors in Escherichia coli K-12 MG1655, Nucleic Acids Research, 46(20), 2018, 10682–10696.</a></li>
                <li><b>deepTools</b>: <a class="ref_link" href="http://doi.org/10.1093/nar/gkw257" target="_blank">Ramírez, Fidel, Devon P. Ryan, Björn Grüning, Vivek Bhardwaj, Fabian Kilpert, Andreas S. Richter, Steffen Heyne, Friederike Dündar, and Thomas Manke. deepTools2: A next Generation Web Server for Deep-Sequencing Data Analysis. Nucleic Acids Research (2016), Volume 44, Issue W1, 8 July 2016, Pages W160–W165.</a></li>
                <li><b>MACE</b>: <a class="ref_link" href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">Liguo Wang, Junsheng Chen, Chen Wang, Liis Uusküla-Reimand, Kaifu Chen, Alejandra Medina-Rivera, Edwin J. Young, Michael T. Zimmermann, Huihuang Yan, Zhifu Sun, Yuji Zhang, Stephen T. Wu, Haojie Huang, Michael D. Wilson, Jean-Pierre A. Kocher, Wei Li, MACE: model based analysis of ChIP-exo, Nucleic Acids Research, Volume 42, Issue 20, 10 November 2014, Page e156.</a></li>
                <li><b>IGV</b>: <a class="ref_link" href="http://www.nature.com/nbt/journal/v29/n1/abs/nbt.1754.html" target="_blank">James T. Robinson, Helga Thorvaldsdóttir, Wendy Winckler, Mitchell Guttman, Eric S. Lander, Gad Getz, Jill P. Mesirov. Integrative Genomics Viewer. Nature Biotechnology 29, 24–26 (2011).</a></li>
                <li><b>IGV.js</b>: <a class="ref_link" href="https://www.biorxiv.org/content/10.1101/2020.05.03.075499v1" target="_blank">James T. Robinson, Helga Thorvaldsdóttir, Douglass Turner, Jill P. Mesirov. igv.js: an embeddable JavaScript implementation of the Integrative Genomics Viewer (IGV). bioRxiv 2020.05.03075499.</a></li>
                <li><b>MEME</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkp335" target="_blank">Timothy L. Bailey, Mikael Bodén, Fabian A. Buske, Martin Frith, Charles E. Grant, Luca Clementi, Jingyuan Ren, Wilfred W. Li, William S. Noble, "MEME SUITE: tools for motif discovery and searching", Nucleic Acids Research, 37:W202-W208, 2009.</a></li>
            </ul>
            <h4 class="text-center py-3">
                Annotation Sources
            </h4>
            <ul>
                <li><b>The Bitome</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkaa774" target="_blank">Cameron R Lamoureux, Kumari Sonal Choudhary, Zachary A King, Troy E Sandberg, Ye Gao, Anand V Sastry, Patrick V Phaneuf, Donghui Choe, Byung-Kwan Cho, Bernhard O Palsson, The Bitome: digitized genomic features reveal fundamental genome organization, Nucleic Acids Research, Volume 48, Issue 18, 09 October 2020, Pages 10157–10163.</a></li>
                <li><b>EcoCyc</b>: <a class="ref_link" href="https://doi.org/10.1128/ecosalplus.esp-0006-2018" target="_blank">Karp PD, Ong WK, Paley S, Billington R, Caspi R, Fulcher C, Kothari A, Krummenacker M, Latendresse M, Midford PE, Subhraveti P, Gama-Castro S, Muñiz-Rascado L, Bonavides-Martinez C, Santos-Zavaleta A, Mackie A, Collado-Vides J, Keseler IM, Paulsen I. The EcoCyc Database. EcoSal Plus. 2018 Nov.</a></li>
                <li><b>RefSeq</b>: <a class="ref_link" href="https://doi.org/10.1093/nar/gkv1189" target="_blank">O'Leary NA, et al. Reference sequence (RefSeq) database at NCBI: current status, taxonomic expansion, and functional annotation. Nucleic Acids Res. 2016 Jan.</a></li>
            </ul>
            <h4 class="text-center py-3">
                Additional References
            </h4>
            <ul>
                <li><a class="ref_link" href="https://doi.org/10.1371/journal.pone.0197272" target="_blank">Latif, H., Federowicz, S., Ebrahim, A., Tarasova, J., Szubin, R., Utrilla, J., Zengler, K., & Palsson, B. O. (2018). ChIP-exo interrogation of Crp, DNA, and RNAP holoenzyme interactions. PloS One, 13(5), e0197272.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1093/nar/gky069" target="_blank">Kim, D., Seo, S. W., Gao, Y., Nam, H., Guzman, G. I., Cho, B.-K., & Palsson, B. O. (2018). Systems assessment of transcriptional regulation on central carbon metabolism by Cra and CRP. Nucleic Acids Research, 46(6), 2901–2917.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/ncomms5910" target="_blank">Seo, S. W., Kim, D., Latif, H., O’Brien, E. J., Szubin, R., & Palsson, B. O. (2014). Deciphering Fur transcriptional regulatory network highlights its complex role beyond iron metabolism in Escherichia coli. Nature Communications, 5, 4910.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/ncomms8970" target="_blank">Seo, S. W., Kim, D., O’Brien, E. J., Szubin, R., & Palsson, B. O. (2015). Decoding genome-wide GadEWX-transcriptional regulatory networks reveals multifaceted cellular responses to acid stress in Escherichia coli. Nature Communications, 6, 7970.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1016/j.celrep.2015.07.043" target="_blank">Seo, S. W., Kim, D., Szubin, R., & Palsson, B. O. (2015). Genome-wide Reconstruction of OxyR and SoxRS Transcriptional Regulatory Networks under Oxidative Stress in Escherichia coli K-12 MG1655. Cell Reports, 12(8), 1289–1299.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1038/s41598-017-02110-7" target="_blank">Seo, S. W., Gao, Y., Kim, D., Szubin, R., Yang, J., Cho, B.-K., & Palsson, B. O. (2017). Revealing genome-scale transcriptional regulatory landscape of OmpR highlights its expanded regulatory roles under osmotic stress in Escherichia coli K-12 MG1655. Scientific Reports, 7(1), 2181.</a></li>
                <li><a class="ref_link" href="https://www.nature.com/articles/s41467-019-13483-w" target="_blank">Sastry, A. V., Gao, Y., Szubin, R., Hefner, Y., Xu, S., Kim, D., Choudhary, K. S., Yang, L., King, Z. A., & Palsson, B. O. (2019). The Escherichia coli transcriptome mostly consists of independently regulated modules. Nature Communications, 10(1), 5536.</a></li>
                <li><a class="ref_link" href="https://doi.org/10.1101/2020.05.21.109694" target="_blank">Gao, Y., Bang, I., Seif, Y., Nam, G., Sastry, A. V., Chen, K., Monk, J. M., Choudhary, K. S., Seo, S. W., Lee, E.-Y., Kim, D., & Palsson, B. O. Reconstruction of Fur pan-regulon uncovers the complexity and diversity of transcriptional regulation in E. coli. </a></li>
            </ul>

            <!-- Contact -->
            <h3 id="contact" class="text-center py-3">
                Contact Us
            </h3>
            <p class="text-center" style="font-size:16pt">
                To ask questions, provide feedback, report an issue, or collaborate with us, please email us at <a href="mailto:chip-pro@ucsd.edu">chip-pro@ucsd.edu</a>.
            </p>
            <!-- Footer -->
            <footer>
                <hr>
                <div class="container">
                    <div class="row">
                        <div class="col-md-3">
                            <a href="http://systemsbiology.ucsd.edu/">
                                <img class="img-fluid" src="images/sbrg-logo.png">
                            </a>
                        </div>
                        <div class="col-sm-9 m-auto">
                            <p class="copyright text-muted small">
                                ChIP-pro is maintained by the Systems Biology Research Group at the University of
                                California, San Diego. <br/>
                            </p>
                        </div>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>

<!-- Load Outside Libraries -->
<script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.1.0/papaparse.min.js"></script>
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/venn.js"></script>
<script src="https://code.highcharts.com/modules/histogram-bellcurve.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
<script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>

<!-- Load scripts -->
<script src="js/igv_and_table.js"></script>
<script src="js/peak_gene_positions.js"></script>
<script src="js/width_histogram.js"></script>
<script src="plot_dev/binding_site_table/binding_site_table.js"></script>
<script src="plot_dev/venn/venn.js"></script>

<!-- Determine which TF -->
<script>
    var organism = 'e_coli';
    var TF_name = 'YddM';
    var genome = 'NC_000913_3';
    var index = 17;
    $('#TF_header').append(TF_name);

    // motif
    const motifSVG = '/plot_dev/motifs/images/' + TF_name + '.svg';
    document.getElementById("motif_sequence").src = motifSVG;
    document.getElementById("motif_sequence").onerror = () =>
        document.getElementById("header-tab-3").classList.add('disabled')
    document.getElementById("motif-download").href = motifSVG;


    $.get('/data/' + organism + '/TF_list.csv', function (file_data) {
        var data = Papa.parse(file_data, {dynamicTyping: true}).data;
        const row = data[parseInt(index) + 1];

        // TF details
        $('#meta_text').append('<p class="meta_cat">ORGANISM</p>');
        $('#meta_text').append('<p class="meta_val"><i>' + row[2] + '</i></p>');
        $('#meta_text').append('<p class="meta_cat">STRAIN</p>');
        $('#meta_text').append('<p class="meta_val">' + row[3] + '</p>');
        $('#meta_text').append('<p class="meta_cat">MEDIA</p>');
        $('#meta_text').append('<p class="meta_val">' + row[4] + '</p>');
        $('#meta_text').append('<p class="meta_cat">SUPPLEMENT</p>');
        $('#meta_text').append('<p class="meta_val">' + row[5] + '</p>');


        // IGV and table
        generateIGVandTable(TF_name, row)

        // Venn
        $.getJSON('/data/' + organism + '/' + genome + '/venn/' + TF_name.substr(0, 4) + '_venn.json', function (data) {
            generateVenn(data, 'venn');
        }).catch(() => {
            document.getElementById("header-tab-2").classList.add('disabled')
        });;

        // Binding width histogram
         $.get('/data/' + organism + '/' + genome + '/binding_widths/' + TF_name + "_widths.csv", function (width_data) {
             generateWidthHistogram(width_data, 'widthHist')
         }).catch(() => {
            document.getElementById("widthHist").innerText = 'Binding width histogram not available.'
        });

        // Peak/Gene Position Scatter
        $.get('/data/' + organism + '/' + genome + '/positions/' + TF_name + "_positions.csv", function (positions_data) {
            generatePeakScatter(positions_data, "position_scatter");
        }).catch(() => {
            document.getElementById("header-tab-4").classList.add('disabled')
        });

    });

</script>
</body>
</html>
