<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>TF Dashboard</title>

    <!-- javascript imports -->
    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
    <script src="js/bootstrap.bundle.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.8.1/dist/igv.min.js"></script>
    <script src="js/tabulator.min.js" type="text/javascript"></script>

    <!-- CSS imports -->
    <link rel="stylesheet" href="css/bootstrap.css" type="text/css">
    <link rel="stylesheet" href="css/common.css" type="text/css">
    <link rel="stylesheet" href="css/tf_dashboard.css" type="text/css">
    <link rel="stylesheet" href="css/tabulator_bootstrap4.min.css" rel="stylesheet">

    <!-- font -->
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&family=Sulphur+Point:wght@300;400;700&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <!-- CSS -->
    <style>
        * {
            font-family: 'Lato', 'Roboto', 'Helvetica', 'Segoe UI', sans-serif;
        }
    </style>

</head>

<body>
<!-- Page Content -->
<div class="container-fluid page-container">
    <!-- Top Navigation Bar -->
    <nav class="navbar navbar-expand-md sticky-top header">
        <a class="navbar-brand" href="index.html"> ChIPdb 1.0.0 </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav mr-auto">
                <li class="nav-item dropdown">
                    <button class="btn nav-color dropdown-toggle" type="button" id="dropdownMenu2"
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        Organism Dataset
                    </button>
                    <div class="dropdown-menu" aria-labelledby="dropdownMenu2">
                        <a class="dropdown-item" href="dataset_page.html?organism=e_coli"><i>E. coli</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=k_pneumoniae"><i>K. pneumoniae</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=s_aureus"><i>S. aureus</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=s_flexneri"><i>S. flexneri</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=s_yphimurium"><i>S.typhimurium</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=y_pseudotuberculosis"><i>Y.
                            pseudotuberculosis</i></a>
                    </div>
                </li>
            </ul>
            <ul class="navbar-nav">
                <li class="nav-item">
                    <a class="mr-4" href="about.html"><span><i class="material-icons">info</i></span>
                        <span>About</span>
                    </a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="row row-no-margin">
        <!-- Left Navigation -->
        <div id="left_col" class="col-sm-12 col-md-2">
            <div id="left_col_content">
                <h1 id="TF_header"></h1>
                <div class="card p-2" id="meta_text"></div>
            </div>
        </div>
        <!-- page content -->
        <div class="col-sm-10 middle-spacing mt-3">
            <div class="row">
                <div class="col-md-12 col-lg-8">
                    <div class="card">
                        <div class="section-header"> Binding Site Table
                            <button class="btn btn-sm btn-mod" id="download-table">Download CSV</button>
                        </div>
                        <div id="binding_site_table" style="width: 100%; height:300px;"></div>
                    </div>
                </div>
                <div class="col-md-12 col-lg-4">
                    <div class="card">
                        <div class="section-header"> ChIPdb vs. Literature* Regulon Venn</div>
                        <div id="venn" style="width:100%; height:300px;"></div>
                    </div>
                </div>
            </div>
            <div class="card mt-3 pb-2">
                <div class="section-header"> Genome Viewer
                    <a class="btn btn-sm btn-mod" id="download-bw">Download bigWig files</a>
                </div>
                <div id="igv-div"></div>
            </div>
            <!-- Create a div where the graph will take place -->
<div id="my_dataviz"></div>

<p>
  <label># bins</label>
  <input type="number" min="1" max="100" step="1" value="3" id="nBin">
</p>

            <!-- Footer -->
            <footer>
                <hr>
                <div class="container">
                    <div class="row">
                        <div class="col-md-3">
                            <a href="http://systemsbiology.ucsd.edu/">
                                <img class="img-fluid" src="images/sbrg-logo.png">
                            </a>
                        </div>
                        <div class="col-sm-9 m-auto">
                            <p class="copyright text-muted small">
                                ChIPdb is maintained by the Systems Biology Research Group at the University of
                                California, San Diego. <br/>
                            </p>
                        </div>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>

<!-- Scripts -->

<!-- Load Outside Libraries -->
<script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.1.0/papaparse.min.js"></script>
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/venn.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://d3js.org/d3.v4.js"></script>

<!-- Load scripts -->
<script src="js/peak_table.js"></script>
<script src="plot_dev/binding_site_table/binding_site_table.js"></script>
<script src="plot_dev/venn/venn.js"></script>

<!-- Determine which TF -->
<script>
    var params = new URLSearchParams(window.location.search);
    var organism = params.get('organism');
    var TF_name = params.get('tf');
    var index = params.get('i');
    $('#TF_header').append(TF_name);
    document.title = TF_name;

    //trigger download of big wig files
    document.getElementById("download-bw").setAttribute("href",
        "/ChIPdb/data/e_coli/NC_000913_3/bw/compressed/" + TF_name + "_bw.zip");

    $.get('/ChIPdb/data/' + organism + '/TF_list.csv', function (file_data) {
        var data = Papa.parse(file_data, {dynamicTyping: true}).data;
        const row = data[parseInt(index) + 1];

        // TF details
        $('#meta_text').append('<p class="meta_cat">ORGANISM</p>');
        $('#meta_text').append('<p class="meta_val"><i>' + row[2] + '</i></p>');
        $('#meta_text').append('<p class="meta_cat">STRAIN</p>');
        $('#meta_text').append('<p class="meta_val">' + row[3] + '</p>');
        $('#meta_text').append('<p class="meta_cat">MEDIA</p>');
        $('#meta_text').append('<p class="meta_val">' + row[4] + '</p>');
        $('#meta_text').append('<p class="meta_cat">SUPPLEMENT</p>');
        $('#meta_text').append('<p class="meta_val">' + row[5] + '</p>');

        // IGV
        var igvDiv = document.getElementById("igv-div");
        var options =
            {
                reference:
                    {
                        "id": "MG1655",
                        "name": "E. coli MG1655",
                        "fastaURL": "/ChIPdb/data/e_coli/NC_000913_3/sequence/sequence.fasta",
                        "indexURL": "/ChIPdb/data/e_coli/NC_000913_3/sequence/sequence.fasta.fai",
                        "tracks": [
                            {
                                "type": "annotation",
                                "format": "gff",
                                "url": "/ChIPdb/data/e_coli/NC_000913_3/annotation/genes.gff",
                                "name": "Genes",
                                "gffTags": "on",
                                "color": "#5386E4",
                                "nameField": "gene",
                                "searchable": true
                            },
                            {
                                "type": "annotation",
                                "format": "gff",
                                "url": "/ChIPdb/data/e_coli/NC_000913_3/annotation/tfbs.gff",
                                "name": "Published TFBS",
                                "color": "#F5CB5C",
                                "gffTags": "on",
                                "nameField": "TF"
                            },
                            {
                                "type": "annotation",
                                "format": "gff",
                                "url": "/ChIPdb/data/e_coli/NC_000913_3/annotation/tss.gff",
                                "name": "Published TSS",
                                "color": "#A6B5AE",
                                "gffTags": "on",
                                "nameField": "tss_name"
                            },
                            {
                                "type": "annotation",
                                "format": "gff",
                                "url": "/ChIPdb/data/e_coli/NC_000913_3/annotation/tu.gff",
                                "name": "Published TUs",
                                "color": "#A88132",
                                "gffTags": "on",
                                "nameField": "genes"
                            }
                        ]
                    }
            };
        igv.createBrowser(igvDiv, options)
            .then(async function (browser) {
                console.log("Created IGV browser");
                var i = row.length - 11
                while (row[i] != null) {
                    await browser.loadTrack({
                        "name": row[i - 1],
                        "type": "wig",
                        "url": "/ChIPdb/data/e_coli/NC_000913_3/bw/" + row[i]
                    })
                    i += 2
                }
            })

        // Table
        $.getJSON('plot_dev/binding_site_table/data/' + TF_name.substr(0, 4) + '_binding_site_table.json', function (data) {
            // convert rows into objects
            const container = 'binding_site_table'
            var tabledata = []
            for (let i = 0; i < data.length; i++) {
                var obj = data[i];
                tabledata.push(obj) //adds items to an object
            }

            for (let i = 0; i < tabledata.length; i++) {
                const index = tabledata[i].index;
                const bindCond = tabledata[i].condition;
                const bindStart = tabledata[i].binding_peak_start;
                const bindEnd = tabledata[i].binding_peak_end;
                const peakStrength = tabledata[i].binding_peak_strength;
                const genes = tabledata[i].target_genes;
            }

            //make a header object for the columns
            var headerMenu = [
                {
                    label: "<i class='fas fa-arrow-right'></i> Move Column to End",
                    action: function (e, column) {
                        column.move("end");
                    }
                }, {
                    label: "<i class='fas fa-trash'></i> Hide Column",
                    action: function (e, column) {
                        column.hide();
                    }
                }
            ]

            // columns object: basic info
            var columns = [
                {
                    title: "TF-#",
                    field: "index",
                    sorter: "string",
                    headerContextMenu: headerMenu
                },
                {
                    title: " Binding Condition",
                    field: "condition",
                    sorter: "string",
                    hozAlign: "left",
                    formatter: "string",
                    formatterParams: {precision: 4},
                    headerContextMenu: headerMenu
                },
                {
                    title: "Start Position",
                    field: "binding_peak_start",
                    sorter: "number",
                    hozAlign: "left",
                    formatter: "money",
                    formatterParams: {precision: 0},
                    headerContextMenu: headerMenu
                },
                {
                    title: "End Position",
                    field: "binding_peak_end",
                    sorter: "number",
                    hozAlign: "left",
                    formatter: "money",
                    formatterParams: {precision: 0},
                    headerContextMenu: headerMenu
                },
                {
                    title: "Peak Intensity",
                    field: "binding_peak_strength",
                    sorter: "number",
                    hozAlign: "left",
                    formatter: "money",
                    formatterParams: {precision: 2},
                    headerContextMenu: headerMenu
                },
                {
                    title: "Target genes",
                    field: "target_genes",
                    sorter: "string",
                    hozAlign: "left",
                    headerContextMenu: headerMenu
                },
            ]
            console.log(row)
            var table = new Tabulator("#" + container, {
                maxHeight: "100%",
                data: tabledata,
                columns: columns,
                layout: "fitDataFill",
                tooltips: true,
                movableColumns: true,
                rowClick: function (e, clicked_row) { //link to the page in a database
                    var start = clicked_row.getData().binding_peak_start;
                    var end = clicked_row.getData().binding_peak_end;
                    igv.removeAllBrowsers()
                    igv.createBrowser(igvDiv, options)
                        .then(async function (browser) {
                            var i = row.length - 11
                            while (row[i] != null) {
                                await browser.loadTrack({
                                    "name": row[i - 1],
                                    "type": "wig",
                                    "color": "#333533",
                                    "url": "/ChIPdb/data/e_coli/NC_000913_3/bw/" + row[i]
                                })
                                i += 2
                            }
                            browser.search('NC_000913.3:'.concat(numberWithCommas(start - 1000), '-',
                                numberWithCommas(end + 1000)));
                            console.log("Created IGV browser 2");
                        });
                },
                tooltips: function (cell) {
                    return "Click to jump to peak in genome viewer below";
                }
            });
            table.redraw();

            //trigger download of binding site table .csv file
            document.getElementById("download-table").addEventListener("click", function () {
                table.download("csv", TF_name + "_binding_site_table.csv");
            });


        });

        // Venn
        $.getJSON('plot_dev/venn/data/' + TF_name.substr(0, 4) + '.json', function (data) {
            generateVenn(data, 'venn');
        });

        // Binding width histogram


// set the dimensions and margins of the graph
        var margin = {top: 10, right: 30, bottom: 30, left: 40},
            width = 460 - margin.left - margin.right,
            height = 400 - margin.top - margin.bottom;

// append the svg object to the body of the page
        var svg = d3.select("#my_dataviz")
            .append("svg")
            .attr("width", width + margin.left + margin.right)
            .attr("height", height + margin.top + margin.bottom)
            .append("g")
            .attr("transform",
                "translate(" + margin.left + "," + margin.top + ")");

// get the data
        d3.csv("/ChIPdb/data/e_coli/NC_000913_3/binding_widths/"+TF_name+"_widths.csv", function (data) {
            console.log("data", data)
            // X axis: scale and draw:
            const min = d3.min(data, function(d) { return +d.binding_width});
            const max = d3.max(data, function(d) {return +d.binding_width});
            var x = d3.scaleLinear()
                .domain([min, max])
                .range([0, width]);
            svg.append("g")
                .attr("transform", "translate(0," + height + ")")
                .call(d3.axisBottom(x));

            // Y axis: initialization
            var y = d3.scaleLinear()
                .range([height, 0]);
            var yAxis = svg.append("g")

            // A function that builds the graph for a specific value of bin
            function update(nBin) {

                // thresholds
                const thresholds = d3.range(min, max, (max - min) / nBin);

                // set the parameters for the histogram
                var histogram = d3.histogram()
                    .value(function (d) {
                        return d.binding_width;
                    })   // I need to give the vector of value
                    .domain(x.domain())  // then the domain of the graphic
                    .thresholds(thresholds); // then the numbers of bins

                // And apply this function to data to get the bins
                var bins = histogram(data);

                // Y axis: update now that we know the domain
                ymax = d3.max(bins, function (d) {
                    return d.length;
                });
                y.domain([0, ymax]);   // d3.hist has to be called before the Y axis obviously
                yAxis
                    .transition()
                    .duration(1000)
                    .call(d3.axisLeft(y).tickFormat(d3.format("d")).tickValues(d3.range(0, ymax+1, Math.max(1,parseInt(ymax/5)))));

                // Join the rect with the bins data
                var u = svg.selectAll("rect")
                    .data(bins)

                // Manage the existing bars and eventually the new ones:
                u
                    .enter()
                    .append("rect") // Add a new rect for each new elements
                    .merge(u) // get the already existing elements as well
                    .transition() // and apply changes to all of them
                    .duration(1000)
                    .attr("x", 1)
                    .attr("transform", function (d) {
                        return "translate(" + x(d.x0) + "," + y(d.length) + ")";
                    })
                    .attr("width", function (d) {
                        console.log("x1", d.x1)
                        console.log("x0", d.x0)
                        console.log("width", width)
                        return x(d.x1) - x(d.x0) - 1;
                    })
                    .attr("height", function (d) {
                        return height - y(d.length);
                    })
                    .style("fill", "#69b3a2")


                // If less bar in the new histogram, I delete the ones not in use anymore
                u
                    .exit()
                    .remove()

            }

            // Initialize with 5 bins
            update(3)

            // Listen to the button -> update if user change it
            d3.select("#nBin").on("input", function () {
                update(+this.value);
            });

        });


        // Formatting commas into numbers
        function numberWithCommas(x) {
            return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
        }
    });

</script>
</body>

</html>
