<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>TF Dashboard</title>

    <!-- javascript imports -->
    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
    <script src="js/bootstrap.bundle.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.8.1/dist/igv.min.js"></script>
    <script src="js/tabulator.min.js" type="text/javascript"></script>

    <!-- CSS imports -->
    <link rel="stylesheet" href="css/bootstrap.css" type="text/css">
    <link rel="stylesheet" href="css/common.css" type="text/css">
    <link rel="stylesheet" href="css/tf_dashboard.css" type="text/css">
    <link rel="stylesheet" href="css/tabulator_bootstrap4.min.css" rel="stylesheet">

    <!-- font -->
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&family=Sulphur+Point:wght@300;400;700&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <!-- CSS -->
    <style>
        * {
            font-family: 'Lato', 'Roboto', 'Helvetica', 'Segoe UI', sans-serif;
        }
    </style>

</head>

<body>
<!-- Page Content -->
<div class="container-fluid page-container">
    <!-- Top Navigation Bar -->
    <nav class="navbar navbar-expand-md sticky-top header">
        <a class="navbar-brand" href="index.html">
            <img src="images/ChIPdb_logo.png" height=35 class="d-inline-block align-top"
                 alt="" loading="lazy">
            proChIPdb 1.0.0 </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav mr-auto">
                <li class="nav-item dropdown">
                    <button class="btn nav-color dropdown-toggle" type="button" id="dropdownMenu2"
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        Organism Dataset
                    </button>
                    <div class="dropdown-menu" aria-labelledby="dropdownMenu2">
                        <a class="dropdown-item" href="dataset_page.html?organism=e_coli"><i>E. coli</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=all_other">All Other Organisms</a>
                    </div>
                </li>
            </ul>
            <ul class="navbar-nav">
                <li class="nav-item">
                    <a class="mr-4" href="about.html"><span><i class="material-icons">info</i></span>
                        <span>About</span>
                    </a>
                </li>
                <li class="nav-item">
                    <a class="mr-4" href="search.html"><span><i class="material-icons">search</i></span>
                        <span>Search</span>
                    </a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="row row-no-margin">
        <!-- Left Navigation -->
        <div id="left_col" class="col-sm-12 col-md-2">
            <div id="left_col_content">
                <div class="row row-no-margin">
                    <div class="col-10>"><h1 class="bd-lead" id="TF_header"></h1></div>
                    <div class="col-2">
                        <button class="panel-info-btn" id="table-info" data-toggle="popover" type="button">
                            <i class="material-icons meta-icons">help</i>
                        </button>
                    </div>
                </div>
                <div class="card p-2" id="meta_text"></div>
            </div>
        </div>
        <!-- page content -->
        <div class="col-sm-10 middle-spacing mt-3">
            <div class="row">
                <div class="col-md-12 col-lg-8">
                    <div class="card">
                        <div class="section-header"> Binding Site Table
                            <button class="panel-info-btn" id="table-info" data-toggle="popover" type="button">
                                <i class="material-icons meta-icons">help</i>
                            </button>
                            <button class="btn btn-sm btn-mod" id="download-table">Download CSV</button>
                        </div>
                        <ul class="nav nav-tabs tab-strip" id="TableTab" role="tablist"></ul>
                        <div id="binding_site_table" style="width: 100%; height:300px;"></div>
                    </div>
                </div>
                <div class="col-md-12 col-lg-4">
                    <div class="card">
                        <div class="section-header"> Feature Visualization
                            <button class="panel-info-btn" id="visual-info" type="button">
                                <i class="material-icons meta-icons">help</i>
                            </button>
                        </div>

                        <ul class="nav nav-tabs tab-strip" id="myTab" role="tablist">

                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab active" id="header-tab-1" data-bs-toggle="tab"
                                        data-bs-target="#tab-1" data-toggle="tab" data-target="#tab-1"
                                        type="button" role="tab" aria-controls="tab-1"
                                        aria-selected="true">Width
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab" id="header-tab-4" data-bs-toggle="tab"
                                        data-bs-target="#tab-4" data-toggle="tab" data-target="#tab-4"
                                        type="button" role="tab" aria-controls="tab-4" aria-selected="false">Position
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab disabled" id="header-tab-3" data-bs-toggle="tab"
                                        data-bs-target="#tab-3" data-toggle="tab" data-target="#tab-3"
                                        type="button" role="tab" aria-controls="tab-3" aria-selected="false">Motif
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab" id="header-tab-2" data-bs-toggle="tab"
                                        data-bs-target="#tab-2" data-toggle="tab" data-target="#tab-2"
                                        type="button" role="tab" aria-controls="tab-2" aria-selected="false">Venn
                                </button>
                            </li>
                        </ul>
                        <!-- Binding width histogram-->
                        <div class="tab-content" id="myTabContent">
                            <div class="tab-pane fade show active" id="tab-1" role="tabpanel"
                                 aria-labelledby="hist-tab">
                                <div id="widthHist" style="width:100%; height:300px;"></div>
                            </div>

                            <!--Venn Diagram-->
                            <div class="tab-pane fade" id="tab-2" role="tabpanel" aria-labelledby="venn-tab">
                                <div id="venn" style="width:100%; height:300px;"></div>
                            </div>

                            <!-- Motif Sequence-->
                            <div class="tab-pane fade" id="tab-3" role="tabpanel" aria-labelledby="contact-tab">

                                <!--Hamburger Dropdown-->
                                <div class="dropdown menu-right">
                                    <!--Trigger-->
                                    <button class="btn menu-hamburger-height" type="button" id="dropdownMenuMotif"
                                            data-toggle="dropdown"
                                            aria-haspopup="true" aria-expanded="false">
                                        <i class="material-icons">menu</i>
                                    </button>
                                    <!--Menu-->
                                    <div class="dropdown-menu dropdown-primary">
                                        <a id="motif-download-png" class="dropdown-item" download>Download PNG image</a>
                                        <a id="motif-download-svg" class="dropdown-item" download>Download SVG vector
                                            image</a>
                                        <a id="motif-download-csv" class="dropdown-item" download>Download PWM CSV</a>
                                    </div>
                                </div>
                                <!--/Hamburger Dropdown-->

                                <div style="height:300px" class="fill">
                                    <img style="max-width:95%; max-height:95%" id="motif_sequence" src=""/>
                                </div>
                                <div id="motif_selector_buttons"></div>
                            </div>

                            <!-- Position Scatterplot-->
                            <div class="tab-pane fade" id="tab-4" role="tabpanel" aria-labelledby="hist-tab">
                                <div id="position_scatter" style="width:100%; height:300px;"></div>
                            </div>
                        </div>

                    </div>
                </div>
            </div>
            <div class="card mt-3 pb-2">
                <div class="section-header"> Genome Viewer
                    <button class="panel-info-btn" id="igv-info" type="button">
                        <i class="material-icons meta-icons">help</i>
                    </button>
                    <a class="btn btn-sm btn-mod" id="download-bw">Download bigWig files</a>
                </div>
                <div id="igv-div"></div>
            </div>

            <!-- Footer -->
            <footer>
                <hr>
                <div class="container">
                    <div class="row">
                        <div class="col-md-3">
                            <a href="http://systemsbiology.ucsd.edu/">
                                <img class="img-fluid" src="images/sbrg-logo.png">
                            </a>
                        </div>
                        <div class="col-sm-9 m-auto">
                            <p class="copyright text-muted small">
                                proChIPdb is maintained by the Systems Biology Research Group at the University of
                                California, San Diego. <br/>
                            </p>
                        </div>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>

<!-- Scripts -->

<!-- Load Outside Libraries -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.1.0/papaparse.min.js"></script>
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/venn.js"></script>
<script src="https://code.highcharts.com/modules/histogram-bellcurve.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
<script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>

<!-- Load scripts -->
<script src="js/igv_and_table.js"></script>
<script src="js/peak_gene_positions.js"></script>
<script src="js/width_histogram.js"></script>
<script src="js/venn2.js"></script>

<!-- Determine which TF -->
<script>
    var params = new URLSearchParams(window.location.search);
    var organism = params.get('organism');
    var TF_name = params.get('tf');
    var genome = params.get('genome');
    var index = params.get('i');

    $('#TF_header').append(TF_name);

    document.title = TF_name;


    $.get('/data/' + organism + '/TF_list.csv', function (file_data) {
        var data = Papa.parse(file_data, {dynamicTyping: true}).data;
        const row = data[parseInt(index) + 1];

        // TF details
        $('#meta_text').append('<p class="meta_cat">ORGANISM</p>');
        $('#meta_text').append('<p class="meta_val"><i>' + row[2] + '</i></p>');
        $('#meta_text').append('<p class="meta_cat">STRAIN</p>');
        $('#meta_text').append('<p class="meta_val">' + row[3] + '</p>');
        $('#meta_text').append('<p class="meta_cat">GENOME ID</p>');
        $('#meta_text').append('<p class="meta_val">' + row[6] + '</p>');
        $('#meta_text').append('<p class="meta_cat">MEDIA</p>');
        $('#meta_text').append('<p class="meta_val">' + row[20] + '<button class="meta-info-btn" id="media-info" type="button" ><i class="material-icons meta-icons">info</i></button></p>');
        $('#meta_text').append('<p class="meta_cat">SUPPLEMENT</p>');
        $('#meta_text').append('<p class="meta_val">' + row[21] + '</p>');


        $('#media-info').popover({
            container: "body",
            title: "additional media information",
            html: true,
            trigger: "hover focus",
            content: function () {
                return `<div class="popover-message">${row[22]}</div>`;
            }
        });


        // out links
        if (organism == 'e_coli') {
            var link_link = '/data/e_coli/outlinks/' + TF_name + '_links.csv';
            var db_num = 4
        } else {
            var link_link = '/data/all_other/outlinks/' + organism + '_' + TF_name + '_links.csv';
            var db_num = 4 //may change as addl data is added
        }
        $.get(link_link, function (link_data) {
            link_table = Papa.parse(link_data, {dynamicTyping: true, skipEmptyLines: true}).data;
            console.log(link_table)

            var link_text = '<p class="meta_cat">LINKS</p>';
            var include_links = false;
            for (let i = 0; i < db_num; i++) {
                if (link_table[i][1]) {

                    link_text += '<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>';
                    include_links = true;
                }
            }
            if (include_links) {
                $('#meta_text').append(link_text);
            }
            if (link_table.length > db_num) {
                $('#meta_text').append('<p class="meta_cat mt-3">iMODULONS</p>');
                for (let i = db_num; i < link_table.length; i++) {
                    $('#meta_text').append('<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>');
                }
            }

        })
        $('#meta_text').append('<p class="meta_cat">RAW DATA ACCESSION</p>');
        $('#meta_text').append('<p class="meta_val">' + row[14] + '</p>');
        if (row[15]) {
            $('#meta_text').append('<p class="meta_cat">DOI REFERENCE</p>');
            $('#meta_text').append('<p class="meta_val"><a href="' + row[15] + '">' + row[15] + '</a></p>');
        }
        if (row[26]) {
            $('#meta_text').append('<p class="meta_cat">EXTRA NOTES</p>');
            $('#meta_text').append('<p class="meta_val">' + row[26] + '</p>');
        }

        // IGV and table
        const [title, fileName] = row[12].split(";")[0].split("|");
        generateIGVandTable(TF_name, row, fileName);
        // First, get row[12] to get list of filenames to create the tabs for x number of files
        const conditions = row[12].split(";");
        // parse the table row of Title A|xyz.json;Title B|def.json;...
        // access the tabs parent component by ID
        $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab active" id="table-tab-1" data-bs-toggle="tab"
                    data-bs-target="#table-tab-1" data-toggle="tab" data-target="#table-tab-1"
                    type="button" role="tab" aria-controls="table-tab-1"
                    aria-selected="true">${title}
            </button>
        </li>
        `);
        $('#table-tab-1').click(() => {
            generateIGVandTable(TF_name, row, fileName);
        });
        // loop through the loaded values of `Title A|xyz.json` and create a tab for each
        conditions.slice(1, conditions.length).forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab" id="table-tab-${i + 2}" data-bs-toggle="tab"
                    data-bs-target="#table-tab-${i + 2}" data-toggle="tab" data-target="#table-tab-${i + 2}"
                    type="button" role="tab" aria-controls="table-tab-${i + 2}"
                    aria-selected="true">${conditionTitle}
            </button>
        </li>
        `);
            $(`#table-tab-${i + 2}`).click(() => {
                generateIGVandTable(TF_name, row, conditionFile);
            });
        });

        // Loop through it all again and create buttons to choose the motif graphic
        let hasFoundFirstImageToRender = false;
        const clearAllActiveMotifButtonFunctions = [];

        conditions.forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            const motifImageAndCSVBase = conditionFile.replace("_binding_table.json", "");
            const imageFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.png`;
            const csvFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.csv`;
            const spacelessConditionTitle = conditionTitle.replace(/ /g, "_");

            $('#motif_selector_buttons').append(`
                <button class="btn btn-primary btn-sm" id="motif-button-${spacelessConditionTitle}">${conditionTitle}</button>
            `);

            $.get(imageFilePath, function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = imageFilePath;
                    document.getElementById("motif-download-svg").classList.add('d-none');
                    document.getElementById("motif-download-png").href = imageFilePath;
                    document.getElementById("motif-download-csv").href = csvFilePath;
                    document.getElementById("motif-download-csv").classList.remove('d-none');

                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.add('active');
                };

                clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.remove('active'));
                $(`#motif-button-${spacelessConditionTitle}`).click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }
            }).catch(() => {
                //     $('#motif_selector_buttons').append(`
                //     <button disabled title="No motif logo found using MEME-ChIP">${conditionTitle}</button>
                // `);
                $(`#motif-button-${spacelessConditionTitle}`).addClass('disabled gray-btn');
                // $(`#motif-button-${spacelessConditionTitle}`).addClass('gray-btn');
                $(`#motif-button-${spacelessConditionTitle}`).attr('disabled', true);
                $(`#motif-button-${spacelessConditionTitle}`).attr("title", "No motif logo found using MEME-ChIP");
            });
        });

        // Add literature motif information, if available
        const motifSVG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.svg';
        const motifPNG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.png';

        $('#motif_selector_buttons').append(`
            <button class="btn btn-primary btn-sm" id="motif-button-lit">Literature</button>
        `);
        clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-lit`).classList.remove('active'));


        $.get(motifSVG)
            .done(function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = motifSVG;
                    document.getElementById("motif-download-svg").href = motifSVG;
                    document.getElementById("motif-download-svg").classList.remove('d-none');
                    document.getElementById("motif-download-png").href = motifPNG;
                    document.getElementById("motif-download-csv").classList.add('d-none');


                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-lit`).classList.add('active');
                };

                $('#motif-button-lit').click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }

                // document.getElementById("motif_sequence").src = motifSVG;
                // document.getElementById("motif-download-svg").href = motifSVG;
            }).fail(function () {
            $.get(motifPNG)
                .done(function () {
                    // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                    document.getElementById("header-tab-3").classList.remove('disabled');

                    const changeImageHandler = function () {
                        document.getElementById("motif_sequence").src = motifPNG;
                        document.getElementById("motif-download-svg").classList.add('d-none');
                        document.getElementById("motif-download-png").href = motifPNG;
                        document.getElementById("motif-download-csv").classList.add('d-none');


                        clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                        document.getElementById(`motif-button-lit`).classList.add('active');
                    };

                    $('#motif-button-lit').click(changeImageHandler);

                    if (!hasFoundFirstImageToRender) {
                        changeImageHandler();
                        hasFoundFirstImageToRender = true;
                    }
                    // document.getElementById("motif_sequence").src = motifPNG;
                    // document.getElementById("motif-download-svg").remove();
                }).fail(function () {
                // document.getElementById("header-tab-3").classList.add('disabled')
                document.getElementById("motif-button-lit").classList.add('disabled')
            })
        });

        // if (!hasFoundFirstImageToRender) {
        //     document.getElementById("header-tab-3").classList.add('disabled');
        // }

        // deal with disabling header tab here

        // Venn
        $.get('/data/' + organism + '/' + genome + '/venn/' + TF_name.substr(0, 4) + '_venn.csv', function (data) {
            generateVenn2(data, 'venn');
        }).catch(() => {
            document.getElementById("header-tab-2").classList.add('disabled')
        });

        // Binding width histogram
        $.get('/data/' + organism + '/' + genome + '/binding_widths/' + TF_name + "_widths.csv", function (width_data) {
            generateWidthHistogram(width_data, 'widthHist')
        }).catch(() => {
            document.getElementById("widthHist").innerText = 'Binding width histogram not available.'
            document.getElementById("widthHist").classList.add('NA_text')
        });

        // Peak/Gene Position Scatter
        $.get('/data/' + organism + '/' + genome + '/positions/' + TF_name + "_positions.csv", function (positions_data) {
            generatePeakScatter(positions_data, row, "position_scatter");
        }).catch(() => {
            document.getElementById("header-tab-4").classList.add('disabled')
        });

    });

    // populate info buttons
    $('#table-info').popover({
        title: "Help - Binding Site Table",
        html: true,
        trigger: "hover focus",
        content: function () {
            return `<div class="popover-message"><p>This table represents <a href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">MACE</a> results from processing the sequence read data presented in the Genome Viewer. Tabs across the top correspond to each binding condition. Each row represents a curated binding peak. Clicking on a row will update the Genome Viewer to display a zoomed in plot of the corresponding peak. The rows are initially sorted by genome location, and the tallest peaks can easily be found by sorting by descending "Peak Intensity" (clicking the column header twice). The columns are as follows:</p>
            <ul>
                <li><b>TF-#</b>: A unique identifier for the peak, ordered by genome location.</li>
                <li><b>Binding Condtion</b>: Takes the form “TF + condition.”</li>
                <li><b>Start Position & End Position</b>: The genome locations of the peak in base pairs are determined by MACE. For each peak, a single-nucleotide resolution border peak is detected by using the Chebyshev Inequality and the border is matched by using the Gale–Shapley stable matching algorithm. For E. coli K-12 MG1655, these locations refer to the <a href="https://www.ncbi.nlm.nih.gov/nuccore/556503834" target="_blank">NC_000913.3 genome</a>.</li>
                <li><b>Peak Intensity</b>: Peak signal intensity is determined with two replicates using MACE peak calling algorithm, which employs Shannon’s relative entropy (H)to compute signal to noise values across the genome for each replicate. Note that the noise value varies for each condition because it captures variation due to inexact cross-linking, exonuclease digestion, dynamic protein conformations, and PCR amplification.</li>
                <li><b>Target genes</b>: Each peak is assigned to the target gene, according to genomic positions.</li>
            </ul></div>`;
        }
    });

    $('#visual-info').popover({
        container: "body",
        title: "Help - Feature Visualization",
        html: true,
        trigger: "hover focus",
        content: function () {
            return `<div class="popover-message">
            <p>
                The tabs within this panel provide additional characterization of the data. In the upper right hand corner of each tab, the menu button (<i class="material-icons">menu</i>) enables PNG, SVG, and data download.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Width
            </h5>
            <p>
                This tab shows a histogram of the binding peak widths from the binding site table. Hover over each bar to see a count of peaks that fall within its corresponding bin.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Position
            </h5>
            <p>
                In this tab, proChIPdb compares the binding locations relative to each target gene. For each gene (as listed in the Target Genes column of the Binding Site Table), the distance from the gene start site to the binding site is measured in base pairs and normalized to the length of the gene. Points are then plotted with this value on the x axis and the peak intensity (S/N) on the y axis. Genes from the same transcriptional unit will have identical y axis values. Clustering on the x axis indicates the distance at which the transcription factor usually exerts its influence on gene expression. Hover over a point to view more details, such as the gene name.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Motif
            </h5>
            <p>
                If this tab exists, then it contains a sequence logo of a significantly enriched motif for these binding sites and its corresponding E-value. The motif and logo are generated by <a href="https://meme-suite.org/meme/index.html" target="_blank">the MEME Suite</a> (parameters: width = 20bp, E-value cutoff = 0.001). The dot with arrows represents the center of the motif, which assists in identifying palindromic sequences.
            </p>

            <h5 class="text-center py-3">
                Feature Visualization – Venn
            </h5>
            <p>
                The venn diagram compares the target genes obtained in proChIPdb to other target gene sets in the literature. Hover over a section of the venn diagram to see the genes it contains. Areas of agreement indicate strong evidence of direct regulation, and areas of disagreement represent opportunities to improve literature annotation or elucidate condition-specific differences in binding. The specific literature sources are mentioned below each diagram. For <i>E. coli</i>, <a href="https://ecocyc.org/" target="_blank">EcoCyc</a>’s transcriptional regulatory network was used.
            </p></div>`;
        }
    });

    $('#igv-info').popover({
        title: "Help - Genome Viewer",
        html: true,
        trigger: "hover focus",
        content: function () {
            return `<div class="popover-message">
            <p>
                proChIPdb’s genome viewer provides access to a complete, genome-wide view of the transcription factor’s activity under the given conditions. The genome viewer was made using <a href="https://doi.org/10.1101/2020.05.03.075499" target="_blank">igv.js</a>, and it visualizes bigWig ChIP read files generated from bam files using <a href="https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html" target="_blank">deeptools</a>. From top to bottom, the features and tracks of the viewer are:
            </p>
            <ul>
                <li><b>Toolbar</b>: The toolbar contains a search bar, which can be used to search for specific loci by gene location or gene name. It also has useful tools for adding cursor guides, a center line, toggling track labels, downloading the current view, and adjusting the zoom level.</li>
                <li><b>Genome Location</b>: The white bar with red outlines may help orient you to which region of the genome is currently displayed. Underneath that, an axis is labeled with specific base pair numbers.</li>
                <li><b>Nucleotides</b>: If the viewer is zoomed close enough, color-coded nucleotide labels will appear.</li>
                <li><b>Genes</b>: Gene annotations as provided in the genome. Click on a track to see additional details, such as the gene name, locus tag, and product description.</li>
                <li><b>Published TUs</b>: Transcriptional unit annotations, as provided by <a href="https://academic.oup.com/nar/article/48/18/10157/5911745" target="_blank">the Bitome</a>.</li>
                <li><b>Published TSS</b>: Transcription start sites, as provided by the Bitome.</li>
                <li><b>Published TFBS</b>: Transcription factor binding sites, as provided by the Bitome.</li>
                <li><b>Data Tracks</b>: The y-axis of these plots is the number of ChIP reads mapped to the x-axis nucleotide. Replicate numbers and unique conditions (if applicable) are indicated by the track name. To download this data for custom analysis, click “Download bigWig Files” in the upper left corner of the pane.</li>
            </ul></div>`;
        }
    });

</script>
</body>

</html>
