<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>TF Dashboard</title>

    <!-- javascript imports -->
    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
    <script src="js/bootstrap.bundle.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.8.1/dist/igv.min.js"></script>
    <script src="js/tabulator.min.js" type="text/javascript"></script>
    <script src="js/intro.min.js"></script>

    <!-- CSS imports -->
    <link rel="stylesheet" href="css/bootstrap.css" type="text/css">
    <link rel="stylesheet" href="css/common.css" type="text/css">
    <link rel="stylesheet" href="css/tf_dashboard.css" type="text/css">
    <link rel="stylesheet" href="css/tabulator_bootstrap4.min.css" rel="stylesheet">
    <link rel="stylesheet" href="css/introjs.min.css" type="text/css">

    <!-- font -->
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&family=Sulphur+Point:wght@300;400;700&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <!-- CSS -->
    <style>
        * {
            font-family: 'Lato', 'Roboto', 'Helvetica', 'Segoe UI', sans-serif;
        }
    </style>

</head>

<body>
<!-- Page Content -->
<div class="container-fluid page-container">
    <!-- Top Navigation Bar -->
    <nav class="navbar navbar-expand-md sticky-top header">
        <a class="navbar-brand" href="index.html" data-step="5" data-title="Thanks for learning about us üëã"
        data-intro="Thanks for checking out our interactive tour! Let's take you back to the homepage. Remember, you can always get more details by visiting out 'About' page linked in the top navigation bar.">
            <img src="images/ChIPdb_logo.png" height=35 class="d-inline-block align-top"
                 alt="" loading="lazy">
            proChIPdb 1.0.0 </a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav mr-auto">
                <li class="nav-item dropdown">
                    <button class="btn nav-color dropdown-toggle" type="button" id="dropdownMenu2"
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        Organism Dataset
                    </button>
                    <div class="dropdown-menu" aria-labelledby="dropdownMenu2">
                        <a class="dropdown-item" href="dataset_page.html?organism=e_coli"><i>E. coli</i></a>
                        <a class="dropdown-item" href="dataset_page.html?organism=all_other">All Other Organisms</a>
                    </div>
                </li>
            </ul>
            <ul class="navbar-nav">
                <li class="nav-item">
                    <a class="mr-4" href="about.html"><span><i class="material-icons">info</i></span>
                        <span>About</span>
                    </a>
                </li>
                <li class="nav-item">
                    <a class="mr-4" href="search.html"><span><i class="material-icons">search</i></span>
                        <span>Search</span>
                    </a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="row row-no-margin">
        <!-- Left Navigation -->
        <div id="left_col" class="col-sm-12 col-md-2"
        data-step="1" data-title="TF dashboard metadata"
        data-intro="In this left panel, various details on the TF metadata, including information from the data source publication and useful links, will be available for your reference.">
            <div id="left_col_content" class="mb-3">
                <h1 class="bd-lead" id="TF_header"></h1>
                <div class="card p-2" id="meta_text"></div>
            </div>
        </div>
        <!-- page content -->
        <div class="col-sm-10 middle-spacing mt-3">
            <div class="row">
                <div class="col-md-12 col-lg-8">
                    <div class="card" data-step="2" data-title="Binding Site Table"
                         data-intro="The binding site table displays different transcription factor binding sites (TFBSs) that were identified from the experimental source data. Each tab represents the results from a different experimental condition.">
                        <div class="section-header"> Binding Site Table
                            <button class="panel-info-btn" id="table-info" type="button">
                                <i class="material-icons meta-icons">help</i>
                            </button>
                            <button class="btn btn-sm btn-mod" id="download-table">Download CSV</button>
                        </div>
                        <ul class="nav nav-tabs tab-strip" id="TableTab" role="tablist"></ul>
                        <div id="binding_site_table" style="width: 100%; height:300px;"></div>
                    </div>
                </div>
                <div class="col-md-12 col-lg-4">
                    <div class="card" data-step="3" data-title="Feature Visualization"
                    data-intro="This panel includes different visuals to summarize the TFBSs (where available), including: 1) a histogram of binding widths, 2) a scatterplot of peak positions relative to their respective closest gene, 3) the generated binding motif (either re-processed with MEME-ChIP or from the source publication), and 4) the Venn diagram comparison between the proChIPdb identified target genes and published regulon genes. Downloads are available in the hamburger menus (‚â°) in the top right corners of each tab.">
                        <div class="section-header"> Feature Visualization
                            <button class="panel-info-btn" id="visual-info" type="button">
                                <i class="material-icons meta-icons">help</i>
                            </button>
                        </div>

                        <ul class="nav nav-tabs tab-strip" id="myTab" role="tablist">

                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab active" id="header-tab-1" data-bs-toggle="tab"
                                        data-bs-target="#tab-1" data-toggle="tab" data-target="#tab-1"
                                        type="button" role="tab" aria-controls="tab-1"
                                        aria-selected="true">Width
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab" id="header-tab-4" data-bs-toggle="tab"
                                        data-bs-target="#tab-4" data-toggle="tab" data-target="#tab-4"
                                        type="button" role="tab" aria-controls="tab-4" aria-selected="false">Position
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab disabled" id="header-tab-3" data-bs-toggle="tab"
                                        data-bs-target="#tab-3" data-toggle="tab" data-target="#tab-3"
                                        type="button" role="tab" aria-controls="tab-3" aria-selected="false">Motif
                                </button>
                            </li>
                            <li class="nav-item" role="presentation">
                                <button class="nav-link round-tab" id="header-tab-2" data-bs-toggle="tab"
                                        data-bs-target="#tab-2" data-toggle="tab" data-target="#tab-2"
                                        type="button" role="tab" aria-controls="tab-2" aria-selected="false">Venn
                                </button>
                            </li>
                        </ul>
                        <!-- Binding width histogram-->
                        <div class="tab-content" id="myTabContent">
                            <div class="tab-pane fade show active" id="tab-1" role="tabpanel"
                                 aria-labelledby="hist-tab">
                                <div id="widthHist" style="width:100%; height:300px;"></div>
                            </div>

                            <!--Venn Diagram-->
                            <div class="tab-pane fade" id="tab-2" role="tabpanel" aria-labelledby="venn-tab">
                                <div id="venn" style="width:100%; height:300px;"></div>
                            </div>

                            <!-- Motif Sequence-->
                            <div class="tab-pane fade" id="tab-3" role="tabpanel" aria-labelledby="contact-tab">
                                <!--Hamburger Dropdown-->
                                <div class="dropdown menu-right">
                                    <!--Trigger-->
                                    <button class="btn menu-hamburger-height" type="button" id="dropdownMenuMotif"
                                            data-toggle="dropdown"
                                            aria-haspopup="true" aria-expanded="false">
                                        <i class="material-icons">menu</i>
                                    </button>
                                    <!--Menu-->
                                    <div class="dropdown-menu dropdown-primary">
                                        <a id="motif-download-png" class="dropdown-item" download>Download PNG image</a>
                                        <a id="motif-download-svg" class="dropdown-item" download>Download SVG vector
                                            image</a>
                                        <a id="motif-download-csv" class="dropdown-item" download>Download PWM CSV</a>
                                    </div>
                                </div>
                                <div id="motif-banner-title" class="motif-banner-title"></div>

                                <!--Motif Logo and Selector Buttons-->
                                <div class="fill-parent">
                                    <div class="fill">
                                        <img style="max-width:95%; max-height:95%" id="motif_sequence" src="">
                                    </div>
                                    <div class="motif-selector-buttons" id="motif_selector_buttons"></div>
                                </div>

                            </div>

                            <!-- Position Scatterplot-->
                            <div class="tab-pane fade" id="tab-4" role="tabpanel" aria-labelledby="hist-tab">
                                <div id="position_scatter" style="width:100%; height:300px;"></div>
                            </div>
                        </div>

                    </div>
                </div>
            </div>
            <div class="card mt-3 pb-2">
                <div class="section-header" data-step="4" data-title="Genome Viewer"
                     data-intro="proChIPdb‚Äôs genome viewer (below) provides access to a complete, genome-wide view of the transcription factor‚Äôs activity under the given conditions. The genome viewer was made using igv.js, and it visualizes bigWig ChIP read files. The genome viewer also interacts with the Binding Site Table panel, and will zoom in on a peak when it is clicked in the table.">
                    Genome Viewer
                    <button class="panel-info-btn" id="igv-info" type="button">
                        <i class="material-icons meta-icons">help</i>
                    </button>
                    <a class="btn btn-sm btn-mod" id="download-bw">Download bigWig files</a>
                </div>
                <div id="igv-div"></div>
            </div>

            <!-- Footer -->
            <footer>
                <hr>
                <div class="container">
                    <div class="row">
                        <div class="col-md-3">
                            <a href="http://systemsbiology.ucsd.edu/">
                                <img class="img-fluid" src="images/sbrg-logo.png">
                            </a>
                        </div>
                        <div class="col-sm-9 m-auto">
                            <p class="copyright text-muted small">
                                proChIPdb is maintained by the Systems Biology Research Group at the University of
                                California, San Diego. <br/>
                            </p>
                        </div>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>

<!-- Scripts -->

<!-- Load Outside Libraries -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.1.0/papaparse.min.js"></script>
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/venn.js"></script>
<script src="https://code.highcharts.com/modules/histogram-bellcurve.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
<script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>

<!-- Load scripts -->
<script src="js/igv_and_table.js"></script>
<script src="js/peak_gene_positions.js"></script>
<script src="js/width_histogram.js"></script>
<script src="js/venn2.js"></script>

<!-- Determine which TF -->
<script>
    var params = new URLSearchParams(window.location.search);
    var organism = params.get('organism');
    var TF_name = params.get('tf');
    var genome = params.get('genome');
    var index = params.get('i');

    $('#TF_header').append(TF_name);

    document.title = TF_name;


    $.get('/data/' + organism + '/TF_list.csv', function (file_data) {
        var data = Papa.parse(file_data, {dynamicTyping: true}).data;
        const row = data[parseInt(index) + 1];

        // TF details
        $('#meta_text').append('<p class="meta_cat">ORGANISM</p>');
        $('#meta_text').append('<p class="meta_val"><i>' + row[2] + '</i></p>');
        $('#meta_text').append('<p class="meta_cat">STRAIN</p>');
        $('#meta_text').append('<p class="meta_val">' + row[3] + '</p>');
        $('#meta_text').append('<p class="meta_cat">GENOME ID</p>');
        $('#meta_text').append('<p class="meta_val">' + row[6] + '</p>');
        $('#meta_text').append('<p class="meta_cat">MEDIA</p>');
        $('#meta_text').append('<p class="meta_val">' + row[20] + '<button class="meta-info-btn" id="media-info" type="button" ><i class="material-icons meta-icons">info</i></button></p>');
        $('#meta_text').append('<p class="meta_cat">SUPPLEMENT</p>');
        $('#meta_text').append('<p class="meta_val">' + row[21] + '</p>');


        $('#media-info').popover({
            container: "body",
            title: "additional media information",
            html: true,
            trigger: "hover focus",
            content: function () {
                return `<div class="popover-message">${row[22]}</div>`;
            }
        });


        // out links
        if (organism == 'e_coli') {
            var link_link = '/data/e_coli/outlinks/' + TF_name + '_links.csv';
            var db_num = 4
        } else {
            var link_link = '/data/all_other/outlinks/' + organism + '_' + TF_name + '_links.csv';
            var db_num = 4 //may change as addl data is added
        }
        $.get(link_link, function (link_data) {
            link_table = Papa.parse(link_data, {dynamicTyping: true, skipEmptyLines: true}).data;
            
            var link_text = '<p class="meta_cat">LINKS</p>';
            var include_links = false;
            for (let i = 0; i < db_num; i++) {
                if (link_table[i][1]) {

                    link_text += '<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>';
                    include_links = true;
                }
            }
            if (include_links) {
                $('#meta_text').append(link_text);
            }
            if (link_table.length > db_num) {
                $('#meta_text').append('<p class="meta_cat mt-3">iMODULONS</p>');
                for (let i = db_num; i < link_table.length; i++) {
                    $('#meta_text').append('<p class="meta_val mb-0"><a href="' + link_table[i][1] + '" target="_blank">' + link_table[i][0] + '</a></p>');
                }
            }

        })
        $('#meta_text').append('<p class="meta_cat">RAW DATA ACCESSION</p>');
        $('#meta_text').append('<p class="meta_val">' + row[14] + '</p>');
        if (row[15]|row[19]) {
            $('#meta_text').append('<p class="meta_cat">REFERENCE</p>');
            $('#meta_text').append('<p class="meta_val mb-0">PMID: ' + row[19] + '</p>');
            $('#meta_text').append('<p class="meta_val">DOI: <a href="' + row[15] + '">' + row[15].replace("https://doi.org/", "") + '</a></p>');

        }
        if (row[26]) {
            $('#meta_text').append('<p class="meta_cat">EXTRA NOTES</p>');
            $('#meta_text').append('<p class="meta_val">' + row[26] + '</p>');
        }

        // IGV and table
        const [title, fileName] = row[12].split(";")[0].split("|");
        generateIGVandTable(TF_name, row, fileName);
        // First, get row[12] to get list of filenames to create the tabs for x number of files
        const conditions = row[12].split(";");
        // parse the table row of Title A|xyz.json;Title B|def.json;...
        // access the tabs parent component by ID
        $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab active" id="table-tab-1" data-bs-toggle="tab"
                    data-bs-target="#table-tab-1" data-toggle="tab" data-target="#table-tab-1"
                    type="button" role="tab" aria-controls="table-tab-1"
                    aria-selected="true">${title}
            </button>
        </li>
        `);
        $('#table-tab-1').click(() => {
            generateIGVandTable(TF_name, row, fileName);
        });
        // loop through the loaded values of `Title A|xyz.json` and create a tab for each
        conditions.slice(1, conditions.length).forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            $('#TableTab').append(`
        <li class="nav-item" role="presentation">
            <button class="nav-link round-tab" id="table-tab-${i + 2}" data-bs-toggle="tab"
                    data-bs-target="#table-tab-${i + 2}" data-toggle="tab" data-target="#table-tab-${i + 2}"
                    type="button" role="tab" aria-controls="table-tab-${i + 2}"
                    aria-selected="true">${conditionTitle}
            </button>
        </li>
        `);
            $(`#table-tab-${i + 2}`).click(() => {
                generateIGVandTable(TF_name, row, conditionFile);
            });
        });

        // Loop through it all again and create buttons to choose the motif graphic
        let hasFoundFirstImageToRender = false;
        const clearAllActiveMotifButtonFunctions = [];

        conditions.forEach((condition, i) => {
            const [conditionTitle, conditionFile] = condition.split('|');
            const motifImageAndCSVBase = conditionFile.replace("_binding_table.json", "");
            const imageFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.png`;
            const csvFilePath = '/data/' + organism + '/' + genome + '/meme-chip/outputs/' + `${motifImageAndCSVBase}.csv`;
            const spacelessConditionTitle = conditionTitle.replace(/ /g, "_");

            $('#motif_selector_buttons').append(`
                <button class="btn btn-outline-dark btn-sm" id="motif-button-${spacelessConditionTitle}">${conditionTitle}</button>
            `);

            $.get(imageFilePath, function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = imageFilePath;
                    document.getElementById("motif-download-svg").classList.add('d-none');
                    document.getElementById("motif-download-png").href = imageFilePath;
                    document.getElementById("motif-download-csv").href = csvFilePath;
                    document.getElementById("motif-download-csv").classList.remove('d-none');

                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.add('active');
                    document.getElementById('motif-banner-title').innerText = 'Logo identified by MEME-ChIP';
                };

                clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-${spacelessConditionTitle}`).classList.remove('active'));
                $(`#motif-button-${spacelessConditionTitle}`).click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }
            }).catch(() => {
                //     $('#motif_selector_buttons').append(`
                //     <button disabled title="No motif logo found using MEME-ChIP">${conditionTitle}</button>
                // `);
                $(`#motif-button-${spacelessConditionTitle}`).addClass('disabled');
                $(`#motif-button-${spacelessConditionTitle}`).addClass('gray-btn');
                $(`#motif-button-${spacelessConditionTitle}`).attr('disabled', true);
                $(`#motif-button-${spacelessConditionTitle}`).attr("title", "No motif logo found using MEME-ChIP");
            });
        });

        // Add literature motif information, if available
        const motifSVG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.svg';
        const motifPNG = '/data/' + organism + '/' + genome + '/motifs/images/' + TF_name + '.png';

        $('#motif_selector_buttons').append(`
            <button class="btn btn-outline-dark btn-sm" id="motif-button-lit">PMID: ${row[19]}</button>
        `);


        $.get(motifSVG)
            .done(function () {
                // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                document.getElementById("header-tab-3").classList.remove('disabled');

                const changeImageHandler = function () {
                    document.getElementById("motif_sequence").src = motifSVG;
                    document.getElementById("motif-download-svg").href = motifSVG;
                    document.getElementById("motif-download-svg").classList.remove('d-none');
                    document.getElementById("motif-download-png").href = motifPNG;
                    document.getElementById("motif-download-csv").classList.add('d-none');


                    clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-lit`).classList.remove('active'));
                    clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                    document.getElementById(`motif-button-lit`).classList.add('active');
                    document.getElementById('motif-banner-title').innerText = `Logo from PMID ${row[19]}`;
                };

                $('#motif-button-lit').click(changeImageHandler);

                if (!hasFoundFirstImageToRender) {
                    changeImageHandler();
                    hasFoundFirstImageToRender = true;
                }

                // document.getElementById("motif_sequence").src = motifSVG;
                // document.getElementById("motif-download-svg").href = motifSVG;
            }).fail(function () {
            $.get(motifPNG)
                .done(function () {
                    // successfully received some sort of data -- remove 'disabled' class to enable the motifs tab.
                    document.getElementById("header-tab-3").classList.remove('disabled');

                    const changeImageHandler = function () {
                        document.getElementById("motif_sequence").src = motifPNG;
                        document.getElementById("motif-download-svg").classList.add('d-none');
                        document.getElementById("motif-download-png").href = motifPNG;
                        document.getElementById("motif-download-csv").classList.add('d-none');

                        clearAllActiveMotifButtonFunctions.push(() => document.getElementById(`motif-button-lit`).classList.remove('active'));
                        clearAllActiveMotifButtonFunctions.forEach(clearFunc => clearFunc());
                        document.getElementById(`motif-button-lit`).classList.add('active');
                        document.getElementById('motif-banner-title').innerText = `Logo from PMID ${row[19]}`;
                    };

                    $('#motif-button-lit').click(changeImageHandler);

                    if (!hasFoundFirstImageToRender) {
                        changeImageHandler();
                        hasFoundFirstImageToRender = true;
                    }
                    // document.getElementById("motif_sequence").src = motifPNG;
                    // document.getElementById("motif-download-svg").remove();
                }).fail(function () {
                // document.getElementById("header-tab-3").classList.add('disabled')
                // document.getElementById("motif-button-lit").classList.add('disabled');
                // document.getElementById("motif-button-lit").classList.add('gray-btn');
                $('#motif-button-lit').remove();
            })
        });

        // if (!hasFoundFirstImageToRender) {
        //     document.getElementById("header-tab-3").classList.add('disabled');
        // }

        // deal with disabling header tab here

        // Venn
        $.get('/data/' + organism + '/' + genome + '/venn/' + TF_name.substr(0, 4) + '_venn.csv', function (data) {
            generateVenn2(data, 'venn');
        }).catch(() => {
            document.getElementById("header-tab-2").classList.add('disabled')
        });

        // Binding width histogram
        $.get('/data/' + organism + '/' + genome + '/binding_widths/' + TF_name + "_widths.csv", function (width_data) {
            generateWidthHistogram(width_data, 'widthHist')
        }).catch(() => {
            document.getElementById("widthHist").innerText = 'Binding width histogram not available.'
            document.getElementById("widthHist").classList.add('NA_text')
        });

        // Peak/Gene Position Scatter
        $.get('/data/' + organism + '/' + genome + '/positions/' + TF_name + "_positions.csv", function (positions_data) {
            generatePeakScatter(positions_data, row, "position_scatter");
        }).catch(() => {
            document.getElementById("header-tab-4").classList.add('disabled')
        });

    });

    // populate help buttons
    $('#table-info').popover({
        title: "Help - Binding Site Table",
        html: true,
        trigger: "hover focus",
        template: '<div class="popover popover-make-it-wide" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
        content: function () {
            return `            <p>
                This table represents <a href="https://academic.oup.com/nar/article/42/20/e156/2902729" target="_blank">MACE</a> results from processing the sequence read data presented in the Genome Viewer. Tabs across the top correspond to each condition. Each row represents a curated binding peak. Clicking on a row will update the Genome Viewer to display a zoomed in plot of the corresponding peak. The rows are initially sorted by genome location, and the tallest peaks can easily be found by sorting by descending "Peak Intensity" (clicking the column header twice). The columns are as follows:
            </p>

            <ul>
                <li><b>TF-Condition-#</b>: A unique identifier for the peak, ordered by genome location.</li>
                <li><b>Start Position & End Position*</b>: The genome locations of the peak in base pairs are determined by MACE.</li>
                <li><b>Peak Intensity*</b>: Peak signal intensity is presented as a signal/noise (S/N) ratio, with noise levels defined separately for each condition.</li>
                <li><b>Closest gene</b>: Each peak is assigned to its closest gene, according to genomic positions. This column will contain a gene name regardless of the distance to the gene.</li>
                <li><b>TU Target genes</b>: If a transcriptional unit exists downstream of the binding peak and is within 500 base pairs of the peak, its genes will be listed here. If that TU also has a page in an online database, then the TU will appear as a link to that page.</li>
            </ul>
            <p>*See the About page for more details.</p>`;
        }
    });

    $('#visual-info').popover({
        container: "body",
        title: "Help - Feature Visualization",
        html: true,
        trigger: "hover focus",
        template: '<div class="popover popover-make-it-wide" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
        content: function () {
            return `<div class="popover-message">
            <p>
                The tabs within this panel provide additional characterization of the data. In the upper right hand corner of each tab, the menu button (<i class="material-icons">menu</i>) enables PNG, SVG, and data download.
            </p>

            <p><b>
                Width:
            </b>
                This tab shows a histogram of the binding peak widths from the binding site table. Hover over each bar to see a count of peaks that fall within its corresponding bin.
            </p>

            <p><b>
                Position: 
            </b>
                In this tab, proChIPdb compares the binding locations relative to each target gene. For each gene (as listed in the Closest Gene column of the Binding Site Table), the distance from the gene start site to the binding site is measured in base pairs and normalized to the length of the gene. Points are then plotted with this value on the x axis and the peak intensity (S/N) on the y axis. Hover over a point to view more details, such as the gene name.
            </p>

            <p><b>
                Motif:
            </b>
                This tab contains a sequence logo of a significantly enriched motif and its corresponding E-value (E < 0.001). Motifs provide valuable insight about which sequences will bind the TF. Sub-tabs across the bottom of the panel allow you to select from each of the conditions in the dashboard. The menu button in the upper right hand corner of each tab allows download of both the image itself and the position weight matrix (PWM) of the motif. In addition, a final tab named using a PMID may be available; this contains the sequence logo as it appears in the publication from which the data is from. See the About page for more details.
            </p>

            <p><b>
                Venn:
            </b>
                The venn diagram compares the target genes obtained in proChIPdb to other target gene sets in the literature. Hover over a section of the venn diagram to see the genes it contains. Areas of agreement indicate strong evidence of direct regulation, and areas of disagreement represent opportunities to improve literature annotation or elucidate condition-specific differences in binding. The specific literature sources are mentioned below each diagram.
            </p></div>`;
        }
    });

    $('#igv-info').popover({
        title: "Help - Genome Viewer",
        html: true,
        trigger: "hover focus",
        template: '<div class="popover popover-make-it-wide" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
        content: function () {
            return `<div class="popover-message">
            <p>
                proChIPdb‚Äôs genome viewer provides access to a complete, genome-wide view of the transcription factor‚Äôs activity under the given conditions. The genome viewer visualizes bigWig ChIP read files. Click a row in the Binding Site Table to quickly zoom in on a peak. Selected elements are described below.
            </p>
            <ul>
                <li><b>Toolbar</b>: The toolbar contains a search bar, which can be used to search for specific loci by gene location or gene name. It also has useful tools for adding cursor guides, a center line, toggling track labels, downloading the current view, and adjusting the zoom level.</li>
                <li><b>Genes</b>: Gene annotations as provided in the genome. Click on an element to see additional details, such as the gene name, locus tag, and product description.</li>
                <li><b>Published TUs</b>: Transcriptional units.</li>
                <li><b>Published TSS</b>: Transcription start sites.</li>
                <li><b>Published TFBS</b>: Transcription factor binding sites.</li>
                <li><b>Data Tracks</b>: The y-axis of these plots is the number of ChIP reads mapped to the x-axis nucleotide. Replicate numbers and unique conditions (if applicable) are indicated by the track name. To download this data for custom analysis, click ‚ÄúDownload bigWig Files‚Äù in the upper left corner of the pane.</li>
            </ul>
            <p>See the About page for more details.</p></div>`;
        }
    });

</script>

<!-- for interactive tour -->
<script type="text/javascript">
      if (RegExp('multipage', 'gi').test(window.location.search)) {
        introJs().setOption('doneLabel', 'Next page').start().oncomplete(function() {
          window.location.href = 'index.html';
        });
      }
    </script>

</body>

</html>