{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for ChIP-pro, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start > BP) & (stop > BP) & (strand == '+')) | ((start < BP) & (stop < BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start < BP) & (stop > BP)) | ((start > BP) & (stop < BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "                    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen(Peak_DF): \n",
    "    TF_name = Peak_DF['index'][1][:4]\n",
    "    gene_list = [i for i in Peak_DF.target_genes if i != '']\n",
    "    chip_data= ','.join(list(gene_list)).split(',')\n",
    "    reg_data = TRN_data[TF_name]\n",
    "    all_genes = [i  for i in chip_data if i not in reg_data] + reg_data\n",
    "    \n",
    "    \n",
    "    TF = TF_name\n",
    "    reg_genes=reg_data\n",
    "    reg_only = []\n",
    "    chip_genes=chip_data\n",
    "    chip_only = []\n",
    "    shared_genes=[]\n",
    "    for i in all_genes: \n",
    "        if (i in reg_data) & (i not in chip_data):\n",
    "            reg_only.append(i)\n",
    "        elif (i in chip_data) & (i not in reg_data):\n",
    "            chip_only.append(i)\n",
    "        elif (i in chip_data) & (i in reg_data):\n",
    "            shared_genes.append(i)\n",
    "            \n",
    "    values = [TF,\n",
    "          len(reg_genes),\n",
    "          len(reg_only),\n",
    "          len(chip_genes),\n",
    "          len(chip_only),\n",
    "          len(shared_genes),\n",
    "          len(all_genes)]\n",
    "    \n",
    "    index_name = ['TF',\n",
    "              'reg_genes',\n",
    "              'reg_only',\n",
    "              'chip_genes',\n",
    "              'chip_only',\n",
    "              'shared_genes',\n",
    "              'all_genes']\n",
    "    \n",
    "    genes = ['; '.join(precise2_TRN.source[precise2_TRN.index == TF_name].unique()),\n",
    "         reg_genes,\n",
    "         reg_only,\n",
    "         chip_genes,\n",
    "         chip_only,\n",
    "         shared_genes,\n",
    "         all_genes]\n",
    "    \n",
    "    same1 = defaultdict(list)\n",
    "    for i in range(0,len(index_name)):\n",
    "        same1[index_name[i]].append(values[i])\n",
    "\n",
    "    finall = pd.DataFrame.from_dict(same1, orient='index', columns = ['value'])\n",
    "    finall['list'] = genes\n",
    "    for i, row in finall.iterrows(): \n",
    "        if row.value == 0:\n",
    "            finall.list[i] = ''\n",
    "    return finall"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "            try:\n",
    "                if gene_info.strand[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>EtOH</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>EtOH</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fructose, Galactose, or Acetate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe or DPD</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GadW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadw_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GadX</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadx_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadx_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadx_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KdpE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>TMA</td>\n",
       "      <td>KCl</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>KCl|kdpe_KCl_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>KdpeE_R2_S34_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>KdpeE_R2_S34_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>M9P|phob_M9P_binding_table.json</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>PhoB_1_S16_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>PhoB_1_S16_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>PhoB_2_S6_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>PhoB_2_S6_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>RpoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe or DPD</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|rpob_dpd_binding_table.json;fe|rpob_fe_bin...</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>rpob_fe1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>rpob_fe2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>rpob_dpd1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>rpob_dpd2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YbaO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybao_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybao_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybao_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>YbaQ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>41.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybaq_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybaq_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybaq_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>40.0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>ph5|ydci_ph5_binding_table.json;ph7|ydci_ph7_b...</td>\n",
       "      <td>pH5 – R1</td>\n",
       "      <td>ydci_ph5_1.bw</td>\n",
       "      <td>pH5 – R2</td>\n",
       "      <td>ydci_ph5_2.bw</td>\n",
       "      <td>pH7 – R1</td>\n",
       "      <td>ydci_ph7_1.bw</td>\n",
       "      <td>pH7 – R2</td>\n",
       "      <td>ydci_ph7_2.bw</td>\n",
       "      <td>pH8 – R1</td>\n",
       "      <td>ydci_ph8_1.bw</td>\n",
       "      <td>pH8 – R2</td>\n",
       "      <td>ydci_ph8_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>YddM</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yddm_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yddm_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yddm_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YheO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yheo_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yheo_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yheo_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>YiaG</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiag_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiag_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiag_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>YieP</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>3HP</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiep_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiep_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiep_2.bw</td>\n",
       "      <td>3HP – R1</td>\n",
       "      <td>yiep_3hp_1.bw</td>\n",
       "      <td>3HP – R2</td>\n",
       "      <td>yiep_3hp_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>YihL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>YihW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>YjdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yjdc_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>YjhI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjhi_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjhi_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>YneJ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynej_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynej_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>YnfL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynfl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynfl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>YpdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ypdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ypdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>YqhC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>M9|yqhc_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>yqhc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yqhc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ZraR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>ZnCl2</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>ZraR_1_S3_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>ZraR_1_S3_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>ZraR_2_S4_R1.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>O157:H7 EDL933</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE_005174_2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>EDL933</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>edl933furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>edl933furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010498_1</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>smsfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>smsfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CU_928163_2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>umn026furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>umn026furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Dps</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|dps_expo_binding_table.json;sta|dps_sta_b...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>dps_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>dps_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>dps_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>dps_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Fis</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|fis_expo_binding_table.json;sta|fis_sta_b...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>fis_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>fis_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>fis_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>fis_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GyrA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|gyra_expo_binding_table.json;sta|gyra_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyra_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyra_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyra_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyra_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GyrB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|gyrb_expo_binding_table.json;sta|gyrb_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyrb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyrb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyrb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyrb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>H-NS</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|h-ns_expo_binding_table.json;sta|h-ns_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hns_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hns_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hns_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hns_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>HupA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|hupa_expo_binding_table.json;sta|hupa_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>HupB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|hupb_expo_binding_table.json;sta|hupb_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>InfA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|infa_expo_binding_table.json;sta|infa_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>infa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>infa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>infa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>infa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MukB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|mukb_expo_binding_table.json;sta|mukb_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>mukb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>mukb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>mukb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>mukb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>RdgC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|rdgc_expo_binding_table.json;sta|rdgc_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rdgc_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rdgc_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rdgc_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rdgc_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Rob</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|rob_expo_binding_table.json;sta|rob_sta_b...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rob_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rob_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rob_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rob_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>StpA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|stpa_expo_binding_table.json;sta|stpa_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>stpa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>stpa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>stpa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>stpa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TopA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|topa_expo_binding_table.json;sta|topa_sta...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>topa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>topa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>topa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>topa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Tus</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|tus_expo_binding_table.json;sta|tus_sta_b...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>tus_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>tus_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>tus_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>tus_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NarL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Nitrate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>nitrate|narl_nitrate_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>narl_nitrate_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>narl_nitrate_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PuuR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Putrescine</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>putrescine|puur_putrescine_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>puur_putrescine_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>puur_putrescine_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>RcsA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Oxida</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>rcsa_oxida_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>rcsa_oxida_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>RpoB (Cra and Crp KO exps)</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Acetate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>Misc</td>\n",
       "      <td>MG1655 ace|rpob (cra and crp ko exps)_MG1655 a...</td>\n",
       "      <td>control (WT) – R1</td>\n",
       "      <td>mg1655_ace_rpob1.bw</td>\n",
       "      <td>control (WT) – R2</td>\n",
       "      <td>mg1655_ace_rpob2.bw</td>\n",
       "      <td>cra-KO – R1</td>\n",
       "      <td>cra_ko_ace_rpob1.bw</td>\n",
       "      <td>cra-KO – R2</td>\n",
       "      <td>cra_ko_ace_rpob2.bw</td>\n",
       "      <td>crp-KO – R1</td>\n",
       "      <td>crp_ko_ace_rpob1.bw</td>\n",
       "      <td>crp-KO – R2</td>\n",
       "      <td>crp_ko_ace_rpob2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               TF          Organism          Strain Media  \\\n",
       "index                                                                       \n",
       "0                            BaeR  Escherichia coli     K-12 MG1655    LB   \n",
       "1                            CpxR  Escherichia coli     K-12 MG1655    LB   \n",
       "2                             Cra  Escherichia coli     K-12 MG1655    M9   \n",
       "3                             Fur  Escherichia coli     K-12 MG1655    M9   \n",
       "4                            GadE  Escherichia coli     K-12 MG1655    M9   \n",
       "5                            GadW  Escherichia coli     K-12 MG1655    M9   \n",
       "6                            GadX  Escherichia coli     K-12 MG1655    M9   \n",
       "7                            KdpE  Escherichia coli     K-12 MG1655   TMA   \n",
       "8                            PhoB  Escherichia coli     K-12 MG1655   M9P   \n",
       "9                            RpoB  Escherichia coli     K-12 MG1655    M9   \n",
       "10                           YbaO  Escherichia coli     K-12 MG1655    M9   \n",
       "11                           YbaQ  Escherichia coli     K-12 MG1655    M9   \n",
       "12                           YdcI  Escherichia coli     K-12 MG1655    M9   \n",
       "13                           YddM  Escherichia coli     K-12 MG1655    M9   \n",
       "14                           YheO  Escherichia coli     K-12 MG1655    M9   \n",
       "15                           YiaG  Escherichia coli     K-12 MG1655    M9   \n",
       "16                           YieP  Escherichia coli     K-12 MG1655    M9   \n",
       "17                           YihL  Escherichia coli     K-12 MG1655    M9   \n",
       "18                           YihW  Escherichia coli     K-12 MG1655    M9   \n",
       "19                           YjdC  Escherichia coli     K-12 MG1655    M9   \n",
       "20                           YjhI  Escherichia coli     K-12 MG1655    M9   \n",
       "21                           YneJ  Escherichia coli     K-12 MG1655    M9   \n",
       "22                           YnfL  Escherichia coli     K-12 MG1655    M9   \n",
       "23                           YpdC  Escherichia coli     K-12 MG1655    M9   \n",
       "24                           YqhC  Escherichia coli     K-12 MG1655    M9   \n",
       "25                           ZraR  Escherichia coli     K-12 MG1655    LB   \n",
       "26                            Fur  Escherichia coli  O157:H7 EDL933    M9   \n",
       "27                            Fur  Escherichia coli         SMS-3-5    M9   \n",
       "28                            Fur  Escherichia coli          UMN026    M9   \n",
       "29                            Dps  Escherichia coli     K-12 MG1655    M9   \n",
       "30                            Fis  Escherichia coli     K-12 MG1655    M9   \n",
       "31                           GyrA  Escherichia coli     K-12 MG1655    M9   \n",
       "32                           GyrB  Escherichia coli     K-12 MG1655    M9   \n",
       "33                           H-NS  Escherichia coli     K-12 MG1655    M9   \n",
       "34                           HupA  Escherichia coli     K-12 MG1655    M9   \n",
       "35                           HupB  Escherichia coli     K-12 MG1655    M9   \n",
       "36                           InfA  Escherichia coli     K-12 MG1655    M9   \n",
       "37                           MukB  Escherichia coli     K-12 MG1655    M9   \n",
       "38                           RdgC  Escherichia coli     K-12 MG1655    M9   \n",
       "39                            Rob  Escherichia coli     K-12 MG1655    M9   \n",
       "40                           StpA  Escherichia coli     K-12 MG1655    M9   \n",
       "41                           TopA  Escherichia coli     K-12 MG1655    M9   \n",
       "42                            Tus  Escherichia coli     K-12 MG1655    M9   \n",
       "43                           NarL  Escherichia coli     K-12 MG1655    M9   \n",
       "44                           PuuR  Escherichia coli     K-12 MG1655    M9   \n",
       "45                           RcsA  Escherichia coli     K-12 MG1655    M9   \n",
       "46     RpoB (Cra and Crp KO exps)  Escherichia coli     K-12 MG1655    M9   \n",
       "\n",
       "                            Supplement    genome_id organism_id  \\\n",
       "index                                                             \n",
       "0                                 EtOH  NC_000913_3      e_coli   \n",
       "1                                 EtOH  NC_000913_3      e_coli   \n",
       "2      Fructose, Galactose, or Acetate  NC_000913_3      e_coli   \n",
       "3                            Fe or DPD  NC_000913_3      e_coli   \n",
       "4                                  NaN  NC_000913_3      e_coli   \n",
       "5                                  NaN  NC_000913_3      e_coli   \n",
       "6                                  NaN  NC_000913_3      e_coli   \n",
       "7                                  KCl  NC_000913_3      e_coli   \n",
       "8                                  NaN  NC_000913_3      e_coli   \n",
       "9                            Fe or DPD  NC_000913_3      e_coli   \n",
       "10                                 NaN  NC_000913_3      e_coli   \n",
       "11                                 NaN  NC_000913_3      e_coli   \n",
       "12                                 NaN  NC_000913_3      e_coli   \n",
       "13                                 NaN  NC_000913_3      e_coli   \n",
       "14                                 NaN  NC_000913_3      e_coli   \n",
       "15                                 NaN  NC_000913_3      e_coli   \n",
       "16                                 3HP  NC_000913_3      e_coli   \n",
       "17                                 NaN  NC_000913_3      e_coli   \n",
       "18                                 NaN  NC_000913_3      e_coli   \n",
       "19                                 NaN  NC_000913_3      e_coli   \n",
       "20                                 NaN  NC_000913_3      e_coli   \n",
       "21                                 NaN  NC_000913_3      e_coli   \n",
       "22                                 NaN  NC_000913_3      e_coli   \n",
       "23                                 NaN  NC_000913_3      e_coli   \n",
       "24                                 NaN  NC_000913_3      e_coli   \n",
       "25                               ZnCl2  NC_000913_3      e_coli   \n",
       "26                                 NaN  AE_005174_2      e_coli   \n",
       "27                                 NaN  NC_010498_1      e_coli   \n",
       "28                                 NaN  CU_928163_2      e_coli   \n",
       "29                                 NaN  NC_000913_3      e_coli   \n",
       "30                                 NaN  NC_000913_3      e_coli   \n",
       "31                                 NaN  NC_000913_3      e_coli   \n",
       "32                                 NaN  NC_000913_3      e_coli   \n",
       "33                                 NaN  NC_000913_3      e_coli   \n",
       "34                                 NaN  NC_000913_3      e_coli   \n",
       "35                                 NaN  NC_000913_3      e_coli   \n",
       "36                                 NaN  NC_000913_3      e_coli   \n",
       "37                                 NaN  NC_000913_3      e_coli   \n",
       "38                                 NaN  NC_000913_3      e_coli   \n",
       "39                                 NaN  NC_000913_3      e_coli   \n",
       "40                                 NaN  NC_000913_3      e_coli   \n",
       "41                                 NaN  NC_000913_3      e_coli   \n",
       "42                                 NaN  NC_000913_3      e_coli   \n",
       "43                             Nitrate  NC_000913_3      e_coli   \n",
       "44                          Putrescine  NC_000913_3      e_coli   \n",
       "45                               Oxida  NC_000913_3      e_coli   \n",
       "46                             Acetate  NC_000913_3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond       project  \\\n",
       "index                                                           \n",
       "0                    NaN            4         1           TCS   \n",
       "1                    NaN            2         1           TCS   \n",
       "2                    NaN            8         4      known_TF   \n",
       "3                    NaN            4         2      known_TF   \n",
       "4                    NaN            2         1      known_TF   \n",
       "5                    NaN            2         1      known_TF   \n",
       "6                    NaN            2         1      known_TF   \n",
       "7                    NaN            2         1           TCS   \n",
       "8                    NaN            4         1           TCS   \n",
       "9                    NaN            4         2      known_TF   \n",
       "10                   4.0            2         1          yTF1   \n",
       "11                  41.0            2         1          yTF1   \n",
       "12                  40.0            6         3          yTF1   \n",
       "13                  18.0            2         1          yTF1   \n",
       "14                   4.0            2         1          yTF1   \n",
       "15                   NaN            2         1          yTF1   \n",
       "16                  23.0            4         1          yTF1   \n",
       "17                   NaN            2         1          yTF2   \n",
       "18                   NaN            2         1          yTF2   \n",
       "19                   NaN            2         1          yTF1   \n",
       "20                   NaN            2         1          yTF2   \n",
       "21                   NaN            2         1          yTF2   \n",
       "22                   NaN            2         1          yTF2   \n",
       "23                   NaN            2         1          yTF2   \n",
       "24                   NaN            2         1          yTF2   \n",
       "25                   NaN            3         1           TCS   \n",
       "26                   NaN            2         1        EDL933   \n",
       "27                   NaN            2         1       SMS-3-5   \n",
       "28                   NaN            2         1        UMN026   \n",
       "29                   NaN            4         2  conformation   \n",
       "30                   NaN            4         2  conformation   \n",
       "31                   NaN            4         2  conformation   \n",
       "32                   NaN            4         2  conformation   \n",
       "33                   NaN            4         2  conformation   \n",
       "34                   NaN            4         2  conformation   \n",
       "35                   NaN            4         2  conformation   \n",
       "36                   NaN            4         2  conformation   \n",
       "37                   NaN            4         2  conformation   \n",
       "38                   NaN            4         2  conformation   \n",
       "39                   NaN            4         2  conformation   \n",
       "40                   NaN            4         2  conformation   \n",
       "41                   NaN            4         2  conformation   \n",
       "42                   NaN            4         2  conformation   \n",
       "43                   NaN            2         1          Misc   \n",
       "44                   NaN            2         1          Misc   \n",
       "45                   NaN            2         1          Misc   \n",
       "46                   NaN            6         3          Misc   \n",
       "\n",
       "                                          binding tables  \\\n",
       "index                                                      \n",
       "0                      EtOH|baer_EtOH_binding_table.json   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   \n",
       "4                          M9|gade_M9_binding_table.json   \n",
       "5                          M9|gadw_M9_binding_table.json   \n",
       "6                          M9|gadx_M9_binding_table.json   \n",
       "7                        KCl|kdpe_KCl_binding_table.json   \n",
       "8                        M9P|phob_M9P_binding_table.json   \n",
       "9      dpd|rpob_dpd_binding_table.json;fe|rpob_fe_bin...   \n",
       "10                         M9|ybao_M9_binding_table.json   \n",
       "11                         M9|ybaq_M9_binding_table.json   \n",
       "12     ph5|ydci_ph5_binding_table.json;ph7|ydci_ph7_b...   \n",
       "13                         M9|yddm_M9_binding_table.json   \n",
       "14                         M9|yheo_M9_binding_table.json   \n",
       "15                         M9|yiag_M9_binding_table.json   \n",
       "16                         M9|yiep_M9_binding_table.json   \n",
       "17                                                   NaN   \n",
       "18                                                   NaN   \n",
       "19                         M9|yjdc_M9_binding_table.json   \n",
       "20                                                   NaN   \n",
       "21                                                   NaN   \n",
       "22                                                   NaN   \n",
       "23                                                   NaN   \n",
       "24                         M9|yqhc_M9_binding_table.json   \n",
       "25                                                   NaN   \n",
       "26                                                   NaN   \n",
       "27                                                   NaN   \n",
       "28                                                   NaN   \n",
       "29     expo|dps_expo_binding_table.json;sta|dps_sta_b...   \n",
       "30     expo|fis_expo_binding_table.json;sta|fis_sta_b...   \n",
       "31     expo|gyra_expo_binding_table.json;sta|gyra_sta...   \n",
       "32     expo|gyrb_expo_binding_table.json;sta|gyrb_sta...   \n",
       "33     expo|h-ns_expo_binding_table.json;sta|h-ns_sta...   \n",
       "34     expo|hupa_expo_binding_table.json;sta|hupa_sta...   \n",
       "35     expo|hupb_expo_binding_table.json;sta|hupb_sta...   \n",
       "36     expo|infa_expo_binding_table.json;sta|infa_sta...   \n",
       "37     expo|mukb_expo_binding_table.json;sta|mukb_sta...   \n",
       "38     expo|rdgc_expo_binding_table.json;sta|rdgc_sta...   \n",
       "39     expo|rob_expo_binding_table.json;sta|rob_sta_b...   \n",
       "40     expo|stpa_expo_binding_table.json;sta|stpa_sta...   \n",
       "41     expo|topa_expo_binding_table.json;sta|topa_sta...   \n",
       "42     expo|tus_expo_binding_table.json;sta|tus_sta_b...   \n",
       "43               nitrate|narl_nitrate_binding_table.json   \n",
       "44         putrescine|puur_putrescine_binding_table.json   \n",
       "45                                                   NaN   \n",
       "46     MG1655 ace|rpob (cra and crp ko exps)_MG1655 a...   \n",
       "\n",
       "                BWcond1_1_name             BWcond1_1           BWcond1_2_name  \\\n",
       "index                                                                           \n",
       "0                bio-rep1 – R1     BaeR_R1_S31_R1.bw            bio-rep1 – R2   \n",
       "1                           R1       CpxRR1_S1_R1.bw                       R2   \n",
       "2                     Glu – R1          cra_glu_1.bw                 Glu – R2   \n",
       "3                      Fe – R1           fur_fe_1.bw                  Fe – R2   \n",
       "4                           R1             gade_1.bw                       R2   \n",
       "5                           R1             gadw_1.bw                       R2   \n",
       "6                           R1             gadx_1.bw                       R2   \n",
       "7                           R1    KdpeE_R2_S34_R1.bw                       R2   \n",
       "8                bio-rep1 – R1      PhoB_1_S16_R1.bw            bio-rep1 – R2   \n",
       "9                      Fe – R1           rpob_fe1.bw                  Fe – R2   \n",
       "10                          R1             ybao_1.bw                       R2   \n",
       "11                          R1             ybaq_1.bw                       R2   \n",
       "12                    pH5 – R1         ydci_ph5_1.bw                 pH5 – R2   \n",
       "13                          R1             yddm_1.bw                       R2   \n",
       "14                          R1             yheo_1.bw                       R2   \n",
       "15                          R1             yiag_1.bw                       R2   \n",
       "16                          R1             yiep_1.bw                       R2   \n",
       "17                          R1             yihl_1.bw                       R2   \n",
       "18                          R1             yihw_1.bw                       R2   \n",
       "19                          R1             yjdc_1.bw                       R2   \n",
       "20                          R1             yjhi_1.bw                       R2   \n",
       "21                          R1             ynej_1.bw                       R2   \n",
       "22                          R1             ynfl_1.bw                       R2   \n",
       "23                          R1             ypdc_1.bw                       R2   \n",
       "24                          R1             yqhc_1.bw                       R2   \n",
       "25               bio-rep1 – R1       ZraR_1_S3_R1.bw            bio-rep1 – R2   \n",
       "26                          R1    edl933furfinal1.bw                       R2   \n",
       "27                          R1       smsfurfinal1.bw                       R2   \n",
       "28                          R1    umn026furfinal1.bw                       R2   \n",
       "29     exponential growth – R1         dps_expo_1.bw  exponential growth – R2   \n",
       "30     exponential growth – R1         fis_expo_1.bw  exponential growth – R2   \n",
       "31     exponential growth – R1        gyra_expo_1.bw  exponential growth – R2   \n",
       "32     exponential growth – R1        gyrb_expo_1.bw  exponential growth – R2   \n",
       "33     exponential growth – R1         hns_expo_1.bw  exponential growth – R2   \n",
       "34     exponential growth – R1        hupa_expo_1.bw  exponential growth – R2   \n",
       "35     exponential growth – R1        hupb_expo_1.bw  exponential growth – R2   \n",
       "36     exponential growth – R1        infa_expo_1.bw  exponential growth – R2   \n",
       "37     exponential growth – R1        mukb_expo_1.bw  exponential growth – R2   \n",
       "38     exponential growth – R1        rdgc_expo_1.bw  exponential growth – R2   \n",
       "39     exponential growth – R1         rob_expo_1.bw  exponential growth – R2   \n",
       "40     exponential growth – R1        stpa_expo_1.bw  exponential growth – R2   \n",
       "41     exponential growth – R1        topa_expo_1.bw  exponential growth – R2   \n",
       "42     exponential growth – R1         tus_expo_1.bw  exponential growth – R2   \n",
       "43                          R1     narl_nitrate_1.bw                       R2   \n",
       "44                          R1  puur_putrescine_1.bw                       R2   \n",
       "45                          R1       rcsa_oxida_1.bw                       R2   \n",
       "46           control (WT) – R1   mg1655_ace_rpob1.bw        control (WT) – R2   \n",
       "\n",
       "                  BWcond1_2          BWcond2_1_name            BWcond2_1  \\\n",
       "index                                                                      \n",
       "0         BaeR_R1_S31_R2.bw           bio-rep2 – R1   BaerR_R2_S32_R1.bw   \n",
       "1           CpxRR2_S2_R2.bw                     NaN                  NaN   \n",
       "2              cra_glu_2.bw                Fru – R1         cra_fru_1.bw   \n",
       "3               fur_fe_2.bw                DPD – R1         fur_dpd_1.bw   \n",
       "4                 gade_2.bw                     NaN                  NaN   \n",
       "5                 gadw_2.bw                     NaN                  NaN   \n",
       "6                 gadx_2.bw                     NaN                  NaN   \n",
       "7        KdpeE_R2_S34_R2.bw                     NaN                  NaN   \n",
       "8          PhoB_1_S16_R2.bw           bio-rep2 – R1      PhoB_2_S6_R1.bw   \n",
       "9               rpob_fe2.bw                DPD – R1         rpob_dpd1.bw   \n",
       "10                ybao_2.bw                     NaN                  NaN   \n",
       "11                ybaq_2.bw                     NaN                  NaN   \n",
       "12            ydci_ph5_2.bw                pH7 – R1        ydci_ph7_1.bw   \n",
       "13                yddm_2.bw                     NaN                  NaN   \n",
       "14                yheo_2.bw                     NaN                  NaN   \n",
       "15                yiag_2.bw                     NaN                  NaN   \n",
       "16                yiep_2.bw                3HP – R1        yiep_3hp_1.bw   \n",
       "17                yihl_2.bw                     NaN                  NaN   \n",
       "18                yihw_2.bw                     NaN                  NaN   \n",
       "19                yjdc_2.bw                     NaN                  NaN   \n",
       "20                yjhi_2.bw                     NaN                  NaN   \n",
       "21                ynej_2.bw                     NaN                  NaN   \n",
       "22                ynfl_2.bw                     NaN                  NaN   \n",
       "23                ypdc_2.bw                     NaN                  NaN   \n",
       "24                yqhc_2.bw                     NaN                  NaN   \n",
       "25          ZraR_1_S3_R2.bw           bio-rep2 – R1      ZraR_2_S4_R1.bw   \n",
       "26       edl933furfinal2.bw                     NaN                  NaN   \n",
       "27          smsfurfinal2.bw                     NaN                  NaN   \n",
       "28       umn026furfinal2.bw                     NaN                  NaN   \n",
       "29            dps_expo_2.bw  stationary growth – R1         dps_sta_1.bw   \n",
       "30            fis_expo_2.bw  stationary growth – R1         fis_sta_1.bw   \n",
       "31           gyra_expo_2.bw  stationary growth – R1        gyra_sta_1.bw   \n",
       "32           gyrb_expo_2.bw  stationary growth – R1        gyrb_sta_1.bw   \n",
       "33            hns_expo_2.bw  stationary growth – R1         hns_sta_1.bw   \n",
       "34           hupa_expo_2.bw  stationary growth – R1        hupa_sta_1.bw   \n",
       "35           hupb_expo_2.bw  stationary growth – R1        hupb_sta_1.bw   \n",
       "36           infa_expo_2.bw  stationary growth – R1        infa_sta_1.bw   \n",
       "37           mukb_expo_2.bw  stationary growth – R1        mukb_sta_1.bw   \n",
       "38           rdgc_expo_2.bw  stationary growth – R1        rdgc_sta_1.bw   \n",
       "39            rob_expo_2.bw  stationary growth – R1         rob_sta_1.bw   \n",
       "40           stpa_expo_2.bw  stationary growth – R1        stpa_sta_1.bw   \n",
       "41           topa_expo_2.bw  stationary growth – R1        topa_sta_1.bw   \n",
       "42            tus_expo_2.bw  stationary growth – R1         tus_sta_1.bw   \n",
       "43        narl_nitrate_2.bw                     NaN                  NaN   \n",
       "44     puur_putrescine_2.bw                     NaN                  NaN   \n",
       "45          rcsa_oxida_2.bw                     NaN                  NaN   \n",
       "46      mg1655_ace_rpob2.bw             cra-KO – R1  cra_ko_ace_rpob1.bw   \n",
       "\n",
       "               BWcond2_2_name            BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                               \n",
       "0               bio-rep2 – R2   BaerR_R2_S32_R2.bw            NaN   \n",
       "1                         NaN                  NaN            NaN   \n",
       "2                    Fru – R2         cra_fru_2.bw       Gal – R1   \n",
       "3                    DPD – R2         fur_dpd_2.bw            NaN   \n",
       "4                         NaN                  NaN            NaN   \n",
       "5                         NaN                  NaN            NaN   \n",
       "6                         NaN                  NaN            NaN   \n",
       "7                         NaN                  NaN            NaN   \n",
       "8               bio-rep2 – R2      PhoB_2_S6_R2.bw            NaN   \n",
       "9                    DPD – R2         rpob_dpd2.bw            NaN   \n",
       "10                        NaN                  NaN            NaN   \n",
       "11                        NaN                  NaN            NaN   \n",
       "12                   pH7 – R2        ydci_ph7_2.bw       pH8 – R1   \n",
       "13                        NaN                  NaN            NaN   \n",
       "14                        NaN                  NaN            NaN   \n",
       "15                        NaN                  NaN            NaN   \n",
       "16                   3HP – R2        yiep_3hp_2.bw            NaN   \n",
       "17                        NaN                  NaN            NaN   \n",
       "18                        NaN                  NaN            NaN   \n",
       "19                        NaN                  NaN            NaN   \n",
       "20                        NaN                  NaN            NaN   \n",
       "21                        NaN                  NaN            NaN   \n",
       "22                        NaN                  NaN            NaN   \n",
       "23                        NaN                  NaN            NaN   \n",
       "24                        NaN                  NaN            NaN   \n",
       "25                        NaN                  NaN            NaN   \n",
       "26                        NaN                  NaN            NaN   \n",
       "27                        NaN                  NaN            NaN   \n",
       "28                        NaN                  NaN            NaN   \n",
       "29     stationary growth – R2         dps_sta_2.bw            NaN   \n",
       "30     stationary growth – R2         fis_sta_2.bw            NaN   \n",
       "31     stationary growth – R2        gyra_sta_2.bw            NaN   \n",
       "32     stationary growth – R2        gyrb_sta_2.bw            NaN   \n",
       "33     stationary growth – R2         hns_sta_2.bw            NaN   \n",
       "34     stationary growth – R2        hupa_sta_2.bw            NaN   \n",
       "35     stationary growth – R2        hupb_sta_2.bw            NaN   \n",
       "36     stationary growth – R2        infa_sta_2.bw            NaN   \n",
       "37     stationary growth – R2        mukb_sta_2.bw            NaN   \n",
       "38     stationary growth – R2        rdgc_sta_2.bw            NaN   \n",
       "39     stationary growth – R2         rob_sta_2.bw            NaN   \n",
       "40     stationary growth – R2        stpa_sta_2.bw            NaN   \n",
       "41     stationary growth – R2        topa_sta_2.bw            NaN   \n",
       "42     stationary growth – R2         tus_sta_2.bw            NaN   \n",
       "43                        NaN                  NaN            NaN   \n",
       "44                        NaN                  NaN            NaN   \n",
       "45                        NaN                  NaN            NaN   \n",
       "46                cra-KO – R2  cra_ko_ace_rpob2.bw    crp-KO – R1   \n",
       "\n",
       "                 BWcond3_1 BWcond3_2_name            BWcond3_2 BWcond4_1_name  \\\n",
       "index                                                                           \n",
       "0                      NaN            NaN                  NaN            NaN   \n",
       "1                      NaN            NaN                  NaN            NaN   \n",
       "2             cra_gal_1.bw       Gal – R2         cra_gal_2.bw       Ace – R1   \n",
       "3                      NaN            NaN                  NaN            NaN   \n",
       "4                      NaN            NaN                  NaN            NaN   \n",
       "5                      NaN            NaN                  NaN            NaN   \n",
       "6                      NaN            NaN                  NaN            NaN   \n",
       "7                      NaN            NaN                  NaN            NaN   \n",
       "8                      NaN            NaN                  NaN            NaN   \n",
       "9                      NaN            NaN                  NaN            NaN   \n",
       "10                     NaN            NaN                  NaN            NaN   \n",
       "11                     NaN            NaN                  NaN            NaN   \n",
       "12           ydci_ph8_1.bw       pH8 – R2        ydci_ph8_2.bw            NaN   \n",
       "13                     NaN            NaN                  NaN            NaN   \n",
       "14                     NaN            NaN                  NaN            NaN   \n",
       "15                     NaN            NaN                  NaN            NaN   \n",
       "16                     NaN            NaN                  NaN            NaN   \n",
       "17                     NaN            NaN                  NaN            NaN   \n",
       "18                     NaN            NaN                  NaN            NaN   \n",
       "19                     NaN            NaN                  NaN            NaN   \n",
       "20                     NaN            NaN                  NaN            NaN   \n",
       "21                     NaN            NaN                  NaN            NaN   \n",
       "22                     NaN            NaN                  NaN            NaN   \n",
       "23                     NaN            NaN                  NaN            NaN   \n",
       "24                     NaN            NaN                  NaN            NaN   \n",
       "25                     NaN            NaN                  NaN            NaN   \n",
       "26                     NaN            NaN                  NaN            NaN   \n",
       "27                     NaN            NaN                  NaN            NaN   \n",
       "28                     NaN            NaN                  NaN            NaN   \n",
       "29                     NaN            NaN                  NaN            NaN   \n",
       "30                     NaN            NaN                  NaN            NaN   \n",
       "31                     NaN            NaN                  NaN            NaN   \n",
       "32                     NaN            NaN                  NaN            NaN   \n",
       "33                     NaN            NaN                  NaN            NaN   \n",
       "34                     NaN            NaN                  NaN            NaN   \n",
       "35                     NaN            NaN                  NaN            NaN   \n",
       "36                     NaN            NaN                  NaN            NaN   \n",
       "37                     NaN            NaN                  NaN            NaN   \n",
       "38                     NaN            NaN                  NaN            NaN   \n",
       "39                     NaN            NaN                  NaN            NaN   \n",
       "40                     NaN            NaN                  NaN            NaN   \n",
       "41                     NaN            NaN                  NaN            NaN   \n",
       "42                     NaN            NaN                  NaN            NaN   \n",
       "43                     NaN            NaN                  NaN            NaN   \n",
       "44                     NaN            NaN                  NaN            NaN   \n",
       "45                     NaN            NaN                  NaN            NaN   \n",
       "46     crp_ko_ace_rpob1.bw    crp-KO – R2  crp_ko_ace_rpob2.bw            NaN   \n",
       "\n",
       "          BWcond4_1 BWcond4_2_name     BWcond4_2  \n",
       "index                                             \n",
       "0               NaN            NaN           NaN  \n",
       "1               NaN            NaN           NaN  \n",
       "2      cra_ace_1.bw       Ace – R2  cra_ace_2.bw  \n",
       "3               NaN            NaN           NaN  \n",
       "4               NaN            NaN           NaN  \n",
       "5               NaN            NaN           NaN  \n",
       "6               NaN            NaN           NaN  \n",
       "7               NaN            NaN           NaN  \n",
       "8               NaN            NaN           NaN  \n",
       "9               NaN            NaN           NaN  \n",
       "10              NaN            NaN           NaN  \n",
       "11              NaN            NaN           NaN  \n",
       "12              NaN            NaN           NaN  \n",
       "13              NaN            NaN           NaN  \n",
       "14              NaN            NaN           NaN  \n",
       "15              NaN            NaN           NaN  \n",
       "16              NaN            NaN           NaN  \n",
       "17              NaN            NaN           NaN  \n",
       "18              NaN            NaN           NaN  \n",
       "19              NaN            NaN           NaN  \n",
       "20              NaN            NaN           NaN  \n",
       "21              NaN            NaN           NaN  \n",
       "22              NaN            NaN           NaN  \n",
       "23              NaN            NaN           NaN  \n",
       "24              NaN            NaN           NaN  \n",
       "25              NaN            NaN           NaN  \n",
       "26              NaN            NaN           NaN  \n",
       "27              NaN            NaN           NaN  \n",
       "28              NaN            NaN           NaN  \n",
       "29              NaN            NaN           NaN  \n",
       "30              NaN            NaN           NaN  \n",
       "31              NaN            NaN           NaN  \n",
       "32              NaN            NaN           NaN  \n",
       "33              NaN            NaN           NaN  \n",
       "34              NaN            NaN           NaN  \n",
       "35              NaN            NaN           NaN  \n",
       "36              NaN            NaN           NaN  \n",
       "37              NaN            NaN           NaN  \n",
       "38              NaN            NaN           NaN  \n",
       "39              NaN            NaN           NaN  \n",
       "40              NaN            NaN           NaN  \n",
       "41              NaN            NaN           NaN  \n",
       "42              NaN            NaN           NaN  \n",
       "43              NaN            NaN           NaN  \n",
       "44              NaN            NaN           NaN  \n",
       "45              NaN            NaN           NaN  \n",
       "46              NaN            NaN           NaN  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4399</th>\n",
       "      <td>4636695</td>\n",
       "      <td>4638120</td>\n",
       "      <td>+</td>\n",
       "      <td>creC</td>\n",
       "      <td>1425</td>\n",
       "      <td>creABCD</td>\n",
       "      <td>Signal transduction mechanisms</td>\n",
       "      <td>4636695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4400</th>\n",
       "      <td>4638177</td>\n",
       "      <td>4639530</td>\n",
       "      <td>+</td>\n",
       "      <td>creD</td>\n",
       "      <td>1353</td>\n",
       "      <td>creABCD</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>4638177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4401</th>\n",
       "      <td>4639589</td>\n",
       "      <td>4640306</td>\n",
       "      <td>-</td>\n",
       "      <td>arcA</td>\n",
       "      <td>717</td>\n",
       "      <td>arcA</td>\n",
       "      <td>Signal transduction mechanisms</td>\n",
       "      <td>4640306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4402</th>\n",
       "      <td>4640401</td>\n",
       "      <td>4640542</td>\n",
       "      <td>+</td>\n",
       "      <td>yjjY</td>\n",
       "      <td>141</td>\n",
       "      <td>yjjY</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>4640401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4403</th>\n",
       "      <td>4640941</td>\n",
       "      <td>4641628</td>\n",
       "      <td>+</td>\n",
       "      <td>yjtD</td>\n",
       "      <td>687</td>\n",
       "      <td>yjtD</td>\n",
       "      <td>Translation, ribosomal structure and biogenesis</td>\n",
       "      <td>4640941</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4386 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         start     stop strand gene_name  length   operon  \\\n",
       "b0001      189      255      +      thrL      66  thrLABC   \n",
       "b0002      336     2799      +      thrA    2463  thrLABC   \n",
       "b0003     2800     3733      +      thrB     933  thrLABC   \n",
       "b0004     3733     5020      +      thrC    1287  thrLABC   \n",
       "b0005     5233     5530      +      yaaX     297     yaaX   \n",
       "...        ...      ...    ...       ...     ...      ...   \n",
       "b4399  4636695  4638120      +      creC    1425  creABCD   \n",
       "b4400  4638177  4639530      +      creD    1353  creABCD   \n",
       "b4401  4639589  4640306      -      arcA     717     arcA   \n",
       "b4402  4640401  4640542      +      yjjY     141     yjjY   \n",
       "b4403  4640941  4641628      +      yjtD     687     yjtD   \n",
       "\n",
       "                                                   cog  start_codon_pos  \n",
       "b0001                                No COG Annotation              189  \n",
       "b0002                                No COG Annotation              336  \n",
       "b0003              Amino acid transport and metabolism             2800  \n",
       "b0004              Amino acid transport and metabolism             3733  \n",
       "b0005                                 Function unknown             5233  \n",
       "...                                                ...              ...  \n",
       "b4399                   Signal transduction mechanisms          4636695  \n",
       "b4400                               Defense mechanisms          4638177  \n",
       "b4401                   Signal transduction mechanisms          4640306  \n",
       "b4402                                 Function unknown          4640401  \n",
       "b4403  Translation, ribosomal structure and biogenesis          4640941  \n",
       "\n",
       "[4386 rows x 8 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913_3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>condition_name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>.</th>\n",
       "      <th>strand</th>\n",
       "      <th>-</th>\n",
       "      <th>details</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>condition</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>190</td>\n",
       "      <td>255</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene thrL;locus_tag b0001;product thr_operon_l...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>337</td>\n",
       "      <td>2799</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene thrA;locus_tag b0002;product fused_aspart...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2801</td>\n",
       "      <td>3733</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene thrB;locus_tag b0003;product homoserine_k...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3734</td>\n",
       "      <td>5020</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene thrC;locus_tag b0004;product threonine_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>5234</td>\n",
       "      <td>5530</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene yaaX;locus_tag b0005;product DUF2502_doma...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4636696</td>\n",
       "      <td>4638120</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene creC;locus_tag b4399;product sensory_hist...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4638178</td>\n",
       "      <td>4639530</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene creD;locus_tag b4400;product putative_inn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4639590</td>\n",
       "      <td>4640306</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>gene arcA;locus_tag b4401;product DNA-binding_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4640402</td>\n",
       "      <td>4640542</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene yjjY;locus_tag b4402;product uncharacteri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000913.3</th>\n",
       "      <td>RefSeq</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4640942</td>\n",
       "      <td>4641628</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene yjtD;locus_tag b4403;product putative_rRN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4324 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ref condition_name    start     stop  . strand  -  \\\n",
       "condition                                                           \n",
       "NC_000913.3  RefSeq            CDS      190      255  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS      337     2799  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS     2801     3733  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS     3734     5020  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS     5234     5530  .      +  0   \n",
       "...             ...            ...      ...      ... ..    ... ..   \n",
       "NC_000913.3  RefSeq            CDS  4636696  4638120  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS  4638178  4639530  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS  4639590  4640306  .      -  0   \n",
       "NC_000913.3  RefSeq            CDS  4640402  4640542  .      +  0   \n",
       "NC_000913.3  RefSeq            CDS  4640942  4641628  .      +  0   \n",
       "\n",
       "                                                       details  \n",
       "condition                                                       \n",
       "NC_000913.3  gene thrL;locus_tag b0001;product thr_operon_l...  \n",
       "NC_000913.3  gene thrA;locus_tag b0002;product fused_aspart...  \n",
       "NC_000913.3  gene thrB;locus_tag b0003;product homoserine_k...  \n",
       "NC_000913.3  gene thrC;locus_tag b0004;product threonine_sy...  \n",
       "NC_000913.3  gene yaaX;locus_tag b0005;product DUF2502_doma...  \n",
       "...                                                        ...  \n",
       "NC_000913.3  gene creC;locus_tag b4399;product sensory_hist...  \n",
       "NC_000913.3  gene creD;locus_tag b4400;product putative_inn...  \n",
       "NC_000913.3  gene arcA;locus_tag b4401;product DNA-binding_...  \n",
       "NC_000913.3  gene yjjY;locus_tag b4402;product uncharacteri...  \n",
       "NC_000913.3  gene yjtD;locus_tag b4403;product putative_rRN...  \n",
       "\n",
       "[4324 rows x 8 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913_3'\n",
    "genes = pd.read_csv(org_folder+strain+'/annotation/genes.gff',index_col=0,\n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['condition', 'ref', 'condition_name', \n",
    "                                          \"start\",'stop', \n",
    "                                          '.', 'strand', '-','details'])\n",
    "\n",
    "genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    else:\n",
    "        TF_list.loc[TF_index, 'binding tables'] = \"NA\"\n",
    "    TF_list.loc[TF_index, 'num_cond'] = max(1,len(curated_loc))\n",
    "    \n",
    "    #write plot data files to folders\n",
    "#     for file_index in range(len(curated_files)):\n",
    "#         try:\n",
    "#             file = curated_files[file_index]\n",
    "#             df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "#                                      delimiter='\\t', header=None, \n",
    "#                                      names = ['ref','condition', 'condition_name', \n",
    "#                                               \"binding_peak_start\",'binding_peak_end', \n",
    "#                                               'binding_peak_strength', 'direction', '.','ID'])\n",
    "#             df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "#             peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "#             final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "#             final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "#             binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "#             peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "#         except:\n",
    "#             print(TF_name, file)\n",
    "#             continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range (0, len(database)): \n",
    "    peak_annot_DF = annotate_peaks(TF_list[i], TF_condition[i],database[i],500,DF_gene_info).fillna('')\n",
    "    final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "    venn_files = Venn_data_gen(final_annot_DF)\n",
    "    final_annot_DF.to_json(out_data_table+TF_list[i]+'_binding_table.json',orient='records')\n",
    "    venn_files.to_json(path.join(out_data_venn, TF_list[i]+'_venn.json'),orient='records')\n",
    "    binding_width_gen(TF_list[i], database[i], '../../data/e_coli/NC_000913_3/binding_widths/')\n",
    "    peak_position_gen(TF_list[i], final_annot_DF, '../../data/e_coli/NC_000913_3/positions/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = 'phob_M9P_curated.gff'\n",
    "df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Other Organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/all_other/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP_000647_1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002947_4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002947_4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_003197_2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_003197_2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE_014073_1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP_035369_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010079_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010079_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NZ_CP009712_1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism           Strain  Media  \\\n",
       "index                                                                 \n",
       "0          Fur        Klebsiella pneumoniae        MGH 78578    NaN   \n",
       "0          Fur           Pseudomonas putida           KT2440    NaN   \n",
       "1      Sigma D           Pseudomonas putida           KT2440    NaN   \n",
       "0          Fur          Salmonella enterica  Typhimurium LT2    NaN   \n",
       "1         YdcI          Salmonella enterica  Typhimurium LT2    NaN   \n",
       "0          Fur            Shigella flexneri        2a, 2457T    NaN   \n",
       "0         CodY        Staphylococcus aureus   LAC chromosome    NaN   \n",
       "1         SigS        Staphylococcus aureus   USA300_TCH1516    NaN   \n",
       "2         VraR        Staphylococcus aureus   USA300_TCH1516    NaN   \n",
       "0          Fur  Yersinia pseudotuberculosis         IP 32953    NaN   \n",
       "\n",
       "      Supplement      genome_id           organism_id  num_binding_sites  \\\n",
       "index                                                                      \n",
       "0            NaN    CP_000647_1          k_pneumoniae                NaN   \n",
       "0            NaN    NC_002947_4              p_putida                NaN   \n",
       "1            NaN    NC_002947_4              p_putida                NaN   \n",
       "0             Fe    NC_003197_2            s_enterica                NaN   \n",
       "1            NaN    NC_003197_2            s_enterica                NaN   \n",
       "0            NaN    AE_014073_1            s_flexneri                NaN   \n",
       "0            NaN    CP_035369_1              s_aureus                NaN   \n",
       "1            NaN    NC_010079_1              s_aureus                NaN   \n",
       "2            NaN    NC_010079_1              s_aureus                NaN   \n",
       "0            NaN  NZ_CP009712_1  y_pseudotuberculosis                NaN   \n",
       "\n",
       "       num_samples  num_cond         project  binding tables BWcond1_1_name  \\\n",
       "index                                                                         \n",
       "0                2         1        MGH78578             NaN             R1   \n",
       "0                1         1           2457T             NaN             R1   \n",
       "1                2         1           2457T             NaN             R1   \n",
       "0                1         1             LT2             NaN            NaN   \n",
       "1                6         3             LT2             NaN      MES5 – R1   \n",
       "0                2         1           2457T             NaN             R1   \n",
       "0                2         1             LAC             NaN             R1   \n",
       "1                8         4  USA300_TCH1516             NaN        M1 – R1   \n",
       "2                2         1  USA300_TCH1516             NaN             R1   \n",
       "0                2         1           2457T             NaN             R1   \n",
       "\n",
       "                             BWcond1_1 BWcond1_2_name  \\\n",
       "index                                                   \n",
       "0                     klebfurfinal1.bw             R2   \n",
       "0            putidafurfinal1_202003.bw            NaN   \n",
       "1               putidaSigmaD_202003.bw             R2   \n",
       "0                                  NaN             R2   \n",
       "1               salmone_ydcI_MES5_1.bw      MES5 – R2   \n",
       "0                    shigella_fur_1.bw             R2   \n",
       "0                StaphLAC_new_cody1.bw             R2   \n",
       "1                  StaphTCH_sigs_M1.bw        M1 – R2   \n",
       "2      StaphTCH_vraR_7H7C11purified.bw             R2   \n",
       "0                    yersifurfinal1.bw             R2   \n",
       "\n",
       "                              BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                    \n",
       "0                      klebfurfinal2.bw            NaN   \n",
       "0                                   NaN            NaN   \n",
       "1                putidaSigmaD_202005.bw            NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw            NaN   \n",
       "1                salmone_ydcI_MES5_2.bw      MES5 – R3   \n",
       "0                     shigella_fur_2.bw            NaN   \n",
       "0                 StaphLAC_new_cody2.bw            NaN   \n",
       "1                   Staph_SigS_M1_R2.bw        M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw            NaN   \n",
       "0                     yersifurfinal2.bw            NaN   \n",
       "\n",
       "                    BWcond2_1 BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                  \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw      MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw        M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "\n",
       "      BWcond3_1_name               BWcond3_1 BWcond3_2_name  \\\n",
       "index                                                         \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1           ph5 – R1  salmone_ydcI_ph55_1.bw       ph5 – R2   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1            M3 – R1     StaphTCH_sigs_M3.bw        M3 – R2   \n",
       "2                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rpob rpob_dpd_curated.gff\n",
      "rpob rpob_fe_curated.gff\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yiag yiag_M9_curated.gff\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value)\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    \n",
    "#     write plot data files to folders\n",
    "    for file_index in range(len(curated_files)):\n",
    "        try:\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "            final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "            final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "            peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "        except:\n",
    "            print(TF_name, file)\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP_000647_1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>NA</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002947_4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NA</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002947_4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>M9|sigma d_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_003197_2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_003197_2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE_014073_1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NA</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP_035369_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>M9|cody_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010079_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NA</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010079_1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NA</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NZ_CP009712_1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>NA</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism           Strain  Media  \\\n",
       "index                                                                 \n",
       "0          Fur        Klebsiella pneumoniae        MGH 78578    NaN   \n",
       "0          Fur           Pseudomonas putida           KT2440    NaN   \n",
       "1      Sigma D           Pseudomonas putida           KT2440    NaN   \n",
       "0          Fur          Salmonella enterica  Typhimurium LT2    NaN   \n",
       "1         YdcI          Salmonella enterica  Typhimurium LT2    NaN   \n",
       "0          Fur            Shigella flexneri        2a, 2457T    NaN   \n",
       "0         CodY        Staphylococcus aureus   LAC chromosome    NaN   \n",
       "1         SigS        Staphylococcus aureus   USA300_TCH1516    NaN   \n",
       "2         VraR        Staphylococcus aureus   USA300_TCH1516    NaN   \n",
       "0          Fur  Yersinia pseudotuberculosis         IP 32953    NaN   \n",
       "\n",
       "      Supplement      genome_id           organism_id  num_binding_sites  \\\n",
       "index                                                                      \n",
       "0            NaN    CP_000647_1          k_pneumoniae                NaN   \n",
       "0            NaN    NC_002947_4              p_putida                NaN   \n",
       "1            NaN    NC_002947_4              p_putida                NaN   \n",
       "0             Fe    NC_003197_2            s_enterica                NaN   \n",
       "1            NaN    NC_003197_2            s_enterica                NaN   \n",
       "0            NaN    AE_014073_1            s_flexneri                NaN   \n",
       "0            NaN    CP_035369_1              s_aureus                NaN   \n",
       "1            NaN    NC_010079_1              s_aureus                NaN   \n",
       "2            NaN    NC_010079_1              s_aureus                NaN   \n",
       "0            NaN  NZ_CP009712_1  y_pseudotuberculosis                NaN   \n",
       "\n",
       "       num_samples  num_cond         project  \\\n",
       "index                                          \n",
       "0                2         1        MGH78578   \n",
       "0                1         1           2457T   \n",
       "1                2         1           2457T   \n",
       "0                1         1             LT2   \n",
       "1                6         3             LT2   \n",
       "0                2         1           2457T   \n",
       "0                2         1             LAC   \n",
       "1                8         1  USA300_TCH1516   \n",
       "2                2         1  USA300_TCH1516   \n",
       "0                2         1           2457T   \n",
       "\n",
       "                                          binding tables BWcond1_1_name  \\\n",
       "index                                                                     \n",
       "0                                                     NA             R1   \n",
       "0                                                     NA             R1   \n",
       "1                       M9|sigma d_M9_binding_table.json             R1   \n",
       "0                                                     NA            NaN   \n",
       "1      MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...      MES5 – R1   \n",
       "0                                                     NA             R1   \n",
       "0                          M9|cody_M9_binding_table.json             R1   \n",
       "1                                                     NA        M1 – R1   \n",
       "2                                                     NA             R1   \n",
       "0                                                     NA             R1   \n",
       "\n",
       "                             BWcond1_1 BWcond1_2_name  \\\n",
       "index                                                   \n",
       "0                     klebfurfinal1.bw             R2   \n",
       "0            putidafurfinal1_202003.bw            NaN   \n",
       "1               putidaSigmaD_202003.bw             R2   \n",
       "0                                  NaN             R2   \n",
       "1               salmone_ydcI_MES5_1.bw      MES5 – R2   \n",
       "0                    shigella_fur_1.bw             R2   \n",
       "0                StaphLAC_new_cody1.bw             R2   \n",
       "1                  StaphTCH_sigs_M1.bw        M1 – R2   \n",
       "2      StaphTCH_vraR_7H7C11purified.bw             R2   \n",
       "0                    yersifurfinal1.bw             R2   \n",
       "\n",
       "                              BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                    \n",
       "0                      klebfurfinal2.bw            NaN   \n",
       "0                                   NaN            NaN   \n",
       "1                putidaSigmaD_202005.bw            NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw            NaN   \n",
       "1                salmone_ydcI_MES5_2.bw      MES5 – R3   \n",
       "0                     shigella_fur_2.bw            NaN   \n",
       "0                 StaphLAC_new_cody2.bw            NaN   \n",
       "1                   Staph_SigS_M1_R2.bw        M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw            NaN   \n",
       "0                     yersifurfinal2.bw            NaN   \n",
       "\n",
       "                    BWcond2_1 BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                  \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw      MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw        M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "\n",
       "      BWcond3_1_name               BWcond3_1 BWcond3_2_name  \\\n",
       "index                                                         \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1           ph5 – R1  salmone_ydcI_ph55_1.bw       ph5 – R2   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1            M3 – R1     StaphTCH_sigs_M3.bw        M3 – R2   \n",
       "2                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "283.4px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
