{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for ChIP-pro, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "            if len(operon) > 1:\n",
    "                print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start > BP) & (stop > BP) & (strand == '+')) | ((start < BP) & (stop < BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start < BP) & (stop > BP)) | ((start > BP) & (stop < BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "                    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen(Peak_DF): \n",
    "    TF_name = Peak_DF['index'][1][:4]\n",
    "    gene_list = [i for i in Peak_DF.target_genes if i != '']\n",
    "    chip_data= ','.join(list(gene_list)).split(',')\n",
    "    reg_data = TRN_data[TF_name]\n",
    "    all_genes = [i  for i in chip_data if i not in reg_data] + reg_data\n",
    "    \n",
    "    \n",
    "    TF = TF_name\n",
    "    reg_genes=reg_data\n",
    "    reg_only = []\n",
    "    chip_genes=chip_data\n",
    "    chip_only = []\n",
    "    shared_genes=[]\n",
    "    for i in all_genes: \n",
    "        if (i in reg_data) & (i not in chip_data):\n",
    "            reg_only.append(i)\n",
    "        elif (i in chip_data) & (i not in reg_data):\n",
    "            chip_only.append(i)\n",
    "        elif (i in chip_data) & (i in reg_data):\n",
    "            shared_genes.append(i)\n",
    "            \n",
    "    values = [TF,\n",
    "          len(reg_genes),\n",
    "          len(reg_only),\n",
    "          len(chip_genes),\n",
    "          len(chip_only),\n",
    "          len(shared_genes),\n",
    "          len(all_genes)]\n",
    "    \n",
    "    index_name = ['TF',\n",
    "              'reg_genes',\n",
    "              'reg_only',\n",
    "              'chip_genes',\n",
    "              'chip_only',\n",
    "              'shared_genes',\n",
    "              'all_genes']\n",
    "    \n",
    "    genes = ['; '.join(precise2_TRN.source[precise2_TRN.index == TF_name].unique()),\n",
    "         reg_genes,\n",
    "         reg_only,\n",
    "         chip_genes,\n",
    "         chip_only,\n",
    "         shared_genes,\n",
    "         all_genes]\n",
    "    \n",
    "    same1 = defaultdict(list)\n",
    "    for i in range(0,len(index_name)):\n",
    "        same1[index_name[i]].append(values[i])\n",
    "\n",
    "    finall = pd.DataFrame.from_dict(same1, orient='index', columns = ['value'])\n",
    "    finall['list'] = genes\n",
    "    for i, row in finall.iterrows(): \n",
    "        if row.value == 0:\n",
    "            finall.list[i] = ''\n",
    "    return finall"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "            try:\n",
    "                if gene_info.strand[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>EtOH</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_curated.gff</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>EtOH</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fructose, Galactose, or Acetate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>acetate|cra_acetate_curated.gff;M9|cra_M9_cura...</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe or DPD</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>fe|fur_fe_curated.gff;dpd|fur_dpd_curated.gff</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GadW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadw_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GadX</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadx_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadx_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadx_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KdpE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>TMA</td>\n",
       "      <td>KCl</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>KCl|kdpe_KCl_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>KdpeE_R2_S34_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>KdpeE_R2_S34_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>M9P|phob_M9P_curated.gff</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>PhoB_1_S16_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>PhoB_1_S16_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>PhoB_2_S6_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>PhoB_2_S6_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>RpoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe or DPD</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>fe|rpob_fe_curated.gff;dpd|rpob_dpd_curated.gff</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>rpob_fe1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>rpob_fe2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>rpob_dpd1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>rpob_dpd2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YbaO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybao_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybao_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybao_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>YbaQ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>41.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybaq_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybaq_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybaq_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>40.0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>ph7|ydci_ph7_curated.gff;ph8|ydci_ph8_curated....</td>\n",
       "      <td>pH5 – R1</td>\n",
       "      <td>ydci_ph5_1.bw</td>\n",
       "      <td>pH5 – R2</td>\n",
       "      <td>ydci_ph5_2.bw</td>\n",
       "      <td>pH7 – R1</td>\n",
       "      <td>ydci_ph7_1.bw</td>\n",
       "      <td>pH7 – R2</td>\n",
       "      <td>ydci_ph7_2.bw</td>\n",
       "      <td>pH8 – R1</td>\n",
       "      <td>ydci_ph8_1.bw</td>\n",
       "      <td>pH8 – R2</td>\n",
       "      <td>ydci_ph8_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>YddM</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yddm_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yddm_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yddm_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YheO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yheo_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yheo_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yheo_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>YiaG</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiag_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiag_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiag_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>YieP</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>3HP</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiep_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiep_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiep_2.bw</td>\n",
       "      <td>3HP – R1</td>\n",
       "      <td>yiep_3hp_1.bw</td>\n",
       "      <td>3HP – R2</td>\n",
       "      <td>yiep_3hp_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>YihL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>YihW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>YjdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yjdc_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>YjhI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjhi_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjhi_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>YneJ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynej_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynej_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>YnfL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynfl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynfl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>YpdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ypdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ypdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>YqhC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>M9|yqhc_M9_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>yqhc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yqhc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ZraR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>ZnCl2</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>TCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>ZraR_1_S3_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>ZraR_1_S3_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>ZraR_2_S4_R1.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>O157:H7 EDL933</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE_005174_2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>EDL933</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>edl933furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>edl933furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_010498_1</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>smsfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>smsfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CU_928163_2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>umn026furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>umn026furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Dps</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|dps_sta_curated.gff;expo|dps_expo_curated.gff</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>dps_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>dps_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>dps_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>dps_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Fis</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|fis_expo_curated.gff;sta|fis_sta_curated.gff</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>fis_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>fis_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>fis_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>fis_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GyrA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|gyra_sta_curated.gff;expo|gyra_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyra_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyra_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyra_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyra_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GyrB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|gyrb_expo_curated.gff;sta|gyrb_sta_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyrb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyrb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyrb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyrb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>H-NS</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|h-ns_expo_curated.gff;sta|h-ns_sta_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hns_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hns_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hns_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hns_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>HupA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|hupa_sta_curated.gff;expo|hupa_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>HupB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|hupb_sta_curated.gff;expo|hupb_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>InfA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|infa_sta_curated.gff;expo|infa_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>infa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>infa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>infa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>infa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MukB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|mukb_sta_curated.gff;expo|mukb_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>mukb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>mukb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>mukb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>mukb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>RdgC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|rdgc_sta_curated.gff;expo|rdgc_expo_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rdgc_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rdgc_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rdgc_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rdgc_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Rob</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|rob_expo_curated.gff;sta|rob_sta_curated.gff</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rob_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rob_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rob_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rob_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>StpA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|stpa_expo_curated.gff;sta|stpa_sta_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>stpa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>stpa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>stpa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>stpa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TopA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|topa_expo_curated.gff;sta|topa_sta_curate...</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>topa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>topa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>topa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>topa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Tus</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>sta|tus_sta_curated.gff;expo|tus_expo_curated.gff</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>tus_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>tus_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>tus_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>tus_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NarL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Nitrate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>nitrate|narl_nitrate_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>narl_nitrate_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>narl_nitrate_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PuuR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Putrescine</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>putrescine|puur_putrescine_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>puur_putrescine_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>puur_putrescine_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>RcsA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Oxida</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>oxida|rcsa_oxida_curated.gff</td>\n",
       "      <td>R1</td>\n",
       "      <td>rcsa_oxida_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>rcsa_oxida_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>RpoB (Cra and Crp KO exps)</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Acetate</td>\n",
       "      <td>NC_000913_3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>Misc</td>\n",
       "      <td>crp-ko ace|rpob (cra and crp ko exps)_crp-ko a...</td>\n",
       "      <td>control (WT) – R1</td>\n",
       "      <td>mg1655_ace_rpob1.bw</td>\n",
       "      <td>control (WT) – R2</td>\n",
       "      <td>mg1655_ace_rpob2.bw</td>\n",
       "      <td>cra-KO – R1</td>\n",
       "      <td>cra_ko_ace_rpob1.bw</td>\n",
       "      <td>cra-KO – R2</td>\n",
       "      <td>cra_ko_ace_rpob2.bw</td>\n",
       "      <td>crp-KO – R1</td>\n",
       "      <td>crp_ko_ace_rpob1.bw</td>\n",
       "      <td>crp-KO – R2</td>\n",
       "      <td>crp_ko_ace_rpob2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               TF          Organism          Strain Media  \\\n",
       "index                                                                       \n",
       "0                            BaeR  Escherichia coli     K-12 MG1655    LB   \n",
       "1                            CpxR  Escherichia coli     K-12 MG1655    LB   \n",
       "2                             Cra  Escherichia coli     K-12 MG1655    M9   \n",
       "3                             Fur  Escherichia coli     K-12 MG1655    M9   \n",
       "4                            GadE  Escherichia coli     K-12 MG1655    M9   \n",
       "5                            GadW  Escherichia coli     K-12 MG1655    M9   \n",
       "6                            GadX  Escherichia coli     K-12 MG1655    M9   \n",
       "7                            KdpE  Escherichia coli     K-12 MG1655   TMA   \n",
       "8                            PhoB  Escherichia coli     K-12 MG1655   M9P   \n",
       "9                            RpoB  Escherichia coli     K-12 MG1655    M9   \n",
       "10                           YbaO  Escherichia coli     K-12 MG1655    M9   \n",
       "11                           YbaQ  Escherichia coli     K-12 MG1655    M9   \n",
       "12                           YdcI  Escherichia coli     K-12 MG1655    M9   \n",
       "13                           YddM  Escherichia coli     K-12 MG1655    M9   \n",
       "14                           YheO  Escherichia coli     K-12 MG1655    M9   \n",
       "15                           YiaG  Escherichia coli     K-12 MG1655    M9   \n",
       "16                           YieP  Escherichia coli     K-12 MG1655    M9   \n",
       "17                           YihL  Escherichia coli     K-12 MG1655    M9   \n",
       "18                           YihW  Escherichia coli     K-12 MG1655    M9   \n",
       "19                           YjdC  Escherichia coli     K-12 MG1655    M9   \n",
       "20                           YjhI  Escherichia coli     K-12 MG1655    M9   \n",
       "21                           YneJ  Escherichia coli     K-12 MG1655    M9   \n",
       "22                           YnfL  Escherichia coli     K-12 MG1655    M9   \n",
       "23                           YpdC  Escherichia coli     K-12 MG1655    M9   \n",
       "24                           YqhC  Escherichia coli     K-12 MG1655    M9   \n",
       "25                           ZraR  Escherichia coli     K-12 MG1655    LB   \n",
       "26                            Fur  Escherichia coli  O157:H7 EDL933    M9   \n",
       "27                            Fur  Escherichia coli         SMS-3-5    M9   \n",
       "28                            Fur  Escherichia coli          UMN026    M9   \n",
       "29                            Dps  Escherichia coli     K-12 MG1655    M9   \n",
       "30                            Fis  Escherichia coli     K-12 MG1655    M9   \n",
       "31                           GyrA  Escherichia coli     K-12 MG1655    M9   \n",
       "32                           GyrB  Escherichia coli     K-12 MG1655    M9   \n",
       "33                           H-NS  Escherichia coli     K-12 MG1655    M9   \n",
       "34                           HupA  Escherichia coli     K-12 MG1655    M9   \n",
       "35                           HupB  Escherichia coli     K-12 MG1655    M9   \n",
       "36                           InfA  Escherichia coli     K-12 MG1655    M9   \n",
       "37                           MukB  Escherichia coli     K-12 MG1655    M9   \n",
       "38                           RdgC  Escherichia coli     K-12 MG1655    M9   \n",
       "39                            Rob  Escherichia coli     K-12 MG1655    M9   \n",
       "40                           StpA  Escherichia coli     K-12 MG1655    M9   \n",
       "41                           TopA  Escherichia coli     K-12 MG1655    M9   \n",
       "42                            Tus  Escherichia coli     K-12 MG1655    M9   \n",
       "43                           NarL  Escherichia coli     K-12 MG1655    M9   \n",
       "44                           PuuR  Escherichia coli     K-12 MG1655    M9   \n",
       "45                           RcsA  Escherichia coli     K-12 MG1655    M9   \n",
       "46     RpoB (Cra and Crp KO exps)  Escherichia coli     K-12 MG1655    M9   \n",
       "\n",
       "                            Supplement    genome_id organism_id  \\\n",
       "index                                                             \n",
       "0                                 EtOH  NC_000913_3      e_coli   \n",
       "1                                 EtOH  NC_000913_3      e_coli   \n",
       "2      Fructose, Galactose, or Acetate  NC_000913_3      e_coli   \n",
       "3                            Fe or DPD  NC_000913_3      e_coli   \n",
       "4                                  NaN  NC_000913_3      e_coli   \n",
       "5                                  NaN  NC_000913_3      e_coli   \n",
       "6                                  NaN  NC_000913_3      e_coli   \n",
       "7                                  KCl  NC_000913_3      e_coli   \n",
       "8                                  NaN  NC_000913_3      e_coli   \n",
       "9                            Fe or DPD  NC_000913_3      e_coli   \n",
       "10                                 NaN  NC_000913_3      e_coli   \n",
       "11                                 NaN  NC_000913_3      e_coli   \n",
       "12                                 NaN  NC_000913_3      e_coli   \n",
       "13                                 NaN  NC_000913_3      e_coli   \n",
       "14                                 NaN  NC_000913_3      e_coli   \n",
       "15                                 NaN  NC_000913_3      e_coli   \n",
       "16                                 3HP  NC_000913_3      e_coli   \n",
       "17                                 NaN  NC_000913_3      e_coli   \n",
       "18                                 NaN  NC_000913_3      e_coli   \n",
       "19                                 NaN  NC_000913_3      e_coli   \n",
       "20                                 NaN  NC_000913_3      e_coli   \n",
       "21                                 NaN  NC_000913_3      e_coli   \n",
       "22                                 NaN  NC_000913_3      e_coli   \n",
       "23                                 NaN  NC_000913_3      e_coli   \n",
       "24                                 NaN  NC_000913_3      e_coli   \n",
       "25                               ZnCl2  NC_000913_3      e_coli   \n",
       "26                                 NaN  AE_005174_2      e_coli   \n",
       "27                                 NaN  NC_010498_1      e_coli   \n",
       "28                                 NaN  CU_928163_2      e_coli   \n",
       "29                                 NaN  NC_000913_3      e_coli   \n",
       "30                                 NaN  NC_000913_3      e_coli   \n",
       "31                                 NaN  NC_000913_3      e_coli   \n",
       "32                                 NaN  NC_000913_3      e_coli   \n",
       "33                                 NaN  NC_000913_3      e_coli   \n",
       "34                                 NaN  NC_000913_3      e_coli   \n",
       "35                                 NaN  NC_000913_3      e_coli   \n",
       "36                                 NaN  NC_000913_3      e_coli   \n",
       "37                                 NaN  NC_000913_3      e_coli   \n",
       "38                                 NaN  NC_000913_3      e_coli   \n",
       "39                                 NaN  NC_000913_3      e_coli   \n",
       "40                                 NaN  NC_000913_3      e_coli   \n",
       "41                                 NaN  NC_000913_3      e_coli   \n",
       "42                                 NaN  NC_000913_3      e_coli   \n",
       "43                             Nitrate  NC_000913_3      e_coli   \n",
       "44                          Putrescine  NC_000913_3      e_coli   \n",
       "45                               Oxida  NC_000913_3      e_coli   \n",
       "46                             Acetate  NC_000913_3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond       project  \\\n",
       "index                                                           \n",
       "0                    NaN            4         1           TCS   \n",
       "1                    NaN            2         1           TCS   \n",
       "2                    NaN            8         4      known_TF   \n",
       "3                    NaN            4         2      known_TF   \n",
       "4                    NaN            2         1      known_TF   \n",
       "5                    NaN            2         1      known_TF   \n",
       "6                    NaN            2         1      known_TF   \n",
       "7                    NaN            2         1           TCS   \n",
       "8                    NaN            4         1           TCS   \n",
       "9                    NaN            4         2      known_TF   \n",
       "10                   4.0            2         1          yTF1   \n",
       "11                  41.0            2         1          yTF1   \n",
       "12                  40.0            6         3          yTF1   \n",
       "13                  18.0            2         1          yTF1   \n",
       "14                   4.0            2         1          yTF1   \n",
       "15                   NaN            2         1          yTF1   \n",
       "16                  23.0            4         1          yTF1   \n",
       "17                   NaN            2         0          yTF2   \n",
       "18                   NaN            2         0          yTF2   \n",
       "19                   NaN            2         1          yTF1   \n",
       "20                   NaN            2         0          yTF2   \n",
       "21                   NaN            2         0          yTF2   \n",
       "22                   NaN            2         0          yTF2   \n",
       "23                   NaN            2         0          yTF2   \n",
       "24                   NaN            2         1          yTF2   \n",
       "25                   NaN            3         0           TCS   \n",
       "26                   NaN            2         0        EDL933   \n",
       "27                   NaN            2         0       SMS-3-5   \n",
       "28                   NaN            2         0        UMN026   \n",
       "29                   NaN            4         2  conformation   \n",
       "30                   NaN            4         2  conformation   \n",
       "31                   NaN            4         2  conformation   \n",
       "32                   NaN            4         2  conformation   \n",
       "33                   NaN            4         2  conformation   \n",
       "34                   NaN            4         2  conformation   \n",
       "35                   NaN            4         2  conformation   \n",
       "36                   NaN            4         2  conformation   \n",
       "37                   NaN            4         2  conformation   \n",
       "38                   NaN            4         2  conformation   \n",
       "39                   NaN            4         2  conformation   \n",
       "40                   NaN            4         2  conformation   \n",
       "41                   NaN            4         2  conformation   \n",
       "42                   NaN            4         2  conformation   \n",
       "43                   NaN            2         1          Misc   \n",
       "44                   NaN            2         1          Misc   \n",
       "45                   NaN            2         1          Misc   \n",
       "46                   NaN            6         3          Misc   \n",
       "\n",
       "                                          binding tables  \\\n",
       "index                                                      \n",
       "0                             EtOH|baer_EtOH_curated.gff   \n",
       "1                             EtOH|cpxr_EtOH_curated.gff   \n",
       "2      acetate|cra_acetate_curated.gff;M9|cra_M9_cura...   \n",
       "3          fe|fur_fe_curated.gff;dpd|fur_dpd_curated.gff   \n",
       "4                                 M9|gade_M9_curated.gff   \n",
       "5                                 M9|gadw_M9_curated.gff   \n",
       "6                                 M9|gadx_M9_curated.gff   \n",
       "7                               KCl|kdpe_KCl_curated.gff   \n",
       "8                               M9P|phob_M9P_curated.gff   \n",
       "9        fe|rpob_fe_curated.gff;dpd|rpob_dpd_curated.gff   \n",
       "10                                M9|ybao_M9_curated.gff   \n",
       "11                                M9|ybaq_M9_curated.gff   \n",
       "12     ph7|ydci_ph7_curated.gff;ph8|ydci_ph8_curated....   \n",
       "13                                M9|yddm_M9_curated.gff   \n",
       "14                                M9|yheo_M9_curated.gff   \n",
       "15                                M9|yiag_M9_curated.gff   \n",
       "16                                M9|yiep_M9_curated.gff   \n",
       "17                                                   NaN   \n",
       "18                                                   NaN   \n",
       "19                                M9|yjdc_M9_curated.gff   \n",
       "20                                                   NaN   \n",
       "21                                                   NaN   \n",
       "22                                                   NaN   \n",
       "23                                                   NaN   \n",
       "24                                M9|yqhc_M9_curated.gff   \n",
       "25                                                   NaN   \n",
       "26                                                   NaN   \n",
       "27                                                   NaN   \n",
       "28                                                   NaN   \n",
       "29     sta|dps_sta_curated.gff;expo|dps_expo_curated.gff   \n",
       "30     expo|fis_expo_curated.gff;sta|fis_sta_curated.gff   \n",
       "31     sta|gyra_sta_curated.gff;expo|gyra_expo_curate...   \n",
       "32     expo|gyrb_expo_curated.gff;sta|gyrb_sta_curate...   \n",
       "33     expo|h-ns_expo_curated.gff;sta|h-ns_sta_curate...   \n",
       "34     sta|hupa_sta_curated.gff;expo|hupa_expo_curate...   \n",
       "35     sta|hupb_sta_curated.gff;expo|hupb_expo_curate...   \n",
       "36     sta|infa_sta_curated.gff;expo|infa_expo_curate...   \n",
       "37     sta|mukb_sta_curated.gff;expo|mukb_expo_curate...   \n",
       "38     sta|rdgc_sta_curated.gff;expo|rdgc_expo_curate...   \n",
       "39     expo|rob_expo_curated.gff;sta|rob_sta_curated.gff   \n",
       "40     expo|stpa_expo_curated.gff;sta|stpa_sta_curate...   \n",
       "41     expo|topa_expo_curated.gff;sta|topa_sta_curate...   \n",
       "42     sta|tus_sta_curated.gff;expo|tus_expo_curated.gff   \n",
       "43                      nitrate|narl_nitrate_curated.gff   \n",
       "44                putrescine|puur_putrescine_curated.gff   \n",
       "45                          oxida|rcsa_oxida_curated.gff   \n",
       "46     crp-ko ace|rpob (cra and crp ko exps)_crp-ko a...   \n",
       "\n",
       "                BWcond1_1_name             BWcond1_1           BWcond1_2_name  \\\n",
       "index                                                                           \n",
       "0                bio-rep1 – R1     BaeR_R1_S31_R1.bw            bio-rep1 – R2   \n",
       "1                           R1       CpxRR1_S1_R1.bw                       R2   \n",
       "2                     Glu – R1          cra_glu_1.bw                 Glu – R2   \n",
       "3                      Fe – R1           fur_fe_1.bw                  Fe – R2   \n",
       "4                           R1             gade_1.bw                       R2   \n",
       "5                           R1             gadw_1.bw                       R2   \n",
       "6                           R1             gadx_1.bw                       R2   \n",
       "7                           R1    KdpeE_R2_S34_R1.bw                       R2   \n",
       "8                bio-rep1 – R1      PhoB_1_S16_R1.bw            bio-rep1 – R2   \n",
       "9                      Fe – R1           rpob_fe1.bw                  Fe – R2   \n",
       "10                          R1             ybao_1.bw                       R2   \n",
       "11                          R1             ybaq_1.bw                       R2   \n",
       "12                    pH5 – R1         ydci_ph5_1.bw                 pH5 – R2   \n",
       "13                          R1             yddm_1.bw                       R2   \n",
       "14                          R1             yheo_1.bw                       R2   \n",
       "15                          R1             yiag_1.bw                       R2   \n",
       "16                          R1             yiep_1.bw                       R2   \n",
       "17                          R1             yihl_1.bw                       R2   \n",
       "18                          R1             yihw_1.bw                       R2   \n",
       "19                          R1             yjdc_1.bw                       R2   \n",
       "20                          R1             yjhi_1.bw                       R2   \n",
       "21                          R1             ynej_1.bw                       R2   \n",
       "22                          R1             ynfl_1.bw                       R2   \n",
       "23                          R1             ypdc_1.bw                       R2   \n",
       "24                          R1             yqhc_1.bw                       R2   \n",
       "25               bio-rep1 – R1       ZraR_1_S3_R1.bw            bio-rep1 – R2   \n",
       "26                          R1    edl933furfinal1.bw                       R2   \n",
       "27                          R1       smsfurfinal1.bw                       R2   \n",
       "28                          R1    umn026furfinal1.bw                       R2   \n",
       "29     exponential growth – R1         dps_expo_1.bw  exponential growth – R2   \n",
       "30     exponential growth – R1         fis_expo_1.bw  exponential growth – R2   \n",
       "31     exponential growth – R1        gyra_expo_1.bw  exponential growth – R2   \n",
       "32     exponential growth – R1        gyrb_expo_1.bw  exponential growth – R2   \n",
       "33     exponential growth – R1         hns_expo_1.bw  exponential growth – R2   \n",
       "34     exponential growth – R1        hupa_expo_1.bw  exponential growth – R2   \n",
       "35     exponential growth – R1        hupb_expo_1.bw  exponential growth – R2   \n",
       "36     exponential growth – R1        infa_expo_1.bw  exponential growth – R2   \n",
       "37     exponential growth – R1        mukb_expo_1.bw  exponential growth – R2   \n",
       "38     exponential growth – R1        rdgc_expo_1.bw  exponential growth – R2   \n",
       "39     exponential growth – R1         rob_expo_1.bw  exponential growth – R2   \n",
       "40     exponential growth – R1        stpa_expo_1.bw  exponential growth – R2   \n",
       "41     exponential growth – R1        topa_expo_1.bw  exponential growth – R2   \n",
       "42     exponential growth – R1         tus_expo_1.bw  exponential growth – R2   \n",
       "43                          R1     narl_nitrate_1.bw                       R2   \n",
       "44                          R1  puur_putrescine_1.bw                       R2   \n",
       "45                          R1       rcsa_oxida_1.bw                       R2   \n",
       "46           control (WT) – R1   mg1655_ace_rpob1.bw        control (WT) – R2   \n",
       "\n",
       "                  BWcond1_2          BWcond2_1_name            BWcond2_1  \\\n",
       "index                                                                      \n",
       "0         BaeR_R1_S31_R2.bw           bio-rep2 – R1   BaerR_R2_S32_R1.bw   \n",
       "1           CpxRR2_S2_R2.bw                     NaN                  NaN   \n",
       "2              cra_glu_2.bw                Fru – R1         cra_fru_1.bw   \n",
       "3               fur_fe_2.bw                DPD – R1         fur_dpd_1.bw   \n",
       "4                 gade_2.bw                     NaN                  NaN   \n",
       "5                 gadw_2.bw                     NaN                  NaN   \n",
       "6                 gadx_2.bw                     NaN                  NaN   \n",
       "7        KdpeE_R2_S34_R2.bw                     NaN                  NaN   \n",
       "8          PhoB_1_S16_R2.bw           bio-rep2 – R1      PhoB_2_S6_R1.bw   \n",
       "9               rpob_fe2.bw                DPD – R1         rpob_dpd1.bw   \n",
       "10                ybao_2.bw                     NaN                  NaN   \n",
       "11                ybaq_2.bw                     NaN                  NaN   \n",
       "12            ydci_ph5_2.bw                pH7 – R1        ydci_ph7_1.bw   \n",
       "13                yddm_2.bw                     NaN                  NaN   \n",
       "14                yheo_2.bw                     NaN                  NaN   \n",
       "15                yiag_2.bw                     NaN                  NaN   \n",
       "16                yiep_2.bw                3HP – R1        yiep_3hp_1.bw   \n",
       "17                yihl_2.bw                     NaN                  NaN   \n",
       "18                yihw_2.bw                     NaN                  NaN   \n",
       "19                yjdc_2.bw                     NaN                  NaN   \n",
       "20                yjhi_2.bw                     NaN                  NaN   \n",
       "21                ynej_2.bw                     NaN                  NaN   \n",
       "22                ynfl_2.bw                     NaN                  NaN   \n",
       "23                ypdc_2.bw                     NaN                  NaN   \n",
       "24                yqhc_2.bw                     NaN                  NaN   \n",
       "25          ZraR_1_S3_R2.bw           bio-rep2 – R1      ZraR_2_S4_R1.bw   \n",
       "26       edl933furfinal2.bw                     NaN                  NaN   \n",
       "27          smsfurfinal2.bw                     NaN                  NaN   \n",
       "28       umn026furfinal2.bw                     NaN                  NaN   \n",
       "29            dps_expo_2.bw  stationary growth – R1         dps_sta_1.bw   \n",
       "30            fis_expo_2.bw  stationary growth – R1         fis_sta_1.bw   \n",
       "31           gyra_expo_2.bw  stationary growth – R1        gyra_sta_1.bw   \n",
       "32           gyrb_expo_2.bw  stationary growth – R1        gyrb_sta_1.bw   \n",
       "33            hns_expo_2.bw  stationary growth – R1         hns_sta_1.bw   \n",
       "34           hupa_expo_2.bw  stationary growth – R1        hupa_sta_1.bw   \n",
       "35           hupb_expo_2.bw  stationary growth – R1        hupb_sta_1.bw   \n",
       "36           infa_expo_2.bw  stationary growth – R1        infa_sta_1.bw   \n",
       "37           mukb_expo_2.bw  stationary growth – R1        mukb_sta_1.bw   \n",
       "38           rdgc_expo_2.bw  stationary growth – R1        rdgc_sta_1.bw   \n",
       "39            rob_expo_2.bw  stationary growth – R1         rob_sta_1.bw   \n",
       "40           stpa_expo_2.bw  stationary growth – R1        stpa_sta_1.bw   \n",
       "41           topa_expo_2.bw  stationary growth – R1        topa_sta_1.bw   \n",
       "42            tus_expo_2.bw  stationary growth – R1         tus_sta_1.bw   \n",
       "43        narl_nitrate_2.bw                     NaN                  NaN   \n",
       "44     puur_putrescine_2.bw                     NaN                  NaN   \n",
       "45          rcsa_oxida_2.bw                     NaN                  NaN   \n",
       "46      mg1655_ace_rpob2.bw             cra-KO – R1  cra_ko_ace_rpob1.bw   \n",
       "\n",
       "               BWcond2_2_name            BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                               \n",
       "0               bio-rep2 – R2   BaerR_R2_S32_R2.bw            NaN   \n",
       "1                         NaN                  NaN            NaN   \n",
       "2                    Fru – R2         cra_fru_2.bw       Gal – R1   \n",
       "3                    DPD – R2         fur_dpd_2.bw            NaN   \n",
       "4                         NaN                  NaN            NaN   \n",
       "5                         NaN                  NaN            NaN   \n",
       "6                         NaN                  NaN            NaN   \n",
       "7                         NaN                  NaN            NaN   \n",
       "8               bio-rep2 – R2      PhoB_2_S6_R2.bw            NaN   \n",
       "9                    DPD – R2         rpob_dpd2.bw            NaN   \n",
       "10                        NaN                  NaN            NaN   \n",
       "11                        NaN                  NaN            NaN   \n",
       "12                   pH7 – R2        ydci_ph7_2.bw       pH8 – R1   \n",
       "13                        NaN                  NaN            NaN   \n",
       "14                        NaN                  NaN            NaN   \n",
       "15                        NaN                  NaN            NaN   \n",
       "16                   3HP – R2        yiep_3hp_2.bw            NaN   \n",
       "17                        NaN                  NaN            NaN   \n",
       "18                        NaN                  NaN            NaN   \n",
       "19                        NaN                  NaN            NaN   \n",
       "20                        NaN                  NaN            NaN   \n",
       "21                        NaN                  NaN            NaN   \n",
       "22                        NaN                  NaN            NaN   \n",
       "23                        NaN                  NaN            NaN   \n",
       "24                        NaN                  NaN            NaN   \n",
       "25                        NaN                  NaN            NaN   \n",
       "26                        NaN                  NaN            NaN   \n",
       "27                        NaN                  NaN            NaN   \n",
       "28                        NaN                  NaN            NaN   \n",
       "29     stationary growth – R2         dps_sta_2.bw            NaN   \n",
       "30     stationary growth – R2         fis_sta_2.bw            NaN   \n",
       "31     stationary growth – R2        gyra_sta_2.bw            NaN   \n",
       "32     stationary growth – R2        gyrb_sta_2.bw            NaN   \n",
       "33     stationary growth – R2         hns_sta_2.bw            NaN   \n",
       "34     stationary growth – R2        hupa_sta_2.bw            NaN   \n",
       "35     stationary growth – R2        hupb_sta_2.bw            NaN   \n",
       "36     stationary growth – R2        infa_sta_2.bw            NaN   \n",
       "37     stationary growth – R2        mukb_sta_2.bw            NaN   \n",
       "38     stationary growth – R2        rdgc_sta_2.bw            NaN   \n",
       "39     stationary growth – R2         rob_sta_2.bw            NaN   \n",
       "40     stationary growth – R2        stpa_sta_2.bw            NaN   \n",
       "41     stationary growth – R2        topa_sta_2.bw            NaN   \n",
       "42     stationary growth – R2         tus_sta_2.bw            NaN   \n",
       "43                        NaN                  NaN            NaN   \n",
       "44                        NaN                  NaN            NaN   \n",
       "45                        NaN                  NaN            NaN   \n",
       "46                cra-KO – R2  cra_ko_ace_rpob2.bw    crp-KO – R1   \n",
       "\n",
       "                 BWcond3_1 BWcond3_2_name            BWcond3_2 BWcond4_1_name  \\\n",
       "index                                                                           \n",
       "0                      NaN            NaN                  NaN            NaN   \n",
       "1                      NaN            NaN                  NaN            NaN   \n",
       "2             cra_gal_1.bw       Gal – R2         cra_gal_2.bw       Ace – R1   \n",
       "3                      NaN            NaN                  NaN            NaN   \n",
       "4                      NaN            NaN                  NaN            NaN   \n",
       "5                      NaN            NaN                  NaN            NaN   \n",
       "6                      NaN            NaN                  NaN            NaN   \n",
       "7                      NaN            NaN                  NaN            NaN   \n",
       "8                      NaN            NaN                  NaN            NaN   \n",
       "9                      NaN            NaN                  NaN            NaN   \n",
       "10                     NaN            NaN                  NaN            NaN   \n",
       "11                     NaN            NaN                  NaN            NaN   \n",
       "12           ydci_ph8_1.bw       pH8 – R2        ydci_ph8_2.bw            NaN   \n",
       "13                     NaN            NaN                  NaN            NaN   \n",
       "14                     NaN            NaN                  NaN            NaN   \n",
       "15                     NaN            NaN                  NaN            NaN   \n",
       "16                     NaN            NaN                  NaN            NaN   \n",
       "17                     NaN            NaN                  NaN            NaN   \n",
       "18                     NaN            NaN                  NaN            NaN   \n",
       "19                     NaN            NaN                  NaN            NaN   \n",
       "20                     NaN            NaN                  NaN            NaN   \n",
       "21                     NaN            NaN                  NaN            NaN   \n",
       "22                     NaN            NaN                  NaN            NaN   \n",
       "23                     NaN            NaN                  NaN            NaN   \n",
       "24                     NaN            NaN                  NaN            NaN   \n",
       "25                     NaN            NaN                  NaN            NaN   \n",
       "26                     NaN            NaN                  NaN            NaN   \n",
       "27                     NaN            NaN                  NaN            NaN   \n",
       "28                     NaN            NaN                  NaN            NaN   \n",
       "29                     NaN            NaN                  NaN            NaN   \n",
       "30                     NaN            NaN                  NaN            NaN   \n",
       "31                     NaN            NaN                  NaN            NaN   \n",
       "32                     NaN            NaN                  NaN            NaN   \n",
       "33                     NaN            NaN                  NaN            NaN   \n",
       "34                     NaN            NaN                  NaN            NaN   \n",
       "35                     NaN            NaN                  NaN            NaN   \n",
       "36                     NaN            NaN                  NaN            NaN   \n",
       "37                     NaN            NaN                  NaN            NaN   \n",
       "38                     NaN            NaN                  NaN            NaN   \n",
       "39                     NaN            NaN                  NaN            NaN   \n",
       "40                     NaN            NaN                  NaN            NaN   \n",
       "41                     NaN            NaN                  NaN            NaN   \n",
       "42                     NaN            NaN                  NaN            NaN   \n",
       "43                     NaN            NaN                  NaN            NaN   \n",
       "44                     NaN            NaN                  NaN            NaN   \n",
       "45                     NaN            NaN                  NaN            NaN   \n",
       "46     crp_ko_ace_rpob1.bw    crp-KO – R2  crp_ko_ace_rpob2.bw            NaN   \n",
       "\n",
       "          BWcond4_1 BWcond4_2_name     BWcond4_2  \n",
       "index                                             \n",
       "0               NaN            NaN           NaN  \n",
       "1               NaN            NaN           NaN  \n",
       "2      cra_ace_1.bw       Ace – R2  cra_ace_2.bw  \n",
       "3               NaN            NaN           NaN  \n",
       "4               NaN            NaN           NaN  \n",
       "5               NaN            NaN           NaN  \n",
       "6               NaN            NaN           NaN  \n",
       "7               NaN            NaN           NaN  \n",
       "8               NaN            NaN           NaN  \n",
       "9               NaN            NaN           NaN  \n",
       "10              NaN            NaN           NaN  \n",
       "11              NaN            NaN           NaN  \n",
       "12              NaN            NaN           NaN  \n",
       "13              NaN            NaN           NaN  \n",
       "14              NaN            NaN           NaN  \n",
       "15              NaN            NaN           NaN  \n",
       "16              NaN            NaN           NaN  \n",
       "17              NaN            NaN           NaN  \n",
       "18              NaN            NaN           NaN  \n",
       "19              NaN            NaN           NaN  \n",
       "20              NaN            NaN           NaN  \n",
       "21              NaN            NaN           NaN  \n",
       "22              NaN            NaN           NaN  \n",
       "23              NaN            NaN           NaN  \n",
       "24              NaN            NaN           NaN  \n",
       "25              NaN            NaN           NaN  \n",
       "26              NaN            NaN           NaN  \n",
       "27              NaN            NaN           NaN  \n",
       "28              NaN            NaN           NaN  \n",
       "29              NaN            NaN           NaN  \n",
       "30              NaN            NaN           NaN  \n",
       "31              NaN            NaN           NaN  \n",
       "32              NaN            NaN           NaN  \n",
       "33              NaN            NaN           NaN  \n",
       "34              NaN            NaN           NaN  \n",
       "35              NaN            NaN           NaN  \n",
       "36              NaN            NaN           NaN  \n",
       "37              NaN            NaN           NaN  \n",
       "38              NaN            NaN           NaN  \n",
       "39              NaN            NaN           NaN  \n",
       "40              NaN            NaN           NaN  \n",
       "41              NaN            NaN           NaN  \n",
       "42              NaN            NaN           NaN  \n",
       "43              NaN            NaN           NaN  \n",
       "44              NaN            NaN           NaN  \n",
       "45              NaN            NaN           NaN  \n",
       "46              NaN            NaN           NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4399</th>\n",
       "      <td>4636695</td>\n",
       "      <td>4638120</td>\n",
       "      <td>+</td>\n",
       "      <td>creC</td>\n",
       "      <td>1425</td>\n",
       "      <td>creABCD</td>\n",
       "      <td>Signal transduction mechanisms</td>\n",
       "      <td>4636695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4400</th>\n",
       "      <td>4638177</td>\n",
       "      <td>4639530</td>\n",
       "      <td>+</td>\n",
       "      <td>creD</td>\n",
       "      <td>1353</td>\n",
       "      <td>creABCD</td>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>4638177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4401</th>\n",
       "      <td>4639589</td>\n",
       "      <td>4640306</td>\n",
       "      <td>-</td>\n",
       "      <td>arcA</td>\n",
       "      <td>717</td>\n",
       "      <td>arcA</td>\n",
       "      <td>Signal transduction mechanisms</td>\n",
       "      <td>4640306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4402</th>\n",
       "      <td>4640401</td>\n",
       "      <td>4640542</td>\n",
       "      <td>+</td>\n",
       "      <td>yjjY</td>\n",
       "      <td>141</td>\n",
       "      <td>yjjY</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>4640401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b4403</th>\n",
       "      <td>4640941</td>\n",
       "      <td>4641628</td>\n",
       "      <td>+</td>\n",
       "      <td>yjtD</td>\n",
       "      <td>687</td>\n",
       "      <td>yjtD</td>\n",
       "      <td>Translation, ribosomal structure and biogenesis</td>\n",
       "      <td>4640941</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4386 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         start     stop strand gene_name  length   operon  \\\n",
       "b0001      189      255      +      thrL      66  thrLABC   \n",
       "b0002      336     2799      +      thrA    2463  thrLABC   \n",
       "b0003     2800     3733      +      thrB     933  thrLABC   \n",
       "b0004     3733     5020      +      thrC    1287  thrLABC   \n",
       "b0005     5233     5530      +      yaaX     297     yaaX   \n",
       "...        ...      ...    ...       ...     ...      ...   \n",
       "b4399  4636695  4638120      +      creC    1425  creABCD   \n",
       "b4400  4638177  4639530      +      creD    1353  creABCD   \n",
       "b4401  4639589  4640306      -      arcA     717     arcA   \n",
       "b4402  4640401  4640542      +      yjjY     141     yjjY   \n",
       "b4403  4640941  4641628      +      yjtD     687     yjtD   \n",
       "\n",
       "                                                   cog  start_codon_pos  \n",
       "b0001                                No COG Annotation              189  \n",
       "b0002                                No COG Annotation              336  \n",
       "b0003              Amino acid transport and metabolism             2800  \n",
       "b0004              Amino acid transport and metabolism             3733  \n",
       "b0005                                 Function unknown             5233  \n",
       "...                                                ...              ...  \n",
       "b4399                   Signal transduction mechanisms          4636695  \n",
       "b4400                               Defense mechanisms          4638177  \n",
       "b4401                   Signal transduction mechanisms          4640306  \n",
       "b4402                                 Function unknown          4640401  \n",
       "b4403  Translation, ribosomal structure and biogenesis          4640941  \n",
       "\n",
       "[4386 rows x 8 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913_3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+curated_files[i] for i in range(len(curated_loc))])\n",
    "    TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    TF_list.loc[TF_index, 'num_cond'] = len(curated_loc)\n",
    "    \n",
    "    #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range (0, len(database)): \n",
    "    peak_annot_DF = annotate_peaks(TF_list[i], TF_condition[i],database[i],500,DF_gene_info).fillna('')\n",
    "    final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "    venn_files = Venn_data_gen(final_annot_DF)\n",
    "    final_annot_DF.to_json(out_data_table+TF_list[i]+'_binding_table.json',orient='records')\n",
    "    venn_files.to_json(path.join(out_data_venn, TF_list[i]+'_venn.json'),orient='records')\n",
    "    binding_width_gen(TF_list[i], database[i], '../../data/e_coli/NC_000913_3/binding_widths/')\n",
    "    peak_position_gen(TF_list[i], final_annot_DF, '../../data/e_coli/NC_000913_3/positions/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_name</th>\n",
       "      <th>binding_peak_start</th>\n",
       "      <th>binding_peak_end</th>\n",
       "      <th>binding_peak_strength</th>\n",
       "      <th>direction</th>\n",
       "      <th>.</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MACE</td>\n",
       "      <td>_filtered_0.95</td>\n",
       "      <td>417086</td>\n",
       "      <td>417115</td>\n",
       "      <td>1.92</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SFB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MACE</td>\n",
       "      <td>_filtered_0.95</td>\n",
       "      <td>1198372</td>\n",
       "      <td>1198398</td>\n",
       "      <td>1.88</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>GSB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MACE</td>\n",
       "      <td>_filtered_0.95</td>\n",
       "      <td>2774093</td>\n",
       "      <td>2774126</td>\n",
       "      <td>1.82</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SFB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MACE</td>\n",
       "      <td>_filtered_0.95</td>\n",
       "      <td>2774321</td>\n",
       "      <td>2774354</td>\n",
       "      <td>1.82</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SRB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MACE</td>\n",
       "      <td>_filtered_0.95</td>\n",
       "      <td>3284518</td>\n",
       "      <td>3284522</td>\n",
       "      <td>1.23</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>GSB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  condition   condition_name  binding_peak_start  binding_peak_end  \\\n",
       "1      MACE   _filtered_0.95              417086            417115   \n",
       "2      MACE   _filtered_0.95             1198372           1198398   \n",
       "3      MACE   _filtered_0.95             2774093           2774126   \n",
       "4      MACE   _filtered_0.95             2774321           2774354   \n",
       "5      MACE   _filtered_0.95             3284518           3284522   \n",
       "\n",
       "   binding_peak_strength direction  .   ID  \n",
       "1                   1.92         +  .  SFB  \n",
       "2                   1.88         +  .  GSB  \n",
       "3                   1.82         +  .  SFB  \n",
       "4                   1.82         +  .  SRB  \n",
       "5                   1.23         +  .  GSB  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file = 'phob_M9P_curated.gff'\n",
    "df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>condition</th>\n",
       "      <th>binding_peak_start</th>\n",
       "      <th>binding_peak_end</th>\n",
       "      <th>binding_peak_strength</th>\n",
       "      <th>target_locus</th>\n",
       "      <th>target_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PhoB-1</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>417086</td>\n",
       "      <td>417115</td>\n",
       "      <td>1.92</td>\n",
       "      <td>b0399,b0400,b0397,b0398</td>\n",
       "      <td>phoB,phoR,sbcC,sbcD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PhoB-2</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>1198372</td>\n",
       "      <td>1198398</td>\n",
       "      <td>1.88</td>\n",
       "      <td>b1139,b1137,b1138</td>\n",
       "      <td>lit,ymfD,ymfE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PhoB-3</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>2774093</td>\n",
       "      <td>2774126</td>\n",
       "      <td>1.82</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PhoB-4</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>2774321</td>\n",
       "      <td>2774354</td>\n",
       "      <td>1.82</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PhoB-5</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>3284518</td>\n",
       "      <td>3284522</td>\n",
       "      <td>1.23</td>\n",
       "      <td>b3138,b3139,b3140,b3141</td>\n",
       "      <td>agaB,agaC,agaD,agaI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PhoB-6</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>3321253</td>\n",
       "      <td>3321291</td>\n",
       "      <td>1.69</td>\n",
       "      <td>b3173</td>\n",
       "      <td>yhbX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PhoB-7</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>3321485</td>\n",
       "      <td>3321524</td>\n",
       "      <td>2.08</td>\n",
       "      <td>b3173</td>\n",
       "      <td>yhbX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PhoB-8</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>3911549</td>\n",
       "      <td>3911584</td>\n",
       "      <td>8.13</td>\n",
       "      <td>b3724,b3725,b3726,b3727,b3728</td>\n",
       "      <td>phoU,pstB,pstA,pstC,pstS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PhoB-9</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>3911629</td>\n",
       "      <td>3911670</td>\n",
       "      <td>6.93</td>\n",
       "      <td>b3724,b3725,b3726,b3727,b3728</td>\n",
       "      <td>phoU,pstB,pstA,pstC,pstS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PhoB-10</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>4003031</td>\n",
       "      <td>4003074</td>\n",
       "      <td>2.05</td>\n",
       "      <td>b3817,b3818</td>\n",
       "      <td>yigF,yigG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PhoB-11</td>\n",
       "      <td>phob + M9P</td>\n",
       "      <td>4058198</td>\n",
       "      <td>4058242</td>\n",
       "      <td>1.65</td>\n",
       "      <td>b3871,b3868,b3869,b3870</td>\n",
       "      <td>typA,glnG,glnL,glnA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      index   condition  binding_peak_start  binding_peak_end  \\\n",
       "1    PhoB-1  phob + M9P              417086            417115   \n",
       "2    PhoB-2  phob + M9P             1198372           1198398   \n",
       "3    PhoB-3  phob + M9P             2774093           2774126   \n",
       "4    PhoB-4  phob + M9P             2774321           2774354   \n",
       "5    PhoB-5  phob + M9P             3284518           3284522   \n",
       "6    PhoB-6  phob + M9P             3321253           3321291   \n",
       "7    PhoB-7  phob + M9P             3321485           3321524   \n",
       "8    PhoB-8  phob + M9P             3911549           3911584   \n",
       "9    PhoB-9  phob + M9P             3911629           3911670   \n",
       "10  PhoB-10  phob + M9P             4003031           4003074   \n",
       "11  PhoB-11  phob + M9P             4058198           4058242   \n",
       "\n",
       "    binding_peak_strength                   target_locus  \\\n",
       "1                    1.92        b0399,b0400,b0397,b0398   \n",
       "2                    1.88              b1139,b1137,b1138   \n",
       "3                    1.82                                  \n",
       "4                    1.82                                  \n",
       "5                    1.23        b3138,b3139,b3140,b3141   \n",
       "6                    1.69                          b3173   \n",
       "7                    2.08                          b3173   \n",
       "8                    8.13  b3724,b3725,b3726,b3727,b3728   \n",
       "9                    6.93  b3724,b3725,b3726,b3727,b3728   \n",
       "10                   2.05                    b3817,b3818   \n",
       "11                   1.65        b3871,b3868,b3869,b3870   \n",
       "\n",
       "                target_genes  \n",
       "1        phoB,phoR,sbcC,sbcD  \n",
       "2              lit,ymfD,ymfE  \n",
       "3                             \n",
       "4                             \n",
       "5        agaB,agaC,agaD,agaI  \n",
       "6                       yhbX  \n",
       "7                       yhbX  \n",
       "8   phoU,pstB,pstA,pstC,pstS  \n",
       "9   phoU,pstB,pstA,pstC,pstS  \n",
       "10                 yigF,yigG  \n",
       "11       typA,glnG,glnL,glnA  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peak_annot_DF = annotate_peaks('PhoB', 'M9P', df, 500, DF_gene_info)\n",
    "final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "\n",
    "final_annot_DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "136"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "float('nan')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[i for i in peak_annot_DF.genes_m][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>condition</th>\n",
       "      <th>binding_peak_start</th>\n",
       "      <th>binding_peak_end</th>\n",
       "      <th>binding_peak_strength</th>\n",
       "      <th>TU_p</th>\n",
       "      <th>genes_p</th>\n",
       "      <th>TU_m</th>\n",
       "      <th>genes_m</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Dps-1</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>235</td>\n",
       "      <td>261</td>\n",
       "      <td>1.50</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>b0001,b0002,b0003,b0004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Dps-2</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>111536</td>\n",
       "      <td>111543</td>\n",
       "      <td>2.28</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dps-3</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>167456</td>\n",
       "      <td>167466</td>\n",
       "      <td>1.95</td>\n",
       "      <td>fhuACDB</td>\n",
       "      <td>b0150,b0151,b0152,b0153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Dps-4</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>246985</td>\n",
       "      <td>247015</td>\n",
       "      <td>1.13</td>\n",
       "      <td>yafL</td>\n",
       "      <td>b0227</td>\n",
       "      <td>dinJ-yafQ</td>\n",
       "      <td>b0225,b0226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Dps-5</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>256349</td>\n",
       "      <td>256373</td>\n",
       "      <td>1.54</td>\n",
       "      <td>gpt,frsA</td>\n",
       "      <td>b0238,b0239</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>Dps-132</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>4406423</td>\n",
       "      <td>4406463</td>\n",
       "      <td>1.01</td>\n",
       "      <td>nsrR-rnr-rlmB-yjfIJ</td>\n",
       "      <td>b4178,b4179,b4180,b4181,b4182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>Dps-133</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>4474258</td>\n",
       "      <td>4474288</td>\n",
       "      <td>1.11</td>\n",
       "      <td>bdcR</td>\n",
       "      <td>b4251</td>\n",
       "      <td>bdcA</td>\n",
       "      <td>b4249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>Dps-134</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>4527665</td>\n",
       "      <td>4527698</td>\n",
       "      <td>1.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sgcXBCQAER</td>\n",
       "      <td>b4300,b4301,b4302,b4303,b4304,b4565,b4305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>Dps-135</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>4566141</td>\n",
       "      <td>4566162</td>\n",
       "      <td>1.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yjiML</td>\n",
       "      <td>b4334,b4335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>Dps-136</td>\n",
       "      <td>dps + expo</td>\n",
       "      <td>4605695</td>\n",
       "      <td>4605707</td>\n",
       "      <td>1.77</td>\n",
       "      <td>yjjZ</td>\n",
       "      <td>b4567</td>\n",
       "      <td>fhuF</td>\n",
       "      <td>b4367</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       index   condition  binding_peak_start  binding_peak_end  \\\n",
       "1      Dps-1  dps + expo                 235               261   \n",
       "2      Dps-2  dps + expo              111536            111543   \n",
       "3      Dps-3  dps + expo              167456            167466   \n",
       "4      Dps-4  dps + expo              246985            247015   \n",
       "5      Dps-5  dps + expo              256349            256373   \n",
       "..       ...         ...                 ...               ...   \n",
       "132  Dps-132  dps + expo             4406423           4406463   \n",
       "133  Dps-133  dps + expo             4474258           4474288   \n",
       "134  Dps-134  dps + expo             4527665           4527698   \n",
       "135  Dps-135  dps + expo             4566141           4566162   \n",
       "136  Dps-136  dps + expo             4605695           4605707   \n",
       "\n",
       "     binding_peak_strength                 TU_p  \\\n",
       "1                     1.50              thrLABC   \n",
       "2                     2.28                        \n",
       "3                     1.95              fhuACDB   \n",
       "4                     1.13                 yafL   \n",
       "5                     1.54             gpt,frsA   \n",
       "..                     ...                  ...   \n",
       "132                   1.01  nsrR-rnr-rlmB-yjfIJ   \n",
       "133                   1.11                 bdcR   \n",
       "134                   1.11                  NaN   \n",
       "135                   1.19                  NaN   \n",
       "136                   1.77                 yjjZ   \n",
       "\n",
       "                           genes_p        TU_m  \\\n",
       "1          b0001,b0002,b0003,b0004         NaN   \n",
       "2                                                \n",
       "3          b0150,b0151,b0152,b0153         NaN   \n",
       "4                            b0227   dinJ-yafQ   \n",
       "5                      b0238,b0239         NaN   \n",
       "..                             ...         ...   \n",
       "132  b4178,b4179,b4180,b4181,b4182         NaN   \n",
       "133                          b4251        bdcA   \n",
       "134                            NaN  sgcXBCQAER   \n",
       "135                            NaN       yjiML   \n",
       "136                          b4567        fhuF   \n",
       "\n",
       "                                       genes_m  \n",
       "1                                          NaN  \n",
       "2                                               \n",
       "3                                          NaN  \n",
       "4                                  b0225,b0226  \n",
       "5                                          NaN  \n",
       "..                                         ...  \n",
       "132                                        NaN  \n",
       "133                                      b4249  \n",
       "134  b4300,b4301,b4302,b4303,b4304,b4565,b4305  \n",
       "135                                b4334,b4335  \n",
       "136                                      b4367  \n",
       "\n",
       "[136 rows x 9 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peak_annot_DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "            if len(operon) > 1:\n",
    "                print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "283.4px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
