{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for proChIPdb, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "from zipfile import ZipFile\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + TF_condition + \"-\" + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus_nc000913(peak_start, peak_stop, gene_info):\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start >= BP) & (stop >= BP) & (strand == '+')) | ((start <= BP) & (stop <= BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start <= BP) & (stop >= BP)) | ((start >= BP) & (stop <= BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    df_complete['closest_gene'] = [closest_locus_nc000913(df_complete.binding_peak_start.iloc[i], \n",
    "                                           df_complete.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(df_complete))]\n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen2(Peak_DF, TRN_data, gene_info): \n",
    "    TF_name = Peak_DF['index'][1].split(\"-\")[0].split(\" \")[0]\n",
    "    \n",
    "\n",
    "    # TRN locus\n",
    "    reg_genes = set(TRN_data.loc[TF_name].gene_id.to_list())\n",
    "\n",
    "    # chip locus\n",
    "    gene_list = [i for i in Peak_DF.target_locus if i != '']\n",
    "    chip_genes = set(','.join(list(gene_list)).split(','))\n",
    "    \n",
    "    #both\n",
    "    both_genes = set(reg_genes & chip_genes)\n",
    "    \n",
    "    # Get gene and operon counts\n",
    "    reg_gene_count = len(reg_genes)\n",
    "    chip_gene_count = len(chip_genes)\n",
    "    both_gene_count = len(both_genes)\n",
    "    \n",
    "    # Add adjustments for venn plotting (add '2' for alternates)\n",
    "    reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    if reg_genes == chip_genes:\n",
    "        reg_gene_count = 0; chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = len(reg_genes)\n",
    "    elif all(item in chip_genes for item in reg_genes):\n",
    "        reg_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = len(reg_genes); chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    elif all(item in reg_genes for item in chip_genes):\n",
    "        chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = len(chip_genes); both_gene_count2 = 0\n",
    "        \n",
    "    res = pd.DataFrame([TF_name, reg_gene_count, chip_gene_count, both_gene_count,\n",
    "                        reg_gene_count2, chip_gene_count2, both_gene_count2], \n",
    "                        columns=['Value'],\n",
    "                        index=['TF', 'reg_genes', 'chip_genes', 'both_genes',\n",
    "                                'reg_genes2', 'chip_genes2', 'both_genes2'])\n",
    "    \n",
    "    # Add gene lists\n",
    "    just_reg = reg_genes - both_genes\n",
    "    just_chip = chip_genes - both_genes\n",
    "    for i, l in zip(['reg_genes', 'chip_genes', 'both_genes'],[just_reg, just_chip, both_genes]):\n",
    "        gene_list = np.array([gene_info.loc[g,'gene_name'] if g in gene_info.index else g for g in l])\n",
    "        gene_list = np.array2string(gene_list, separator = ' ')\n",
    "        res.loc[i, 'list'] = gene_list\n",
    "    \n",
    "    # Add data sources\n",
    "    res.loc['TF', 'list'] = '; '.join(TRN_data.source[TRN_data.index == TF_name].unique())\n",
    "    \n",
    "    return res"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        gene = peak.closest_gene\n",
    "        gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "        try:\n",
    "            if gene_info.strand[0] == \"-\":\n",
    "                dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "            else:\n",
    "                dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "            peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "            counter+=1\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - NC_000913.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>mixed-TFs-pool3</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+ace|mixed-tfs-pool3_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>Mixed TFs: YqhC, Rob, YcaN, YciT, YfiE, GadX, ...</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool3_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool3_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>mixed-TFs-pool4</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>Mixed TFs: YidZ, YedW, ArcA, YhjB</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool4_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool4_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>mixed-TFs-pool5</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Galactose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>Mixed TFs: YebK, YbeF</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool5_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool5_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ArcA-1</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>monoclonal antibodies (ArcA part1)</td>\n",
       "      <td>1C6</td>\n",
       "      <td>arcA_1C6.bw</td>\n",
       "      <td>1D6</td>\n",
       "      <td>arcA_1D6.bw</td>\n",
       "      <td>2A1</td>\n",
       "      <td>arcA_2A1.bw</td>\n",
       "      <td>2B6</td>\n",
       "      <td>arcA_2B6.bw</td>\n",
       "      <td>2C4</td>\n",
       "      <td>arcA_2C4.bw</td>\n",
       "      <td>2D6</td>\n",
       "      <td>arcA_2D6.bw</td>\n",
       "      <td>4A1</td>\n",
       "      <td>arcA_4A1.bw</td>\n",
       "      <td>4A3</td>\n",
       "      <td>arcA_4A3.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>ArcA-2</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>monoclonal antibodies (ArcA part2)</td>\n",
       "      <td>4D6</td>\n",
       "      <td>arcA_4D6.bw</td>\n",
       "      <td>5A10</td>\n",
       "      <td>arcA_5A10.bw</td>\n",
       "      <td>5B6</td>\n",
       "      <td>arcA_5B6.bw</td>\n",
       "      <td>5B9</td>\n",
       "      <td>arcA_5B9.bw</td>\n",
       "      <td>5F9</td>\n",
       "      <td>arcA_5F9.bw</td>\n",
       "      <td>5G8</td>\n",
       "      <td>arcA_5G8.bw</td>\n",
       "      <td>7B1</td>\n",
       "      <td>arcA_7B10.bw</td>\n",
       "      <td>8B2</td>\n",
       "      <td>arcA_8B2.bw</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    TF          Organism       Strain Media  \\\n",
       "index                                                         \n",
       "0                 BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1                 CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2                  Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3                  Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4                 GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "...                ...               ...          ...   ...   \n",
       "59     mixed-TFs-pool3  Escherichia coli  K-12 MG1655    M9   \n",
       "60     mixed-TFs-pool4  Escherichia coli  K-12 MG1655    M9   \n",
       "61     mixed-TFs-pool5  Escherichia coli  K-12 MG1655    M9   \n",
       "62              ArcA-1  Escherichia coli  K-12 MG1655    M9   \n",
       "63              ArcA-2  Escherichia coli  K-12 MG1655    M9   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "...                                              ...          ...         ...   \n",
       "59                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "60                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "61                                    0.2% Galactose  NC_000913.3      e_coli   \n",
       "62                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "63                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond      project  \\\n",
       "index                                                          \n",
       "0                    NaN            4         1          TCS   \n",
       "1                    NaN            2         1          TCS   \n",
       "2                    NaN            8         4     known_TF   \n",
       "3                    NaN            4         2     known_TF   \n",
       "4                    NaN            2         1     known_TF   \n",
       "...                  ...          ...       ...          ...   \n",
       "59                   NaN            2         1  combination   \n",
       "60                   NaN            2         1  combination   \n",
       "61                   NaN            2         1  combination   \n",
       "62                   NaN            8         1         mAbs   \n",
       "63                   NaN            8         1         mAbs   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "...                                                  ...        ...   \n",
       "59      M9+ace|mixed-tfs-pool3_M9+ace_binding_table.json  GSE182078   \n",
       "60      M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json  GSE182078   \n",
       "61      M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json  GSE182078   \n",
       "62                 M9+ace|arca_M9+ace_binding_table.json  GSE182077   \n",
       "63                 M9+ace|arca_M9+ace_binding_table.json  GSE182077   \n",
       "\n",
       "                                             doi peak_intensity_measure  \\\n",
       "index                                                                     \n",
       "0      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "1      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "2             https://doi.org/10.1093/nar/gky069               MACE S/N   \n",
       "3             https://doi.org/10.1038/ncomms5910               MACE S/N   \n",
       "4             https://doi.org/10.1038/ncomms8970               MACE S/N   \n",
       "...                                          ...                    ...   \n",
       "59                                           NaN               MACE S/N   \n",
       "60                                           NaN               MACE S/N   \n",
       "61                                           NaN               MACE S/N   \n",
       "62                                           NaN               MACE S/N   \n",
       "63                                           NaN               MACE S/N   \n",
       "\n",
       "         Method    TF Category  \\\n",
       "index                            \n",
       "0      ChIP-exo  1 – Single TF   \n",
       "1      ChIP-exo  1 – Single TF   \n",
       "2      ChIP-exo  1 – Single TF   \n",
       "3      ChIP-exo  1 – Single TF   \n",
       "4      ChIP-exo  1 – Single TF   \n",
       "...         ...            ...   \n",
       "59     ChIP-exo   3 – Mixed TF   \n",
       "60     ChIP-exo   3 – Mixed TF   \n",
       "61     ChIP-exo   3 – Mixed TF   \n",
       "62     ChIP-exo       2 – mAbs   \n",
       "63     ChIP-exo       2 – mAbs   \n",
       "\n",
       "                                                    Note BWcond1_1_name  \\\n",
       "index                                                                     \n",
       "0                                                    NaN  bio-rep1 – R1   \n",
       "1                                                    NaN             R1   \n",
       "2                                                    NaN       Glu – R1   \n",
       "3                                                    NaN        Fe – R1   \n",
       "4                                                    NaN             R1   \n",
       "...                                                  ...            ...   \n",
       "59     Mixed TFs: YqhC, Rob, YcaN, YciT, YfiE, GadX, ...             R1   \n",
       "60                     Mixed TFs: YidZ, YedW, ArcA, YhjB             R1   \n",
       "61                                 Mixed TFs: YebK, YbeF             R1   \n",
       "62                    monoclonal antibodies (ArcA part1)            1C6   \n",
       "63                    monoclonal antibodies (ArcA part2)            4D6   \n",
       "\n",
       "               BWcond1_1 BWcond1_2_name          BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                                       \n",
       "0      BaeR_R1_S31_R1.bw  bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1   \n",
       "1        CpxRR1_S1_R1.bw             R2    CpxRR2_S2_R2.bw            NaN   \n",
       "2           cra_glu_1.bw       Glu – R2       cra_glu_2.bw       Fru – R1   \n",
       "3            fur_fe_1.bw        Fe – R2        fur_fe_2.bw       DPD – R1   \n",
       "4              gade_1.bw             R2          gade_2.bw            NaN   \n",
       "...                  ...            ...                ...            ...   \n",
       "59      mixTF_pool3_1.bw             R2   mixTF_pool3_2.bw            NaN   \n",
       "60      mixTF_pool4_1.bw             R2   mixTF_pool4_2.bw            NaN   \n",
       "61      mixTF_pool5_1.bw             R2   mixTF_pool5_2.bw            NaN   \n",
       "62           arcA_1C6.bw            1D6        arcA_1D6.bw            2A1   \n",
       "63           arcA_4D6.bw           5A10       arcA_5A10.bw            5B6   \n",
       "\n",
       "                BWcond2_1 BWcond2_2_name           BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                                         \n",
       "0      BaerR_R2_S32_R1.bw  bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN   \n",
       "1                     NaN            NaN                 NaN            NaN   \n",
       "2            cra_fru_1.bw       Fru – R2        cra_fru_2.bw       Gal – R1   \n",
       "3            fur_dpd_1.bw       DPD – R2        fur_dpd_2.bw            NaN   \n",
       "4                     NaN            NaN                 NaN            NaN   \n",
       "...                   ...            ...                 ...            ...   \n",
       "59                    NaN            NaN                 NaN            NaN   \n",
       "60                    NaN            NaN                 NaN            NaN   \n",
       "61                    NaN            NaN                 NaN            NaN   \n",
       "62            arcA_2A1.bw            2B6         arcA_2B6.bw            2C4   \n",
       "63            arcA_5B6.bw            5B9         arcA_5B9.bw            5F9   \n",
       "\n",
       "          BWcond3_1 BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                                           \n",
       "0               NaN            NaN           NaN            NaN           NaN   \n",
       "1               NaN            NaN           NaN            NaN           NaN   \n",
       "2      cra_gal_1.bw       Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3               NaN            NaN           NaN            NaN           NaN   \n",
       "4               NaN            NaN           NaN            NaN           NaN   \n",
       "...             ...            ...           ...            ...           ...   \n",
       "59              NaN            NaN           NaN            NaN           NaN   \n",
       "60              NaN            NaN           NaN            NaN           NaN   \n",
       "61              NaN            NaN           NaN            NaN           NaN   \n",
       "62      arcA_2C4.bw            2D6   arcA_2D6.bw            4A1   arcA_4A1.bw   \n",
       "63      arcA_5F9.bw            5G8   arcA_5G8.bw            7B1  arcA_7B10.bw   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  \n",
       "...              ...           ...  \n",
       "59               NaN           NaN  \n",
       "60               NaN           NaN  \n",
       "61               NaN           NaN  \n",
       "62               4A3   arcA_4A3.bw  \n",
       "63               8B2   arcA_8B2.bw  \n",
       "\n",
       "[64 rows x 34 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       start  stop strand gene_name  length   operon  \\\n",
       "b0001    189   255      +      thrL      66  thrLABC   \n",
       "b0002    336  2799      +      thrA    2463  thrLABC   \n",
       "b0003   2800  3733      +      thrB     933  thrLABC   \n",
       "b0004   3733  5020      +      thrC    1287  thrLABC   \n",
       "b0005   5233  5530      +      yaaX     297     yaaX   \n",
       "\n",
       "                                       cog  start_codon_pos  \n",
       "b0001                    No COG Annotation              189  \n",
       "b0002                    No COG Annotation              336  \n",
       "b0003  Amino acid transport and metabolism             2800  \n",
       "b0004  Amino acid transport and metabolism             3733  \n",
       "b0005                     Function unknown             5233  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913.3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "baer\n",
      "cpxr\n",
      "cra\n",
      "fur\n",
      "gade\n",
      "gadw\n",
      "gadx\n",
      "kdpe\n",
      "phob\n",
      "rpob\n",
      "ybao\n",
      "ybaq\n",
      "ydci\n",
      "yddm\n",
      "yheo\n",
      "yiag\n",
      "yiep\n",
      "yihl\n",
      "yihw\n",
      "yjdc\n",
      "yjhi\n",
      "ynej\n",
      "ynfl\n",
      "ypdc\n",
      "yqhc\n",
      "zrar\n",
      "fur\n",
      "fur\n",
      "fur\n",
      "dps\n",
      "fis\n",
      "gyra\n",
      "gyrb\n",
      "h-ns\n",
      "hupa\n",
      "hupb\n",
      "infa\n",
      "mukb\n",
      "rdgc\n",
      "rob\n",
      "stpa\n",
      "topa\n",
      "tus\n",
      "narl\n",
      "puur\n",
      "rcsa\n",
      "rpob (cra and crp ko exps)\n",
      "flhc\n",
      "flhd\n",
      "flia\n",
      "fnr\n",
      "rpod\n",
      "argr\n",
      "ompr\n",
      "rpoh\n",
      "mixed-tfs-pool1a\n",
      "mixed-tfs-pool1b\n",
      "mixed-tfs-pool1c\n",
      "mixed-tfs-pool2\n",
      "mixed-tfs-pool3\n",
      "mixed-tfs-pool4\n",
      "mixed-tfs-pool5\n",
      "arca-1\n",
      "arca-2\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    print(TF_name)\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "    \n",
    "    if strain != 'NC_000913.3':\n",
    "        continue\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] # change split to \"/\" on linux\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    else:\n",
    "        TF_list.loc[TF_index, 'binding tables'] = \"NA\"\n",
    "    TF_list.loc[TF_index, 'num_cond'] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    "    # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "            final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "            final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "            \n",
    "    #venn diagrams\n",
    "        try:\n",
    "            if strain == 'NC_000913.3' and TF_list.TF[TF_index].split(\" \")[0] in TRN.index:\n",
    "                Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(org_folder+strain+\"/venn/\"+TF_list.TF[TF_index]+'_venn.csv')\n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(org_folder+strain+\"/bw/compressed/\"+TF_list.TF[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(org_folder+strain+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Other Organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_gff_to_geneinfo(org, strain):\n",
    "    genes = pd.read_csv('../data'+'/'+org+'/'+strain+'/annotation/genes.gff',index_col=0, delimiter='\\t', header=None, \n",
    "                        names = ['genome', 'ref', 'type', 'start','stop','.', 'strand', '-','details'])\n",
    "    genes[\"locus_tag\"] = [gene_detail[0].split(\" \")[1] for gene_detail in genes.details.str.split(\"; \")]\n",
    "    genes.set_index('locus_tag', inplace=True)\n",
    "    genes = genes.drop(['.', '-','ref','type'], axis=1)\n",
    "\n",
    "    return(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus(peak_start, peak_stop, gene_info):\n",
    "    gene_info['gene_name'] = [i.split(\";\")[0].split(\" \")[-1] for i in gene_info.details]\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_table_gen(TF_name, TF_condition, peak_df, gene_info):\n",
    "    \n",
    "    table = peak_df.copy()\n",
    "    table[\"index\"] = [TF_name+\"-\"+TF_condition+\"-\"+str(i+1) for i in range(len(table))]\n",
    "    table[\"condition\"] = [TF_name.lower() + \" + \" + TF_condition for i in range(len(table))]\n",
    "    table.drop([\"condition_name\",\".\", \"ID\"], axis=1, inplace=True)\n",
    "    table[\"target_locus\"] = [closest_locus(table.binding_peak_start.iloc[i], \n",
    "                                           table.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(table))]\n",
    "    table[\"target_genes\"] = table[\"target_locus\"]\n",
    "    table[\"closest_gene\"] = table[\"target_locus\"]\n",
    "\n",
    "    return(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_peak_position_gen(TF, final_annot_DF,out_dir, DF_gene_info): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.iloc[i,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.index == gene]\n",
    "            try:\n",
    "                if gene_info.strand.iloc[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/(gene_info.stop-gene_info.start+1))\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/(gene_info.stop-gene_info.start+1))\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.drop(['direction'], axis=1, inplace=True)\n",
    "    cols = ['index', 'condition', 'binding_peak_start', 'binding_peak_end', 'binding_peak_strength', 'target_locus', \n",
    "            'target_genes', 'closest_gene', 'gene', 'normalized_dist', 'binding_peak_center']\n",
    "    peak_scatter_df = peak_scatter_df[cols]\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/all_other/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>CP000647.1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>fe|fur_fe_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>M9|sigma d_M9_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH5, MES</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...</td>\n",
       "      <td>GSE181791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>AE014073.1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>CP035369.1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>M9|cody_M9_binding_table.json</td>\n",
       "      <td>GSE159856</td>\n",
       "      <td>https://doi.org/10.1101/2021.01.08.426013</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|sigs_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|vrar_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>BHI</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NZ_CP009712.1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FleQ</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|fleq_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742210.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742211.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GbdR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gbdr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742212.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742213.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PchR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|pchr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742214.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742215.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|phob_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742216.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742217.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SphR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|sphr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742218.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742219.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MvfR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mvfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776542.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776543.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SoxR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|soxr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776544.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776545.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RpoN</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|rpon_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776546.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776547.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Vfr</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|vfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776548.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776549.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AlgR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|algr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776550.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776551.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ExsA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|exsa_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776552.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776553.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GacA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gaca_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776554.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776555.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MexT</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mext_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776558.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776559.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>WspR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|wspr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776560.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776561.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HrdB</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>MS</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|hrdb_MS_binding_table.json</td>\n",
       "      <td>E-MTAB-6926</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky1018</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR2639621.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR2639622.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NdgR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>minimal media</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>minimal-media|ndgr_minimal-media_binding_table...</td>\n",
       "      <td>GSE59010</td>\n",
       "      <td>https://doi.org/10.1186/s12864-015-1311-0</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1503874.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ScbR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752661.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ScbR2</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr2_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752662.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AmrZ</td>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>F113</td>\n",
       "      <td>SA (low Fe)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_016830.1</td>\n",
       "      <td>p_fluorescens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRP039494</td>\n",
       "      <td>https://doi.org/10.1186/1471-2164-15-237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1184483.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Lsr2</td>\n",
       "      <td>Streptomyces venezuelae</td>\n",
       "      <td>ATCC 10712</td>\n",
       "      <td>MS</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_018750.1</td>\n",
       "      <td>s_venezuelae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|lsr2_MS_binding_table.json</td>\n",
       "      <td>GSE115439</td>\n",
       "      <td>https://doi.org/10.7554/eLife.47691</td>\n",
       "      <td>Fold difference</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LexA1</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Copenhageni Fiocruz L1-130</td>\n",
       "      <td>EMJH</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_005823.1</td>\n",
       "      <td>l_interrogans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>EMJH|lexa1_EMJH_binding_table.json</td>\n",
       "      <td>GSE73688</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv1536</td>\n",
       "      <td>CisGenome fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2549977.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR2549979.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MksB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NeonGreen 1 – R1</td>\n",
       "      <td>SRR8791699.bw</td>\n",
       "      <td>NeonGreen 1 – R2</td>\n",
       "      <td>SRR8791700.bw</td>\n",
       "      <td>NeonGreen 2 – R1</td>\n",
       "      <td>SRR8791701.bw</td>\n",
       "      <td>NeonGreen 2 – R2</td>\n",
       "      <td>SRR8791702.bw</td>\n",
       "      <td>mCherry – R1</td>\n",
       "      <td>SRR8791680.bw</td>\n",
       "      <td>mCherry – R2</td>\n",
       "      <td>SRR8791681.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ParB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791676.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791677.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791678.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791679.bw</td>\n",
       "      <td>mCherry 3 – R1</td>\n",
       "      <td>SRR8791682.bw</td>\n",
       "      <td>mCherry 3 – R2</td>\n",
       "      <td>SRR8791683.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Smc</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791691.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791692.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791697.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791698.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism                      Strain  \\\n",
       "index                                                                     \n",
       "0          Fur        Klebsiella pneumoniae                   MGH 78578   \n",
       "0          Fur           Pseudomonas putida                      KT2440   \n",
       "1      Sigma D           Pseudomonas putida                      KT2440   \n",
       "0          Fur          Salmonella enterica             Typhimurium LT2   \n",
       "1         YdcI          Salmonella enterica             Typhimurium LT2   \n",
       "0          Fur            Shigella flexneri                   2a, 2457T   \n",
       "0         CodY        Staphylococcus aureus              LAC chromosome   \n",
       "1         SigS        Staphylococcus aureus              USA300_TCH1516   \n",
       "2         VraR        Staphylococcus aureus              USA300_TCH1516   \n",
       "0          Fur  Yersinia pseudotuberculosis                    IP 32953   \n",
       "0         FleQ       Pseudomonas aeruginosa                        PAO1   \n",
       "1         GbdR       Pseudomonas aeruginosa                        PAO1   \n",
       "2         PchR       Pseudomonas aeruginosa                        PAO1   \n",
       "3         PhoB       Pseudomonas aeruginosa                        PAO1   \n",
       "4         SphR       Pseudomonas aeruginosa                        PAO1   \n",
       "5         MvfR       Pseudomonas aeruginosa                        PAO1   \n",
       "6         SoxR       Pseudomonas aeruginosa                        PAO1   \n",
       "7         RpoN       Pseudomonas aeruginosa                        PAO1   \n",
       "8          Vfr       Pseudomonas aeruginosa                        PAO1   \n",
       "9         AlgR       Pseudomonas aeruginosa                        PAO1   \n",
       "10        ExsA       Pseudomonas aeruginosa                        PAO1   \n",
       "11        GacA       Pseudomonas aeruginosa                        PAO1   \n",
       "12        MexT       Pseudomonas aeruginosa                        PAO1   \n",
       "13        WspR       Pseudomonas aeruginosa                        PAO1   \n",
       "0         HrdB      Streptomyces coelicolor                  A3(2) M145   \n",
       "1         NdgR      Streptomyces coelicolor                  A3(2) M145   \n",
       "2         ScbR      Streptomyces coelicolor                  A3(2) M145   \n",
       "3        ScbR2      Streptomyces coelicolor                  A3(2) M145   \n",
       "0         AmrZ      Pseudomonas fluorescens                        F113   \n",
       "0         Lsr2      Streptomyces venezuelae                  ATCC 10712   \n",
       "0        LexA1       Leptospira interrogans  Copenhageni Fiocruz L1-130   \n",
       "0         MksB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "1         ParB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "2          Smc   Corynebacterium glutamicum                  ATCC 13032   \n",
       "\n",
       "               Media Supplement      genome_id           organism_id  \\\n",
       "index                                                                  \n",
       "0                 M9         Fe     CP000647.1          k_pneumoniae   \n",
       "0                 M9         Fe    NC_002947.4              p_putida   \n",
       "1                 M9         --    NC_002947.4              p_putida   \n",
       "0                 M9         Fe    NC_003197.2            s_enterica   \n",
       "1                 M9   pH5, MES    NC_003197.2            s_enterica   \n",
       "0                 M9         Fe     AE014073.1            s_flexneri   \n",
       "0               RPMI     10% LB     CP035369.1              s_aureus   \n",
       "1               RPMI     10% LB      NC_010079              s_aureus   \n",
       "2               RPMI     10% LB      NC_010079              s_aureus   \n",
       "0                BHI         Fe  NZ_CP009712.1  y_pseudotuberculosis   \n",
       "0                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "1                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "2                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "3                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "4                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "5                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "6                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "7                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "8                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "9                 LB         --    NC_002516.2          p_aeruginosa   \n",
       "10                LB         --    NC_002516.2          p_aeruginosa   \n",
       "11                LB         --    NC_002516.2          p_aeruginosa   \n",
       "12                LB         --    NC_002516.2          p_aeruginosa   \n",
       "13                LB         --    NC_002516.2          p_aeruginosa   \n",
       "0                 MS         --    NC_003888.3          s_coelicolor   \n",
       "1      minimal media         --    NC_003888.3          s_coelicolor   \n",
       "2                SMM         --    NC_003888.3          s_coelicolor   \n",
       "3                SMM         --    NC_003888.3          s_coelicolor   \n",
       "0        SA (low Fe)         --    NC_016830.1         p_fluorescens   \n",
       "0                 MS         --    NC_018750.1          s_venezuelae   \n",
       "0               EMJH         --    NC_005823.1         l_interrogans   \n",
       "0                BHI         --     BX927147.1          c_glutamicum   \n",
       "1                BHI         --     BX927147.1          c_glutamicum   \n",
       "2                BHI         --     BX927147.1          c_glutamicum   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond         project  \\\n",
       "index                                                             \n",
       "0                    NaN            2         1        MGH78578   \n",
       "0                    NaN            1         1           2457T   \n",
       "1                    NaN            2         1           2457T   \n",
       "0                    NaN            1         1             LT2   \n",
       "1                    NaN            6         3             LT2   \n",
       "0                    NaN            2         1           2457T   \n",
       "0                    NaN            2         1             LAC   \n",
       "1                    NaN            8         1  USA300_TCH1516   \n",
       "2                    NaN            2         1  USA300_TCH1516   \n",
       "0                    NaN            2         1           2457T   \n",
       "0                    NaN            2         1     public_data   \n",
       "1                    NaN            2         1     public_data   \n",
       "2                    NaN            2         1     public_data   \n",
       "3                    NaN            2         1     public_data   \n",
       "4                    NaN            2         1     public_data   \n",
       "5                    NaN            2         1     public_data   \n",
       "6                    NaN            2         1     public_data   \n",
       "7                    NaN            2         1     public_data   \n",
       "8                    NaN            2         1     public_data   \n",
       "9                    NaN            2         1     public_data   \n",
       "10                   NaN            2         1     public_data   \n",
       "11                   NaN            2         1     public_data   \n",
       "12                   NaN            2         1     public_data   \n",
       "13                   NaN            2         1     public_data   \n",
       "0                    NaN            3         1     public_data   \n",
       "1                    NaN            1         1     public_data   \n",
       "2                    NaN            1         1     public_data   \n",
       "3                    NaN            1         1     public_data   \n",
       "0                    NaN            1         1     public_data   \n",
       "0                    NaN            2         1     public_data   \n",
       "0                    NaN            2         1     public_data   \n",
       "0                    NaN            2         1     public_data   \n",
       "1                    NaN            1         1     public_data   \n",
       "2                    NaN            1         1     public_data   \n",
       "\n",
       "                                          binding tables    accession  \\\n",
       "index                                                                   \n",
       "0                           Fe|fur_Fe_binding_table.json    GSE181770   \n",
       "0                           fe|fur_fe_binding_table.json    GSE181775   \n",
       "1                       M9|sigma d_M9_binding_table.json    GSE181775   \n",
       "0                           Fe|fur_Fe_binding_table.json    GSE181770   \n",
       "1      MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...    GSE181791   \n",
       "0                           Fe|fur_Fe_binding_table.json    GSE181770   \n",
       "0                          M9|cody_M9_binding_table.json    GSE159856   \n",
       "1                      RPMI|sigs_RPMI_binding_table.json    GSE182080   \n",
       "2                      RPMI|vrar_RPMI_binding_table.json    GSE182080   \n",
       "0                           Fe|fur_Fe_binding_table.json    GSE181770   \n",
       "0                          LB|fleq_LB_binding_table.json    GSE128430   \n",
       "1                          LB|gbdr_LB_binding_table.json    GSE128430   \n",
       "2                          LB|pchr_LB_binding_table.json    GSE128430   \n",
       "3                          LB|phob_LB_binding_table.json    GSE128430   \n",
       "4                          LB|sphr_LB_binding_table.json    GSE128430   \n",
       "5                          LB|mvfr_LB_binding_table.json     GSE65356   \n",
       "6                          LB|soxr_LB_binding_table.json     GSE65356   \n",
       "7                          LB|rpon_LB_binding_table.json     GSE65356   \n",
       "8                           LB|vfr_LB_binding_table.json     GSE65356   \n",
       "9                          LB|algr_LB_binding_table.json     GSE65356   \n",
       "10                         LB|exsa_LB_binding_table.json     GSE65356   \n",
       "11                         LB|gaca_LB_binding_table.json     GSE65356   \n",
       "12                         LB|mext_LB_binding_table.json     GSE65356   \n",
       "13                         LB|wspr_LB_binding_table.json     GSE65356   \n",
       "0                          MS|hrdb_MS_binding_table.json  E-MTAB-6926   \n",
       "1      minimal-media|ndgr_minimal-media_binding_table...     GSE59010   \n",
       "2                        SMM|scbr_SMM_binding_table.json     GSE64903   \n",
       "3                       SMM|scbr2_SMM_binding_table.json     GSE64903   \n",
       "0                                                    NaN    SRP039494   \n",
       "0                          MS|lsr2_MS_binding_table.json    GSE115439   \n",
       "0                     EMJH|lexa1_EMJH_binding_table.json     GSE73688   \n",
       "0                                                    NaN  PRJNA529385   \n",
       "1                                                    NaN  PRJNA529385   \n",
       "2                                                    NaN  PRJNA529385   \n",
       "\n",
       "                                              doi     peak_intensity_measure  \\\n",
       "index                                                                          \n",
       "0                                             NaN                   MACE S/N   \n",
       "0                                             NaN                   MACE S/N   \n",
       "1                                             NaN                   MACE S/N   \n",
       "0                                             NaN                   MACE S/N   \n",
       "1                                             NaN                   MACE S/N   \n",
       "0                                             NaN                   MACE S/N   \n",
       "0       https://doi.org/10.1101/2021.01.08.426013                   MACE S/N   \n",
       "1                                             NaN                   MACE S/N   \n",
       "2                                             NaN                   MACE S/N   \n",
       "0                                             NaN                   MACE S/N   \n",
       "0      https://doi.org/10.1038/s41467-019-10778-w       MACS fold enrichment   \n",
       "1      https://doi.org/10.1038/s41467-019-10778-w       MACS fold enrichment   \n",
       "2      https://doi.org/10.1038/s41467-019-10778-w       MACS fold enrichment   \n",
       "3      https://doi.org/10.1038/s41467-019-10778-w       MACS fold enrichment   \n",
       "4      https://doi.org/10.1038/s41467-019-10778-w       MACS fold enrichment   \n",
       "5              https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "6              https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "7              https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "8              https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "9              https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "10             https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "11             https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "12             https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "13             https://doi.org/10.1093/nar/gkv747       MACS fold enrichment   \n",
       "0             https://doi.org/10.1093/nar/gky1018       MACS fold enrichment   \n",
       "1       https://doi.org/10.1186/s12864-015-1311-0       MACS fold enrichment   \n",
       "2               https://doi.org/10.1038/srep14831       MACS fold enrichment   \n",
       "3               https://doi.org/10.1038/srep14831       MACS fold enrichment   \n",
       "0        https://doi.org/10.1186/1471-2164-15-237                        NaN   \n",
       "0             https://doi.org/10.7554/eLife.47691            Fold difference   \n",
       "0             https://doi.org/10.1093/nar/gkv1536  CisGenome fold enrichment   \n",
       "0      https://doi.org/10.1038/s41467-020-15238-4                        NaN   \n",
       "1      https://doi.org/10.1038/s41467-020-15238-4                        NaN   \n",
       "2      https://doi.org/10.1038/s41467-020-15238-4                        NaN   \n",
       "\n",
       "         Method    TF Category  Note    BWcond1_1_name  \\\n",
       "index                                                    \n",
       "0      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "0      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "1      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "0      ChIP-exo  1 – Single TF   NaN               NaN   \n",
       "1      ChIP-exo  1 – Single TF   NaN         MES5 – R1   \n",
       "0      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "0      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "1      ChIP-exo  1 – Single TF   NaN           M1 – R1   \n",
       "2      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "0      ChIP-exo  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "1      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "2      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "3      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "4      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "5      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "6      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "7      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "8      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "9      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "10     ChIP-seq  1 – Single TF   NaN                R1   \n",
       "11     ChIP-seq  1 – Single TF   NaN                R1   \n",
       "12     ChIP-seq  1 – Single TF   NaN                R1   \n",
       "13     ChIP-seq  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "1      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "2      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "3      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN                R1   \n",
       "0      ChIP-seq  1 – Single TF   NaN  NeonGreen 1 – R1   \n",
       "1      ChIP-seq  1 – Single TF   NaN    mCherry 1 – R1   \n",
       "2      ChIP-seq  1 – Single TF   NaN    mCherry 1 – R1   \n",
       "\n",
       "                             BWcond1_1    BWcond1_2_name  \\\n",
       "index                                                      \n",
       "0                     klebfurfinal1.bw                R2   \n",
       "0            putidafurfinal1_202003.bw               NaN   \n",
       "1               putidaSigmaD_202003.bw                R2   \n",
       "0                                  NaN                R2   \n",
       "1               salmone_ydcI_MES5_1.bw         MES5 – R2   \n",
       "0                    shigella_fur_1.bw                R2   \n",
       "0                StaphLAC_new_cody1.bw                R2   \n",
       "1                  StaphTCH_sigs_M1.bw           M1 – R2   \n",
       "2      StaphTCH_vraR_7H7C11purified.bw                R2   \n",
       "0                    yersifurfinal1.bw                R2   \n",
       "0                        SRR8742210.bw                R2   \n",
       "1                        SRR8742212.bw                R2   \n",
       "2                        SRR8742214.bw                R2   \n",
       "3                        SRR8742216.bw                R2   \n",
       "4                        SRR8742218.bw                R2   \n",
       "5                        SRR1776542.bw                R2   \n",
       "6                        SRR1776544.bw                R2   \n",
       "7                        SRR1776546.bw                R2   \n",
       "8                        SRR1776548.bw                R2   \n",
       "9                        SRR1776550.bw                R2   \n",
       "10                       SRR1776552.bw                R2   \n",
       "11                       SRR1776554.bw                R2   \n",
       "12                       SRR1776558.bw                R2   \n",
       "13                       SRR1776560.bw                R2   \n",
       "0                        ERR2639621.bw                R2   \n",
       "1                        SRR1503874.bw               NaN   \n",
       "2                        SRR1752661.bw               NaN   \n",
       "3                        SRR1752662.bw               NaN   \n",
       "0                        SRR1184483.bw               NaN   \n",
       "0                        SRR7282569.bw                R2   \n",
       "0                        SRR2549977.bw                R2   \n",
       "0                        SRR8791699.bw  NeonGreen 1 – R2   \n",
       "1                        SRR8791676.bw    mCherry 1 – R2   \n",
       "2                        SRR8791691.bw    mCherry 1 – R2   \n",
       "\n",
       "                              BWcond1_2    BWcond2_1_name  \\\n",
       "index                                                       \n",
       "0                      klebfurfinal2.bw               NaN   \n",
       "0                                   NaN               NaN   \n",
       "1                putidaSigmaD_202005.bw               NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw               NaN   \n",
       "1                salmone_ydcI_MES5_2.bw         MES5 – R3   \n",
       "0                     shigella_fur_2.bw               NaN   \n",
       "0                 StaphLAC_new_cody2.bw               NaN   \n",
       "1                   Staph_SigS_M1_R2.bw           M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw               NaN   \n",
       "0                     yersifurfinal2.bw               NaN   \n",
       "0                         SRR8742211.bw               NaN   \n",
       "1                         SRR8742213.bw               NaN   \n",
       "2                         SRR8742215.bw               NaN   \n",
       "3                         SRR8742217.bw               NaN   \n",
       "4                         SRR8742219.bw               NaN   \n",
       "5                         SRR1776543.bw               NaN   \n",
       "6                         SRR1776545.bw               NaN   \n",
       "7                         SRR1776547.bw               NaN   \n",
       "8                         SRR1776549.bw               NaN   \n",
       "9                         SRR1776551.bw               NaN   \n",
       "10                        SRR1776553.bw               NaN   \n",
       "11                        SRR1776555.bw               NaN   \n",
       "12                        SRR1776559.bw               NaN   \n",
       "13                        SRR1776561.bw               NaN   \n",
       "0                         ERR2639622.bw               NaN   \n",
       "1                                   NaN               NaN   \n",
       "2                                   NaN               NaN   \n",
       "3                                   NaN               NaN   \n",
       "0                                   NaN               NaN   \n",
       "0                         SRR7282569.bw               NaN   \n",
       "0                         SRR2549979.bw               NaN   \n",
       "0                         SRR8791700.bw  NeonGreen 2 – R1   \n",
       "1                         SRR8791677.bw    mCherry 2 – R1   \n",
       "2                         SRR8791692.bw    mCherry 2 – R1   \n",
       "\n",
       "                    BWcond2_1    BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                     \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw         MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw           M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "4                         NaN               NaN                     NaN   \n",
       "5                         NaN               NaN                     NaN   \n",
       "6                         NaN               NaN                     NaN   \n",
       "7                         NaN               NaN                     NaN   \n",
       "8                         NaN               NaN                     NaN   \n",
       "9                         NaN               NaN                     NaN   \n",
       "10                        NaN               NaN                     NaN   \n",
       "11                        NaN               NaN                     NaN   \n",
       "12                        NaN               NaN                     NaN   \n",
       "13                        NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0               SRR8791701.bw  NeonGreen 2 – R2           SRR8791702.bw   \n",
       "1               SRR8791678.bw    mCherry 2 – R2           SRR8791679.bw   \n",
       "2               SRR8791697.bw    mCherry 2 – R2           SRR8791698.bw   \n",
       "\n",
       "       BWcond3_1_name               BWcond3_1  BWcond3_2_name  \\\n",
       "index                                                           \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1            ph5 – R1  salmone_ydcI_ph55_1.bw        ph5 – R2   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1             M3 – R1     StaphTCH_sigs_M3.bw         M3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "4                 NaN                     NaN             NaN   \n",
       "5                 NaN                     NaN             NaN   \n",
       "6                 NaN                     NaN             NaN   \n",
       "7                 NaN                     NaN             NaN   \n",
       "8                 NaN                     NaN             NaN   \n",
       "9                 NaN                     NaN             NaN   \n",
       "10                NaN                     NaN             NaN   \n",
       "11                NaN                     NaN             NaN   \n",
       "12                NaN                     NaN             NaN   \n",
       "13                NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0        mCherry – R1           SRR8791680.bw    mCherry – R2   \n",
       "1      mCherry 3 – R1           SRR8791682.bw  mCherry 3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "4                         NaN            NaN                  NaN   \n",
       "5                         NaN            NaN                  NaN   \n",
       "6                         NaN            NaN                  NaN   \n",
       "7                         NaN            NaN                  NaN   \n",
       "8                         NaN            NaN                  NaN   \n",
       "9                         NaN            NaN                  NaN   \n",
       "10                        NaN            NaN                  NaN   \n",
       "11                        NaN            NaN                  NaN   \n",
       "12                        NaN            NaN                  NaN   \n",
       "13                        NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0               SRR8791681.bw            NaN                  NaN   \n",
       "1               SRR8791683.bw            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "4                NaN                  NaN  \n",
       "5                NaN                  NaN  \n",
       "6                NaN                  NaN  \n",
       "7                NaN                  NaN  \n",
       "8                NaN                  NaN  \n",
       "9                NaN                  NaN  \n",
       "10               NaN                  NaN  \n",
       "11               NaN                  NaN  \n",
       "12               NaN                  NaN  \n",
       "13               NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "amrz\n",
      "mksb\n",
      "parb\n",
      "smc\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] \n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "\n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        print(TF_name)\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list.insert(12, \"curated information\", TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')\n",
    "\n",
    "#individual file updates\n",
    "for org in new_TF_list.organism_id.unique():\n",
    "    new_TF_list[new_TF_list.organism_id == org].to_csv(\"../data/\"+org+\"/TF_list.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - other strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         TF          Organism       Strain Media  \\\n",
       "index                                              \n",
       "0      BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1      CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2       Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3       Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4      GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond   project  \\\n",
       "index                                                       \n",
       "0                    NaN          4.0       1.0       TCS   \n",
       "1                    NaN          2.0       1.0       TCS   \n",
       "2                    NaN          8.0       4.0  known_TF   \n",
       "3                    NaN          4.0       2.0  known_TF   \n",
       "4                    NaN          2.0       1.0  known_TF   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "\n",
       "                                             doi peak_intensity_measure  \\\n",
       "index                                                                     \n",
       "0      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "1      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "2             https://doi.org/10.1093/nar/gky069               MACE S/N   \n",
       "3             https://doi.org/10.1038/ncomms5910               MACE S/N   \n",
       "4             https://doi.org/10.1038/ncomms8970               MACE S/N   \n",
       "\n",
       "         Method    TF Category Note BWcond1_1_name          BWcond1_1  \\\n",
       "index                                                                   \n",
       "0      ChIP-exo  1 – Single TF  NaN  bio-rep1 – R1  BaeR_R1_S31_R1.bw   \n",
       "1      ChIP-exo  1 – Single TF  NaN             R1    CpxRR1_S1_R1.bw   \n",
       "2      ChIP-exo  1 – Single TF  NaN       Glu – R1       cra_glu_1.bw   \n",
       "3      ChIP-exo  1 – Single TF  NaN        Fe – R1        fur_fe_1.bw   \n",
       "4      ChIP-exo  1 – Single TF  NaN             R1          gade_1.bw   \n",
       "\n",
       "      BWcond1_2_name          BWcond1_2 BWcond2_1_name           BWcond2_1  \\\n",
       "index                                                                        \n",
       "0      bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1  BaerR_R2_S32_R1.bw   \n",
       "1                 R2    CpxRR2_S2_R2.bw            NaN                 NaN   \n",
       "2           Glu – R2       cra_glu_2.bw       Fru – R1        cra_fru_1.bw   \n",
       "3            Fe – R2        fur_fe_2.bw       DPD – R1        fur_dpd_1.bw   \n",
       "4                 R2          gade_2.bw            NaN                 NaN   \n",
       "\n",
       "      BWcond2_2_name           BWcond2_2 BWcond3_1_name     BWcond3_1  \\\n",
       "index                                                                   \n",
       "0      bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN           NaN   \n",
       "1                NaN                 NaN            NaN           NaN   \n",
       "2           Fru – R2        cra_fru_2.bw       Gal – R1  cra_gal_1.bw   \n",
       "3           DPD – R2        fur_dpd_2.bw            NaN           NaN   \n",
       "4                NaN                 NaN            NaN           NaN   \n",
       "\n",
       "      BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                             \n",
       "0                NaN           NaN            NaN           NaN   \n",
       "1                NaN           NaN            NaN           NaN   \n",
       "2           Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3                NaN           NaN            NaN           NaN   \n",
       "4                NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_folder = \"../data/e_coli/\"\n",
    "\n",
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "    \n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "    \n",
    "    if genome_id == 'NC_000913.3':\n",
    "            continue\n",
    "            \n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "            \n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "\n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Metadata for manuscript table S1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._set_item(key, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return func(*args, **kwargs)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:10: FutureWarning: As the xlwt package is no longer maintained, the xlwt engine will be removed in a future version of pandas. This is the only engine in pandas that supports writing in the xls format. Install openpyxl and write to an xlsx file instead. You can set the option io.excel.xls.writer to 'xlwt' to silence this warning. While this option is deprecated and will also raise a warning, it can be globally set and the warning suppressed.\n",
      "  # Remove the CWD from sys.path while we load stuff.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>TF</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>v1.0.0 TF dashboard link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Organism</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>MksB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>ParB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>Smc</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArcA</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArgR</td>\n",
       "      <td>GSE60546</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv150</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Strain    TF    accession  \\\n",
       "Organism                                                     \n",
       "Corynebacterium glutamicum   ATCC 13032  MksB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032  ParB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032   Smc  PRJNA529385   \n",
       "Escherichia coli            K-12 MG1655  ArcA    GSE182077   \n",
       "Escherichia coli            K-12 MG1655  ArgR     GSE60546   \n",
       "\n",
       "                                                                   doi  \\\n",
       "Organism                                                                 \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Escherichia coli                                                   NaN   \n",
       "Escherichia coli                    https://doi.org/10.1093/nar/gkv150   \n",
       "\n",
       "                                                     v1.0.0 TF dashboard link  \n",
       "Organism                                                                       \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.org/tf_dashboard.html?organi...  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_ecoli = pd.read_csv('../data/e_coli/TF_list.csv', index_col=0)\n",
    "meta_other = pd.read_csv('../data/all_other/TF_list.csv', index_col=0)\n",
    "\n",
    "meta_combined = pd.concat([meta_ecoli, meta_other], axis=0)\n",
    "table_s1 = meta_combined[['Organism', 'Strain', 'TF', 'accession', 'doi']]\n",
    "table_s1['v1.0.0 TF dashboard link'] = 'https://prochipdb.org/tf_dashboard.html?organism='+meta_combined.organism_id+'&tf='+meta_combined.TF+'&genome='+meta_combined.genome_id+'&i='+[str(i) for i in meta_combined.index.to_list()]\n",
    "table_s1.sort_values(by=['Organism','Strain','TF'], inplace=True)\n",
    "table_s1.set_index('Organism', inplace=True)\n",
    "\n",
    "table_s1.to_excel('../../Supplementary Table 1.xls')\n",
    "table_s1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "e coli dataset\n",
      "num org 1\n",
      "num strain 4\n",
      "num TF 63\n",
      "num ChIP-exo 167.0 92.3%\n",
      "num ChIP-exo 14.0 7.7%\n",
      "num sample 181.0\n",
      "percent curated 92.1 58\n"
     ]
    }
   ],
   "source": [
    "print('e coli dataset')\n",
    "print('num org', len(meta_ecoli.Organism.unique()))\n",
    "print('num strain', len(meta_ecoli.Strain.unique()))\n",
    "print('num TF', len(meta_ecoli))\n",
    "num_samples = meta_ecoli.num_samples.sum()\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_ecoli['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 12\n",
      "num strain 13\n",
      "num TF 34\n",
      "num ChIP-exo 28 39.4%\n",
      "num ChIP-exo 43 60.6%\n",
      "num sample 71\n",
      "percent curated 88.2 30\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_other.Organism.unique()))\n",
    "print('num strain', len(meta_other.Strain.unique()))\n",
    "print('num TF', len(meta_other))\n",
    "num_samples = meta_other.num_samples.sum()\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_other['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 13\n",
      "num strain 17\n",
      "num TF 97\n",
      "num ChIP-exo 195.0 77.4%\n",
      "num ChIP-exo 57.0 22.6%\n",
      "num sample 252.0\n",
      "percent curated 90.7 88\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_combined.Organism.unique()))\n",
    "print('num strain', len(meta_combined.Strain.unique()))\n",
    "print('num TF', len(meta_combined))\n",
    "num_samples = meta_combined.num_samples.sum()\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_combined['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "205.4px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
