{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for proChIPdb, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "from zipfile import ZipFile\n",
    "from Bio import SeqIO\n",
    "import xml.etree.ElementTree as ET\n",
    "from os.path import exists\n",
    "from PIL import Image, ImageFont, ImageDraw, ImageOps\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + TF_condition + \"-\" + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus_nc000913(peak_start, peak_stop, gene_info):\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start >= BP) & (stop >= BP) & (strand == '+')) | ((start <= BP) & (stop <= BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start <= BP) & (stop >= BP)) | ((start >= BP) & (stop <= BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "        \n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    df_complete['closest_gene'] = [closest_locus_nc000913(df_complete.binding_peak_start.iloc[i], \n",
    "                                           df_complete.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(df_complete))]\n",
    "#     df_complete['target_genes'] = [df_complete['target_genes'].to_list()[i]+\", \"+df_complete['closest_gene'].to_list()[i] if df_complete['closest_gene'].to_list()[i] not in df_complete['target_genes'].to_list()[i] else df_complete['target_genes'].to_list()[i] for i in range(len(df_complete))]\n",
    "\n",
    "            \n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen2(Peak_DF, TRN_data, gene_info): \n",
    "    TF_name = Peak_DF['index'][1].split(\"-\")[0].split(\" \")[0]\n",
    "    \n",
    "\n",
    "    # TRN locus\n",
    "    reg_genes = set(TRN_data.loc[TF_name].gene_id.to_list())\n",
    "\n",
    "    # chip locus\n",
    "    gene_list = [i for i in Peak_DF.target_locus if i != '']\n",
    "    chip_genes = set(','.join(list(gene_list)).split(','))\n",
    "    \n",
    "    #both\n",
    "    both_genes = set(reg_genes & chip_genes)\n",
    "    \n",
    "    # Get gene and operon counts\n",
    "    reg_gene_count = len(reg_genes)\n",
    "    chip_gene_count = len(chip_genes)\n",
    "    both_gene_count = len(both_genes)\n",
    "    \n",
    "    # Add adjustments for venn plotting (add '2' for alternates)\n",
    "    reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    if reg_genes == chip_genes:\n",
    "        reg_gene_count = 0; chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = len(reg_genes)\n",
    "    elif all(item in chip_genes for item in reg_genes):\n",
    "        reg_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = len(reg_genes); chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    elif all(item in reg_genes for item in chip_genes):\n",
    "        chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = len(chip_genes); both_gene_count2 = 0\n",
    "        \n",
    "    res = pd.DataFrame([TF_name, reg_gene_count, chip_gene_count, both_gene_count,\n",
    "                        reg_gene_count2, chip_gene_count2, both_gene_count2], \n",
    "                        columns=['Value'],\n",
    "                        index=['TF', 'reg_genes', 'chip_genes', 'both_genes',\n",
    "                                'reg_genes2', 'chip_genes2', 'both_genes2'])\n",
    "    \n",
    "    # Add gene lists\n",
    "    just_reg = reg_genes - both_genes\n",
    "    just_chip = chip_genes - both_genes\n",
    "    for i, l in zip(['reg_genes', 'chip_genes', 'both_genes'],[just_reg, just_chip, both_genes]):\n",
    "        gene_list = np.array([gene_info.loc[g,'gene_name'] if g in gene_info.index else g for g in l])\n",
    "        gene_list = np.array2string(gene_list, separator = ' ')\n",
    "        res.loc[i, 'list'] = gene_list\n",
    "    \n",
    "    # Add data sources\n",
    "    res.loc['TF', 'list'] = '; '.join(TRN_data.source[TRN_data.index == TF_name].unique())\n",
    "    \n",
    "    return res"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        gene = peak.closest_gene\n",
    "        gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "        try:\n",
    "            if gene_info.strand[0] == \"-\":\n",
    "                dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "            else:\n",
    "                dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "            peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "            counter+=1\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - NC_000913.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>mixed-TFs-pool4</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>Mixed TFs: YidZ, YedW, ArcA, YhjB</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool4_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool4_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>mixed-TFs-pool5</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Galactose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>Mixed TFs: YebK, YbeF</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool5_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool5_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ArcA-1</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca-1_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>monoclonal antibodies (ArcA part1)</td>\n",
       "      <td>1C6</td>\n",
       "      <td>arcA_1C6.bw</td>\n",
       "      <td>1D6</td>\n",
       "      <td>arcA_1D6.bw</td>\n",
       "      <td>2A1</td>\n",
       "      <td>arcA_2A1.bw</td>\n",
       "      <td>2B6</td>\n",
       "      <td>arcA_2B6.bw</td>\n",
       "      <td>2C4</td>\n",
       "      <td>arcA_2C4.bw</td>\n",
       "      <td>2D6</td>\n",
       "      <td>arcA_2D6.bw</td>\n",
       "      <td>4A1</td>\n",
       "      <td>arcA_4A1.bw</td>\n",
       "      <td>4A3</td>\n",
       "      <td>arcA_4A3.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>ArcA-2</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca-2_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>monoclonal antibodies (ArcA part2)</td>\n",
       "      <td>4D6</td>\n",
       "      <td>arcA_4D6.bw</td>\n",
       "      <td>5A10</td>\n",
       "      <td>arcA_5A10.bw</td>\n",
       "      <td>5B6</td>\n",
       "      <td>arcA_5B6.bw</td>\n",
       "      <td>5B9</td>\n",
       "      <td>arcA_5B9.bw</td>\n",
       "      <td>5F9</td>\n",
       "      <td>arcA_5F9.bw</td>\n",
       "      <td>5G8</td>\n",
       "      <td>arcA_5G8.bw</td>\n",
       "      <td>7B1</td>\n",
       "      <td>arcA_7B10.bw</td>\n",
       "      <td>8B2</td>\n",
       "      <td>arcA_8B2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>UvrY</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE74810</td>\n",
       "      <td>https://dx.doi.org/10.1371%2Fjournal.pone.0145035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2917944.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    TF          Organism       Strain Media  \\\n",
       "index                                                         \n",
       "0                 BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1                 CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2                  Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3                  Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4                 GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "...                ...               ...          ...   ...   \n",
       "60     mixed-TFs-pool4  Escherichia coli  K-12 MG1655    M9   \n",
       "61     mixed-TFs-pool5  Escherichia coli  K-12 MG1655    M9   \n",
       "62              ArcA-1  Escherichia coli  K-12 MG1655    M9   \n",
       "63              ArcA-2  Escherichia coli  K-12 MG1655    M9   \n",
       "64                UvrY  Escherichia coli  K-12 MG1655    LB   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "...                                              ...          ...         ...   \n",
       "60                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "61                                    0.2% Galactose  NC_000913.3      e_coli   \n",
       "62                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "63                                      0.2% Acetate  NC_000913.3      e_coli   \n",
       "64                                                --  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond      project  \\\n",
       "index                                                          \n",
       "0                    NaN            4         1          TCS   \n",
       "1                    NaN            2         1          TCS   \n",
       "2                    NaN            8         4     known_TF   \n",
       "3                    NaN            4         2     known_TF   \n",
       "4                    NaN            2         1     known_TF   \n",
       "...                  ...          ...       ...          ...   \n",
       "60                   NaN            2         1  combination   \n",
       "61                   NaN            2         1  combination   \n",
       "62                   NaN            8         1         mAbs   \n",
       "63                   NaN            8         1         mAbs   \n",
       "64                   NaN            1         1       public   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "...                                                  ...        ...   \n",
       "60      M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json  GSE182078   \n",
       "61      M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json  GSE182078   \n",
       "62               M9+ace|arca-1_M9+ace_binding_table.json  GSE182077   \n",
       "63               M9+ace|arca-2_M9+ace_binding_table.json  GSE182077   \n",
       "64                                                   NaN   GSE74810   \n",
       "\n",
       "                                                     doi  \\\n",
       "index                                                      \n",
       "0              https://doi.org/10.1128/mSystems.00980-20   \n",
       "1              https://doi.org/10.1128/mSystems.00980-20   \n",
       "2                     https://doi.org/10.1093/nar/gky069   \n",
       "3                     https://doi.org/10.1038/ncomms5910   \n",
       "4                     https://doi.org/10.1038/ncomms8970   \n",
       "...                                                  ...   \n",
       "60                                                   NaN   \n",
       "61                                                   NaN   \n",
       "62                                                   NaN   \n",
       "63                                                   NaN   \n",
       "64     https://dx.doi.org/10.1371%2Fjournal.pone.0145035   \n",
       "\n",
       "      peak_intensity_measure    Method    TF Category  \\\n",
       "index                                                   \n",
       "0                   MACE S/N  ChIP-exo  1 – Single TF   \n",
       "1                   MACE S/N  ChIP-exo  1 – Single TF   \n",
       "2                   MACE S/N  ChIP-exo  1 – Single TF   \n",
       "3                   MACE S/N  ChIP-exo  1 – Single TF   \n",
       "4                   MACE S/N  ChIP-exo  1 – Single TF   \n",
       "...                      ...       ...            ...   \n",
       "60                  MACE S/N  ChIP-exo   3 – Mixed TF   \n",
       "61                  MACE S/N  ChIP-exo   3 – Mixed TF   \n",
       "62                  MACE S/N  ChIP-exo       2 – mAbs   \n",
       "63                  MACE S/N  ChIP-exo       2 – mAbs   \n",
       "64                       NaN  ChIP-exo  1 – Single TF   \n",
       "\n",
       "                                     Note BWcond1_1_name          BWcond1_1  \\\n",
       "index                                                                         \n",
       "0                                     NaN  bio-rep1 – R1  BaeR_R1_S31_R1.bw   \n",
       "1                                     NaN             R1    CpxRR1_S1_R1.bw   \n",
       "2                                     NaN       Glu – R1       cra_glu_1.bw   \n",
       "3                                     NaN        Fe – R1        fur_fe_1.bw   \n",
       "4                                     NaN             R1          gade_1.bw   \n",
       "...                                   ...            ...                ...   \n",
       "60      Mixed TFs: YidZ, YedW, ArcA, YhjB             R1   mixTF_pool4_1.bw   \n",
       "61                  Mixed TFs: YebK, YbeF             R1   mixTF_pool5_1.bw   \n",
       "62     monoclonal antibodies (ArcA part1)            1C6        arcA_1C6.bw   \n",
       "63     monoclonal antibodies (ArcA part2)            4D6        arcA_4D6.bw   \n",
       "64                                    NaN             R1      SRR2917944.bw   \n",
       "\n",
       "      BWcond1_2_name          BWcond1_2 BWcond2_1_name           BWcond2_1  \\\n",
       "index                                                                        \n",
       "0      bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1  BaerR_R2_S32_R1.bw   \n",
       "1                 R2    CpxRR2_S2_R2.bw            NaN                 NaN   \n",
       "2           Glu – R2       cra_glu_2.bw       Fru – R1        cra_fru_1.bw   \n",
       "3            Fe – R2        fur_fe_2.bw       DPD – R1        fur_dpd_1.bw   \n",
       "4                 R2          gade_2.bw            NaN                 NaN   \n",
       "...              ...                ...            ...                 ...   \n",
       "60                R2   mixTF_pool4_2.bw            NaN                 NaN   \n",
       "61                R2   mixTF_pool5_2.bw            NaN                 NaN   \n",
       "62               1D6        arcA_1D6.bw            2A1         arcA_2A1.bw   \n",
       "63              5A10       arcA_5A10.bw            5B6         arcA_5B6.bw   \n",
       "64               NaN                NaN            NaN                 NaN   \n",
       "\n",
       "      BWcond2_2_name           BWcond2_2 BWcond3_1_name     BWcond3_1  \\\n",
       "index                                                                   \n",
       "0      bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN           NaN   \n",
       "1                NaN                 NaN            NaN           NaN   \n",
       "2           Fru – R2        cra_fru_2.bw       Gal – R1  cra_gal_1.bw   \n",
       "3           DPD – R2        fur_dpd_2.bw            NaN           NaN   \n",
       "4                NaN                 NaN            NaN           NaN   \n",
       "...              ...                 ...            ...           ...   \n",
       "60               NaN                 NaN            NaN           NaN   \n",
       "61               NaN                 NaN            NaN           NaN   \n",
       "62               2B6         arcA_2B6.bw            2C4   arcA_2C4.bw   \n",
       "63               5B9         arcA_5B9.bw            5F9   arcA_5F9.bw   \n",
       "64               NaN                 NaN            NaN           NaN   \n",
       "\n",
       "      BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                             \n",
       "0                NaN           NaN            NaN           NaN   \n",
       "1                NaN           NaN            NaN           NaN   \n",
       "2           Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3                NaN           NaN            NaN           NaN   \n",
       "4                NaN           NaN            NaN           NaN   \n",
       "...              ...           ...            ...           ...   \n",
       "60               NaN           NaN            NaN           NaN   \n",
       "61               NaN           NaN            NaN           NaN   \n",
       "62               2D6   arcA_2D6.bw            4A1   arcA_4A1.bw   \n",
       "63               5G8   arcA_5G8.bw            7B1  arcA_7B10.bw   \n",
       "64               NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  \n",
       "...              ...           ...  \n",
       "60               NaN           NaN  \n",
       "61               NaN           NaN  \n",
       "62               4A3   arcA_4A3.bw  \n",
       "63               8B2   arcA_8B2.bw  \n",
       "64               NaN           NaN  \n",
       "\n",
       "[65 rows x 34 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       start  stop strand gene_name  length   operon  \\\n",
       "b0001    189   255      +      thrL      66  thrLABC   \n",
       "b0002    336  2799      +      thrA    2463  thrLABC   \n",
       "b0003   2800  3733      +      thrB     933  thrLABC   \n",
       "b0004   3733  5020      +      thrC    1287  thrLABC   \n",
       "b0005   5233  5530      +      yaaX     297     yaaX   \n",
       "\n",
       "                                       cog  start_codon_pos  \n",
       "b0001                    No COG Annotation              189  \n",
       "b0002                    No COG Annotation              336  \n",
       "b0003  Amino acid transport and metabolism             2800  \n",
       "b0004  Amino acid transport and metabolism             3733  \n",
       "b0005                     Function unknown             5233  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913.3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "baer\n",
      "cpxr\n",
      "cra\n",
      "fur\n",
      "gade\n",
      "gadw\n",
      "gadx\n",
      "kdpe\n",
      "phob\n",
      "rpob\n",
      "ybao\n",
      "ybaq\n",
      "ydci\n",
      "yddm\n",
      "yheo\n",
      "yiag\n",
      "yiep\n",
      "yihl\n",
      "yihw\n",
      "yjdc\n",
      "yjhi\n",
      "ynej\n",
      "ynfl\n",
      "ypdc\n",
      "yqhc\n",
      "zrar\n",
      "fur\n",
      "fur\n",
      "fur\n",
      "dps\n",
      "fis\n",
      "gyra\n",
      "gyrb\n",
      "h-ns\n",
      "hupa\n",
      "hupb\n",
      "infa\n",
      "mukb\n",
      "rdgc\n",
      "rob\n",
      "stpa\n",
      "topa\n",
      "tus\n",
      "narl\n",
      "puur\n",
      "rcsa\n",
      "rpob (cra and crp ko exps)\n",
      "flhc\n",
      "flhd\n",
      "flia\n",
      "fnr\n",
      "rpod\n",
      "argr\n",
      "ompr\n",
      "rpoh\n",
      "mixed-tfs-pool1a\n",
      "mixed-tfs-pool1b\n",
      "mixed-tfs-pool1c\n",
      "mixed-tfs-pool2\n",
      "mixed-tfs-pool3\n",
      "mixed-tfs-pool4\n",
      "mixed-tfs-pool5\n",
      "arca-1\n",
      "arca-2\n",
      "uvry\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    print(TF_name)\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "    \n",
    "    if strain != 'NC_000913.3':\n",
    "        continue\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] # change split to \"/\" on linux\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    else:\n",
    "        TF_list.loc[TF_index, 'binding tables'] = \"NA\"\n",
    "    TF_list.loc[TF_index, 'num_cond'] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    "    # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "            final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "            final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "\n",
    "    #venn diagrams\n",
    "        try:\n",
    "            if strain == 'NC_000913.3' and TF_list.TF[TF_index].split(\" \")[0] in TRN.index:\n",
    "                Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(org_folder+strain+\"/venn/\"+TF_list.TF[TF_index]+'_venn.csv')\n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(org_folder+strain+\"/bw/compressed/\"+TF_list.TF[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(org_folder+strain+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Other Organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_gff_to_geneinfo(org, strain):\n",
    "    genes = pd.read_csv('../data'+'/'+org+'/'+strain+'/annotation/genes.gff',index_col=0, delimiter='\\t', header=None, \n",
    "                        names = ['genome', 'ref', 'type', 'start','stop','.', 'strand', '-','details'])\n",
    "    genes[\"locus_tag\"] = [gene_detail[0].split(\" \")[1] for gene_detail in genes.details.str.split(\"; \")]\n",
    "    genes.set_index('locus_tag', inplace=True)\n",
    "    genes = genes.drop(['.', '-','ref','type'], axis=1)\n",
    "\n",
    "    return(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus(peak_start, peak_stop, gene_info):\n",
    "    gene_info['gene_name'] = [i.split(\";\")[0].split(\" \")[-1] for i in gene_info.details]\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_table_gen(TF_name, TF_condition, peak_df, gene_info):\n",
    "    \n",
    "    table = peak_df.copy()\n",
    "    table[\"index\"] = [TF_name+\"-\"+TF_condition+\"-\"+str(i+1) for i in range(len(table))]\n",
    "    table[\"condition\"] = [TF_name.lower() + \" + \" + TF_condition for i in range(len(table))]\n",
    "    table.drop([\"condition_name\",\".\", \"ID\"], axis=1, inplace=True)\n",
    "    table[\"target_locus\"] = [closest_locus(table.binding_peak_start.iloc[i], \n",
    "                                           table.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(table))]\n",
    "    table[\"target_genes\"] = table[\"target_locus\"]\n",
    "    table[\"closest_gene\"] = table[\"target_locus\"]\n",
    "\n",
    "    return(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_peak_position_gen(TF, final_annot_DF,out_dir, DF_gene_info): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.iloc[i,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.index == gene]\n",
    "            try:\n",
    "                if gene_info.strand.iloc[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/(gene_info.stop-gene_info.start+1))\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/(gene_info.stop-gene_info.start+1))\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.drop(['direction'], axis=1, inplace=True)\n",
    "    cols = ['index', 'condition', 'binding_peak_start', 'binding_peak_end', 'binding_peak_strength', 'target_locus', \n",
    "            'target_genes', 'closest_gene', 'gene', 'normalized_dist', 'binding_peak_center']\n",
    "    peak_scatter_df = peak_scatter_df[cols]\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/all_other/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Media_new</th>\n",
       "      <th>Supplement_new</th>\n",
       "      <th>Contents</th>\n",
       "      <th>Additional_Info</th>\n",
       "      <th>Motif_Available?</th>\n",
       "      <th>Motif_Parameters</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>CP000647.1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>fe|fur_fe_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>M9|sigma d_M9_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH5, MES</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...</td>\n",
       "      <td>GSE181791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>AE014073.1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>CP035369.1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>M9|cody_M9_binding_table.json</td>\n",
       "      <td>GSE159856</td>\n",
       "      <td>https://doi.org/10.1101/2021.01.08.426013</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>{RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...</td>\n",
       "      <td>All  S.  aureus  strains  were  grown  in  try...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|sigs_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>{RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|vrar_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>{RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>BHI</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NZ_CP009712.1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M9</td>\n",
       "      <td>FeCl2 (0.1 mM), 0.2%(w/v) glucose</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FleQ</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|fleq_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>yes</td>\n",
       "      <td>not available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742210.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742211.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GbdR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gbdr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>yes</td>\n",
       "      <td>not available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742212.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742213.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PchR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|pchr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>yes</td>\n",
       "      <td>not available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742214.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742215.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|phob_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>yes</td>\n",
       "      <td>not available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742216.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742217.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SphR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|sphr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>yes</td>\n",
       "      <td>not available</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742218.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742219.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MvfR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mvfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776542.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776543.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SoxR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|soxr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776544.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776545.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RpoN</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|rpon_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776546.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776547.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Vfr</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|vfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776548.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776549.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AlgR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|algr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/mL); tetracycline (150 μg/mL...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776550.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776551.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ExsA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|exsa_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776552.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776553.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GacA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gaca_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776554.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776555.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MexT</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mext_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776558.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776559.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>WspR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|wspr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>gentamicin (50 μg/ml); tetracycline (150 μg/ml...</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776560.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776561.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HrdB</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>Na-glutamate medium</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|hrdb_MS_binding_table.json</td>\n",
       "      <td>E-MTAB-6926</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky1018</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>30371884.0</td>\n",
       "      <td>Na-glutamate medium</td>\n",
       "      <td>trace element solution (8 mL/L); TMS1 (5.6 mL/L)</td>\n",
       "      <td>FILL</td>\n",
       "      <td>For ChIP-seq analysis, spores stocks were prep...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR2639621.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR2639622.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NdgR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>minimal media</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>minimal-media|ndgr_minimal-media_binding_table...</td>\n",
       "      <td>GSE59010</td>\n",
       "      <td>https://doi.org/10.1186/s12864-015-1311-0</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>25766138.0</td>\n",
       "      <td>Solid minimal media</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{Solid minimal media: {0.05% K2HPO4, 0.02% MgS...</td>\n",
       "      <td>All strains used are S. coelicolor A3 (2) M145...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1503874.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ScbR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26442964.0</td>\n",
       "      <td>SMM (Supplemented liquid minimal media)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{SMM: {81.9 mL PEG 6000 (BDH 6.1% w/v in disti...</td>\n",
       "      <td>S. coelicolor strains were incubated on MS sol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752661.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ScbR2</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr2_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26442964.0</td>\n",
       "      <td>SMM (Supplemented liquid minimal media)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{SMM: {81.9 mL PEG 6000 (BDH 6.1% w/v in disti...</td>\n",
       "      <td>S. coelicolor strains were incubated on MS sol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752662.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AmrZ</td>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>F113</td>\n",
       "      <td>SA (low Fe)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_016830.1</td>\n",
       "      <td>p_fluorescens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRP039494</td>\n",
       "      <td>https://doi.org/10.1186/1471-2164-15-237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>24670089.0</td>\n",
       "      <td>SA medium or LB (Luria-Bertani) medium</td>\n",
       "      <td>IPTG [Isopropyl-β-D-Thiogalactopyranoside] (1 ...</td>\n",
       "      <td>{SA: {}, LB: {1% NaCl, 1% tryptone, 0.5% yeast...</td>\n",
       "      <td>P. fluorescens strains were grown in SA medium...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1184483.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Lsr2</td>\n",
       "      <td>Streptomyces venezuelae</td>\n",
       "      <td>ATCC 10712</td>\n",
       "      <td>MS</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_018750.1</td>\n",
       "      <td>s_venezuelae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|lsr2_MS_binding_table.json</td>\n",
       "      <td>GSE115439</td>\n",
       "      <td>https://doi.org/10.7554/eLife.47691</td>\n",
       "      <td>Fold difference</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31215866.0</td>\n",
       "      <td>MYM (maltose-yeast extract-mannitol) medium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{MYM: {4g/L Maltose, 4g/L Yeast Extract Oxoid,...</td>\n",
       "      <td>Each culture was then grown in 300 mL MYM cult...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LexA1</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Copenhageni Fiocruz L1-130</td>\n",
       "      <td>EMJH</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_005823.1</td>\n",
       "      <td>l_interrogans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>EMJH|lexa1_EMJH_binding_table.json</td>\n",
       "      <td>GSE73688</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv1536</td>\n",
       "      <td>CisGenome fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26762976.0</td>\n",
       "      <td>EMJH</td>\n",
       "      <td>ciprofloxacin (5 μg/mL)</td>\n",
       "      <td>{EMJH: {1 g/L Disodium Phosphate, 0.3g/L Monop...</td>\n",
       "      <td>L. interrogans serovar Copenhageni Fiocruz L1–...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2549977.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR2549979.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MksB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>glucose (4%)</td>\n",
       "      <td>{BHI: {12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NeonGreen 1 – R1</td>\n",
       "      <td>SRR8791699.bw</td>\n",
       "      <td>NeonGreen 1 – R2</td>\n",
       "      <td>SRR8791700.bw</td>\n",
       "      <td>NeonGreen 2 – R1</td>\n",
       "      <td>SRR8791701.bw</td>\n",
       "      <td>NeonGreen 2 – R2</td>\n",
       "      <td>SRR8791702.bw</td>\n",
       "      <td>mCherry – R1</td>\n",
       "      <td>SRR8791680.bw</td>\n",
       "      <td>mCherry – R2</td>\n",
       "      <td>SRR8791681.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ParB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>glucose (4%)</td>\n",
       "      <td>{BHI: {12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791676.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791677.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791678.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791679.bw</td>\n",
       "      <td>mCherry 3 – R1</td>\n",
       "      <td>SRR8791682.bw</td>\n",
       "      <td>mCherry 3 – R2</td>\n",
       "      <td>SRR8791683.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Smc</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>glucose (4%)</td>\n",
       "      <td>{BHI: {12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791691.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791692.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791697.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791698.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EspR</td>\n",
       "      <td>Mycobacterium tuberculosis</td>\n",
       "      <td>H37Rv</td>\n",
       "      <td>7H9 Broth (Difco)</td>\n",
       "      <td>0.2% gly, ADC, 0.05% Tween 80</td>\n",
       "      <td>NC_000962.2</td>\n",
       "      <td>m_tuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE35149</td>\n",
       "      <td>https://doi.org/10.1371/journal.ppat.1002621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>22479184.0</td>\n",
       "      <td>7H9 broth</td>\n",
       "      <td>glycerol (0.2%); Tween 80 (0.05%);  Middlebroo...</td>\n",
       "      <td>{7H9: {Ammonium Sulfate 0.5g/L, L-Glutamic Aci...</td>\n",
       "      <td>Mycobacterium tuberculosis (Mtb) H37Rv and Erd...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR400040.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR400041.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism                      Strain  \\\n",
       "index                                                                     \n",
       "0          Fur        Klebsiella pneumoniae                   MGH 78578   \n",
       "0          Fur           Pseudomonas putida                      KT2440   \n",
       "1      Sigma D           Pseudomonas putida                      KT2440   \n",
       "0          Fur          Salmonella enterica             Typhimurium LT2   \n",
       "1         YdcI          Salmonella enterica             Typhimurium LT2   \n",
       "0          Fur            Shigella flexneri                   2a, 2457T   \n",
       "0         CodY        Staphylococcus aureus              LAC chromosome   \n",
       "1         SigS        Staphylococcus aureus              USA300_TCH1516   \n",
       "2         VraR        Staphylococcus aureus              USA300_TCH1516   \n",
       "0          Fur  Yersinia pseudotuberculosis                    IP 32953   \n",
       "0         FleQ       Pseudomonas aeruginosa                        PAO1   \n",
       "1         GbdR       Pseudomonas aeruginosa                        PAO1   \n",
       "2         PchR       Pseudomonas aeruginosa                        PAO1   \n",
       "3         PhoB       Pseudomonas aeruginosa                        PAO1   \n",
       "4         SphR       Pseudomonas aeruginosa                        PAO1   \n",
       "5         MvfR       Pseudomonas aeruginosa                        PAO1   \n",
       "6         SoxR       Pseudomonas aeruginosa                        PAO1   \n",
       "7         RpoN       Pseudomonas aeruginosa                        PAO1   \n",
       "8          Vfr       Pseudomonas aeruginosa                        PAO1   \n",
       "9         AlgR       Pseudomonas aeruginosa                        PAO1   \n",
       "10        ExsA       Pseudomonas aeruginosa                        PAO1   \n",
       "11        GacA       Pseudomonas aeruginosa                        PAO1   \n",
       "12        MexT       Pseudomonas aeruginosa                        PAO1   \n",
       "13        WspR       Pseudomonas aeruginosa                        PAO1   \n",
       "0         HrdB      Streptomyces coelicolor                  A3(2) M145   \n",
       "1         NdgR      Streptomyces coelicolor                  A3(2) M145   \n",
       "2         ScbR      Streptomyces coelicolor                  A3(2) M145   \n",
       "3        ScbR2      Streptomyces coelicolor                  A3(2) M145   \n",
       "0         AmrZ      Pseudomonas fluorescens                        F113   \n",
       "0         Lsr2      Streptomyces venezuelae                  ATCC 10712   \n",
       "0        LexA1       Leptospira interrogans  Copenhageni Fiocruz L1-130   \n",
       "0         MksB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "1         ParB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "2          Smc   Corynebacterium glutamicum                  ATCC 13032   \n",
       "0         EspR   Mycobacterium tuberculosis                       H37Rv   \n",
       "\n",
       "                     Media                     Supplement      genome_id  \\\n",
       "index                                                                      \n",
       "0                       M9                             Fe     CP000647.1   \n",
       "0                       M9                             Fe    NC_002947.4   \n",
       "1                       M9                             --    NC_002947.4   \n",
       "0                       M9                             Fe    NC_003197.2   \n",
       "1                       M9                       pH5, MES    NC_003197.2   \n",
       "0                       M9                             Fe     AE014073.1   \n",
       "0                     RPMI                         10% LB     CP035369.1   \n",
       "1                     RPMI                         10% LB      NC_010079   \n",
       "2                     RPMI                         10% LB      NC_010079   \n",
       "0                      BHI                             Fe  NZ_CP009712.1   \n",
       "0                       LB                             --    NC_002516.2   \n",
       "1                       LB                             --    NC_002516.2   \n",
       "2                       LB                             --    NC_002516.2   \n",
       "3                       LB                             --    NC_002516.2   \n",
       "4                       LB                             --    NC_002516.2   \n",
       "5                       LB                             --    NC_002516.2   \n",
       "6                       LB                             --    NC_002516.2   \n",
       "7                       LB                             --    NC_002516.2   \n",
       "8                       LB                             --    NC_002516.2   \n",
       "9                       LB                             --    NC_002516.2   \n",
       "10                      LB                             --    NC_002516.2   \n",
       "11                      LB                             --    NC_002516.2   \n",
       "12                      LB                             --    NC_002516.2   \n",
       "13                      LB                             --    NC_002516.2   \n",
       "0      Na-glutamate medium                             --    NC_003888.3   \n",
       "1            minimal media                             --    NC_003888.3   \n",
       "2                      SMM                             --    NC_003888.3   \n",
       "3                      SMM                             --    NC_003888.3   \n",
       "0              SA (low Fe)                             --    NC_016830.1   \n",
       "0                       MS                             --    NC_018750.1   \n",
       "0                     EMJH                             --    NC_005823.1   \n",
       "0                      BHI                             --     BX927147.1   \n",
       "1                      BHI                             --     BX927147.1   \n",
       "2                      BHI                             --     BX927147.1   \n",
       "0        7H9 Broth (Difco)  0.2% gly, ADC, 0.05% Tween 80    NC_000962.2   \n",
       "\n",
       "                organism_id  num_binding_sites  num_samples  num_cond  \\\n",
       "index                                                                   \n",
       "0              k_pneumoniae                NaN            2         1   \n",
       "0                  p_putida                NaN            1         1   \n",
       "1                  p_putida                NaN            2         1   \n",
       "0                s_enterica                NaN            1         1   \n",
       "1                s_enterica                NaN            6         3   \n",
       "0                s_flexneri                NaN            2         1   \n",
       "0                  s_aureus                NaN            2         1   \n",
       "1                  s_aureus                NaN            8         1   \n",
       "2                  s_aureus                NaN            2         1   \n",
       "0      y_pseudotuberculosis                NaN            2         1   \n",
       "0              p_aeruginosa                NaN            2         1   \n",
       "1              p_aeruginosa                NaN            2         1   \n",
       "2              p_aeruginosa                NaN            2         1   \n",
       "3              p_aeruginosa                NaN            2         1   \n",
       "4              p_aeruginosa                NaN            2         1   \n",
       "5              p_aeruginosa                NaN            2         1   \n",
       "6              p_aeruginosa                NaN            2         1   \n",
       "7              p_aeruginosa                NaN            2         1   \n",
       "8              p_aeruginosa                NaN            2         1   \n",
       "9              p_aeruginosa                NaN            2         1   \n",
       "10             p_aeruginosa                NaN            2         1   \n",
       "11             p_aeruginosa                NaN            2         1   \n",
       "12             p_aeruginosa                NaN            2         1   \n",
       "13             p_aeruginosa                NaN            2         1   \n",
       "0              s_coelicolor                NaN            3         1   \n",
       "1              s_coelicolor                NaN            1         1   \n",
       "2              s_coelicolor                NaN            1         1   \n",
       "3              s_coelicolor                NaN            1         1   \n",
       "0             p_fluorescens                NaN            1         1   \n",
       "0              s_venezuelae                NaN            2         1   \n",
       "0             l_interrogans                NaN            2         1   \n",
       "0              c_glutamicum                NaN            2         1   \n",
       "1              c_glutamicum                NaN            1         1   \n",
       "2              c_glutamicum                NaN            1         1   \n",
       "0            m_tuberculosis                NaN            2         1   \n",
       "\n",
       "              project                                     binding tables  \\\n",
       "index                                                                      \n",
       "0            MGH78578                       Fe|fur_Fe_binding_table.json   \n",
       "0               2457T                       fe|fur_fe_binding_table.json   \n",
       "1               2457T                   M9|sigma d_M9_binding_table.json   \n",
       "0                 LT2                       Fe|fur_Fe_binding_table.json   \n",
       "1                 LT2  MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...   \n",
       "0               2457T                       Fe|fur_Fe_binding_table.json   \n",
       "0                 LAC                      M9|cody_M9_binding_table.json   \n",
       "1      USA300_TCH1516                  RPMI|sigs_RPMI_binding_table.json   \n",
       "2      USA300_TCH1516                  RPMI|vrar_RPMI_binding_table.json   \n",
       "0               2457T                       Fe|fur_Fe_binding_table.json   \n",
       "0         public_data                      LB|fleq_LB_binding_table.json   \n",
       "1         public_data                      LB|gbdr_LB_binding_table.json   \n",
       "2         public_data                      LB|pchr_LB_binding_table.json   \n",
       "3         public_data                      LB|phob_LB_binding_table.json   \n",
       "4         public_data                      LB|sphr_LB_binding_table.json   \n",
       "5         public_data                      LB|mvfr_LB_binding_table.json   \n",
       "6         public_data                      LB|soxr_LB_binding_table.json   \n",
       "7         public_data                      LB|rpon_LB_binding_table.json   \n",
       "8         public_data                       LB|vfr_LB_binding_table.json   \n",
       "9         public_data                      LB|algr_LB_binding_table.json   \n",
       "10        public_data                      LB|exsa_LB_binding_table.json   \n",
       "11        public_data                      LB|gaca_LB_binding_table.json   \n",
       "12        public_data                      LB|mext_LB_binding_table.json   \n",
       "13        public_data                      LB|wspr_LB_binding_table.json   \n",
       "0         public_data                      MS|hrdb_MS_binding_table.json   \n",
       "1         public_data  minimal-media|ndgr_minimal-media_binding_table...   \n",
       "2         public_data                    SMM|scbr_SMM_binding_table.json   \n",
       "3         public_data                   SMM|scbr2_SMM_binding_table.json   \n",
       "0         public_data                                                NaN   \n",
       "0         public_data                      MS|lsr2_MS_binding_table.json   \n",
       "0         public_data                 EMJH|lexa1_EMJH_binding_table.json   \n",
       "0         public_data                                                NaN   \n",
       "1         public_data                                                NaN   \n",
       "2         public_data                                                NaN   \n",
       "0         public_data                                                NaN   \n",
       "\n",
       "         accession                                           doi  \\\n",
       "index                                                              \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE181775                                           NaN   \n",
       "1        GSE181775                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "1        GSE181791                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE159856     https://doi.org/10.1101/2021.01.08.426013   \n",
       "1        GSE182080                                           NaN   \n",
       "2        GSE182080                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "1        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "2        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "3        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "4        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "5         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "6         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "7         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "8         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "9         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "10        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "11        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "12        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "13        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "0      E-MTAB-6926           https://doi.org/10.1093/nar/gky1018   \n",
       "1         GSE59010     https://doi.org/10.1186/s12864-015-1311-0   \n",
       "2         GSE64903             https://doi.org/10.1038/srep14831   \n",
       "3         GSE64903             https://doi.org/10.1038/srep14831   \n",
       "0        SRP039494      https://doi.org/10.1186/1471-2164-15-237   \n",
       "0        GSE115439           https://doi.org/10.7554/eLife.47691   \n",
       "0         GSE73688           https://doi.org/10.1093/nar/gkv1536   \n",
       "0      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "1      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "2      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "0         GSE35149  https://doi.org/10.1371/journal.ppat.1002621   \n",
       "\n",
       "          peak_intensity_measure    Method    TF Category        PMID  \\\n",
       "index                                                                   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "2                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "1           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "2           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "3           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "4           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "5           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "6           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "7           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "8           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "9           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "10          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "11          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "12          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "13          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "0           MACS fold enrichment  ChIP-seq  1 – Single TF  30371884.0   \n",
       "1           MACS fold enrichment  ChIP-seq  1 – Single TF  25766138.0   \n",
       "2           MACS fold enrichment  ChIP-seq  1 – Single TF  26442964.0   \n",
       "3           MACS fold enrichment  ChIP-seq  1 – Single TF  26442964.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  24670089.0   \n",
       "0                Fold difference  ChIP-seq  1 – Single TF  31215866.0   \n",
       "0      CisGenome fold enrichment  ChIP-seq  1 – Single TF  26762976.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "1                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "2                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  22479184.0   \n",
       "\n",
       "                                         Media_new  \\\n",
       "index                                                \n",
       "0                                               M9   \n",
       "0                                               M9   \n",
       "1                                               M9   \n",
       "0                                               M9   \n",
       "1                                               M9   \n",
       "0                                               M9   \n",
       "0                                RPMI-1640 (Gibco)   \n",
       "1                                RPMI-1640 (Gibco)   \n",
       "2                                RPMI-1640 (Gibco)   \n",
       "0                                               M9   \n",
       "0                         LB (Luria-Bertani) broth   \n",
       "1                         LB (Luria-Bertani) broth   \n",
       "2                         LB (Luria-Bertani) broth   \n",
       "3                         LB (Luria-Bertani) broth   \n",
       "4                         LB (Luria-Bertani) broth   \n",
       "5                         LB (Luria-Bertani) broth   \n",
       "6                         LB (Luria-Bertani) broth   \n",
       "7                         LB (Luria-Bertani) broth   \n",
       "8                         LB (Luria-Bertani) broth   \n",
       "9                         LB (Luria-Bertani) broth   \n",
       "10                        LB (Luria-Bertani) broth   \n",
       "11                        LB (Luria-Bertani) broth   \n",
       "12                        LB (Luria-Bertani) broth   \n",
       "13                        LB (Luria-Bertani) broth   \n",
       "0                              Na-glutamate medium   \n",
       "1                              Solid minimal media   \n",
       "2          SMM (Supplemented liquid minimal media)   \n",
       "3          SMM (Supplemented liquid minimal media)   \n",
       "0           SA medium or LB (Luria-Bertani) medium   \n",
       "0      MYM (maltose-yeast extract-mannitol) medium   \n",
       "0                                             EMJH   \n",
       "0                BHI (brain heart infusion medium)   \n",
       "1                BHI (brain heart infusion medium)   \n",
       "2                BHI (brain heart infusion medium)   \n",
       "0                                        7H9 broth   \n",
       "\n",
       "                                          Supplement_new  \\\n",
       "index                                                      \n",
       "0                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "0                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "1                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "0                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "1                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "0                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "0                           10% LB (Luria-Bertani) broth   \n",
       "1                           10% LB (Luria-Bertani) broth   \n",
       "2                           10% LB (Luria-Bertani) broth   \n",
       "0                      FeCl2 (0.1 mM), 0.2%(w/v) glucose   \n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "5      gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "6      gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "7      gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "8      gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "9      gentamicin (50 μg/mL); tetracycline (150 μg/mL...   \n",
       "10     gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "11     gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "12     gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "13     gentamicin (50 μg/ml); tetracycline (150 μg/ml...   \n",
       "0       trace element solution (8 mL/L); TMS1 (5.6 mL/L)   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "0      IPTG [Isopropyl-β-D-Thiogalactopyranoside] (1 ...   \n",
       "0                                                    NaN   \n",
       "0                                ciprofloxacin (5 μg/mL)   \n",
       "0                                           glucose (4%)   \n",
       "1                                           glucose (4%)   \n",
       "2                                           glucose (4%)   \n",
       "0      glycerol (0.2%); Tween 80 (0.05%);  Middlebroo...   \n",
       "\n",
       "                                                Contents  \\\n",
       "index                                                      \n",
       "0      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "0      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "1      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "0      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "1      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "0      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "0      {RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...   \n",
       "1      {RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...   \n",
       "2      {RPMI-1640: {2 mM L-glutamine, 10 mM HEPES, 1 ...   \n",
       "0      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...   \n",
       "0       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "1       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "2       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "3       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "4       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "5       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "6       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "7       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "8       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "9       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "10      {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "11      {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "12      {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "13      {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}   \n",
       "0                                                   FILL   \n",
       "1      {Solid minimal media: {0.05% K2HPO4, 0.02% MgS...   \n",
       "2      {SMM: {81.9 mL PEG 6000 (BDH 6.1% w/v in disti...   \n",
       "3      {SMM: {81.9 mL PEG 6000 (BDH 6.1% w/v in disti...   \n",
       "0      {SA: {}, LB: {1% NaCl, 1% tryptone, 0.5% yeast...   \n",
       "0      {MYM: {4g/L Maltose, 4g/L Yeast Extract Oxoid,...   \n",
       "0      {EMJH: {1 g/L Disodium Phosphate, 0.3g/L Monop...   \n",
       "0      {BHI: {12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "1      {BHI: {12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "2      {BHI: {12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "0      {7H9: {Ammonium Sulfate 0.5g/L, L-Glutamic Aci...   \n",
       "\n",
       "                                         Additional_Info Motif_Available?  \\\n",
       "index                                                                       \n",
       "0                                                    NaN              NaN   \n",
       "0                                                    NaN              NaN   \n",
       "1                                                    NaN              NaN   \n",
       "0                                                    NaN              NaN   \n",
       "1                                                    NaN              NaN   \n",
       "0                                                    NaN              NaN   \n",
       "0      All  S.  aureus  strains  were  grown  in  try...              NaN   \n",
       "1                                                    NaN              NaN   \n",
       "2                                                    NaN              NaN   \n",
       "0                                                    NaN              NaN   \n",
       "0      The P. aeruginosa PAO1 strain and its derivati...              yes   \n",
       "1      The P. aeruginosa PAO1 strain and its derivati...              yes   \n",
       "2      The P. aeruginosa PAO1 strain and its derivati...              yes   \n",
       "3      The P. aeruginosa PAO1 strain and its derivati...              yes   \n",
       "4      The P. aeruginosa PAO1 strain and its derivati...              yes   \n",
       "5      The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "6      The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "7      The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "8      The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "9      The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "10     The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "11     The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "12     The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "13     The P. aeruginosa PAO1 strain and derivatives ...              NaN   \n",
       "0      For ChIP-seq analysis, spores stocks were prep...              NaN   \n",
       "1      All strains used are S. coelicolor A3 (2) M145...              NaN   \n",
       "2      S. coelicolor strains were incubated on MS sol...              NaN   \n",
       "3      S. coelicolor strains were incubated on MS sol...              NaN   \n",
       "0      P. fluorescens strains were grown in SA medium...              NaN   \n",
       "0      Each culture was then grown in 300 mL MYM cult...              NaN   \n",
       "0      L. interrogans serovar Copenhageni Fiocruz L1–...              NaN   \n",
       "0      Growth experiments of C. glutamicum cells were...              NaN   \n",
       "1      Growth experiments of C. glutamicum cells were...              NaN   \n",
       "2      Growth experiments of C. glutamicum cells were...              NaN   \n",
       "0      Mycobacterium tuberculosis (Mtb) H37Rv and Erd...              NaN   \n",
       "\n",
       "      Motif_Parameters  Note    BWcond1_1_name  \\\n",
       "index                                            \n",
       "0                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "1                  NaN   NaN                R1   \n",
       "0                  NaN   NaN               NaN   \n",
       "1                  NaN   NaN         MES5 – R1   \n",
       "0                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "1                  NaN   NaN           M1 – R1   \n",
       "2                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "0        not available   NaN                R1   \n",
       "1        not available   NaN                R1   \n",
       "2        not available   NaN                R1   \n",
       "3        not available   NaN                R1   \n",
       "4        not available   NaN                R1   \n",
       "5                  NaN   NaN                R1   \n",
       "6                  NaN   NaN                R1   \n",
       "7                  NaN   NaN                R1   \n",
       "8                  NaN   NaN                R1   \n",
       "9                  NaN   NaN                R1   \n",
       "10                 NaN   NaN                R1   \n",
       "11                 NaN   NaN                R1   \n",
       "12                 NaN   NaN                R1   \n",
       "13                 NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "1                  NaN   NaN                R1   \n",
       "2                  NaN   NaN                R1   \n",
       "3                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "0                  NaN   NaN                R1   \n",
       "0                  NaN   NaN  NeonGreen 1 – R1   \n",
       "1                  NaN   NaN    mCherry 1 – R1   \n",
       "2                  NaN   NaN    mCherry 1 – R1   \n",
       "0                  NaN   NaN                R1   \n",
       "\n",
       "                             BWcond1_1    BWcond1_2_name  \\\n",
       "index                                                      \n",
       "0                     klebfurfinal1.bw                R2   \n",
       "0            putidafurfinal1_202003.bw               NaN   \n",
       "1               putidaSigmaD_202003.bw                R2   \n",
       "0                                  NaN                R2   \n",
       "1               salmone_ydcI_MES5_1.bw         MES5 – R2   \n",
       "0                    shigella_fur_1.bw                R2   \n",
       "0                StaphLAC_new_cody1.bw                R2   \n",
       "1                  StaphTCH_sigs_M1.bw           M1 – R2   \n",
       "2      StaphTCH_vraR_7H7C11purified.bw                R2   \n",
       "0                    yersifurfinal1.bw                R2   \n",
       "0                        SRR8742210.bw                R2   \n",
       "1                        SRR8742212.bw                R2   \n",
       "2                        SRR8742214.bw                R2   \n",
       "3                        SRR8742216.bw                R2   \n",
       "4                        SRR8742218.bw                R2   \n",
       "5                        SRR1776542.bw                R2   \n",
       "6                        SRR1776544.bw                R2   \n",
       "7                        SRR1776546.bw                R2   \n",
       "8                        SRR1776548.bw                R2   \n",
       "9                        SRR1776550.bw                R2   \n",
       "10                       SRR1776552.bw                R2   \n",
       "11                       SRR1776554.bw                R2   \n",
       "12                       SRR1776558.bw                R2   \n",
       "13                       SRR1776560.bw                R2   \n",
       "0                        ERR2639621.bw                R2   \n",
       "1                        SRR1503874.bw               NaN   \n",
       "2                        SRR1752661.bw               NaN   \n",
       "3                        SRR1752662.bw               NaN   \n",
       "0                        SRR1184483.bw               NaN   \n",
       "0                        SRR7282569.bw                R2   \n",
       "0                        SRR2549977.bw                R2   \n",
       "0                        SRR8791699.bw  NeonGreen 1 – R2   \n",
       "1                        SRR8791676.bw    mCherry 1 – R2   \n",
       "2                        SRR8791691.bw    mCherry 1 – R2   \n",
       "0                         SRR400040.bw                R2   \n",
       "\n",
       "                              BWcond1_2    BWcond2_1_name  \\\n",
       "index                                                       \n",
       "0                      klebfurfinal2.bw               NaN   \n",
       "0                                   NaN               NaN   \n",
       "1                putidaSigmaD_202005.bw               NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw               NaN   \n",
       "1                salmone_ydcI_MES5_2.bw         MES5 – R3   \n",
       "0                     shigella_fur_2.bw               NaN   \n",
       "0                 StaphLAC_new_cody2.bw               NaN   \n",
       "1                   Staph_SigS_M1_R2.bw           M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw               NaN   \n",
       "0                     yersifurfinal2.bw               NaN   \n",
       "0                         SRR8742211.bw               NaN   \n",
       "1                         SRR8742213.bw               NaN   \n",
       "2                         SRR8742215.bw               NaN   \n",
       "3                         SRR8742217.bw               NaN   \n",
       "4                         SRR8742219.bw               NaN   \n",
       "5                         SRR1776543.bw               NaN   \n",
       "6                         SRR1776545.bw               NaN   \n",
       "7                         SRR1776547.bw               NaN   \n",
       "8                         SRR1776549.bw               NaN   \n",
       "9                         SRR1776551.bw               NaN   \n",
       "10                        SRR1776553.bw               NaN   \n",
       "11                        SRR1776555.bw               NaN   \n",
       "12                        SRR1776559.bw               NaN   \n",
       "13                        SRR1776561.bw               NaN   \n",
       "0                         ERR2639622.bw               NaN   \n",
       "1                                   NaN               NaN   \n",
       "2                                   NaN               NaN   \n",
       "3                                   NaN               NaN   \n",
       "0                                   NaN               NaN   \n",
       "0                         SRR7282569.bw               NaN   \n",
       "0                         SRR2549979.bw               NaN   \n",
       "0                         SRR8791700.bw  NeonGreen 2 – R1   \n",
       "1                         SRR8791677.bw    mCherry 2 – R1   \n",
       "2                         SRR8791692.bw    mCherry 2 – R1   \n",
       "0                          SRR400041.bw               NaN   \n",
       "\n",
       "                    BWcond2_1    BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                     \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw         MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw           M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "4                         NaN               NaN                     NaN   \n",
       "5                         NaN               NaN                     NaN   \n",
       "6                         NaN               NaN                     NaN   \n",
       "7                         NaN               NaN                     NaN   \n",
       "8                         NaN               NaN                     NaN   \n",
       "9                         NaN               NaN                     NaN   \n",
       "10                        NaN               NaN                     NaN   \n",
       "11                        NaN               NaN                     NaN   \n",
       "12                        NaN               NaN                     NaN   \n",
       "13                        NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0               SRR8791701.bw  NeonGreen 2 – R2           SRR8791702.bw   \n",
       "1               SRR8791678.bw    mCherry 2 – R2           SRR8791679.bw   \n",
       "2               SRR8791697.bw    mCherry 2 – R2           SRR8791698.bw   \n",
       "0                         NaN               NaN                     NaN   \n",
       "\n",
       "       BWcond3_1_name               BWcond3_1  BWcond3_2_name  \\\n",
       "index                                                           \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1            ph5 – R1  salmone_ydcI_ph55_1.bw        ph5 – R2   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1             M3 – R1     StaphTCH_sigs_M3.bw         M3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "4                 NaN                     NaN             NaN   \n",
       "5                 NaN                     NaN             NaN   \n",
       "6                 NaN                     NaN             NaN   \n",
       "7                 NaN                     NaN             NaN   \n",
       "8                 NaN                     NaN             NaN   \n",
       "9                 NaN                     NaN             NaN   \n",
       "10                NaN                     NaN             NaN   \n",
       "11                NaN                     NaN             NaN   \n",
       "12                NaN                     NaN             NaN   \n",
       "13                NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0        mCherry – R1           SRR8791680.bw    mCherry – R2   \n",
       "1      mCherry 3 – R1           SRR8791682.bw  mCherry 3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "4                         NaN            NaN                  NaN   \n",
       "5                         NaN            NaN                  NaN   \n",
       "6                         NaN            NaN                  NaN   \n",
       "7                         NaN            NaN                  NaN   \n",
       "8                         NaN            NaN                  NaN   \n",
       "9                         NaN            NaN                  NaN   \n",
       "10                        NaN            NaN                  NaN   \n",
       "11                        NaN            NaN                  NaN   \n",
       "12                        NaN            NaN                  NaN   \n",
       "13                        NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0               SRR8791681.bw            NaN                  NaN   \n",
       "1               SRR8791683.bw            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "4                NaN                  NaN  \n",
       "5                NaN                  NaN  \n",
       "6                NaN                  NaN  \n",
       "7                NaN                  NaN  \n",
       "8                NaN                  NaN  \n",
       "9                NaN                  NaN  \n",
       "10               NaN                  NaN  \n",
       "11               NaN                  NaN  \n",
       "12               NaN                  NaN  \n",
       "13               NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "amrz\n",
      "mksb\n",
      "parb\n",
      "smc\n",
      "espr\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] \n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "\n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        print(TF_name)\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list.insert(12, \"curated information\", TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')\n",
    "\n",
    "#individual file updates\n",
    "for org in new_TF_list.organism_id.unique():\n",
    "    new_TF_list[new_TF_list.organism_id == org].to_csv(\"../data/\"+org+\"/TF_list.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - other strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         TF          Organism       Strain Media  \\\n",
       "index                                              \n",
       "0      BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1      CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2       Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3       Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4      GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond   project  \\\n",
       "index                                                       \n",
       "0                    NaN            4         1       TCS   \n",
       "1                    NaN            2         1       TCS   \n",
       "2                    NaN            8         4  known_TF   \n",
       "3                    NaN            4         2  known_TF   \n",
       "4                    NaN            2         1  known_TF   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "\n",
       "                                             doi peak_intensity_measure  \\\n",
       "index                                                                     \n",
       "0      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "1      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "2             https://doi.org/10.1093/nar/gky069               MACE S/N   \n",
       "3             https://doi.org/10.1038/ncomms5910               MACE S/N   \n",
       "4             https://doi.org/10.1038/ncomms8970               MACE S/N   \n",
       "\n",
       "         Method    TF Category Note BWcond1_1_name          BWcond1_1  \\\n",
       "index                                                                   \n",
       "0      ChIP-exo  1 – Single TF  NaN  bio-rep1 – R1  BaeR_R1_S31_R1.bw   \n",
       "1      ChIP-exo  1 – Single TF  NaN             R1    CpxRR1_S1_R1.bw   \n",
       "2      ChIP-exo  1 – Single TF  NaN       Glu – R1       cra_glu_1.bw   \n",
       "3      ChIP-exo  1 – Single TF  NaN        Fe – R1        fur_fe_1.bw   \n",
       "4      ChIP-exo  1 – Single TF  NaN             R1          gade_1.bw   \n",
       "\n",
       "      BWcond1_2_name          BWcond1_2 BWcond2_1_name           BWcond2_1  \\\n",
       "index                                                                        \n",
       "0      bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1  BaerR_R2_S32_R1.bw   \n",
       "1                 R2    CpxRR2_S2_R2.bw            NaN                 NaN   \n",
       "2           Glu – R2       cra_glu_2.bw       Fru – R1        cra_fru_1.bw   \n",
       "3            Fe – R2        fur_fe_2.bw       DPD – R1        fur_dpd_1.bw   \n",
       "4                 R2          gade_2.bw            NaN                 NaN   \n",
       "\n",
       "      BWcond2_2_name           BWcond2_2 BWcond3_1_name     BWcond3_1  \\\n",
       "index                                                                   \n",
       "0      bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN           NaN   \n",
       "1                NaN                 NaN            NaN           NaN   \n",
       "2           Fru – R2        cra_fru_2.bw       Gal – R1  cra_gal_1.bw   \n",
       "3           DPD – R2        fur_dpd_2.bw            NaN           NaN   \n",
       "4                NaN                 NaN            NaN           NaN   \n",
       "\n",
       "      BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                             \n",
       "0                NaN           NaN            NaN           NaN   \n",
       "1                NaN           NaN            NaN           NaN   \n",
       "2           Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3                NaN           NaN            NaN           NaN   \n",
       "4                NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_folder = \"../data/e_coli/\"\n",
    "\n",
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "    \n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "    \n",
    "    if genome_id == 'NC_000913.3':\n",
    "            continue\n",
    "            \n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "            \n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "\n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Go through all binding sequences and perform new run of MEME-ChIP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../data/s_aureus/NC_010079/curated_input/sigs_RPMI_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_aureus/NC_010079/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/sigs_RPMI.txt 1> $metrics\n",
      "getsize ran successfully in 0.002663 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/sigs_RPMI.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030278 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/sigs_RPMI.txt 1> ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-centered\n",
      "fasta-center ran successfully in 0.035498 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-centered ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.0022 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/sigs_RPMI.txt ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/background\n",
      "fasta-get-markov ran successfully in 0.001531 seconds\n",
      "Starting meme: meme ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-centered -oc ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.57025 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/dreme_out -png -dna -p ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-centered -n ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/centrimo_out -bfile ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/background ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/sigs_RPMI.txt ../data/s_aureus/NC_010079/meme-chip/sigs_RPMI/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/s_aureus/NC_010079/curated_input/vrar_RPMI_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_aureus/NC_010079/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/vrar_RPMI.txt 1> $metrics\n",
      "getsize ran successfully in 0.002459 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/vrar_RPMI.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032788 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/vrar_RPMI.txt 1> ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-centered\n",
      "fasta-center ran successfully in 0.035393 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-centered ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001626 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/vrar_RPMI.txt ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/background\n",
      "fasta-get-markov ran successfully in 0.00261 seconds\n",
      "Starting meme: meme ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-centered -oc ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.603955 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/dreme_out -png -dna -p ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-centered -n ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/centrimo_out -bfile ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/background ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/vrar_RPMI.txt ../data/s_aureus/NC_010079/meme-chip/vrar_RPMI/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/s_aureus/CP035369.1/curated_input/cody_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_aureus/CP035369.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.004074 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031254 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt 1> ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035171 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-centered ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003233 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt ../data/s_aureus/CP035369.1/meme-chip/cody_M9/background\n",
      "fasta-get-markov ran successfully in 0.003331 seconds\n",
      "Starting meme: meme ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-centered -oc ../data/s_aureus/CP035369.1/meme-chip/cody_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_aureus/CP035369.1/meme-chip/cody_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 121.640662 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_aureus/CP035369.1/meme-chip/cody_M9/dreme_out -png -dna -p ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-centered -n ../data/s_aureus/CP035369.1/meme-chip/cody_M9/seqs-shuffled -t 5651\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_aureus/CP035369.1/meme-chip/cody_M9/centrimo_out -bfile ../data/s_aureus/CP035369.1/meme-chip/cody_M9/background ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt ../data/s_aureus/CP035369.1/meme-chip/cody_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_aureus/CP035369.1/meme-chip/cody_M9/combined.meme ../data/s_aureus/CP035369.1/meme-chip/cody_M9/combined.meme 1> ../data/s_aureus/CP035369.1/meme-chip/cody_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.021633 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_aureus/CP035369.1/meme-chip/cody_M9/spamo_out_1 -bgfile ../data/s_aureus/CP035369.1/meme-chip/cody_M9/background -primary 1 ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt ../data/s_aureus/CP035369.1/meme-chip/cody_M9/meme_out/meme.xml ../data/s_aureus/CP035369.1/meme-chip/cody_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_aureus/CP035369.1/meme-chip/cody_M9/fimo_out_1 --bgfile ../data/s_aureus/CP035369.1/meme-chip/cody_M9/background --motif 1 ../data/s_aureus/CP035369.1/meme-chip/cody_M9/meme_out/meme.xml ../data/s_aureus/CP035369.1/meme-chip/cody_M9/cody_M9.txt\n",
      "fimo ran successfully in 0.067401 seconds\n",
      "mkdir: cannot create directory ‘../data/s_aureus/CP035369.1/meme-chip/outputs/’: File exists\n",
      "../data/s_flexneri/AE014073.1/curated_input/fur_Fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_flexneri/AE014073.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.003351 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030618 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.035716 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-centered ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003238 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002886 seconds\n",
      "Starting meme: meme ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-centered -oc ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 34.049328 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-centered -n ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/seqs-shuffled -t 5686\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/centrimo_out -bfile ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/background ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/combined.meme ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/combined.meme 1> ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.014937 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/background -primary 1 ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/background --motif 1 ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/s_flexneri/AE014073.1/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.044304 seconds\n",
      "mkdir: cannot create directory ‘../data/s_flexneri/AE014073.1/meme-chip/outputs/’: File exists\n",
      "../data/l_interrogans/NC_005823.1/curated_input/lexa1_EMJH_curated.gff\n",
      "mkdir: cannot create directory ‘../data/l_interrogans/NC_005823.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt 1> $metrics\n",
      "getsize ran successfully in 0.002567 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032964 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt 1> ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-centered\n",
      "fasta-center ran successfully in 0.036758 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-centered ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002581 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/background\n",
      "fasta-get-markov ran successfully in 0.00242 seconds\n",
      "Starting meme: meme ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-centered -oc ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.416709 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/dreme_out -png -dna -p ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-centered -n ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/centrimo_out -bfile ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/background ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/combined.meme ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/combined.meme 1> ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/motif_alignment.txt\n",
      "tomtom ran successfully in 0.017728 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/spamo_out_1 -bgfile ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/background -primary 1 ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/meme_out/meme.xml ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/fimo_out_1 --bgfile ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/background --motif 1 ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/meme_out/meme.xml ../data/l_interrogans/NC_005823.1/meme-chip/lexa1_EMJH/lexa1_EMJH.txt\n",
      "fimo ran successfully in 0.028457 seconds\n",
      "mkdir: cannot create directory ‘../data/l_interrogans/NC_005823.1/meme-chip/outputs/’: File exists\n",
      "../data/k_pneumoniae/CP000647.1/curated_input/fur_Fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/k_pneumoniae/CP000647.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.002618 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031707 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.03076 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-centered ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002561 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002822 seconds\n",
      "Starting meme: meme ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-centered -oc ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 3.930388 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-centered -n ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/centrimo_out -bfile ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/background ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/combined.meme ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/combined.meme 1> ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.026376 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/background -primary 1 ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/background --motif 1 ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/k_pneumoniae/CP000647.1/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.033308 seconds\n",
      "mkdir: cannot create directory ‘../data/k_pneumoniae/CP000647.1/meme-chip/outputs/’: File exists\n",
      "../data/s_venezuelae/NC_018750.1/curated_input/lsr2_MS_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_venezuelae/NC_018750.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt 1> $metrics\n",
      "getsize ran successfully in 0.00387 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030777 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt 1> ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-centered\n",
      "fasta-center ran successfully in 0.036818 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-centered ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00514 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/background\n",
      "fasta-get-markov ran successfully in 0.003383 seconds\n",
      "Starting meme: meme ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-centered -oc ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 90.57467 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/dreme_out -png -dna -p ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-centered -n ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/seqs-shuffled -t 5663\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/centrimo_out -bfile ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/background ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/combined.meme ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/combined.meme 1> ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/motif_alignment.txt\n",
      "tomtom ran successfully in 0.008313 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/spamo_out_1 -bgfile ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/background -primary 1 ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/meme_out/meme.xml ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/fimo_out_1 --bgfile ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/background --motif 1 ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/meme_out/meme.xml ../data/s_venezuelae/NC_018750.1/meme-chip/lsr2_MS/lsr2_MS.txt\n",
      "fimo ran successfully in 0.053704 seconds\n",
      "mkdir: cannot create directory ‘../data/s_venezuelae/NC_018750.1/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/CU928163.2/curated_input/fur_Fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/CU928163.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.002809 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029826 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.037109 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-centered ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002423 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/CU928163.2/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002218 seconds\n",
      "Starting meme: meme ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-centered -oc ../data/e_coli/CU928163.2/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/CU928163.2/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 6.022989 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/CU928163.2/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-centered -n ../data/e_coli/CU928163.2/meme-chip/fur_Fe/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/CU928163.2/meme-chip/fur_Fe/centrimo_out -bfile ../data/e_coli/CU928163.2/meme-chip/fur_Fe/background ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/CU928163.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/CU928163.2/meme-chip/fur_Fe/combined.meme ../data/e_coli/CU928163.2/meme-chip/fur_Fe/combined.meme 1> ../data/e_coli/CU928163.2/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.078496 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/CU928163.2/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/e_coli/CU928163.2/meme-chip/fur_Fe/background -primary 1 ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/CU928163.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/CU928163.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/e_coli/CU928163.2/meme-chip/fur_Fe/background --motif 1 ../data/e_coli/CU928163.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/CU928163.2/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.038737 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/CU928163.2/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_010498.1/curated_input/fur_Fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_010498.1/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.003562 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031187 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.035484 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-centered ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002803 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002725 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-centered -oc ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 23.205131 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-centered -n ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/centrimo_out -bfile ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/background ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/combined.meme ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/combined.meme 1> ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.021201 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/background -primary 1 ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/background --motif 1 ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/NC_010498.1/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.045459 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_010498.1/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/AE005174.2/curated_input/fur_Fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/AE005174.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.002721 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032224 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.036665 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-centered ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002611 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/AE005174.2/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002577 seconds\n",
      "Starting meme: meme ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-centered -oc ../data/e_coli/AE005174.2/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/AE005174.2/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.965754 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/AE005174.2/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-centered -n ../data/e_coli/AE005174.2/meme-chip/fur_Fe/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/AE005174.2/meme-chip/fur_Fe/centrimo_out -bfile ../data/e_coli/AE005174.2/meme-chip/fur_Fe/background ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/AE005174.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/AE005174.2/meme-chip/fur_Fe/combined.meme ../data/e_coli/AE005174.2/meme-chip/fur_Fe/combined.meme 1> ../data/e_coli/AE005174.2/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.019139 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/AE005174.2/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/e_coli/AE005174.2/meme-chip/fur_Fe/background -primary 1 ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt ../data/e_coli/AE005174.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/AE005174.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/e_coli/AE005174.2/meme-chip/fur_Fe/background --motif 1 ../data/e_coli/AE005174.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/e_coli/AE005174.2/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.037143 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/AE005174.2/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ygbi_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/ygbi_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002492 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/ygbi_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033692 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/ygbi_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.033791 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002044 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/ygbi_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/background\n",
      "fasta-get-markov ran successfully in 0.001901 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "You must specify '-mod anr' to set the motif site model to 'Any Number of Repetitions [per sequence]' when only providing one sequence\n",
      "meme exited with error code 1\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ygbi_M9/seqs-shuffled -t 5700\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ydcr_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/ydcr_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002627 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/ydcr_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031728 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/ydcr_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035363 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001582 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/ydcr_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/background\n",
      "fasta-get-markov ran successfully in 0.002445 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.680741 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/background ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/ydcr_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydcr_M9/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rdgc_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.005923 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034891 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.038219 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004504 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background\n",
      "fasta-get-markov ran successfully in 0.004126 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 569.708165 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/seqs-shuffled -t 5472\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.553222 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt\n",
      "fimo ran successfully in 0.129438 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_sta/rdgc_sta.txt\n",
      "fimo ran successfully in 0.115223 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/phob_M9P_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/phob_M9P.txt 1> $metrics\n",
      "getsize ran successfully in 0.002578 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/phob_M9P.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032633 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/phob_M9P.txt 1> ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-centered\n",
      "fasta-center ran successfully in 0.037266 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002139 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/phob_M9P.txt ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/background\n",
      "fasta-get-markov ran successfully in 0.001989 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.021724 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/background ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/phob_M9P.txt ../data/e_coli/NC_000913.3/meme-chip/phob_M9P/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/infa_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.008989 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034956 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.041149 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.007876 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.036048 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background\n",
      "fasta-get-markov ran successfully in 0.009629 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 617.494285 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/infa_sta/seqs-shuffled -t 5452\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/infa_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/infa_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.876139 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt\n",
      "fimo ran successfully in 0.159443 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt\n",
      "fimo ran successfully in 0.172327 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/infa_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/infa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_sta/infa_sta.txt\n",
      "fimo ran successfully in 0.171442 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ygav_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/ygav_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.00226 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/ygav_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032106 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/ygav_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.03609 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002298 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/ygav_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/background\n",
      "fasta-get-markov ran successfully in 0.002321 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.796875 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/background ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/ygav_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ygav_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rcsb_RcsD-KO_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt 1> $metrics\n",
      "getsize ran successfully in 0.002693 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033018 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-centered\n",
      "fasta-center ran successfully in 0.036358 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002496 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/background\n",
      "fasta-get-markov ran successfully in 0.00267 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.32771 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/background ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/motif_alignment.txt\n",
      "tomtom ran successfully in 0.042617 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_RcsD-KO/rcsb_RcsD-KO.txt\n",
      "fimo ran successfully in 0.0324 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mukb_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.00882 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035074 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.03643 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004632 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.033197 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background\n",
      "fasta-get-markov ran successfully in 0.00952 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 625.152165 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/seqs-shuffled -t 5450\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.60524 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt\n",
      "fimo ran successfully in 0.184224 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt\n",
      "fimo ran successfully in 0.151627 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt\n",
      "fimo ran successfully in 0.182001 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_sta/mukb_sta.txt\n",
      "fimo ran successfully in 0.197386 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/cpxr_EtOH_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt 1> $metrics\n",
      "getsize ran successfully in 0.003079 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031539 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-centered\n",
      "fasta-center ran successfully in 0.034964 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001747 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background\n",
      "fasta-get-markov ran successfully in 0.001638 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 24.303465 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/motif_alignment.txt\n",
      "tomtom ran successfully in 0.683963 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt\n",
      "fimo ran successfully in 0.036154 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt\n",
      "fimo ran successfully in 0.049773 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt\n",
      "fimo ran successfully in 0.049406 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cpxr_EtOH/cpxr_EtOH.txt\n",
      "fimo ran successfully in 0.050087 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yeht_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/yeht_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002445 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/yeht_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029334 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/yeht_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.033357 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002429 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/yeht_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/background\n",
      "fasta-get-markov ran successfully in 0.002303 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.593907 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/background ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/yeht_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yeht_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/hupb_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.006535 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031641 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.040634 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005418 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.035495 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background\n",
      "fasta-get-markov ran successfully in 0.006024 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 618.512035 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/seqs-shuffled -t 5452\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.48692 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt\n",
      "fimo ran successfully in 0.118094 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt\n",
      "fimo ran successfully in 0.109342 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_sta/hupb_sta.txt\n",
      "fimo ran successfully in 0.133175 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybaq_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.003103 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032593 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036883 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002182 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/background\n",
      "fasta-get-markov ran successfully in 0.002417 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.561545 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/background ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.017557 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ybaq_M9/ybaq_M9.txt\n",
      "fimo ran successfully in 0.030301 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yfie_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/yfie_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002441 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/yfie_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029851 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/yfie_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036555 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002011 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/yfie_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/background\n",
      "fasta-get-markov ran successfully in 0.00218399999999996 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.624865 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/background ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/yfie_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfie_M9/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rpob (cra and crp ko exps)_crp-ko ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "../data/e_coli/NC_000913.3/curated_input/topa_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.012134 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029152 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.045107 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006668 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.039567 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/topa_expo/background\n",
      "fasta-get-markov ran successfully in 0.014687 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/topa_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/topa_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 546.328732 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/topa_expo/seqs-shuffled -t 5481\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/topa_expo/background ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/topa_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Skipping sequence peak1967 as its length (482) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/topa_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/topa_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/topa_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.011921 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/topa_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/topa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/topa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/topa_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/topa_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/topa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/topa_expo/topa_expo.txt\n",
      "fimo ran successfully in 0.256085 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/rpob_fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.013694 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.0349 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-centered\n",
      "fasta-center ran successfully in 0.046052 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.011525 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.040599 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background\n",
      "fasta-get-markov ran successfully in 0.013661 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 626.525488 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/seqs-shuffled -t 5449\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/motif_alignment.txt\n",
      "tomtom ran successfully in 1.107775 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt\n",
      "fimo ran successfully in 0.268833 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt\n",
      "fimo ran successfully in 0.278853 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt\n",
      "fimo ran successfully in 0.274801 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_fe/rpob_fe.txt\n",
      "fimo ran successfully in 0.277648 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yheo_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002405 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031306 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036637 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00217 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/background\n",
      "fasta-get-markov ran successfully in 0.002182 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.842642 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/background ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.091057 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yheo_M9/yheo_M9.txt\n",
      "fimo ran successfully in 0.031823 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybhd_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/ybhd_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002385 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/ybhd_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029466 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/ybhd_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.034071 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001907 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/ybhd_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/background\n",
      "fasta-get-markov ran successfully in 0.001419 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.684044 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/background ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/ybhd_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybhd_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/h-ns_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/h-ns_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.010493 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/h-ns_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035366 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/h-ns_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.035924 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004367 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.038042 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/h-ns_expo.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/background\n",
      "fasta-get-markov ran successfully in 0.008112 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 625.783193 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/seqs-shuffled -t 5449\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/background ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/h-ns_expo.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ydci_ph7_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/ydci_ph7.txt 1> $metrics\n",
      "getsize ran successfully in 0.002511 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/ydci_ph7.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032936 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/ydci_ph7.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-centered\n",
      "fasta-center ran successfully in 0.036145 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002248 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/ydci_ph7.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/background\n",
      "fasta-get-markov ran successfully in 0.001987 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.194497 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/background ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/ydci_ph7.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph7/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool4_M9+ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt 1> $metrics\n",
      "getsize ran successfully in 0.008192 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034809 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-centered\n",
      "fasta-center ran successfully in 0.039313 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006789 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.036619 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background\n",
      "fasta-get-markov ran successfully in 0.007413 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 614.213788 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/seqs-shuffled -t 5454\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/motif_alignment.txt\n",
      "tomtom ran successfully in 0.764197 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt\n",
      "fimo ran successfully in 0.173825 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt\n",
      "fimo ran successfully in 0.144671 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt\n",
      "fimo ran successfully in 0.168101 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool4_M9+ace/mixed-tfs-pool4_M9+ace.txt\n",
      "fimo ran successfully in 0.161748 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ynej_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/ynej_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.002535 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/ynej_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031839 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/ynej_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.035381 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002337 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/ynej_glc.txt ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/background\n",
      "fasta-get-markov ran successfully in 0.002072 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.464679 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/background ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/ynej_glc.txt ../data/e_coli/NC_000913.3/meme-chip/ynej_glc/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/cusr_CuSO4_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt 1> $metrics\n",
      "getsize ran successfully in 0.002816 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033063 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-centered\n",
      "fasta-center ran successfully in 0.033946 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002642 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background\n",
      "fasta-get-markov ran successfully in 0.002653 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.032626 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/motif_alignment.txt\n",
      "tomtom ran successfully in 0.230801 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt\n",
      "fimo ran successfully in 0.042917 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt\n",
      "fimo ran successfully in 0.038149 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/baer_EtOH_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt 1> $metrics\n",
      "getsize ran successfully in 0.004001 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028209 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt 1> ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-centered\n",
      "fasta-center ran successfully in 0.036007 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003032 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background\n",
      "fasta-get-markov ran successfully in 0.003489 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 59.428547 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/seqs-shuffled -t 5676\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/combined.meme ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/motif_alignment.txt\n",
      "tomtom ran successfully in 0.241331 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt\n",
      "fimo ran successfully in 0.053866 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/baer_EtOH/baer_EtOH.txt\n",
      "fimo ran successfully in 0.092599 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/atoc_acetoacetate_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt 1> $metrics\n",
      "getsize ran successfully in 0.002905 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031777 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt 1> ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-centered\n",
      "fasta-center ran successfully in 0.036294 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002509 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background\n",
      "fasta-get-markov ran successfully in 0.001821 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 13.92087 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/seqs-shuffled -t 5694\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/combined.meme ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/motif_alignment.txt\n",
      "tomtom ran successfully in 0.289548 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt\n",
      "fimo ran successfully in 0.038713 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt\n",
      "fimo ran successfully in 0.038687 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/atoc_acetoacetate/atoc_acetoacetate.txt\n",
      "fimo ran successfully in 0.042547 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/arca-2_M9+ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt 1> $metrics\n",
      "getsize ran successfully in 0.002826 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030809 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt 1> ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-centered\n",
      "fasta-center ran successfully in 0.035039 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002286 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/background\n",
      "fasta-get-markov ran successfully in 0.002285 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 5.598603 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/background ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/combined.meme ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/motif_alignment.txt\n",
      "tomtom ran successfully in 0.014406 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/arca-2_M9+ace/arca-2_M9+ace.txt\n",
      "fimo ran successfully in 0.028728 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool2_M9+glu_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt 1> $metrics\n",
      "getsize ran successfully in 0.006753 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032581 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-centered\n",
      "fasta-center ran successfully in 0.037387 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003211 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.03516 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background\n",
      "fasta-get-markov ran successfully in 0.006533 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 609.712113 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/seqs-shuffled -t 5456\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/motif_alignment.txt\n",
      "tomtom ran successfully in 0.49046 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt\n",
      "fimo ran successfully in 0.130734 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt\n",
      "fimo ran successfully in 0.129249 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt\n",
      "fimo ran successfully in 0.129306 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool2_M9+glu/mixed-tfs-pool2_M9+glu.txt\n",
      "fimo ran successfully in 0.126298 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yidz_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.003912 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031391 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036881 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003244 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background\n",
      "fasta-get-markov ran successfully in 0.002858 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 43.643448 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/seqs-shuffled -t 5682\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.230247 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt\n",
      "fimo ran successfully in 0.057268 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt\n",
      "fimo ran successfully in 0.072416 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yidz_M9/yidz_M9.txt\n",
      "fimo ran successfully in 0.057546 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool5_M9+gal_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt 1> $metrics\n",
      "getsize ran successfully in 0.00756 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034417 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-centered\n",
      "fasta-center ran successfully in 0.038937 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00688 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.037023 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background\n",
      "fasta-get-markov ran successfully in 0.00752 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 626.352204 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/seqs-shuffled -t 5449\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/motif_alignment.txt\n",
      "tomtom ran successfully in 0.509739 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt\n",
      "fimo ran successfully in 0.139954 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt\n",
      "fimo ran successfully in 0.14389 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt\n",
      "fimo ran successfully in 0.154122 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool5_M9+gal/mixed-tfs-pool5_M9+gal.txt\n",
      "fimo ran successfully in 0.142874 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/fis_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.012971 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035337 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.046846 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.010788 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.042381 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background\n",
      "fasta-get-markov ran successfully in 0.013068 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 505.095648 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/fis_expo/seqs-shuffled -t 5498\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/fis_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/fis_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.759508 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt\n",
      "fimo ran successfully in 0.287298 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt\n",
      "fimo ran successfully in 0.338254 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/fis_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_expo/fis_expo.txt\n",
      "fimo ran successfully in 0.341356 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gadx_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.00271 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032649 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.038034 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.0026 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/background\n",
      "fasta-get-markov ran successfully in 0.002227 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 5.552867 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/background ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.032856 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gadx_M9/gadx_M9.txt\n",
      "fimo ran successfully in 0.029274 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/infa_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.014269 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.036651 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.040478 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.010606 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.039046 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background\n",
      "fasta-get-markov ran successfully in 0.007462 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 548.090407 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/infa_expo/seqs-shuffled -t 5480\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/infa_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/infa_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.136169 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt\n",
      "fimo ran successfully in 0.248343 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/infa_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/infa_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/infa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/infa_expo/infa_expo.txt\n",
      "fimo ran successfully in 0.29129 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gyrb_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.00407 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.027874 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.029725 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001968 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background\n",
      "fasta-get-markov ran successfully in 0.003241 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 70.443795 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/seqs-shuffled -t 5671\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.205936 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt\n",
      "fimo ran successfully in 0.052989 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_expo/gyrb_expo.txt\n",
      "fimo ran successfully in 0.057259 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/hupa_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.00557 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03161 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.039407 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004508 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background\n",
      "fasta-get-markov ran successfully in 0.003038 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 406.02952 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/seqs-shuffled -t 5537\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.368574 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt\n",
      "fimo ran successfully in 0.103005 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt\n",
      "fimo ran successfully in 0.099063 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_sta/hupa_sta.txt\n",
      "fimo ran successfully in 0.11288 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/rcsb_EtOH_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt 1> $metrics\n",
      "getsize ran successfully in 0.002915 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03113 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-centered\n",
      "fasta-center ran successfully in 0.029247 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001583 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background\n",
      "fasta-get-markov ran successfully in 0.002486 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 12.327266 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/seqs-shuffled -t 5695\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/motif_alignment.txt\n",
      "tomtom ran successfully in 0.396201 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt\n",
      "fimo ran successfully in 0.041765 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt\n",
      "fimo ran successfully in 0.035899 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rcsb_EtOH/rcsb_EtOH.txt\n",
      "fimo ran successfully in 0.035136 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/stpa_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.009027 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033657 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.042886 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.008811 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.037546 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background\n",
      "fasta-get-markov ran successfully in 0.009341 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 621.343576 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/seqs-shuffled -t 5451\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.56749 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt\n",
      "fimo ran successfully in 0.200122 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt\n",
      "fimo ran successfully in 0.235897 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_expo/stpa_expo.txt\n",
      "fimo ran successfully in 0.182839 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybcm_oxidative_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/ybcm_oxidative.txt 1> $metrics\n",
      "getsize ran successfully in 0.002513 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/ybcm_oxidative.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029624 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/ybcm_oxidative.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-centered\n",
      "fasta-center ran successfully in 0.029952 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002266 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/ybcm_oxidative.txt ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/background\n",
      "fasta-get-markov ran successfully in 0.001593 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.180115 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/background ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/ybcm_oxidative.txt ../data/e_coli/NC_000913.3/meme-chip/ybcm_oxidative/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/hupa_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.003686 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032655 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.036917 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003136 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background\n",
      "fasta-get-markov ran successfully in 0.002692 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 94.127399 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/seqs-shuffled -t 5662\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.726816 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt\n",
      "fimo ran successfully in 0.062796 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt\n",
      "fimo ran successfully in 0.058907 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt\n",
      "fimo ran successfully in 0.068455 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupa_expo/hupa_expo.txt\n",
      "fimo ran successfully in 0.058691 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ycan_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/ycan_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002406 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/ycan_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030388 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/ycan_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.030543 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002219 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/ycan_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/background\n",
      "fasta-get-markov ran successfully in 0.0018 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.743373 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/background ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/ycan_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycan_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool1b_M9+glu_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt 1> $metrics\n",
      "getsize ran successfully in 0.003024 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032094 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-centered\n",
      "fasta-center ran successfully in 0.039294 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002294 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background\n",
      "fasta-get-markov ran successfully in 0.001716 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 16.230276 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/seqs-shuffled -t 5693\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/motif_alignment.txt\n",
      "tomtom ran successfully in 0.109655 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt\n",
      "fimo ran successfully in 0.042503 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1b_M9+glu/mixed-tfs-pool1b_M9+glu.txt\n",
      "fimo ran successfully in 0.055327 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yihl_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/yihl_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.002532 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/yihl_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03133 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/yihl_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.03579 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002179 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/yihl_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/background\n",
      "fasta-get-markov ran successfully in 0.002244 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.726154 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/background ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/yihl_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yihl_glc/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ynfl_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/ynfl_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.002302 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/ynfl_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034822 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/ynfl_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.038149 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002257 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/ynfl_glc.txt ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/background\n",
      "fasta-get-markov ran successfully in 0.002134 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "You must specify '-mod anr' to set the motif site model to 'Any Number of Repetitions [per sequence]' when only providing one sequence\n",
      "meme exited with error code 1\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ynfl_glc/seqs-shuffled -t 5700\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ompr_NaCl_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/ompr_NaCl.txt 1> $metrics\n",
      "getsize ran successfully in 0.002949 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/ompr_NaCl.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03214 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/ompr_NaCl.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-centered\n",
      "fasta-center ran successfully in 0.03511 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002385 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/ompr_NaCl.txt ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/background\n",
      "fasta-get-markov ran successfully in 0.002328 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.250002 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/background ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/ompr_NaCl.txt ../data/e_coli/NC_000913.3/meme-chip/ompr_NaCl/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ydcn_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002704 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029618 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036017 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002229 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/background\n",
      "fasta-get-markov ran successfully in 0.002233 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.481008 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/background ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.011657 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ydcn_M9/ydcn_M9.txt\n",
      "fimo ran successfully in 0.032695 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool1a_M9+glu_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt 1> $metrics\n",
      "getsize ran successfully in 0.002703 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030762 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-centered\n",
      "fasta-center ran successfully in 0.036336 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002593 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background\n",
      "fasta-get-markov ran successfully in 0.001757 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.041243 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/motif_alignment.txt\n",
      "tomtom ran successfully in 0.147047 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt\n",
      "fimo ran successfully in 0.030777 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1a_M9+glu/mixed-tfs-pool1a_M9+glu.txt\n",
      "fimo ran successfully in 0.039099 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gyra_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.006245 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.0339 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.041565 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005128 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.03298 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background\n",
      "fasta-get-markov ran successfully in 0.006202 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 625.012794 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/seqs-shuffled -t 5450\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.415491 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt\n",
      "fimo ran successfully in 0.132842 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_sta/gyra_sta.txt\n",
      "fimo ran successfully in 0.120237 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gyra_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.006141 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031961 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.039781 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005013 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background\n",
      "fasta-get-markov ran successfully in 0.005547 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 585.968955 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/seqs-shuffled -t 5465\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.291901 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt\n",
      "fimo ran successfully in 0.118932 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt\n",
      "fimo ran successfully in 0.122976 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyra_expo/gyra_expo.txt\n",
      "fimo ran successfully in 0.103221 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yqhc_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/yqhc_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002615 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/yqhc_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030827 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/yqhc_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.034372 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00235 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/yqhc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/background\n",
      "fasta-get-markov ran successfully in 0.002154 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.375267 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/background ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/yqhc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yqhc_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rpob_dpd_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt 1> $metrics\n",
      "getsize ran successfully in 0.012429 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031976 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-centered\n",
      "fasta-center ran successfully in 0.048063 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.008248 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.04033 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background\n",
      "fasta-get-markov ran successfully in 0.012534 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 613.597561 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/seqs-shuffled -t 5454\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Skipping sequence peak1 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/motif_alignment.txt\n",
      "tomtom ran successfully in 0.395276 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt\n",
      "fimo ran successfully in 0.234533 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt\n",
      "fimo ran successfully in 0.203545 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rpob_dpd/rpob_dpd.txt\n",
      "fimo ran successfully in 0.238771 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/fur_fe_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.002905 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029595 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt 1> ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-centered\n",
      "fasta-center ran successfully in 0.036411 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002575 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt ../data/e_coli/NC_000913.3/meme-chip/fur_fe/background\n",
      "fasta-get-markov ran successfully in 0.001823 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/fur_fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/fur_fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.32091 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_fe/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/fur_fe/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_fe/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/fur_fe/background ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt ../data/e_coli/NC_000913.3/meme-chip/fur_fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/fur_fe/combined.meme ../data/e_coli/NC_000913.3/meme-chip/fur_fe/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/fur_fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.020038 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_fe/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_fe/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt ../data/e_coli/NC_000913.3/meme-chip/fur_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_fe/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/fur_fe/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_fe/fur_fe.txt\n",
      "fimo ran successfully in 0.036121 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ydip_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/ydip_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002399 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/ydip_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03288 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/ydip_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.03716 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002132 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/ydip_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/background\n",
      "fasta-get-markov ran successfully in 0.002013 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.707072 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/background ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/ydip_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydip_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yjhi_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.002554 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030134 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.034034 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002348 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/background\n",
      "fasta-get-markov ran successfully in 0.002298 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.208826 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/background ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/motif_alignment.txt\n",
      "tomtom ran successfully in 0.04583 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjhi_glc/yjhi_glc.txt\n",
      "fimo ran successfully in 0.026854 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/rpob (cra and crp ko exps)_cra-ko ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "../data/e_coli/NC_000913.3/curated_input/fur_dpd_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt 1> $metrics\n",
      "getsize ran successfully in 0.003328 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03055 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt 1> ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-centered\n",
      "fasta-center ran successfully in 0.034158 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002689 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background\n",
      "fasta-get-markov ran successfully in 0.00265 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 24.970789 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/combined.meme ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/motif_alignment.txt\n",
      "tomtom ran successfully in 0.486262 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt\n",
      "fimo ran successfully in 0.043898 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt\n",
      "fimo ran successfully in 0.054805 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fur_dpd/fur_dpd.txt\n",
      "fimo ran successfully in 0.056083 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yfed_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002753 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029717 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036254 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002233 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/background\n",
      "fasta-get-markov ran successfully in 0.002314 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 6.300944 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/background ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.013886 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yfed_M9/yfed_M9.txt\n",
      "fimo ran successfully in 0.032388 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ycit_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002698 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032585 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.034698 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002333 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/background\n",
      "fasta-get-markov ran successfully in 0.002171 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 6.493235 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/background ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.013192 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ycit_M9/ycit_M9.txt\n",
      "fimo ran successfully in 0.033649 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yiag_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.006324 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033473 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036438 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005077 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.031582 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background\n",
      "fasta-get-markov ran successfully in 0.005577 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 611.982086 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/seqs-shuffled -t 5455\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 1.21504 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt\n",
      "fimo ran successfully in 0.121003 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt\n",
      "fimo ran successfully in 0.125183 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt\n",
      "fimo ran successfully in 0.125669 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiag_M9/yiag_M9.txt\n",
      "fimo ran successfully in 0.127102 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/narl_nitrate_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt 1> $metrics\n",
      "getsize ran successfully in 0.005437 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031418 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt 1> ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-centered\n",
      "fasta-center ran successfully in 0.0334 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004748 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background\n",
      "fasta-get-markov ran successfully in 0.004216 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 377.5194 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/seqs-shuffled -t 5548\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/combined.meme ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/motif_alignment.txt\n",
      "tomtom ran successfully in 0.260322 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt\n",
      "fimo ran successfully in 0.097442 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt\n",
      "fimo ran successfully in 0.09211 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/narl_nitrate/narl_nitrate.txt\n",
      "fimo ran successfully in 0.093497 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gade_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002612 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.02866 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.037752 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002422 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gade_M9/background\n",
      "fasta-get-markov ran successfully in 0.002325 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/gade_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gade_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.07141 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gade_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gade_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gade_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gade_M9/background ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gade_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gade_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gade_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gade_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.062389 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gade_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gade_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gade_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gade_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gade_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gade_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gade_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gade_M9/gade_M9.txt\n",
      "fimo ran successfully in 0.03343 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yeam_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/yeam_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002376 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/yeam_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030593 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/yeam_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.03336 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002167 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/yeam_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/background\n",
      "fasta-get-markov ran successfully in 0.001507 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.592688 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/background ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/yeam_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yeam_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rdgc_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.005737 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032801 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.039424 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002861 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background\n",
      "fasta-get-markov ran successfully in 0.004156 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 529.366933 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/seqs-shuffled -t 5488\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.411716 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt\n",
      "fimo ran successfully in 0.108431 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt\n",
      "fimo ran successfully in 0.108552 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt\n",
      "fimo ran successfully in 0.115454 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rdgc_expo/rdgc_expo.txt\n",
      "fimo ran successfully in 0.103653 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yebk_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/yebk_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002445 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/yebk_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031934 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/yebk_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.034612 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002343 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/yebk_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/background\n",
      "fasta-get-markov ran successfully in 0.002184 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.710896 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/background ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/yebk_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yebk_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/flhd_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003101 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030697 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt 1> ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.03558 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002402 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background\n",
      "fasta-get-markov ran successfully in 0.002422 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.525884 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/combined.meme ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.784903 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt\n",
      "fimo ran successfully in 0.039676 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt\n",
      "fimo ran successfully in 0.040418 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt\n",
      "fimo ran successfully in 0.040219 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhd_LB/flhd_LB.txt\n",
      "fimo ran successfully in 0.031708 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/dps_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.004117 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032645 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.038304 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003786 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background\n",
      "fasta-get-markov ran successfully in 0.003987 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 116.952713 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/dps_sta/seqs-shuffled -t 5653\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/dps_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/dps_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.17383 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt\n",
      "fimo ran successfully in 0.074112 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/dps_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/dps_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_sta/dps_sta.txt\n",
      "fimo ran successfully in 0.070594 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/dps_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.003301 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032104 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.034137 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002761 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background\n",
      "fasta-get-markov ran successfully in 0.004576 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 36.282268 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/dps_expo/seqs-shuffled -t 5685\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/dps_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/dps_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.476535 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt\n",
      "fimo ran successfully in 0.05813 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt\n",
      "fimo ran successfully in 0.049301 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/dps_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/dps_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/dps_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/dps_expo/dps_expo.txt\n",
      "fimo ran successfully in 0.049953 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybdo_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/ybdo_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002414 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/ybdo_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030308 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/ybdo_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036026 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002059 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/ybdo_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/background\n",
      "fasta-get-markov ran successfully in 0.001955 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.727332 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/background ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/ybdo_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybdo_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/zrar_ZnCl2_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt 1> $metrics\n",
      "getsize ran successfully in 0.003333 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029173 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt 1> ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-centered\n",
      "fasta-center ran successfully in 0.03542 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003093 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background\n",
      "fasta-get-markov ran successfully in 0.002072 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 22.938411 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/combined.meme ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/motif_alignment.txt\n",
      "tomtom ran successfully in 0.114109 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt\n",
      "fimo ran successfully in 0.04388 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/zrar_ZnCl2/zrar_ZnCl2.txt\n",
      "fimo ran successfully in 0.04437 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/kdpe_KCl_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/kdpe_KCl.txt 1> $metrics\n",
      "getsize ran successfully in 0.002967 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/kdpe_KCl.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032454 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/kdpe_KCl.txt 1> ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-centered\n",
      "fasta-center ran successfully in 0.038445 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002676 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/kdpe_KCl.txt ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/background\n",
      "fasta-get-markov ran successfully in 0.001795 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 14.463225 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/seqs-shuffled -t 5694\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/background ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/kdpe_KCl.txt ../data/e_coli/NC_000913.3/meme-chip/kdpe_KCl/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rcsA_oxida_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/rcsA_oxida.txt 1> $metrics\n",
      "getsize ran successfully in 0.003319 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/rcsA_oxida.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032979 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/rcsA_oxida.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-centered\n",
      "fasta-center ran successfully in 0.034865 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003182 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/rcsA_oxida.txt ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/background\n",
      "fasta-get-markov ran successfully in 0.001904 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 25.7144 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/seqs-shuffled -t 5689\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/background ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/rcsA_oxida.txt ../data/e_coli/NC_000913.3/meme-chip/rcsA_oxida/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/btsr_pyruvate_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt 1> $metrics\n",
      "getsize ran successfully in 0.003007 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033744 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt 1> ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-centered\n",
      "fasta-center ran successfully in 0.035165 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002189 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/background\n",
      "fasta-get-markov ran successfully in 0.001814 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 7.220706 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/background ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/combined.meme ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/motif_alignment.txt\n",
      "tomtom ran successfully in 0.124806 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/btsr_pyruvate/btsr_pyruvate.txt\n",
      "fimo ran successfully in 0.033896 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/stpa_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.00897 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033244 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.039669 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00746 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.037775 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background\n",
      "fasta-get-markov ran successfully in 0.008616 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 619.621005 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/seqs-shuffled -t 5452\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.69678 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt\n",
      "fimo ran successfully in 0.178114 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt\n",
      "fimo ran successfully in 0.172129 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt\n",
      "fimo ran successfully in 0.159955 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/stpa_sta/stpa_sta.txt\n",
      "fimo ran successfully in 0.144152 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yjdc_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.00681 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030328 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.040865 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006079 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.034306 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background\n",
      "fasta-get-markov ran successfully in 0.006537 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 602.586538 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/seqs-shuffled -t 5458\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.86802 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt\n",
      "fimo ran successfully in 0.135402 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt\n",
      "fimo ran successfully in 0.132982 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt\n",
      "fimo ran successfully in 0.141804 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yjdc_M9/yjdc_M9.txt\n",
      "fimo ran successfully in 0.126699 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/cra_acetate_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt 1> $metrics\n",
      "getsize ran successfully in 0.002875 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031415 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-centered\n",
      "fasta-center ran successfully in 0.034219 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002501 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background\n",
      "fasta-get-markov ran successfully in 0.002504 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.503622 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/motif_alignment.txt\n",
      "tomtom ran successfully in 0.293115 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt\n",
      "fimo ran successfully in 0.037726 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt\n",
      "fimo ran successfully in 0.045024 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_acetate/cra_acetate.txt\n",
      "fimo ran successfully in 0.04307 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/h-ns_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.014961 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03601 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.052133 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.014903 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.046913 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background\n",
      "fasta-get-markov ran successfully in 0.016126 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 520.860499 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/seqs-shuffled -t 5491\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.106202 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt\n",
      "fimo ran successfully in 0.33179 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/h-ns_sta/h-ns_sta.txt\n",
      "fimo ran successfully in 0.358275 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/soxs_pq_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt 1> $metrics\n",
      "getsize ran successfully in 0.003697 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030265 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt 1> ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-centered\n",
      "fasta-center ran successfully in 0.035861 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003092 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.037968 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background\n",
      "fasta-get-markov ran successfully in 0.005614 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 594.59612 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/seqs-shuffled -t 5462\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/combined.meme ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/motif_alignment.txt\n",
      "tomtom ran successfully in 0.58246 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt\n",
      "fimo ran successfully in 0.117301 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt\n",
      "fimo ran successfully in 0.112441 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt\n",
      "fimo ran successfully in 0.129441 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/soxs_pq/soxs_pq.txt\n",
      "fimo ran successfully in 0.120412 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/argr_arginine_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/argr_arginine.txt 1> $metrics\n",
      "getsize ran successfully in 0.00269 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/argr_arginine.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030765 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/argr_arginine.txt 1> ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-centered\n",
      "fasta-center ran successfully in 0.035924 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002201 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/argr_arginine.txt ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/background\n",
      "fasta-get-markov ran successfully in 0.001616 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.810576 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/background ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/argr_arginine.txt ../data/e_coli/NC_000913.3/meme-chip/argr_arginine/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/tus_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.006398 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032383 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.04084 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003535 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.0338 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background\n",
      "fasta-get-markov ran successfully in 0.006139 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 614.295838 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/tus_sta/seqs-shuffled -t 5454\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/tus_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/tus_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.738134 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt\n",
      "fimo ran successfully in 0.131553 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt\n",
      "fimo ran successfully in 0.142268 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt\n",
      "fimo ran successfully in 0.132295 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_sta/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/tus_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_sta/tus_sta.txt\n",
      "fimo ran successfully in 0.115481 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gadw_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002594 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031104 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035183 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002388 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/background\n",
      "fasta-get-markov ran successfully in 0.00271 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.448117 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/background ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.076201 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gadw_M9/gadw_M9.txt\n",
      "fimo ran successfully in 0.038285 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yiep_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002188 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032794 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.037488 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002134 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/background\n",
      "fasta-get-markov ran successfully in 0.00203 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.616982 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/background ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.023041 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yiep_M9/yiep_M9.txt\n",
      "fimo ran successfully in 0.025953 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/topa_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.00814 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033063 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.039967 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.007827 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.032829 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/topa_sta/background\n",
      "fasta-get-markov ran successfully in 0.008589 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/topa_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/topa_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 595.898965 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/topa_sta/seqs-shuffled -t 5461\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/topa_sta/background ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/topa_sta/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Skipping sequence peak1056 as its length (482) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/topa_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/topa_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/topa_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.005321 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/topa_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/topa_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt ../data/e_coli/NC_000913.3/meme-chip/topa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/topa_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/topa_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/topa_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/topa_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/topa_sta/topa_sta.txt\n",
      "fimo ran successfully in 0.146467 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/puur_putrescine_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt 1> $metrics\n",
      "getsize ran successfully in 0.001699 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029071 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt 1> ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-centered\n",
      "fasta-center ran successfully in 0.036029 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002489 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background\n",
      "fasta-get-markov ran successfully in 0.002635 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 12.786558 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/seqs-shuffled -t 5694\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/combined.meme ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/motif_alignment.txt\n",
      "tomtom ran successfully in 0.807096 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt\n",
      "fimo ran successfully in 0.040887 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt\n",
      "fimo ran successfully in 0.039166 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt\n",
      "fimo ran successfully in 0.042301 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/puur_putrescine/puur_putrescine.txt\n",
      "fimo ran successfully in 0.038984 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mukb_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.003871 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031298 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.039362 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005272 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.032742 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background\n",
      "fasta-get-markov ran successfully in 0.006254 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 614.477362 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/seqs-shuffled -t 5454\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.352486 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt\n",
      "fimo ran successfully in 0.134384 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt\n",
      "fimo ran successfully in 0.126072 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt\n",
      "fimo ran successfully in 0.137154 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mukb_expo/mukb_expo.txt\n",
      "fimo ran successfully in 0.109247 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yhjb_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/yhjb_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002457 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/yhjb_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031515 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/yhjb_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.037469 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002246 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/yhjb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/background\n",
      "fasta-get-markov ran successfully in 0.00232 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.672919 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/background ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/yhjb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yhjb_M9/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/cra_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002688 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028555 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036665 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002368 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background\n",
      "fasta-get-markov ran successfully in 0.001606 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.573537 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cra_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cra_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cra_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.159961 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt\n",
      "fimo ran successfully in 0.030891 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_M9/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_M9/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/cra_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_M9/cra_M9.txt\n",
      "fimo ran successfully in 0.041182 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/fis_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.014968 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034687 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.044028 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006788 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.039536 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background\n",
      "fasta-get-markov ran successfully in 0.014252 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 530.954553 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/fis_sta/seqs-shuffled -t 5487\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/fis_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/fis_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.025889 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt\n",
      "fimo ran successfully in 0.249857 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/fis_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/fis_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/fis_sta/fis_sta.txt\n",
      "fimo ran successfully in 0.258987 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/rob_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.0066 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03157 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.040942 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004663 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background\n",
      "fasta-get-markov ran successfully in 0.004801 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 480.887999 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rob_expo/seqs-shuffled -t 5507\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Skipping sequence peak1 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rob_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rob_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.694592 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt\n",
      "fimo ran successfully in 0.105123 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt\n",
      "fimo ran successfully in 0.119851 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt\n",
      "fimo ran successfully in 0.117849 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_expo/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/rob_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_expo/rob_expo.txt\n",
      "fimo ran successfully in 0.118949 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybao_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/ybao_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.00242 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/ybao_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.02766 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/ybao_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.029081 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002164 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/ybao_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/background\n",
      "fasta-get-markov ran successfully in 0.001465 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.571851 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/background ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/ybao_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybao_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool1c_M9+glu_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/mixed-tfs-pool1c_M9+glu.txt 1> $metrics\n",
      "getsize ran successfully in 0.011747 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/mixed-tfs-pool1c_M9+glu.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035516 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/mixed-tfs-pool1c_M9+glu.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-centered\n",
      "fasta-center ran successfully in 0.047682 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00881 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.039768 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/mixed-tfs-pool1c_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/background\n",
      "fasta-get-markov ran successfully in 0.010557 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 619.192108 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/seqs-shuffled -t 5452\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/mixed-tfs-pool1c_M9+glu.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool1c_M9+glu/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ypdc_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/ypdc_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.00262 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/ypdc_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031192 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/ypdc_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.035614 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001479 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/ypdc_glc.txt ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/background\n",
      "fasta-get-markov ran successfully in 0.001904 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.619342 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/background ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/ypdc_glc.txt ../data/e_coli/NC_000913.3/meme-chip/ypdc_glc/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ygfi_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/ygfi_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002403 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/ygfi_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031008 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/ygfi_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.033958 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002134 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/ygfi_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/background\n",
      "fasta-get-markov ran successfully in 0.001949 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.568699 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/background ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/ygfi_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ygfi_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yiau_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/yiau_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002558 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/yiau_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032613 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/yiau_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035473 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002209 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/yiau_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/background\n",
      "fasta-get-markov ran successfully in 0.001557 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.3589 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/background ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/yiau_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yiau_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/flhc_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002881 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029292 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt 1> ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.034856 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002463 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background\n",
      "fasta-get-markov ran successfully in 0.001678 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 6.201511 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/combined.meme ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.827879 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt\n",
      "fimo ran successfully in 0.037538 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt\n",
      "fimo ran successfully in 0.036839 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt\n",
      "fimo ran successfully in 0.037885 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flhc_LB/flhc_LB.txt\n",
      "fimo ran successfully in 0.027733 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ybef_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/ybef_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002705 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/ybef_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030684 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/ybef_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036669 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002245 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/ybef_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/background\n",
      "fasta-get-markov ran successfully in 0.001966 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.629066 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/background ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/ybef_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ybef_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yidl_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/yidl_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002531 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/yidl_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03177 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/yidl_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035461 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002027 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/yidl_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/background\n",
      "fasta-get-markov ran successfully in 0.002142 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.723672 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/background ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/yidl_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yidl_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yhjc_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002423 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030293 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.028801 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002515 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/background\n",
      "fasta-get-markov ran successfully in 0.001588 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.05477 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/background ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.149292 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yhjc_M9/yhjc_M9.txt\n",
      "fimo ran successfully in 0.025871 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/mixed-tfs-pool3_M9+ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt 1> $metrics\n",
      "getsize ran successfully in 0.007597 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035755 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-centered\n",
      "fasta-center ran successfully in 0.042877 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006152 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.034903 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background\n",
      "fasta-get-markov ran successfully in 0.007057 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 632.180408 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/seqs-shuffled -t 5447\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/combined.meme ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/motif_alignment.txt\n",
      "tomtom ran successfully in 0.797445 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt\n",
      "fimo ran successfully in 0.148147 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt\n",
      "fimo ran successfully in 0.149021 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt\n",
      "fimo ran successfully in 0.15567 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/mixed-tfs-pool3_M9+ace/mixed-tfs-pool3_M9+ace.txt\n",
      "fimo ran successfully in 0.145196 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/gyrb_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.006836 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.037571 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.036314 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.006816 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.036229 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background\n",
      "fasta-get-markov ran successfully in 0.00836 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 624.213866 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/seqs-shuffled -t 5450\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.21751 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt\n",
      "fimo ran successfully in 0.141569 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt\n",
      "fimo ran successfully in 0.167545 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/gyrb_sta/gyrb_sta.txt\n",
      "fimo ran successfully in 0.129493 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/rob_sta_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt 1> $metrics\n",
      "getsize ran successfully in 0.008893 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034277 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt 1> ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-centered\n",
      "fasta-center ran successfully in 0.042238 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.008506 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.032936 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background\n",
      "fasta-get-markov ran successfully in 0.009473 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 600.439209 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/rob_sta/seqs-shuffled -t 5459\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/combined.meme ../data/e_coli/NC_000913.3/meme-chip/rob_sta/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/rob_sta/motif_alignment.txt\n",
      "tomtom ran successfully in 0.299361 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt\n",
      "fimo ran successfully in 0.203328 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt\n",
      "fimo ran successfully in 0.181109 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/rob_sta/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/rob_sta/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/rob_sta/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/rob_sta/rob_sta.txt\n",
      "fimo ran successfully in 0.155279 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ydci_ph8_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/ydci_ph8.txt 1> $metrics\n",
      "getsize ran successfully in 0.002531 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/ydci_ph8.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031286 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/ydci_ph8.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-centered\n",
      "fasta-center ran successfully in 0.03347 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002247 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/ydci_ph8.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/background\n",
      "fasta-get-markov ran successfully in 0.002135 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.79356 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/background ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/ydci_ph8.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph8/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yahb_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/yahb_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002365 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/yahb_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030805 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/yahb_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.03786 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002089 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/yahb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/background\n",
      "fasta-get-markov ran successfully in 0.002305 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.582287 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/background ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/yahb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yahb_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ycfq_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/ycfq_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002379 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/ycfq_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031735 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/ycfq_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.030774 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002153 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/ycfq_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/background\n",
      "fasta-get-markov ran successfully in 0.002215 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 0.609191 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/background ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/ycfq_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycfq_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/ydhb_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002615 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03019 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035742 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002427 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/background\n",
      "fasta-get-markov ran successfully in 0.002351 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.714077 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/background ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.017073 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ydhb_M9/ydhb_M9.txt\n",
      "fimo ran successfully in 0.026151 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/hupb_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.004669 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030925 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.037678 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003625 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background\n",
      "fasta-get-markov ran successfully in 0.003128 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 177.003892 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/seqs-shuffled -t 5629\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.382836 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt\n",
      "fimo ran successfully in 0.07755 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt\n",
      "fimo ran successfully in 0.072792 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/hupb_expo/hupb_expo.txt\n",
      "fimo ran successfully in 0.073967 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/flia_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003768 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031728 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt 1> ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.038059 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002931 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background\n",
      "fasta-get-markov ran successfully in 0.003239 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 82.099434 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/flia_LB/seqs-shuffled -t 5667\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/combined.meme ../data/e_coli/NC_000913.3/meme-chip/flia_LB/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/flia_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.701099 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt\n",
      "fimo ran successfully in 0.085439 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt\n",
      "fimo ran successfully in 0.053906 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt\n",
      "fimo ran successfully in 0.070859 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/flia_LB/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/flia_LB/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/flia_LB/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/flia_LB/flia_LB.txt\n",
      "fimo ran successfully in 0.060287 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/oxyr_pq_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt 1> $metrics\n",
      "getsize ran successfully in 0.002502 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031528 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt 1> ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-centered\n",
      "fasta-center ran successfully in 0.035738 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001539 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/background\n",
      "fasta-get-markov ran successfully in 0.002349 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.270998 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/background ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/combined.meme ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/motif_alignment.txt\n",
      "tomtom ran successfully in 0.123199 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/oxyr_pq/oxyr_pq.txt\n",
      "fimo ran successfully in 0.034634 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yihw_glc_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt 1> $metrics\n",
      "getsize ran successfully in 0.002453 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032398 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-centered\n",
      "fasta-center ran successfully in 0.035399 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002288 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/background\n",
      "fasta-get-markov ran successfully in 0.002295 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.75348 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/background ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/motif_alignment.txt\n",
      "tomtom ran successfully in 0.126613 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yihw_glc/yihw_glc.txt\n",
      "fimo ran successfully in 0.032726 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yddm_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002508 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030647 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035121 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.0021 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/background\n",
      "fasta-get-markov ran successfully in 0.002137 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.434615 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/background ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.016057 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yddm_M9/yddm_M9.txt\n",
      "fimo ran successfully in 0.026486 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/cra_galactose_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt 1> $metrics\n",
      "getsize ran successfully in 0.002131 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031213 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-centered\n",
      "fasta-center ran successfully in 0.035572 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001552 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/background\n",
      "fasta-get-markov ran successfully in 0.002442 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.149562 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/background ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/motif_alignment.txt\n",
      "tomtom ran successfully in 0.011847 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_galactose/cra_galactose.txt\n",
      "fimo ran successfully in 0.029006 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/soxr_pq_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/soxr_pq.txt 1> $metrics\n",
      "getsize ran successfully in 0.002583 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/soxr_pq.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031872 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/soxr_pq.txt 1> ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-centered\n",
      "fasta-center ran successfully in 0.037464 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002159 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/soxr_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/background\n",
      "fasta-get-markov ran successfully in 0.001344 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.632811 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/background ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/soxr_pq.txt ../data/e_coli/NC_000913.3/meme-chip/soxr_pq/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/arca-1_M9+ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt 1> $metrics\n",
      "getsize ran successfully in 0.002668 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031639 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt 1> ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-centered\n",
      "fasta-center ran successfully in 0.038298 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001588 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/background\n",
      "fasta-get-markov ran successfully in 0.002297 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 5.604839 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/background ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/combined.meme ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/motif_alignment.txt\n",
      "tomtom ran successfully in 0.012536 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/arca-1_M9+ace/arca-1_M9+ace.txt\n",
      "fimo ran successfully in 0.032141 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/yedw_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/yedw_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002588 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/yedw_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03374 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/yedw_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.034647 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002177 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/yedw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/background\n",
      "fasta-get-markov ran successfully in 0.002258 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.074505 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/background ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/yedw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yedw_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/yfec_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002757 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029957 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.036637 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001534 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/background\n",
      "fasta-get-markov ran successfully in 0.002318 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.740393 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/background ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.022288 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/yfec_M9/yfec_M9.txt\n",
      "fimo ran successfully in 0.033227 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/tus_expo_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt 1> $metrics\n",
      "getsize ran successfully in 0.00642 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033775 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt 1> ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-centered\n",
      "fasta-center ran successfully in 0.038571 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005012 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-centered 600 -rest ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-discarded -seed 1 1> ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.034765 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background\n",
      "fasta-get-markov ran successfully in 0.004434 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-sampled -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 611.240506 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/tus_expo/seqs-shuffled -t 5455\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/combined.meme ../data/e_coli/NC_000913.3/meme-chip/tus_expo/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/tus_expo/motif_alignment.txt\n",
      "tomtom ran successfully in 0.825396 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/spamo_out_3 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background -primary 3 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/spamo_out_4 -bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background -primary 4 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt\n",
      "fimo ran successfully in 0.129206 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt\n",
      "fimo ran successfully in 0.122054 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/fimo_out_3 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background --motif 3 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt\n",
      "fimo ran successfully in 0.122691 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/tus_expo/fimo_out_4 --bgfile ../data/e_coli/NC_000913.3/meme-chip/tus_expo/background --motif 4 ../data/e_coli/NC_000913.3/meme-chip/tus_expo/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/tus_expo/tus_expo.txt\n",
      "fimo ran successfully in 0.129617 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ycjw_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt 1> $metrics\n",
      "getsize ran successfully in 0.002868 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031884 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-centered\n",
      "fasta-center ran successfully in 0.035553 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002237 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/background\n",
      "fasta-get-markov ran successfully in 0.002185 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 11.201462 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/seqs-shuffled -t 5695\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/background ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/combined.meme ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/motif_alignment.txt\n",
      "tomtom ran successfully in 0.014127 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/ycjw_M9/ycjw_M9.txt\n",
      "fimo ran successfully in 0.183438 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/e_coli/NC_000913.3/curated_input/ydci_ph5_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/ydci_ph5.txt 1> $metrics\n",
      "getsize ran successfully in 0.002522 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/ydci_ph5.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030675 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/ydci_ph5.txt 1> ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-centered\n",
      "fasta-center ran successfully in 0.042969 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002158 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/ydci_ph5.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/background\n",
      "fasta-get-markov ran successfully in 0.002054 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.300404 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/background ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/ydci_ph5.txt ../data/e_coli/NC_000913.3/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/e_coli/NC_000913.3/curated_input/rpob (cra and crp ko exps)_MG1655 ace_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "../data/e_coli/NC_000913.3/curated_input/cra_fructose_curated.gff\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt 1> $metrics\n",
      "getsize ran successfully in 0.002618 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031021 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt 1> ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-centered\n",
      "fasta-center ran successfully in 0.043398 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-centered ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002421 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background\n",
      "fasta-get-markov ran successfully in 0.002588 seconds\n",
      "Starting meme: meme ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-centered -oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 3.036223 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/dreme_out -png -dna -p ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-centered -n ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/centrimo_out -bfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/combined.meme ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/combined.meme 1> ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/motif_alignment.txt\n",
      "tomtom ran successfully in 0.151062 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/spamo_out_1 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background -primary 1 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/spamo_out_2 -bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background -primary 2 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt\n",
      "fimo ran successfully in 0.041471 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/fimo_out_2 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/background --motif 2 ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cra_fructose/cra_fructose.txt\n",
      "fimo ran successfully in 0.028129 seconds\n",
      "mkdir: cannot create directory ‘../data/e_coli/NC_000913.3/meme-chip/outputs/’: File exists\n",
      "../data/s_coelicolor/NC_003888.3/curated_input/scbr_SMM_curated.gff\n",
      "Starting getsize: getsize ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt 1> $metrics\n",
      "getsize ran successfully in 0.003317 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034545 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt 1> ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-centered\n",
      "fasta-center ran successfully in 0.039023 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-centered ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003163 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/background\n",
      "fasta-get-markov ran successfully in 0.002771 seconds\n",
      "Starting meme: meme ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-centered -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 36.723619 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/dreme_out -png -dna -p ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-centered -n ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/seqs-shuffled -t 5685\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/centrimo_out -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/background ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/combined.meme ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/combined.meme 1> ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/motif_alignment.txt\n",
      "tomtom ran successfully in 0.021708 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/spamo_out_1 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/background -primary 1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/fimo_out_1 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/background --motif 1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr_SMM/scbr_SMM.txt\n",
      "fimo ran successfully in 0.050793 seconds\n",
      "../data/s_coelicolor/NC_003888.3/curated_input/hrdb_MS_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_coelicolor/NC_003888.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt 1> $metrics\n",
      "getsize ran successfully in 0.010545 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.034225 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt 1> ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-centered\n",
      "fasta-center ran successfully in 0.04348 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-centered ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.008907 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-centered 600 -rest ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-discarded -seed 1 1> ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.039966 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background\n",
      "fasta-get-markov ran successfully in 0.010108 seconds\n",
      "Starting meme: meme ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-sampled -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 627.369217 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/dreme_out -png -dna -p ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-centered -n ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/seqs-shuffled -t 5448\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/centrimo_out -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/combined.meme ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/combined.meme 1> ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/motif_alignment.txt\n",
      "tomtom ran successfully in 0.421637 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/spamo_out_1 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background -primary 1 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/spamo_out_2 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background -primary 2 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/spamo_out_3 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background -primary 4 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/spamo_out_4 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background -primary 3 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/fimo_out_1 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background --motif 1 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt\n",
      "fimo ran successfully in 0.235276 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/fimo_out_2 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background --motif 2 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt\n",
      "fimo ran successfully in 0.184077 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/fimo_out_3 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background --motif 4 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt\n",
      "fimo ran successfully in 0.237724 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/fimo_out_4 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/background --motif 3 ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/hrdb_MS/hrdb_MS.txt\n",
      "fimo ran successfully in 0.18117 seconds\n",
      "mkdir: cannot create directory ‘../data/s_coelicolor/NC_003888.3/meme-chip/outputs/’: File exists\n",
      "../data/s_coelicolor/NC_003888.3/curated_input/scbr2_SMM_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_coelicolor/NC_003888.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt 1> $metrics\n",
      "getsize ran successfully in 0.006341 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03366 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt 1> ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-centered\n",
      "fasta-center ran successfully in 0.038285 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-centered ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004853 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background\n",
      "fasta-get-markov ran successfully in 0.005162 seconds\n",
      "Starting meme: meme ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-centered -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 364.947204 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/dreme_out -png -dna -p ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-centered -n ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/seqs-shuffled -t 5554\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/centrimo_out -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/combined.meme ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/combined.meme 1> ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/motif_alignment.txt\n",
      "tomtom ran successfully in 0.013505 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/spamo_out_1 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background -primary 1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/spamo_out_2 -bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background -primary 2 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/fimo_out_1 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background --motif 1 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt\n",
      "fimo ran successfully in 0.11951 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/fimo_out_2 --bgfile ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/background --motif 2 ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/meme_out/meme.xml ../data/s_coelicolor/NC_003888.3/meme-chip/scbr2_SMM/scbr2_SMM.txt\n",
      "fimo ran successfully in 0.116392 seconds\n",
      "mkdir: cannot create directory ‘../data/s_coelicolor/NC_003888.3/meme-chip/outputs/’: File exists\n",
      "../data/s_coelicolor/NC_003888.3/curated_input/ndgr_minimal-media_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_coelicolor/NC_003888.3/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/ndgr_minimal-media.txt 1> $metrics\n",
      "getsize ran successfully in 0.002588 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/ndgr_minimal-media.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.029926 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/ndgr_minimal-media.txt 1> ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-centered\n",
      "fasta-center ran successfully in 0.030755 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-centered ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002389 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/ndgr_minimal-media.txt ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/background\n",
      "fasta-get-markov ran successfully in 0.002115 seconds\n",
      "Starting meme: meme ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-centered -oc ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.619727 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/dreme_out -png -dna -p ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-centered -n ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/centrimo_out -bfile ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/background ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/ndgr_minimal-media.txt ../data/s_coelicolor/NC_003888.3/meme-chip/ndgr_minimal-media/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/p_putida/NC_002947.4/curated_input/fur_fe_curated.gff\n",
      "Starting getsize: getsize ../data/p_putida/NC_002947.4/meme-chip/fur_fe/fur_fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.002691 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_putida/NC_002947.4/meme-chip/fur_fe/fur_fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028545 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_putida/NC_002947.4/meme-chip/fur_fe/fur_fe.txt 1> ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-centered\n",
      "fasta-center ran successfully in 0.037365 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-centered ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001933 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_putida/NC_002947.4/meme-chip/fur_fe/fur_fe.txt ../data/p_putida/NC_002947.4/meme-chip/fur_fe/background\n",
      "fasta-get-markov ran successfully in 0.002282 seconds\n",
      "Starting meme: meme ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-centered -oc ../data/p_putida/NC_002947.4/meme-chip/fur_fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_putida/NC_002947.4/meme-chip/fur_fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.818823 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_putida/NC_002947.4/meme-chip/fur_fe/dreme_out -png -dna -p ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-centered -n ../data/p_putida/NC_002947.4/meme-chip/fur_fe/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_putida/NC_002947.4/meme-chip/fur_fe/centrimo_out -bfile ../data/p_putida/NC_002947.4/meme-chip/fur_fe/background ../data/p_putida/NC_002947.4/meme-chip/fur_fe/fur_fe.txt ../data/p_putida/NC_002947.4/meme-chip/fur_fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/p_putida/NC_002947.4/curated_input/sigma d_M9_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_putida/NC_002947.4/meme-chip’: File exists\n",
      "The sequences specified do not exist.\n",
      "Usage:\n",
      "    meme-chip [options] <sequences>\n",
      "\n",
      "     Options:\n",
      "      -o                <dir>   : output to the specified directory, failing if the directory exists\n",
      "      -oc               <dir>   : output to the specified directory, overwriting if the directory exists\n",
      "      -index-name       <name>  : name of html index file; default: index.html\n",
      "      -db               <path>  : target database for use by TOMTOM and CentriMo, if not present then \n",
      "                                  TOMTOM and CentriMo are not run\n",
      "      -neg              <path>  : negative (control) sequence file name;\n",
      "                                  sequences are assumed to originate from the same\n",
      "                                  alphabet as the positive sequence set and should\n",
      "                                  be the same length as those;\n",
      "                                  default: no negative sequences are used for MEME and\n",
      "                                  for DREME, the positive sequences are shuffled\n",
      "                                  to create the negative set\n",
      "      -dna                        set the alphabet to DNA; this is the default\n",
      "      -rna                        set the alphabet to RNA\n",
      "      -[x]alph          <path>  : alphabet file; when the x is specified the motif\n",
      "                                  databases are converted to the specified alphabet;\n",
      "                                  default: DNA\n",
      "      -bfile            <path>  : background file\n",
      "      -order            <ord>   : set the order of the Markov background model\n",
      "                                  that is generated from the sequences when a\n",
      "                                  background file is not given; default: 1\n",
      "      -nmeme            <num>   : limit of sequences to pass to MEME\n",
      "      -seed             <seed>  : seed for the randomized selection of sequences\n",
      "                                  for MEME and the shuffling of sequences for DREME;\n",
      "                                  default: seed randomly\n",
      "      -norand                   : disable random selection of sequences and\n",
      "                                  select in file order\n",
      "      -ccut             <num>   : maximum size of a sequence before it is cut down \n",
      "                                  to a centered section; a value of 0 indicates the\n",
      "                                  sequences should not be cut down; default: 100\n",
      "      -group-thresh     <gthr>  : primary threshold for clustering motifs; default: 0.05\n",
      "      -group-weak       <gwk>   : secondary threshold for clustering motifs; default: 2*gthr\n",
      "      -filter-thresh    <fthr>  : E-value threshold for including motifs; default: 0.05\n",
      "      -time          <minutes>  : maximum time that this program has to run and \n",
      "                                  create output in; default: no limit\n",
      "      -desc             <text>  : description of the job\n",
      "      -fdesc            <file>  : file containing plain text description of the job\n",
      "      -norc                     : search given strand only\n",
      "      -old-clustering           : pick cluster seed motifs based only on significance;\n",
      "                                  default: preferentially select discovered motifs as\n",
      "                                  clustering seeds even if there is a library motif\n",
      "                                  that appears more enriched\n",
      "      -noecho                   : don't echo the commands run\n",
      "      -help                     : display this help message\n",
      "      -version                  : print the version and exit\n",
      "\n",
      "     MEME Specific Options:\n",
      "      -meme-mod [oops|zoops|anr]: sites used in a single sequence\n",
      "      -meme-minw        <num>   : minimum motif width\n",
      "      -meme-maxw        <num>   : maximum motif width\n",
      "      -meme-nmotifs     <num>   : maximum number of motifs to find\n",
      "      -meme-minsites    <num>   : minimum number of sites per motif\n",
      "      -meme-maxsites    <num>   : maximum number of sites per motif\n",
      "      -meme-p           <np>    : use parallel version with <np> processors\n",
      "      -meme-pal                 : look for palindromes only\n",
      "      -meme-maxsize     <num>   : change the maximum dataset size; note that the\n",
      "                                  default maximum size exists to warn users that\n",
      "                                  their dataset is possibly too large to process\n",
      "                                  in a reasonable time so please consider carefully\n",
      "                                  before increasing this value. It should not be\n",
      "                                  required to change this value unless you are\n",
      "                                  changing -ccut or -nmeme options.\n",
      "\n",
      "     DREME Specific Options:\n",
      "      -dreme-e          <num>   : stop searching after reaching this E-value threshold\n",
      "      -dreme-m          <num>   : stop searching after finding this many motifs\n",
      "\n",
      "     CentriMo Specific Options:\n",
      "      -centrimo-local           : compute enrichment of all regions (not only central)\n",
      "      -centrimo-score   <num>   : set the minimum allowed match score\n",
      "      -centrimo-maxreg  <num>   : set the maximum region size to be considered\n",
      "      -centrimo-ethresh <num>   : set the E-value threshold for reporting\n",
      "      -centrimo-noseq           : don't store sequence IDs in the output\n",
      "      -centrimo-flip            : reflect matches on reverse strand around center\n",
      "\n",
      "     SpaMo Specific Options:\n",
      "      -spamo-skip               : don't run SpaMo\n",
      "\n",
      "     FIMO Specific Options:\n",
      "      -fimo-skip                : don't run FIMO\n",
      "\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/rsal_LB_curated.gff\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002282 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032098 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.033767 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002262 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/background\n",
      "fasta-get-markov ran successfully in 0.002199 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 6.107566 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/seqs-shuffled -t 5697\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.019212 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rsal_LB/rsal_LB.txt\n",
      "fimo ran successfully in 0.034427 seconds\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/sphr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003821 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03039 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.036233 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001996 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/background\n",
      "fasta-get-markov ran successfully in 0.003359 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 79.65131 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/seqs-shuffled -t 5668\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.022187 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/sphr_LB/sphr_LB.txt\n",
      "fimo ran successfully in 0.057199 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/mext_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002483 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.024108 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.035112 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00237 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/background\n",
      "fasta-get-markov ran successfully in 0.002252 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 1.075232 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/seqs-shuffled -t 5699\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.045945 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/mext_LB/mext_LB.txt\n",
      "fimo ran successfully in 0.029533 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/mvfr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002919 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03232 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.035055 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002451 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002169 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 2.743425 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.022122 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/mvfr_LB/mvfr_LB.txt\n",
      "fimo ran successfully in 0.030436 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/rpon_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.004526 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03231 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.038069 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003829 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background\n",
      "fasta-get-markov ran successfully in 0.003311 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 168.225332 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/seqs-shuffled -t 5632\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.031292 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/spamo_out_2 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background -primary 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt\n",
      "fimo ran successfully in 0.09469 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/fimo_out_2 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/background --motif 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/rpon_LB/rpon_LB.txt\n",
      "fimo ran successfully in 0.099122 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/algr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002783 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032559 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.035272 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002578 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/background\n",
      "fasta-get-markov ran successfully in 0.00263 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 10.870923 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/seqs-shuffled -t 5695\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.011822 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/algr_LB/algr_LB.txt\n",
      "fimo ran successfully in 0.030598 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/vfrv_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003596 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030405 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.036988 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002969 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background\n",
      "fasta-get-markov ran successfully in 0.003053 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 48.105024 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/seqs-shuffled -t 5680\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.032372 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/spamo_out_2 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background -primary 3 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt\n",
      "fimo ran successfully in 0.053574 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/fimo_out_2 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/background --motif 3 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfrv_LB/vfrv_LB.txt\n",
      "fimo ran successfully in 0.053479 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/gbdr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002738 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031164 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.037964 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.001713 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002553 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.550793 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.02066 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/gbdr_LB/gbdr_LB.txt\n",
      "fimo ran successfully in 0.035667 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/cdpr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002712 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030385 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.037779 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002287 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002382 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 5.284051 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.019747 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/spamo_out_2 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background -primary 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt\n",
      "fimo ran successfully in 0.031297 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/fimo_out_2 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/background --motif 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/cdpr_LB/cdpr_LB.txt\n",
      "fimo ran successfully in 0.034191 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/pchr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/pchr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002983 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/pchr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028997 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/pchr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.036269 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002661 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/pchr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002392 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 23.313323 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/pchr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/pchr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/wspr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/wspr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002586 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/wspr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.031631 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/wspr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.031546 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002201 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/wspr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002267 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "You must specify '-mod anr' to set the motif site model to 'Any Number of Repetitions [per sequence]' when only providing one sequence\n",
      "meme exited with error code 1\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/wspr_LB/seqs-shuffled -t 5700\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/exsa_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002515 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.03245 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.034578 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002905 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/background\n",
      "fasta-get-markov ran successfully in 0.001588 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 3.366595 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.045358 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/exsa_LB/exsa_LB.txt\n",
      "fimo ran successfully in 0.031423 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/fleq_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003407 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028226 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.035827 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.003225 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/background\n",
      "fasta-get-markov ran successfully in 0.00307000000000002 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 50.872735 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/seqs-shuffled -t 5679\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.012696 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/fleq_LB/fleq_LB.txt\n",
      "fimo ran successfully in 0.050346 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/vqsm_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003008 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.028791 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.036213 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002666 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/background\n",
      "fasta-get-markov ran successfully in 0.002719 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 23.212218 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/seqs-shuffled -t 5690\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.007864 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vqsm_LB/vqsm_LB.txt\n",
      "fimo ran successfully in 0.038772 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/gaca_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002731 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032288 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.036249 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002245 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/background\n",
      "fasta-get-markov ran successfully in 0.002215 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.925136 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.022019 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/gaca_LB/gaca_LB.txt\n",
      "fimo ran successfully in 0.032745 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/vfr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003698 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030099 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.031338 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002836 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background\n",
      "fasta-get-markov ran successfully in 0.00179 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 46.103541 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/seqs-shuffled -t 5681\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.033097 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/spamo_out_2 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background -primary 3 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt\n",
      "fimo ran successfully in 0.052105 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/fimo_out_2 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/background --motif 3 ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/vfr_LB/vfr_LB.txt\n",
      "fimo ran successfully in 0.056386 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/phob_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.003007 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.030659 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.037893 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002743 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/background\n",
      "fasta-get-markov ran successfully in 0.001707 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 9.768869 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/seqs-shuffled -t 5696\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.019835 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/phob_LB/phob_LB.txt\n",
      "fimo ran successfully in 0.035439 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/p_aeruginosa/NC_002516.2/curated_input/soxr_LB_curated.gff\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt 1> $metrics\n",
      "getsize ran successfully in 0.002693 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032356 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-centered\n",
      "fasta-center ran successfully in 0.034808 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-centered ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002353 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/background\n",
      "fasta-get-markov ran successfully in 0.002237 seconds\n",
      "Starting meme: meme ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-centered -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 4.839408 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/dreme_out -png -dna -p ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-centered -n ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/seqs-shuffled -t 5698\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/centrimo_out -bfile ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/background ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/combined.meme ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/combined.meme 1> ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/motif_alignment.txt\n",
      "tomtom ran successfully in 0.021295 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/spamo_out_1 -bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/background -primary 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/fimo_out_1 --bgfile ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/background --motif 1 ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/meme_out/meme.xml ../data/p_aeruginosa/NC_002516.2/meme-chip/soxr_LB/soxr_LB.txt\n",
      "fimo ran successfully in 0.031901 seconds\n",
      "mkdir: cannot create directory ‘../data/p_aeruginosa/NC_002516.2/meme-chip/outputs/’: File exists\n",
      "../data/y_pseudotuberculosis/NZ_CP009712.1/curated_input/fur_Fe_curated.gff\n",
      "Starting getsize: getsize ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.003519 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033102 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.034866 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-centered ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00255 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.001879 seconds\n",
      "Starting meme: meme ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-centered -oc ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 26.056005 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-centered -n ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/seqs-shuffled -t 5689\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/centrimo_out -bfile ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/background ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/combined.meme ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/combined.meme 1> ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.017984 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/background -primary 1 ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/background --motif 1 ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/meme_out/meme.xml ../data/y_pseudotuberculosis/NZ_CP009712.1/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.046008 seconds\n",
      "../data/s_enterica/NC_003197.2/curated_input/fur_Fe_curated.gff\n",
      "Starting getsize: getsize ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "getsize ran successfully in 0.003159 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.033466 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt 1> ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-centered\n",
      "fasta-center ran successfully in 0.035771 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-centered ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.002517 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/background\n",
      "fasta-get-markov ran successfully in 0.002289 seconds\n",
      "Starting meme: meme ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-centered -oc ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 10.814409 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/dreme_out -png -dna -p ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-centered -n ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/seqs-shuffled -t 5695\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/centrimo_out -bfile ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/background ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/combined.meme ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/combined.meme 1> ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/motif_alignment.txt\n",
      "tomtom ran successfully in 0.019853 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/spamo_out_1 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/background -primary 1 ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fimo_out_1 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/background --motif 1 ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/fur_Fe/fur_Fe.txt\n",
      "fimo ran successfully in 0.03939 seconds\n",
      "../data/s_enterica/NC_003197.2/curated_input/ydci_MES5_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt 1> $metrics\n",
      "getsize ran successfully in 0.005995 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032545 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-centered\n",
      "fasta-center ran successfully in 0.038396 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-centered ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.005061 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background\n",
      "fasta-get-markov ran successfully in 0.006039 seconds\n",
      "Starting meme: meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-centered -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 608.531897 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/dreme_out -png -dna -p ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-centered -n ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/seqs-shuffled -t 5456\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/centrimo_out -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/combined.meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/combined.meme 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/motif_alignment.txt\n",
      "tomtom ran successfully in 0.789078 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/spamo_out_1 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background -primary 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/spamo_out_2 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background -primary 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/spamo_out_3 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background -primary 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/spamo_out_4 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background -primary 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/fimo_out_1 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background --motif 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt\n",
      "fimo ran successfully in 0.119417 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/fimo_out_2 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background --motif 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt\n",
      "fimo ran successfully in 0.117325 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/fimo_out_3 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background --motif 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt\n",
      "fimo ran successfully in 0.112478 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/fimo_out_4 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/background --motif 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES5/ydci_MES5.txt\n",
      "fimo ran successfully in 0.113996 seconds\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip/outputs/’: File exists\n",
      "../data/s_enterica/NC_003197.2/curated_input/ydci_MES2_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt 1> $metrics\n",
      "getsize ran successfully in 0.00579600000000002 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.032627 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-centered\n",
      "fasta-center ran successfully in 0.03867 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-centered ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.004854 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background\n",
      "fasta-get-markov ran successfully in 0.005603 seconds\n",
      "Starting meme: meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-centered -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 514.674267 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/dreme_out -png -dna -p ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-centered -n ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/seqs-shuffled -t 5494\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/centrimo_out -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml\n",
      "Skipping sequence peak0 as its length (0) does not match the expected length (500).\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/combined.meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/combined.meme 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/motif_alignment.txt\n",
      "tomtom ran successfully in 0.997894 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/spamo_out_1 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background -primary 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/spamo_out_2 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background -primary 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/spamo_out_3 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background -primary 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/spamo_out_4 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background -primary 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/fimo_out_1 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background --motif 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt\n",
      "fimo ran successfully in 0.111854 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/fimo_out_2 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background --motif 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt\n",
      "fimo ran successfully in 0.116695 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/fimo_out_3 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background --motif 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt\n",
      "fimo ran successfully in 0.113498 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/fimo_out_4 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/background --motif 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_MES2/ydci_MES2.txt\n",
      "fimo ran successfully in 0.11486 seconds\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip/outputs/’: File exists\n",
      "../data/s_enterica/NC_003197.2/curated_input/ydci_ph5_curated.gff\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip’: File exists\n",
      "Starting getsize: getsize ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt 1> $metrics\n",
      "getsize ran successfully in 0.00955 seconds\n",
      "Starting fasta-most: fasta-most -min 50 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt 1> $metrics\n",
      "fasta-most ran successfully in 0.035009 seconds\n",
      "Starting fasta-center: fasta-center -dna -len 100 < ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-centered\n",
      "fasta-center ran successfully in 0.040351 seconds\n",
      "Starting fasta-shuffle-letters: fasta-shuffle-letters ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-centered ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-shuffled -kmer 2 -tag -dinuc -dna -seed 1\n",
      "fasta-shuffle-letters ran successfully in 0.00662 seconds\n",
      "Starting fasta-subsample: fasta-subsample ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-centered 600 -rest ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-discarded -seed 1 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-sampled\n",
      "fasta-subsample ran successfully in 0.035511 seconds\n",
      "Starting fasta-get-markov: fasta-get-markov -nostatus -nosummary -dna -m 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background\n",
      "fasta-get-markov ran successfully in 0.007596 seconds\n",
      "Starting meme: meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-sampled -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out -mod zoops -nmotifs 4 -minw 6 -maxw 40 -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background -dna -time 4054 -revcomp -nostatus\n",
      "meme ran successfully in 615.578677 seconds\n",
      "Starting dreme: dreme -v 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/dreme_out -png -dna -p ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-centered -n ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/seqs-shuffled -t 5453\n",
      "  File \"/home/katherine/anaconda3/bin/dreme\", line 760\n",
      "    print \"Finding secondary RE in left flank...\"\n",
      "                                                ^\n",
      "SyntaxError: Missing parentheses in call to 'print'. Did you mean print(\"Finding secondary RE in left flank...\")?\n",
      "dreme exited with error code 1\n",
      "Starting centrimo: centrimo -seqlen 500 -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/centrimo_out -bfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "Bad file name.\n",
      "FATAL: Template does not contain data section.\n",
      "\n",
      "centrimo exited with error code 1\n",
      "Starting tomtom: tomtom -verbosity 1 -text -thresh 0.1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/combined.meme ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/combined.meme 1> ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/motif_alignment.txt\n",
      "tomtom ran successfully in 0.707211 seconds\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/spamo_out_1 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background -primary 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/spamo_out_2 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background -primary 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "spamo process died with signal 6, with coredump\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/spamo_out_3 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background -primary 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting spamo: spamo -verbosity 1 -oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/spamo_out_4 -bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background -primary 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml\n",
      "spamo exited with error code 1\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/fimo_out_1 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background --motif 1 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt\n",
      "fimo ran successfully in 0.154268 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/fimo_out_2 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background --motif 2 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt\n",
      "fimo ran successfully in 0.141935 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/fimo_out_3 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background --motif 3 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt\n",
      "fimo ran successfully in 0.152516 seconds\n",
      "Starting fimo: fimo --parse-genomic-coord --verbosity 1 --oc ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/fimo_out_4 --bgfile ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/background --motif 4 ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/meme_out/meme.xml ../data/s_enterica/NC_003197.2/meme-chip/ydci_ph5/ydci_ph5.txt\n",
      "fimo ran successfully in 0.146434 seconds\n",
      "mkdir: cannot create directory ‘../data/s_enterica/NC_003197.2/meme-chip/outputs/’: File exists\n"
     ]
    }
   ],
   "source": [
    "db_dir = '../data/'\n",
    "\n",
    "all_peaks = glob.glob(db_dir+\"*/*/curated_input/*.gff\")\n",
    "\n",
    "for peak_file in all_peaks:\n",
    "    print(peak_file)\n",
    "    name= peak_file.split('/')[5][0:-12]\n",
    "    tfbs = pd.read_csv(peak_file, sep='\\t', header=None,\n",
    "                      names=['genome', 'program','filter', 'Start Position', 'End Position', '1', '2', '3', '4'])\n",
    "    tfbs['mid'] = 0.5*(tfbs['End Position'] + tfbs['Start Position'])\n",
    "    tfbs['length'] = tfbs['End Position'] - tfbs['Start Position']\n",
    "    \n",
    "    strain_folder = \"/\".join(peak_file.split('/')[0:4])\n",
    "    fasta = strain_folder+\"/sequence/sequence.fasta\"\n",
    "    for seq_record in SeqIO.parse(fasta, \"fasta\"):\n",
    "        sequence = seq_record.seq\n",
    "\n",
    "    seq_list = []\n",
    "    for i,row in tfbs.iterrows():\n",
    "        start = int(row.mid-250)\n",
    "        stop = int(row.mid+250)\n",
    "        seq = str(sequence[start:stop])\n",
    "        seq_list.append(seq)\n",
    "        \n",
    "\n",
    "    !{'mkdir '+strain_folder+'/meme-chip'}\n",
    "\n",
    "    filename = strain_folder+'/meme-chip/'+name+'.txt'\n",
    "    textfile = open(filename, \"w\")\n",
    "    for i,seq in enumerate(seq_list):\n",
    "        textfile.write(\">peak\"+str(i)+ \"\\n\")\n",
    "        textfile.write(seq + \"\\n\")\n",
    "    textfile.close()\n",
    "\n",
    "    out_folder= strain_folder+'/meme-chip/'+name\n",
    "    !{\"meme-chip -oc \"+out_folder+\" -time 240 -ccut 100 -dna -order 2 -meme-minw 6 -meme-maxw 40 \\\n",
    "      -filter-thresh 0.001 \\\n",
    "      -meme-mod zoops -meme-nmotifs 4 \"+filename}\n",
    "    \n",
    "    # parse XML output, if available\n",
    "    if exists(out_folder+'/meme_out/meme.xml'):\n",
    "        tree = ET.parse(out_folder+'/meme_out/meme.xml')\n",
    "        root = tree.getroot()\n",
    "\n",
    "        e_thres = 0.001\n",
    "        for motif in root.findall('./motifs/motif'):\n",
    "            probs = motif.findall('probabilities/alphabet_matrix/alphabet_array/value')\n",
    "            e_value = float(motif.get('e_value'))\n",
    "\n",
    "            if e_value < e_thres:\n",
    "                final_outputs = strain_folder+'/meme-chip/outputs/'\n",
    "                !{'mkdir '+final_outputs}\n",
    "                # Get probabilities\n",
    "                prob_list = []\n",
    "                for prob in probs:\n",
    "                    prob_list.append(float(prob.text))\n",
    "                # make PWM\n",
    "                pwm = pd.DataFrame(index=['A', 'C', 'G', 'T'])\n",
    "                for i,four_start in enumerate(range(0, len(prob_list),4)):\n",
    "                    pwm['position-'+str(i+1)] = prob_list[four_start:four_start+4]\n",
    "                pwm.to_csv(final_outputs+name+\".csv\")\n",
    "                # reformat image and save\n",
    "                motif_num = motif.get('name')\n",
    "                img = Image.open(out_folder+\"/meme_out/logo\"+motif_num+\".png\")\n",
    "                img = ImageOps.expand(img, border=30)\n",
    "                draw = ImageDraw.Draw(img)\n",
    "                font = ImageFont.truetype(\"FONTS/ariali.ttf\", 36)\n",
    "                font_bold = ImageFont.truetype(\"FONTS/arial.ttf\", 36)\n",
    "                draw.text((30,310),\"E-value = \"+\"{:.1e}\".format(e_value),'#000000',font=font)\n",
    "                draw.text((30,0),\"Condition: \"+name.split(\"_\")[1],'#000000',font=font_bold)\n",
    "                img.save(final_outputs+name+\".png\")\n",
    "                #reflect change in e-value\n",
    "                e_thres = e_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Metadata for manuscript table S1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._set_item(key, value)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return func(*args, **kwargs)\n",
      "/home/katherine/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:10: FutureWarning: As the xlwt package is no longer maintained, the xlwt engine will be removed in a future version of pandas. This is the only engine in pandas that supports writing in the xls format. Install openpyxl and write to an xlsx file instead. You can set the option io.excel.xls.writer to 'xlwt' to silence this warning. While this option is deprecated and will also raise a warning, it can be globally set and the warning suppressed.\n",
      "  # Remove the CWD from sys.path while we load stuff.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>TF</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>v1.0.0 TF dashboard link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Organism</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>MksB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>ParB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>Smc</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArcA-1</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArcA-2</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://prochipdb.org/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Strain      TF    accession  \\\n",
       "Organism                                                       \n",
       "Corynebacterium glutamicum   ATCC 13032    MksB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032    ParB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032     Smc  PRJNA529385   \n",
       "Escherichia coli            K-12 MG1655  ArcA-1    GSE182077   \n",
       "Escherichia coli            K-12 MG1655  ArcA-2    GSE182077   \n",
       "\n",
       "                                                                   doi  \\\n",
       "Organism                                                                 \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Escherichia coli                                                   NaN   \n",
       "Escherichia coli                                                   NaN   \n",
       "\n",
       "                                                     v1.0.0 TF dashboard link  \n",
       "Organism                                                                       \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.org/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.org/tf_dashboard.html?organi...  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_ecoli = pd.read_csv('../data/e_coli/TF_list.csv', index_col=0)\n",
    "meta_other = pd.read_csv('../data/all_other/TF_list.csv', index_col=0)\n",
    "\n",
    "meta_combined = pd.concat([meta_ecoli, meta_other], axis=0)\n",
    "table_s1 = meta_combined[['Organism', 'Strain', 'TF', 'accession', 'doi']]\n",
    "table_s1['v1.0.0 TF dashboard link'] = 'https://prochipdb.org/tf_dashboard.html?organism='+meta_combined.organism_id+'&tf='+meta_combined.TF+'&genome='+meta_combined.genome_id+'&i='+[str(i) for i in meta_combined.index.to_list()]\n",
    "table_s1.sort_values(by=['Organism','Strain','TF'], inplace=True)\n",
    "table_s1.set_index('Organism', inplace=True)\n",
    "\n",
    "table_s1.to_excel('../../Supplementary Table 1.xls')\n",
    "table_s1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "e coli dataset\n",
      "num org 1\n",
      "num strain 4\n",
      "num TF 65\n",
      "num ChIP-exo 184 92.9%\n",
      "num ChIP-exo 14 7.1%\n",
      "num sample 198\n",
      "percent curated 92.3 60\n"
     ]
    }
   ],
   "source": [
    "print('e coli dataset')\n",
    "print('num org', len(meta_ecoli.Organism.unique()))\n",
    "print('num strain', len(meta_ecoli.Strain.unique()))\n",
    "print('num TF', len(meta_ecoli))\n",
    "num_samples = meta_ecoli.num_samples.sum()\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_ecoli['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 13\n",
      "num strain 14\n",
      "num TF 35\n",
      "num ChIP-exo 28 38.4%\n",
      "num ChIP-exo 45 61.6%\n",
      "num sample 73\n",
      "percent curated 85.7 30\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_other.Organism.unique()))\n",
    "print('num strain', len(meta_other.Strain.unique()))\n",
    "print('num TF', len(meta_other))\n",
    "num_samples = meta_other.num_samples.sum()\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_other['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 14\n",
      "num strain 18\n",
      "num TF 100\n",
      "num ChIP-exo 212 78.2%\n",
      "num ChIP-exo 59 21.8%\n",
      "num sample 271\n",
      "percent curated 90.0 90\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_combined.Organism.unique()))\n",
    "print('num strain', len(meta_combined.Strain.unique()))\n",
    "print('num TF', len(meta_combined))\n",
    "num_samples = meta_combined.num_samples.sum()\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_combined['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
