{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for proChIPdb, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "from zipfile import ZipFile\n",
    "from Bio import SeqIO\n",
    "import xml.etree.ElementTree as ET\n",
    "from os.path import exists\n",
    "from PIL import Image, ImageFont, ImageDraw, ImageOps\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + TF_condition + \"-\" + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus_nc000913(peak_start, peak_stop, gene_info):\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start >= BP) & (stop >= BP) & (strand == '+')) | ((start <= BP) & (stop <= BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start <= BP) & (stop >= BP)) | ((start >= BP) & (stop <= BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "        \n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    df_complete['closest_gene'] = [closest_locus_nc000913(df_complete.binding_peak_start.iloc[i], \n",
    "                                           df_complete.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(df_complete))]\n",
    "#     df_complete['target_genes'] = [df_complete['target_genes'].to_list()[i]+\", \"+df_complete['closest_gene'].to_list()[i] if df_complete['closest_gene'].to_list()[i] not in df_complete['target_genes'].to_list()[i] else df_complete['target_genes'].to_list()[i] for i in range(len(df_complete))]\n",
    "\n",
    "            \n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen2(Peak_DF, TRN_data, gene_info): \n",
    "    TF_name = Peak_DF['index'][1].split(\"-\")[0].split(\" \")[0]\n",
    "    \n",
    "\n",
    "    # TRN locus\n",
    "    reg_genes = set(TRN_data.loc[TF_name].gene_id.to_list())\n",
    "\n",
    "    # chip locus\n",
    "    gene_list = [i for i in Peak_DF.target_locus if i != '']\n",
    "    chip_genes = set(','.join(list(gene_list)).split(','))\n",
    "    \n",
    "    #both\n",
    "    both_genes = set(reg_genes & chip_genes)\n",
    "    \n",
    "    # Get gene and operon counts\n",
    "    reg_gene_count = len(reg_genes)\n",
    "    chip_gene_count = len(chip_genes)\n",
    "    both_gene_count = len(both_genes)\n",
    "    \n",
    "    # Add adjustments for venn plotting (add '2' for alternates)\n",
    "    reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    if reg_genes == chip_genes:\n",
    "        reg_gene_count = 0; chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = len(reg_genes)\n",
    "    elif all(item in chip_genes for item in reg_genes):\n",
    "        reg_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = len(reg_genes); chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    elif all(item in reg_genes for item in chip_genes):\n",
    "        chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = len(chip_genes); both_gene_count2 = 0\n",
    "        \n",
    "    res = pd.DataFrame([TF_name, reg_gene_count, chip_gene_count, both_gene_count,\n",
    "                        reg_gene_count2, chip_gene_count2, both_gene_count2], \n",
    "                        columns=['Value'],\n",
    "                        index=['TF', 'reg_genes', 'chip_genes', 'both_genes',\n",
    "                                'reg_genes2', 'chip_genes2', 'both_genes2'])\n",
    "    \n",
    "    # Add gene lists\n",
    "    just_reg = reg_genes - both_genes\n",
    "    just_chip = chip_genes - both_genes\n",
    "    for i, l in zip(['reg_genes', 'chip_genes', 'both_genes'],[just_reg, just_chip, both_genes]):\n",
    "        gene_list = np.array([gene_info.loc[g,'gene_name'] if g in gene_info.index else g for g in l])\n",
    "        gene_list = np.array2string(gene_list, separator = ' ')\n",
    "        res.loc[i, 'list'] = gene_list\n",
    "    \n",
    "    # Add data sources\n",
    "    res.loc['TF', 'list'] = '; '.join(TRN_data.source[TRN_data.index == TF_name].unique())\n",
    "    \n",
    "    return res"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        gene = peak.closest_gene\n",
    "        gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "        try:\n",
    "            if gene_info.strand[0] == \"-\":\n",
    "                dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "            else:\n",
    "                dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "            peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "            counter+=1\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - NC_000913.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>PMID</th>\n",
       "      <th>additional_carbon_source</th>\n",
       "      <th>media_info</th>\n",
       "      <th>supplement_info</th>\n",
       "      <th>Additional_Info</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>33172971.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>33172971.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>1mL trace element solution</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>glucose|cra_glucose_binding_table.json;acetate...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>29394395.0</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Trace element solution (100x): 1 g EDTA, 29 mg...</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>25222563.0</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>HCl</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26258987.0</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Acidic stress condition: M9 media adjusted to ...</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>mixed-TFs-pool4</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>1mL trace element solution</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Trace element solution (100x): 1 g EDTA, 29 mg...</td>\n",
       "      <td>--</td>\n",
       "      <td>Mixed TFs: YidZ, YedW, ArcA, YhjB</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool4_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool4_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>mixed-TFs-pool5</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>1mL trace element solution</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>combination</td>\n",
       "      <td>M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json</td>\n",
       "      <td>GSE182078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>3 – Mixed TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Galactose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Trace element solution (100x): 1 g EDTA, 29 mg...</td>\n",
       "      <td>--</td>\n",
       "      <td>Mixed TFs: YebK, YbeF</td>\n",
       "      <td>R1</td>\n",
       "      <td>mixTF_pool5_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>mixTF_pool5_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ArcA-1</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>1mL trace element solution</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca-1_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Trace element solution (100x): 1 g EDTA, 29 mg...</td>\n",
       "      <td>--</td>\n",
       "      <td>monoclonal antibodies (ArcA part1)</td>\n",
       "      <td>1C6</td>\n",
       "      <td>arcA_1C6.bw</td>\n",
       "      <td>1D6</td>\n",
       "      <td>arcA_1D6.bw</td>\n",
       "      <td>2A1</td>\n",
       "      <td>arcA_2A1.bw</td>\n",
       "      <td>2B6</td>\n",
       "      <td>arcA_2B6.bw</td>\n",
       "      <td>2C4</td>\n",
       "      <td>arcA_2C4.bw</td>\n",
       "      <td>2D6</td>\n",
       "      <td>arcA_2D6.bw</td>\n",
       "      <td>4A1</td>\n",
       "      <td>arcA_4A1.bw</td>\n",
       "      <td>4A3</td>\n",
       "      <td>arcA_4A3.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>ArcA-2</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>1mL trace element solution</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>mAbs</td>\n",
       "      <td>M9+ace|arca-2_M9+ace_binding_table.json</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>2 – mAbs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Trace element solution (100x): 1 g EDTA, 29 mg...</td>\n",
       "      <td>--</td>\n",
       "      <td>monoclonal antibodies (ArcA part2)</td>\n",
       "      <td>4D6</td>\n",
       "      <td>arcA_4D6.bw</td>\n",
       "      <td>5A10</td>\n",
       "      <td>arcA_5A10.bw</td>\n",
       "      <td>5B6</td>\n",
       "      <td>arcA_5B6.bw</td>\n",
       "      <td>5B9</td>\n",
       "      <td>arcA_5B9.bw</td>\n",
       "      <td>5F9</td>\n",
       "      <td>arcA_5F9.bw</td>\n",
       "      <td>5G8</td>\n",
       "      <td>arcA_5G8.bw</td>\n",
       "      <td>7B1</td>\n",
       "      <td>arcA_7B10.bw</td>\n",
       "      <td>8B2</td>\n",
       "      <td>arcA_8B2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>UvrY</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE74810</td>\n",
       "      <td>https://doi.org/10.1371/journal.pone.0145035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2917944.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    TF          Organism       Strain                Media  \\\n",
       "index                                                                        \n",
       "0                 BaeR  Escherichia coli  K-12 MG1655  LB (Lysogeny Broth)   \n",
       "1                 CpxR  Escherichia coli  K-12 MG1655  LB (Lysogeny Broth)   \n",
       "2                  Cra  Escherichia coli  K-12 MG1655                   M9   \n",
       "3                  Fur  Escherichia coli  K-12 MG1655                   M9   \n",
       "4                 GadE  Escherichia coli  K-12 MG1655                   M9   \n",
       "...                ...               ...          ...                  ...   \n",
       "60     mixed-TFs-pool4  Escherichia coli  K-12 MG1655                   M9   \n",
       "61     mixed-TFs-pool5  Escherichia coli  K-12 MG1655                   M9   \n",
       "62              ArcA-1  Escherichia coli  K-12 MG1655                   M9   \n",
       "63              ArcA-2  Escherichia coli  K-12 MG1655                   M9   \n",
       "64                UvrY  Escherichia coli  K-12 MG1655  LB (Lysogeny Broth)   \n",
       "\n",
       "                       Supplement    genome_id organism_id  num_binding_sites  \\\n",
       "index                                                                           \n",
       "0                         5% EtOH  NC_000913.3      e_coli                NaN   \n",
       "1                         5% EtOH  NC_000913.3      e_coli                NaN   \n",
       "2      1mL trace element solution  NC_000913.3      e_coli                NaN   \n",
       "3                     0.1mM FeCl2  NC_000913.3      e_coli                NaN   \n",
       "4                             HCl  NC_000913.3      e_coli                NaN   \n",
       "...                           ...          ...         ...                ...   \n",
       "60     1mL trace element solution  NC_000913.3      e_coli                NaN   \n",
       "61     1mL trace element solution  NC_000913.3      e_coli                NaN   \n",
       "62     1mL trace element solution  NC_000913.3      e_coli                NaN   \n",
       "63     1mL trace element solution  NC_000913.3      e_coli                NaN   \n",
       "64                             --  NC_000913.3      e_coli                NaN   \n",
       "\n",
       "       num_samples  num_cond      project  \\\n",
       "index                                       \n",
       "0                4         1          TCS   \n",
       "1                2         1          TCS   \n",
       "2                8         4     known_TF   \n",
       "3                4         2     known_TF   \n",
       "4                2         1     known_TF   \n",
       "...            ...       ...          ...   \n",
       "60               2         1  combination   \n",
       "61               2         1  combination   \n",
       "62               8         1         mAbs   \n",
       "63               8         1         mAbs   \n",
       "64               1         1       public   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      glucose|cra_glucose_binding_table.json;acetate...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "...                                                  ...        ...   \n",
       "60      M9+ace|mixed-tfs-pool4_M9+ace_binding_table.json  GSE182078   \n",
       "61      M9+gal|mixed-tfs-pool5_M9+gal_binding_table.json  GSE182078   \n",
       "62               M9+ace|arca-1_M9+ace_binding_table.json  GSE182077   \n",
       "63               M9+ace|arca-2_M9+ace_binding_table.json  GSE182077   \n",
       "64                                                   NaN   GSE74810   \n",
       "\n",
       "                                                doi peak_intensity_measure  \\\n",
       "index                                                                        \n",
       "0         https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "1         https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "2                https://doi.org/10.1093/nar/gky069               MACE S/N   \n",
       "3                https://doi.org/10.1038/ncomms5910               MACE S/N   \n",
       "4                https://doi.org/10.1038/ncomms8970               MACE S/N   \n",
       "...                                             ...                    ...   \n",
       "60                                              NaN               MACE S/N   \n",
       "61                                              NaN               MACE S/N   \n",
       "62                                              NaN               MACE S/N   \n",
       "63                                              NaN               MACE S/N   \n",
       "64     https://doi.org/10.1371/journal.pone.0145035                    NaN   \n",
       "\n",
       "         Method    TF Category        PMID  \\\n",
       "index                                        \n",
       "0      ChIP-exo  1 – Single TF  33172971.0   \n",
       "1      ChIP-exo  1 – Single TF  33172971.0   \n",
       "2      ChIP-exo  1 – Single TF  29394395.0   \n",
       "3      ChIP-exo  1 – Single TF  25222563.0   \n",
       "4      ChIP-exo  1 – Single TF  26258987.0   \n",
       "...         ...            ...         ...   \n",
       "60     ChIP-exo   3 – Mixed TF         NaN   \n",
       "61     ChIP-exo   3 – Mixed TF         NaN   \n",
       "62     ChIP-exo       2 – mAbs         NaN   \n",
       "63     ChIP-exo       2 – mAbs         NaN   \n",
       "64     ChIP-exo  1 – Single TF         NaN   \n",
       "\n",
       "                            additional_carbon_source  \\\n",
       "index                                                  \n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose   \n",
       "3                                       0.2% Glucose   \n",
       "4                                       0.2% Glucose   \n",
       "...                                              ...   \n",
       "60                                      0.2% Acetate   \n",
       "61                                    0.2% Galactose   \n",
       "62                                      0.2% Acetate   \n",
       "63                                      0.2% Acetate   \n",
       "64                                               NaN   \n",
       "\n",
       "                                              media_info  \\\n",
       "index                                                      \n",
       "0       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "1       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "2      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "3      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "4      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "...                                                  ...   \n",
       "60     {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "61     {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "62     {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "63     {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "64      {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "\n",
       "                                         supplement_info Additional_Info  \\\n",
       "index                                                                      \n",
       "0                                                    NaN              --   \n",
       "1                                                    NaN              --   \n",
       "2      Trace element solution (100x): 1 g EDTA, 29 mg...              --   \n",
       "3                                 Iron-replete condition              --   \n",
       "4      Acidic stress condition: M9 media adjusted to ...              --   \n",
       "...                                                  ...             ...   \n",
       "60     Trace element solution (100x): 1 g EDTA, 29 mg...              --   \n",
       "61     Trace element solution (100x): 1 g EDTA, 29 mg...              --   \n",
       "62     Trace element solution (100x): 1 g EDTA, 29 mg...              --   \n",
       "63     Trace element solution (100x): 1 g EDTA, 29 mg...              --   \n",
       "64                                                   NaN              --   \n",
       "\n",
       "                                     Note BWcond1_1_name          BWcond1_1  \\\n",
       "index                                                                         \n",
       "0                                     NaN  bio-rep1 – R1  BaeR_R1_S31_R1.bw   \n",
       "1                                     NaN             R1    CpxRR1_S1_R1.bw   \n",
       "2                                     NaN       Glu – R1       cra_glu_1.bw   \n",
       "3                                     NaN        Fe – R1        fur_fe_1.bw   \n",
       "4                                     NaN             R1          gade_1.bw   \n",
       "...                                   ...            ...                ...   \n",
       "60      Mixed TFs: YidZ, YedW, ArcA, YhjB             R1   mixTF_pool4_1.bw   \n",
       "61                  Mixed TFs: YebK, YbeF             R1   mixTF_pool5_1.bw   \n",
       "62     monoclonal antibodies (ArcA part1)            1C6        arcA_1C6.bw   \n",
       "63     monoclonal antibodies (ArcA part2)            4D6        arcA_4D6.bw   \n",
       "64                                    NaN             R1      SRR2917944.bw   \n",
       "\n",
       "      BWcond1_2_name          BWcond1_2 BWcond2_1_name           BWcond2_1  \\\n",
       "index                                                                        \n",
       "0      bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1  BaerR_R2_S32_R1.bw   \n",
       "1                 R2    CpxRR2_S2_R2.bw            NaN                 NaN   \n",
       "2           Glu – R2       cra_glu_2.bw       Fru – R1        cra_fru_1.bw   \n",
       "3            Fe – R2        fur_fe_2.bw       DPD – R1        fur_dpd_1.bw   \n",
       "4                 R2          gade_2.bw            NaN                 NaN   \n",
       "...              ...                ...            ...                 ...   \n",
       "60                R2   mixTF_pool4_2.bw            NaN                 NaN   \n",
       "61                R2   mixTF_pool5_2.bw            NaN                 NaN   \n",
       "62               1D6        arcA_1D6.bw            2A1         arcA_2A1.bw   \n",
       "63              5A10       arcA_5A10.bw            5B6         arcA_5B6.bw   \n",
       "64               NaN                NaN            NaN                 NaN   \n",
       "\n",
       "      BWcond2_2_name           BWcond2_2 BWcond3_1_name     BWcond3_1  \\\n",
       "index                                                                   \n",
       "0      bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN           NaN   \n",
       "1                NaN                 NaN            NaN           NaN   \n",
       "2           Fru – R2        cra_fru_2.bw       Gal – R1  cra_gal_1.bw   \n",
       "3           DPD – R2        fur_dpd_2.bw            NaN           NaN   \n",
       "4                NaN                 NaN            NaN           NaN   \n",
       "...              ...                 ...            ...           ...   \n",
       "60               NaN                 NaN            NaN           NaN   \n",
       "61               NaN                 NaN            NaN           NaN   \n",
       "62               2B6         arcA_2B6.bw            2C4   arcA_2C4.bw   \n",
       "63               5B9         arcA_5B9.bw            5F9   arcA_5F9.bw   \n",
       "64               NaN                 NaN            NaN           NaN   \n",
       "\n",
       "      BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                             \n",
       "0                NaN           NaN            NaN           NaN   \n",
       "1                NaN           NaN            NaN           NaN   \n",
       "2           Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3                NaN           NaN            NaN           NaN   \n",
       "4                NaN           NaN            NaN           NaN   \n",
       "...              ...           ...            ...           ...   \n",
       "60               NaN           NaN            NaN           NaN   \n",
       "61               NaN           NaN            NaN           NaN   \n",
       "62               2D6   arcA_2D6.bw            4A1   arcA_4A1.bw   \n",
       "63               5G8   arcA_5G8.bw            7B1  arcA_7B10.bw   \n",
       "64               NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  \n",
       "...              ...           ...  \n",
       "60               NaN           NaN  \n",
       "61               NaN           NaN  \n",
       "62               4A3   arcA_4A3.bw  \n",
       "63               8B2   arcA_8B2.bw  \n",
       "64               NaN           NaN  \n",
       "\n",
       "[65 rows x 39 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "TF_list['binding tables'] = TF_list['binding tables'].fillna('NA')\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>gene_product</th>\n",
       "      <th>evidence</th>\n",
       "      <th>PMID_list</th>\n",
       "      <th>confidence</th>\n",
       "      <th>all_bnumber</th>\n",
       "      <th>regulonDB_geneID</th>\n",
       "      <th>operon</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>locus_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>190.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>65</td>\n",
       "      <td>operon leader peptide</td>\n",
       "      <td>[IDA|S|Inferred from direct assay],[IMP|W|Infe...</td>\n",
       "      <td>3112412,318654,6811557,</td>\n",
       "      <td>Strong</td>\n",
       "      <td>b0001</td>\n",
       "      <td>ECK120001251</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>190.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>337.0</td>\n",
       "      <td>2799.0</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2462</td>\n",
       "      <td>fused aspartate kinase/homoserine dehydrogenase 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b0002</td>\n",
       "      <td>ECK120000987</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>337.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2801.0</td>\n",
       "      <td>3733.0</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>932</td>\n",
       "      <td>homoserine kinase</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b0003</td>\n",
       "      <td>ECK120000988</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>2801.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3734.0</td>\n",
       "      <td>5020.0</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1286</td>\n",
       "      <td>threonine synthase</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b0004</td>\n",
       "      <td>ECK120000989</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>3734.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5234.0</td>\n",
       "      <td>5530.0</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>296</td>\n",
       "      <td>DUF2502 domain-containing protein YaaX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24391520,</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b0005</td>\n",
       "      <td>ECK120002701</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>5234.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            start    stop strand gene_name  length  \\\n",
       "locus_tag                                            \n",
       "b0001       190.0   255.0      +      thrL      65   \n",
       "b0002       337.0  2799.0      +      thrA    2462   \n",
       "b0003      2801.0  3733.0      +      thrB     932   \n",
       "b0004      3734.0  5020.0      +      thrC    1286   \n",
       "b0005      5234.0  5530.0      +      yaaX     296   \n",
       "\n",
       "                                                gene_product  \\\n",
       "locus_tag                                                      \n",
       "b0001                                  operon leader peptide   \n",
       "b0002      fused aspartate kinase/homoserine dehydrogenase 1   \n",
       "b0003                                      homoserine kinase   \n",
       "b0004                                     threonine synthase   \n",
       "b0005                 DUF2502 domain-containing protein YaaX   \n",
       "\n",
       "                                                    evidence  \\\n",
       "locus_tag                                                      \n",
       "b0001      [IDA|S|Inferred from direct assay],[IMP|W|Infe...   \n",
       "b0002                                                    NaN   \n",
       "b0003                                                    NaN   \n",
       "b0004                                                    NaN   \n",
       "b0005                                                    NaN   \n",
       "\n",
       "                         PMID_list confidence all_bnumber regulonDB_geneID  \\\n",
       "locus_tag                                                                    \n",
       "b0001      3112412,318654,6811557,     Strong       b0001     ECK120001251   \n",
       "b0002                          NaN        NaN       b0002     ECK120000987   \n",
       "b0003                          NaN        NaN       b0003     ECK120000988   \n",
       "b0004                          NaN        NaN       b0004     ECK120000989   \n",
       "b0005                    24391520,        NaN       b0005     ECK120002701   \n",
       "\n",
       "            operon  start_codon_pos  \n",
       "locus_tag                            \n",
       "b0001      thrLABC            190.0  \n",
       "b0002      thrLABC            337.0  \n",
       "b0003      thrLABC           2801.0  \n",
       "b0004      thrLABC           3734.0  \n",
       "b0005         yaaX           5234.0  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913.3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info_final.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "baer\n",
      "cpxr\n",
      "cra\n",
      "fur\n",
      "gade\n",
      "gadw\n",
      "gadx\n",
      "kdpe\n",
      "phob\n",
      "rpob\n",
      "ybao\n",
      "ybaq\n",
      "ydci\n",
      "yddm\n",
      "yheo\n",
      "yiag\n",
      "yiep\n",
      "yihl\n",
      "yihw\n",
      "yjdc\n",
      "yjhi\n",
      "ynej\n",
      "ynfl\n",
      "ypdc\n",
      "yqhc\n",
      "zrar\n",
      "fur\n",
      "fur\n",
      "fur\n",
      "dps\n",
      "fis\n",
      "gyra\n",
      "gyrb\n",
      "h-ns\n",
      "hupa\n",
      "hupb\n",
      "infa\n",
      "mukb\n",
      "rdgc\n",
      "rob\n",
      "stpa\n",
      "topa\n",
      "tus\n",
      "narl\n",
      "puur\n",
      "rcsa\n",
      "rpob (cra and crp ko exps)\n",
      "flhc\n",
      "flhd\n",
      "flia\n",
      "fnr\n",
      "rpod\n",
      "argr\n",
      "ompr\n",
      "rpoh\n",
      "mixed-tfs-pool1a\n",
      "mixed-tfs-pool1b\n",
      "mixed-tfs-pool1c\n",
      "mixed-tfs-pool2\n",
      "mixed-tfs-pool3\n",
      "mixed-tfs-pool4\n",
      "mixed-tfs-pool5\n",
      "arca-1\n",
      "arca-2\n",
      "uvry\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    print(TF_name)\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "    \n",
    "    if strain != 'NC_000913.3':\n",
    "        continue\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] # change split to \"/\" on linux\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    else:\n",
    "        TF_list.loc[TF_index, 'binding tables'] = \"NA\"\n",
    "    TF_list.loc[TF_index, 'num_cond'] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    "    # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "            final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "            final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "\n",
    "       #venn diagrams\n",
    "        try:\n",
    "            if strain == 'NC_000913.3' and TF_list.TF[TF_index].split(\" \")[0] in TRN.index:\n",
    "                Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(org_folder+strain+\"/venn/\"+TF_list.TF[TF_index]+'_venn.csv')\n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(org_folder+strain+\"/bw/compressed/\"+TF_list.TF[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(org_folder+strain+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Other Organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_gff_to_geneinfo(org, strain):\n",
    "    genes = pd.read_csv('../data'+'/'+org+'/'+strain+'/annotation/genes.gff',index_col=0, delimiter='\\t', header=None, \n",
    "                        names = ['genome', 'ref', 'type', 'start','stop','.', 'strand', '-','details'])\n",
    "    genes[\"locus_tag\"] = [gene_detail[1].split(\" \")[1] for gene_detail in genes.details.str.split(\"; \")]\n",
    "    genes['gene_name'] = [gene_detail[0].split(\" \")[1] for gene_detail in genes.details.str.split(\"; \")]\n",
    "    genes.set_index('locus_tag', inplace=True)\n",
    "    genes = genes.drop(['.', '-','ref','type'], axis=1)\n",
    "\n",
    "    return(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus(peak_start, peak_stop, gene_info):\n",
    "    gene_info_temp = gene_info.copy()\n",
    "    gene_info_temp.reset_index(inplace=True, drop=False)\n",
    "    df_pos = gene_info_temp[gene_info_temp.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = (df_pos.iloc[0,:]['locus_tag'], df_pos.iloc[0,:]['gene_name'])\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info_temp[gene_info_temp.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = (df_neg.iloc[-1,:]['locus_tag'], df_neg.iloc[-1,:]['gene_name'])\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_table_gen(TF_name, TF_condition, peak_df, gene_info):\n",
    "    \n",
    "    table = peak_df.copy()\n",
    "    table[\"index\"] = [TF_name+\"-\"+TF_condition+\"-\"+str(i+1) for i in range(len(table))]\n",
    "    table[\"condition\"] = [TF_name.lower() + \" + \" + TF_condition for i in range(len(table))]\n",
    "    table.drop([\"condition_name\",\".\", \"ID\"], axis=1, inplace=True)\n",
    "    closest = [closest_locus(table.binding_peak_start.iloc[i], \n",
    "                                           table.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(table))]\n",
    "    table[\"target_locus\"] = [i[0] for i in closest]\n",
    "    table[\"target_genes\"] = [i[1] for i in closest]\n",
    "    table[\"closest_gene\"] = table[\"target_genes\"]\n",
    "\n",
    "    return(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_venn(Peak_DF, TRN_data, gene_info): \n",
    "    TF_name = Peak_DF['index'][1].split(\"-\")[0].split(\" \")[0]\n",
    "    \n",
    "\n",
    "    # TRN locus\n",
    "    reg_genes = set(TRN_data.loc[TF_name].gene_id.to_list())\n",
    "\n",
    "    # chip locus\n",
    "    gene_list = [i for i in Peak_DF.target_locus if i != '']\n",
    "    chip_genes = set(','.join(list(gene_list)).split(','))\n",
    "    \n",
    "    #both\n",
    "    both_genes = set(reg_genes & chip_genes)\n",
    "    \n",
    "    # Get gene and operon counts\n",
    "    reg_gene_count = len(reg_genes)\n",
    "    chip_gene_count = len(chip_genes)\n",
    "    both_gene_count = len(both_genes)\n",
    "    \n",
    "    # Add adjustments for venn plotting (add '2' for alternates)\n",
    "    reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    if reg_genes == chip_genes:\n",
    "        reg_gene_count = 0; chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = 0; both_gene_count2 = len(reg_genes)\n",
    "    elif all(item in chip_genes for item in reg_genes):\n",
    "        reg_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = len(reg_genes); chip_gene_count2 = 0; both_gene_count2 = 0\n",
    "    elif all(item in reg_genes for item in chip_genes):\n",
    "        chip_gene_count = 0; both_gene_count = 0\n",
    "        reg_gene_count2 = 0; chip_gene_count2 = len(chip_genes); both_gene_count2 = 0\n",
    "        \n",
    "    res = pd.DataFrame([TF_name, reg_gene_count, chip_gene_count, both_gene_count,\n",
    "                        reg_gene_count2, chip_gene_count2, both_gene_count2], \n",
    "                        columns=['Value'],\n",
    "                        index=['TF', 'reg_genes', 'chip_genes', 'both_genes',\n",
    "                                'reg_genes2', 'chip_genes2', 'both_genes2'])\n",
    "    \n",
    "    # Add gene lists\n",
    "    just_reg = reg_genes - both_genes\n",
    "    just_chip = chip_genes - both_genes\n",
    "    for i, l in zip(['reg_genes', 'chip_genes', 'both_genes'],[just_reg, just_chip, both_genes]):\n",
    "        gene_list = np.array([gene_info.loc[g,'gene_name'] if g in gene_info.index else g for g in l])\n",
    "        gene_list = np.array2string(gene_list, separator = ' ')\n",
    "        res.loc[i, 'list'] = gene_list\n",
    "    \n",
    "    # Add data sources\n",
    "    res.loc['TF', 'list'] = '; '.join(TRN_data.source[TRN_data.index == TF_name].unique())\n",
    "    \n",
    "    return res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_peak_position_gen(TF, final_annot_DF,out_dir, DF_gene_info): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.iloc[i,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.index == gene]\n",
    "            try:\n",
    "                if gene_info.strand.iloc[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/(gene_info.stop-gene_info.start+1))\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/(gene_info.stop-gene_info.start+1))\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.drop(['direction'], axis=1, inplace=True)\n",
    "    cols = ['index', 'condition', 'binding_peak_start', 'binding_peak_end', 'binding_peak_strength', 'target_locus', \n",
    "            'target_genes', 'closest_gene', 'gene', 'normalized_dist', 'binding_peak_center']\n",
    "    peak_scatter_df = peak_scatter_df[cols]\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/all_other/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>PMID</th>\n",
       "      <th>additional_carbon_source</th>\n",
       "      <th>media_info</th>\n",
       "      <th>supplement_info</th>\n",
       "      <th>Additional_Info</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>CP000647.1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>fe|fur_fe_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>M9|sigma d_M9_binding_table.json</td>\n",
       "      <td>GSE181775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>150 mM MES buffer</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...</td>\n",
       "      <td>GSE181791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>For acid stress with MES buffer, the culture m...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>AE014073.1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2% Glucose</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>CP035369.1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>M9|cody_M9_binding_table.json</td>\n",
       "      <td>GSE159856</td>\n",
       "      <td>https://doi.org/10.1101/2021.01.08.426013</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>All  S.  aureus  strains  were  grown  in  try...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|sigs_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI-1640 (Gibco)</td>\n",
       "      <td>10% LB (Luria-Bertani) broth</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|vrar_RPMI_binding_table.json</td>\n",
       "      <td>GSE182080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>0.1mM FeCl2</td>\n",
       "      <td>NZ_CP009712.1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>GSE181770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...</td>\n",
       "      <td>Iron-replete condition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FleQ</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|fleq_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742210.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742211.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GbdR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gbdr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742212.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742213.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PchR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|pchr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742214.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742215.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|phob_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742216.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742217.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SphR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|sphr_LB_binding_table.json</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31270321.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and its derivati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742218.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742219.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MvfR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mvfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776542.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776543.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SoxR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|soxr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776544.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776545.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RpoN</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|rpon_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776546.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776547.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Vfr</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|vfr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776548.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776549.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AlgR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|algr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776550.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776551.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ExsA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|exsa_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776552.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776553.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GacA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|gaca_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776554.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776555.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MexT</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|mext_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776558.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776559.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>WspR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB (Lysogeny Broth)</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_002516.2</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>LB|wspr_LB_binding_table.json</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26206672.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The P. aeruginosa PAO1 strain and derivatives ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776560.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776561.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HrdB</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>Na-glutamate medium</td>\n",
       "      <td>trace element solution (8 mL/L); TMS1 (5.6 mL/L)</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|hrdb_MS_binding_table.json</td>\n",
       "      <td>E-MTAB-6926</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky1018</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>30371884.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{Na-glutamate media: [Na-glutamate, 61 g/l; gl...</td>\n",
       "      <td>{Trace element solution: [ZnSO4 x 7 H2O, 0.1 g...</td>\n",
       "      <td>For ChIP-seq analysis, spores stocks were prep...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR2639621.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR2639622.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NdgR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>Solid minimal media</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>minimal-media|ndgr_minimal-media_binding_table...</td>\n",
       "      <td>GSE59010</td>\n",
       "      <td>https://doi.org/10.1186/s12864-015-1311-0</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>25766138.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{Solid minimal media: [0.05% K2HPO4, 0.02% MgS...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>All strains used are S. coelicolor A3 (2) M145...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1503874.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ScbR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM (Supplemented liquid minimal media)</td>\n",
       "      <td>0.1 mL Trace element solution</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26442964.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{SMM: [81.9 mL PEG 6000 (BDH 6.1% w/v in disti...</td>\n",
       "      <td>Trace element solution (0.1g/L of each): ZnSO4...</td>\n",
       "      <td>S. coelicolor strains were incubated on MS sol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752661.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ScbR2</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>SMM (Supplemented liquid minimal media)</td>\n",
       "      <td>0.1 mL Trace element solution</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>SMM|scbr2_SMM_binding_table.json</td>\n",
       "      <td>GSE64903</td>\n",
       "      <td>https://doi.org/10.1038/srep14831</td>\n",
       "      <td>MACS fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26442964.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{SMM: [81.9 mL PEG 6000 (BDH 6.1% w/v in disti...</td>\n",
       "      <td>Trace element solution (0.1g/L of each): ZnSO4...</td>\n",
       "      <td>S. coelicolor strains were incubated on MS sol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1752662.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AmrZ</td>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>F113</td>\n",
       "      <td>SA medium or LB (Luria-Bertani) medium</td>\n",
       "      <td>IPTG [Isopropyl-β-D-Thiogalactopyranoside] (1 ...</td>\n",
       "      <td>NC_016830.1</td>\n",
       "      <td>p_fluorescens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NA</td>\n",
       "      <td>SRP039494</td>\n",
       "      <td>https://doi.org/10.1186/1471-2164-15-237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>24670089.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{SA: [20g/L sucrose, 2g/L L-asparagine, 1g/L K...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P. fluorescens strains were grown in SA medium...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1184483.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Lsr2</td>\n",
       "      <td>Streptomyces venezuelae</td>\n",
       "      <td>ATCC 10712</td>\n",
       "      <td>MYM (maltose-yeast extract-mannitol) medium</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_018750.1</td>\n",
       "      <td>s_venezuelae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>MS|lsr2_MS_binding_table.json</td>\n",
       "      <td>GSE115439</td>\n",
       "      <td>https://doi.org/10.7554/eLife.47691</td>\n",
       "      <td>Fold difference</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>31215866.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{MYM: {4g/L Maltose, 4g/L Yeast Extract Oxoid,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Each culture was then grown in 300 mL MYM cult...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LexA1</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Copenhageni Fiocruz L1-130</td>\n",
       "      <td>EMJH</td>\n",
       "      <td>--</td>\n",
       "      <td>NC_005823.1</td>\n",
       "      <td>l_interrogans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>EMJH|lexa1_EMJH_binding_table.json</td>\n",
       "      <td>GSE73688</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv1536</td>\n",
       "      <td>CisGenome fold enrichment</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26762976.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{EMJH: [1 g/L Disodium Phosphate, 0.3g/L Monop...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L. interrogans serovar Copenhageni Fiocruz L1–...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2549977.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR2549979.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MksB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NA</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>4% Glucose</td>\n",
       "      <td>{BHI: [12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NeonGreen 1 – R1</td>\n",
       "      <td>SRR8791699.bw</td>\n",
       "      <td>NeonGreen 1 – R2</td>\n",
       "      <td>SRR8791700.bw</td>\n",
       "      <td>NeonGreen 2 – R1</td>\n",
       "      <td>SRR8791701.bw</td>\n",
       "      <td>NeonGreen 2 – R2</td>\n",
       "      <td>SRR8791702.bw</td>\n",
       "      <td>mCherry – R1</td>\n",
       "      <td>SRR8791680.bw</td>\n",
       "      <td>mCherry – R2</td>\n",
       "      <td>SRR8791681.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ParB</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NA</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>4% Glucose</td>\n",
       "      <td>{BHI: [12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791676.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791677.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791678.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791679.bw</td>\n",
       "      <td>mCherry 3 – R1</td>\n",
       "      <td>SRR8791682.bw</td>\n",
       "      <td>mCherry 3 – R2</td>\n",
       "      <td>SRR8791683.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Smc</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI (brain heart infusion medium)</td>\n",
       "      <td>--</td>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>c_glutamicum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NA</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>32198399.0</td>\n",
       "      <td>4% Glucose</td>\n",
       "      <td>{BHI: [12.5 g/L brain infusion solids, 5.0 g/L...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Growth experiments of C. glutamicum cells were...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mCherry 1 – R1</td>\n",
       "      <td>SRR8791691.bw</td>\n",
       "      <td>mCherry 1 – R2</td>\n",
       "      <td>SRR8791692.bw</td>\n",
       "      <td>mCherry 2 – R1</td>\n",
       "      <td>SRR8791697.bw</td>\n",
       "      <td>mCherry 2 – R2</td>\n",
       "      <td>SRR8791698.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EspR</td>\n",
       "      <td>Mycobacterium tuberculosis</td>\n",
       "      <td>H37Rv</td>\n",
       "      <td>7H9 broth</td>\n",
       "      <td>Tween 80 (0.05%);  Middlebrook albumin-dextros...</td>\n",
       "      <td>NC_000962.2</td>\n",
       "      <td>m_tuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NA</td>\n",
       "      <td>GSE35149</td>\n",
       "      <td>https://doi.org/10.1371/journal.ppat.1002621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>22479184.0</td>\n",
       "      <td>0.2% Glycerol</td>\n",
       "      <td>{7H9: [Ammonium Sulfate 0.5g/L, L-Glutamic Aci...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mycobacterium tuberculosis (Mtb) H37Rv and Erd...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR400040.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR400041.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism                      Strain  \\\n",
       "index                                                                     \n",
       "0          Fur        Klebsiella pneumoniae                   MGH 78578   \n",
       "0          Fur           Pseudomonas putida                      KT2440   \n",
       "1      Sigma D           Pseudomonas putida                      KT2440   \n",
       "0          Fur          Salmonella enterica             Typhimurium LT2   \n",
       "1         YdcI          Salmonella enterica             Typhimurium LT2   \n",
       "0          Fur            Shigella flexneri                   2a, 2457T   \n",
       "0         CodY        Staphylococcus aureus              LAC chromosome   \n",
       "1         SigS        Staphylococcus aureus              USA300_TCH1516   \n",
       "2         VraR        Staphylococcus aureus              USA300_TCH1516   \n",
       "0          Fur  Yersinia pseudotuberculosis                    IP 32953   \n",
       "0         FleQ       Pseudomonas aeruginosa                        PAO1   \n",
       "1         GbdR       Pseudomonas aeruginosa                        PAO1   \n",
       "2         PchR       Pseudomonas aeruginosa                        PAO1   \n",
       "3         PhoB       Pseudomonas aeruginosa                        PAO1   \n",
       "4         SphR       Pseudomonas aeruginosa                        PAO1   \n",
       "5         MvfR       Pseudomonas aeruginosa                        PAO1   \n",
       "6         SoxR       Pseudomonas aeruginosa                        PAO1   \n",
       "7         RpoN       Pseudomonas aeruginosa                        PAO1   \n",
       "8          Vfr       Pseudomonas aeruginosa                        PAO1   \n",
       "9         AlgR       Pseudomonas aeruginosa                        PAO1   \n",
       "10        ExsA       Pseudomonas aeruginosa                        PAO1   \n",
       "11        GacA       Pseudomonas aeruginosa                        PAO1   \n",
       "12        MexT       Pseudomonas aeruginosa                        PAO1   \n",
       "13        WspR       Pseudomonas aeruginosa                        PAO1   \n",
       "0         HrdB      Streptomyces coelicolor                  A3(2) M145   \n",
       "1         NdgR      Streptomyces coelicolor                  A3(2) M145   \n",
       "2         ScbR      Streptomyces coelicolor                  A3(2) M145   \n",
       "3        ScbR2      Streptomyces coelicolor                  A3(2) M145   \n",
       "0         AmrZ      Pseudomonas fluorescens                        F113   \n",
       "0         Lsr2      Streptomyces venezuelae                  ATCC 10712   \n",
       "0        LexA1       Leptospira interrogans  Copenhageni Fiocruz L1-130   \n",
       "0         MksB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "1         ParB   Corynebacterium glutamicum                  ATCC 13032   \n",
       "2          Smc   Corynebacterium glutamicum                  ATCC 13032   \n",
       "0         EspR   Mycobacterium tuberculosis                       H37Rv   \n",
       "\n",
       "                                             Media  \\\n",
       "index                                                \n",
       "0                                               M9   \n",
       "0                                               M9   \n",
       "1                                               M9   \n",
       "0                                               M9   \n",
       "1                                               M9   \n",
       "0                                               M9   \n",
       "0                                RPMI-1640 (Gibco)   \n",
       "1                                RPMI-1640 (Gibco)   \n",
       "2                                RPMI-1640 (Gibco)   \n",
       "0                BHI (brain heart infusion medium)   \n",
       "0                              LB (Lysogeny Broth)   \n",
       "1                              LB (Lysogeny Broth)   \n",
       "2                              LB (Lysogeny Broth)   \n",
       "3                              LB (Lysogeny Broth)   \n",
       "4                              LB (Lysogeny Broth)   \n",
       "5                              LB (Lysogeny Broth)   \n",
       "6                              LB (Lysogeny Broth)   \n",
       "7                              LB (Lysogeny Broth)   \n",
       "8                              LB (Lysogeny Broth)   \n",
       "9                              LB (Lysogeny Broth)   \n",
       "10                             LB (Lysogeny Broth)   \n",
       "11                             LB (Lysogeny Broth)   \n",
       "12                             LB (Lysogeny Broth)   \n",
       "13                             LB (Lysogeny Broth)   \n",
       "0                              Na-glutamate medium   \n",
       "1                              Solid minimal media   \n",
       "2          SMM (Supplemented liquid minimal media)   \n",
       "3          SMM (Supplemented liquid minimal media)   \n",
       "0           SA medium or LB (Luria-Bertani) medium   \n",
       "0      MYM (maltose-yeast extract-mannitol) medium   \n",
       "0                                             EMJH   \n",
       "0                BHI (brain heart infusion medium)   \n",
       "1                BHI (brain heart infusion medium)   \n",
       "2                BHI (brain heart infusion medium)   \n",
       "0                                        7H9 broth   \n",
       "\n",
       "                                              Supplement      genome_id  \\\n",
       "index                                                                     \n",
       "0                                            0.1mM FeCl2     CP000647.1   \n",
       "0                                            0.1mM FeCl2    NC_002947.4   \n",
       "1                                                     --    NC_002947.4   \n",
       "0                                            0.1mM FeCl2    NC_003197.2   \n",
       "1                                      150 mM MES buffer    NC_003197.2   \n",
       "0                                            0.1mM FeCl2     AE014073.1   \n",
       "0                           10% LB (Luria-Bertani) broth     CP035369.1   \n",
       "1                           10% LB (Luria-Bertani) broth      NC_010079   \n",
       "2                           10% LB (Luria-Bertani) broth      NC_010079   \n",
       "0                                            0.1mM FeCl2  NZ_CP009712.1   \n",
       "0                                                     --    NC_002516.2   \n",
       "1                                                     --    NC_002516.2   \n",
       "2                                                     --    NC_002516.2   \n",
       "3                                                     --    NC_002516.2   \n",
       "4                                                     --    NC_002516.2   \n",
       "5                                                     --    NC_002516.2   \n",
       "6                                                     --    NC_002516.2   \n",
       "7                                                     --    NC_002516.2   \n",
       "8                                                     --    NC_002516.2   \n",
       "9                                                     --    NC_002516.2   \n",
       "10                                                    --    NC_002516.2   \n",
       "11                                                    --    NC_002516.2   \n",
       "12                                                    --    NC_002516.2   \n",
       "13                                                    --    NC_002516.2   \n",
       "0       trace element solution (8 mL/L); TMS1 (5.6 mL/L)    NC_003888.3   \n",
       "1                                                     --    NC_003888.3   \n",
       "2                          0.1 mL Trace element solution    NC_003888.3   \n",
       "3                          0.1 mL Trace element solution    NC_003888.3   \n",
       "0      IPTG [Isopropyl-β-D-Thiogalactopyranoside] (1 ...    NC_016830.1   \n",
       "0                                                     --    NC_018750.1   \n",
       "0                                                     --    NC_005823.1   \n",
       "0                                                     --     BX927147.1   \n",
       "1                                                     --     BX927147.1   \n",
       "2                                                     --     BX927147.1   \n",
       "0      Tween 80 (0.05%);  Middlebrook albumin-dextros...    NC_000962.2   \n",
       "\n",
       "                organism_id  num_binding_sites  num_samples  num_cond  \\\n",
       "index                                                                   \n",
       "0              k_pneumoniae                NaN            2         1   \n",
       "0                  p_putida                NaN            1         1   \n",
       "1                  p_putida                NaN            2         1   \n",
       "0                s_enterica                NaN            1         1   \n",
       "1                s_enterica                NaN            6         3   \n",
       "0                s_flexneri                NaN            2         1   \n",
       "0                  s_aureus                NaN            2         1   \n",
       "1                  s_aureus                NaN            8         1   \n",
       "2                  s_aureus                NaN            2         1   \n",
       "0      y_pseudotuberculosis                NaN            2         1   \n",
       "0              p_aeruginosa                NaN            2         1   \n",
       "1              p_aeruginosa                NaN            2         1   \n",
       "2              p_aeruginosa                NaN            2         1   \n",
       "3              p_aeruginosa                NaN            2         1   \n",
       "4              p_aeruginosa                NaN            2         1   \n",
       "5              p_aeruginosa                NaN            2         1   \n",
       "6              p_aeruginosa                NaN            2         1   \n",
       "7              p_aeruginosa                NaN            2         1   \n",
       "8              p_aeruginosa                NaN            2         1   \n",
       "9              p_aeruginosa                NaN            2         1   \n",
       "10             p_aeruginosa                NaN            2         1   \n",
       "11             p_aeruginosa                NaN            2         1   \n",
       "12             p_aeruginosa                NaN            2         1   \n",
       "13             p_aeruginosa                NaN            2         1   \n",
       "0              s_coelicolor                NaN            3         1   \n",
       "1              s_coelicolor                NaN            1         1   \n",
       "2              s_coelicolor                NaN            1         1   \n",
       "3              s_coelicolor                NaN            1         1   \n",
       "0             p_fluorescens                NaN            1         1   \n",
       "0              s_venezuelae                NaN            2         1   \n",
       "0             l_interrogans                NaN            2         1   \n",
       "0              c_glutamicum                NaN            2         1   \n",
       "1              c_glutamicum                NaN            1         1   \n",
       "2              c_glutamicum                NaN            1         1   \n",
       "0            m_tuberculosis                NaN            2         1   \n",
       "\n",
       "              project                                     binding tables  \\\n",
       "index                                                                      \n",
       "0            MGH78578                       Fe|fur_Fe_binding_table.json   \n",
       "0               2457T                       fe|fur_fe_binding_table.json   \n",
       "1               2457T                   M9|sigma d_M9_binding_table.json   \n",
       "0                 LT2                       Fe|fur_Fe_binding_table.json   \n",
       "1                 LT2  MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...   \n",
       "0               2457T                       Fe|fur_Fe_binding_table.json   \n",
       "0                 LAC                      M9|cody_M9_binding_table.json   \n",
       "1      USA300_TCH1516                  RPMI|sigs_RPMI_binding_table.json   \n",
       "2      USA300_TCH1516                  RPMI|vrar_RPMI_binding_table.json   \n",
       "0               2457T                       Fe|fur_Fe_binding_table.json   \n",
       "0         public_data                      LB|fleq_LB_binding_table.json   \n",
       "1         public_data                      LB|gbdr_LB_binding_table.json   \n",
       "2         public_data                      LB|pchr_LB_binding_table.json   \n",
       "3         public_data                      LB|phob_LB_binding_table.json   \n",
       "4         public_data                      LB|sphr_LB_binding_table.json   \n",
       "5         public_data                      LB|mvfr_LB_binding_table.json   \n",
       "6         public_data                      LB|soxr_LB_binding_table.json   \n",
       "7         public_data                      LB|rpon_LB_binding_table.json   \n",
       "8         public_data                       LB|vfr_LB_binding_table.json   \n",
       "9         public_data                      LB|algr_LB_binding_table.json   \n",
       "10        public_data                      LB|exsa_LB_binding_table.json   \n",
       "11        public_data                      LB|gaca_LB_binding_table.json   \n",
       "12        public_data                      LB|mext_LB_binding_table.json   \n",
       "13        public_data                      LB|wspr_LB_binding_table.json   \n",
       "0         public_data                      MS|hrdb_MS_binding_table.json   \n",
       "1         public_data  minimal-media|ndgr_minimal-media_binding_table...   \n",
       "2         public_data                    SMM|scbr_SMM_binding_table.json   \n",
       "3         public_data                   SMM|scbr2_SMM_binding_table.json   \n",
       "0         public_data                                                 NA   \n",
       "0         public_data                      MS|lsr2_MS_binding_table.json   \n",
       "0         public_data                 EMJH|lexa1_EMJH_binding_table.json   \n",
       "0         public_data                                                 NA   \n",
       "1         public_data                                                 NA   \n",
       "2         public_data                                                 NA   \n",
       "0         public_data                                                 NA   \n",
       "\n",
       "         accession                                           doi  \\\n",
       "index                                                              \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE181775                                           NaN   \n",
       "1        GSE181775                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "1        GSE181791                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE159856     https://doi.org/10.1101/2021.01.08.426013   \n",
       "1        GSE182080                                           NaN   \n",
       "2        GSE182080                                           NaN   \n",
       "0        GSE181770                                           NaN   \n",
       "0        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "1        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "2        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "3        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "4        GSE128430    https://doi.org/10.1038/s41467-019-10778-w   \n",
       "5         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "6         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "7         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "8         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "9         GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "10        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "11        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "12        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "13        GSE65356            https://doi.org/10.1093/nar/gkv747   \n",
       "0      E-MTAB-6926           https://doi.org/10.1093/nar/gky1018   \n",
       "1         GSE59010     https://doi.org/10.1186/s12864-015-1311-0   \n",
       "2         GSE64903             https://doi.org/10.1038/srep14831   \n",
       "3         GSE64903             https://doi.org/10.1038/srep14831   \n",
       "0        SRP039494      https://doi.org/10.1186/1471-2164-15-237   \n",
       "0        GSE115439           https://doi.org/10.7554/eLife.47691   \n",
       "0         GSE73688           https://doi.org/10.1093/nar/gkv1536   \n",
       "0      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "1      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "2      PRJNA529385    https://doi.org/10.1038/s41467-020-15238-4   \n",
       "0         GSE35149  https://doi.org/10.1371/journal.ppat.1002621   \n",
       "\n",
       "          peak_intensity_measure    Method    TF Category        PMID  \\\n",
       "index                                                                   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "1                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "2                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0                       MACE S/N  ChIP-exo  1 – Single TF         NaN   \n",
       "0           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "1           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "2           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "3           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "4           MACS fold enrichment  ChIP-seq  1 – Single TF  31270321.0   \n",
       "5           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "6           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "7           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "8           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "9           MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "10          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "11          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "12          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "13          MACS fold enrichment  ChIP-seq  1 – Single TF  26206672.0   \n",
       "0           MACS fold enrichment  ChIP-seq  1 – Single TF  30371884.0   \n",
       "1           MACS fold enrichment  ChIP-seq  1 – Single TF  25766138.0   \n",
       "2           MACS fold enrichment  ChIP-seq  1 – Single TF  26442964.0   \n",
       "3           MACS fold enrichment  ChIP-seq  1 – Single TF  26442964.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  24670089.0   \n",
       "0                Fold difference  ChIP-seq  1 – Single TF  31215866.0   \n",
       "0      CisGenome fold enrichment  ChIP-seq  1 – Single TF  26762976.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "1                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "2                            NaN  ChIP-seq  1 – Single TF  32198399.0   \n",
       "0                            NaN  ChIP-seq  1 – Single TF  22479184.0   \n",
       "\n",
       "      additional_carbon_source  \\\n",
       "index                            \n",
       "0                 0.2% Glucose   \n",
       "0                 0.2% Glucose   \n",
       "1                 0.2% Glucose   \n",
       "0                 0.2% Glucose   \n",
       "1                 0.2% Glucose   \n",
       "0                 0.2% Glucose   \n",
       "0                          NaN   \n",
       "1                          NaN   \n",
       "2                          NaN   \n",
       "0                          NaN   \n",
       "0                          NaN   \n",
       "1                          NaN   \n",
       "2                          NaN   \n",
       "3                          NaN   \n",
       "4                          NaN   \n",
       "5                          NaN   \n",
       "6                          NaN   \n",
       "7                          NaN   \n",
       "8                          NaN   \n",
       "9                          NaN   \n",
       "10                         NaN   \n",
       "11                         NaN   \n",
       "12                         NaN   \n",
       "13                         NaN   \n",
       "0                          NaN   \n",
       "1                          NaN   \n",
       "2                          NaN   \n",
       "3                          NaN   \n",
       "0                          NaN   \n",
       "0                          NaN   \n",
       "0                          NaN   \n",
       "0                   4% Glucose   \n",
       "1                   4% Glucose   \n",
       "2                   4% Glucose   \n",
       "0                0.2% Glycerol   \n",
       "\n",
       "                                              media_info  \\\n",
       "index                                                      \n",
       "0      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "0      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "1      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "0      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "1      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "0      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "0      {RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...   \n",
       "1      {RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...   \n",
       "2      {RPMI-1640: [11.11 mM glucose, 0.003 mM Glutat...   \n",
       "0      {M9: [47.8 mM Na2HPO4, 22 mM KH2PO4, 8.6 mM Na...   \n",
       "0       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "1       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "2       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "3       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "4       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "5       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "6       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "7       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "8       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "9       {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "10      {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "11      {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "12      {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "13      {LB: [1% NaCl, 1% tryptone, 0.5% yeast extract]}   \n",
       "0      {Na-glutamate media: [Na-glutamate, 61 g/l; gl...   \n",
       "1      {Solid minimal media: [0.05% K2HPO4, 0.02% MgS...   \n",
       "2      {SMM: [81.9 mL PEG 6000 (BDH 6.1% w/v in disti...   \n",
       "3      {SMM: [81.9 mL PEG 6000 (BDH 6.1% w/v in disti...   \n",
       "0      {SA: [20g/L sucrose, 2g/L L-asparagine, 1g/L K...   \n",
       "0      {MYM: {4g/L Maltose, 4g/L Yeast Extract Oxoid,...   \n",
       "0      {EMJH: [1 g/L Disodium Phosphate, 0.3g/L Monop...   \n",
       "0      {BHI: [12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "1      {BHI: [12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "2      {BHI: [12.5 g/L brain infusion solids, 5.0 g/L...   \n",
       "0      {7H9: [Ammonium Sulfate 0.5g/L, L-Glutamic Aci...   \n",
       "\n",
       "                                         supplement_info  \\\n",
       "index                                                      \n",
       "0                                 Iron-replete condition   \n",
       "0                                 Iron-replete condition   \n",
       "1                                                    NaN   \n",
       "0                                 Iron-replete condition   \n",
       "1      For acid stress with MES buffer, the culture m...   \n",
       "0                                 Iron-replete condition   \n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "0                                 Iron-replete condition   \n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "5                                                    NaN   \n",
       "6                                                    NaN   \n",
       "7                                                    NaN   \n",
       "8                                                    NaN   \n",
       "9                                                    NaN   \n",
       "10                                                   NaN   \n",
       "11                                                   NaN   \n",
       "12                                                   NaN   \n",
       "13                                                   NaN   \n",
       "0      {Trace element solution: [ZnSO4 x 7 H2O, 0.1 g...   \n",
       "1                                                    NaN   \n",
       "2      Trace element solution (0.1g/L of each): ZnSO4...   \n",
       "3      Trace element solution (0.1g/L of each): ZnSO4...   \n",
       "0                                                    NaN   \n",
       "0                                                    NaN   \n",
       "0                                                    NaN   \n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "0                                                    NaN   \n",
       "\n",
       "                                         Additional_Info  Note  \\\n",
       "index                                                            \n",
       "0                                                    NaN   NaN   \n",
       "0                                                    NaN   NaN   \n",
       "1                                                    NaN   NaN   \n",
       "0                                                    NaN   NaN   \n",
       "1                                                    NaN   NaN   \n",
       "0                                                    NaN   NaN   \n",
       "0      All  S.  aureus  strains  were  grown  in  try...   NaN   \n",
       "1                                                    NaN   NaN   \n",
       "2                                                    NaN   NaN   \n",
       "0                                                    NaN   NaN   \n",
       "0      The P. aeruginosa PAO1 strain and its derivati...   NaN   \n",
       "1      The P. aeruginosa PAO1 strain and its derivati...   NaN   \n",
       "2      The P. aeruginosa PAO1 strain and its derivati...   NaN   \n",
       "3      The P. aeruginosa PAO1 strain and its derivati...   NaN   \n",
       "4      The P. aeruginosa PAO1 strain and its derivati...   NaN   \n",
       "5      The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "6      The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "7      The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "8      The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "9      The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "10     The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "11     The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "12     The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "13     The P. aeruginosa PAO1 strain and derivatives ...   NaN   \n",
       "0      For ChIP-seq analysis, spores stocks were prep...   NaN   \n",
       "1      All strains used are S. coelicolor A3 (2) M145...   NaN   \n",
       "2      S. coelicolor strains were incubated on MS sol...   NaN   \n",
       "3      S. coelicolor strains were incubated on MS sol...   NaN   \n",
       "0      P. fluorescens strains were grown in SA medium...   NaN   \n",
       "0      Each culture was then grown in 300 mL MYM cult...   NaN   \n",
       "0      L. interrogans serovar Copenhageni Fiocruz L1–...   NaN   \n",
       "0      Growth experiments of C. glutamicum cells were...   NaN   \n",
       "1      Growth experiments of C. glutamicum cells were...   NaN   \n",
       "2      Growth experiments of C. glutamicum cells were...   NaN   \n",
       "0      Mycobacterium tuberculosis (Mtb) H37Rv and Erd...   NaN   \n",
       "\n",
       "         BWcond1_1_name                        BWcond1_1    BWcond1_2_name  \\\n",
       "index                                                                        \n",
       "0                    R1                 klebfurfinal1.bw                R2   \n",
       "0                    R1        putidafurfinal1_202003.bw               NaN   \n",
       "1                    R1           putidaSigmaD_202003.bw                R2   \n",
       "0                   NaN                              NaN                R2   \n",
       "1             MES5 – R1           salmone_ydcI_MES5_1.bw         MES5 – R2   \n",
       "0                    R1                shigella_fur_1.bw                R2   \n",
       "0                    R1            StaphLAC_new_cody1.bw                R2   \n",
       "1               M1 – R1              StaphTCH_sigs_M1.bw           M1 – R2   \n",
       "2                    R1  StaphTCH_vraR_7H7C11purified.bw                R2   \n",
       "0                    R1                yersifurfinal1.bw                R2   \n",
       "0                    R1                    SRR8742210.bw                R2   \n",
       "1                    R1                    SRR8742212.bw                R2   \n",
       "2                    R1                    SRR8742214.bw                R2   \n",
       "3                    R1                    SRR8742216.bw                R2   \n",
       "4                    R1                    SRR8742218.bw                R2   \n",
       "5                    R1                    SRR1776542.bw                R2   \n",
       "6                    R1                    SRR1776544.bw                R2   \n",
       "7                    R1                    SRR1776546.bw                R2   \n",
       "8                    R1                    SRR1776548.bw                R2   \n",
       "9                    R1                    SRR1776550.bw                R2   \n",
       "10                   R1                    SRR1776552.bw                R2   \n",
       "11                   R1                    SRR1776554.bw                R2   \n",
       "12                   R1                    SRR1776558.bw                R2   \n",
       "13                   R1                    SRR1776560.bw                R2   \n",
       "0                    R1                    ERR2639621.bw                R2   \n",
       "1                    R1                    SRR1503874.bw               NaN   \n",
       "2                    R1                    SRR1752661.bw               NaN   \n",
       "3                    R1                    SRR1752662.bw               NaN   \n",
       "0                    R1                    SRR1184483.bw               NaN   \n",
       "0                    R1                    SRR7282569.bw                R2   \n",
       "0                    R1                    SRR2549977.bw                R2   \n",
       "0      NeonGreen 1 – R1                    SRR8791699.bw  NeonGreen 1 – R2   \n",
       "1        mCherry 1 – R1                    SRR8791676.bw    mCherry 1 – R2   \n",
       "2        mCherry 1 – R1                    SRR8791691.bw    mCherry 1 – R2   \n",
       "0                    R1                     SRR400040.bw                R2   \n",
       "\n",
       "                              BWcond1_2    BWcond2_1_name  \\\n",
       "index                                                       \n",
       "0                      klebfurfinal2.bw               NaN   \n",
       "0                                   NaN               NaN   \n",
       "1                putidaSigmaD_202005.bw               NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw               NaN   \n",
       "1                salmone_ydcI_MES5_2.bw         MES5 – R3   \n",
       "0                     shigella_fur_2.bw               NaN   \n",
       "0                 StaphLAC_new_cody2.bw               NaN   \n",
       "1                   Staph_SigS_M1_R2.bw           M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw               NaN   \n",
       "0                     yersifurfinal2.bw               NaN   \n",
       "0                         SRR8742211.bw               NaN   \n",
       "1                         SRR8742213.bw               NaN   \n",
       "2                         SRR8742215.bw               NaN   \n",
       "3                         SRR8742217.bw               NaN   \n",
       "4                         SRR8742219.bw               NaN   \n",
       "5                         SRR1776543.bw               NaN   \n",
       "6                         SRR1776545.bw               NaN   \n",
       "7                         SRR1776547.bw               NaN   \n",
       "8                         SRR1776549.bw               NaN   \n",
       "9                         SRR1776551.bw               NaN   \n",
       "10                        SRR1776553.bw               NaN   \n",
       "11                        SRR1776555.bw               NaN   \n",
       "12                        SRR1776559.bw               NaN   \n",
       "13                        SRR1776561.bw               NaN   \n",
       "0                         ERR2639622.bw               NaN   \n",
       "1                                   NaN               NaN   \n",
       "2                                   NaN               NaN   \n",
       "3                                   NaN               NaN   \n",
       "0                                   NaN               NaN   \n",
       "0                         SRR7282569.bw               NaN   \n",
       "0                         SRR2549979.bw               NaN   \n",
       "0                         SRR8791700.bw  NeonGreen 2 – R1   \n",
       "1                         SRR8791677.bw    mCherry 2 – R1   \n",
       "2                         SRR8791692.bw    mCherry 2 – R1   \n",
       "0                          SRR400041.bw               NaN   \n",
       "\n",
       "                    BWcond2_1    BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                     \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw         MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw           M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "4                         NaN               NaN                     NaN   \n",
       "5                         NaN               NaN                     NaN   \n",
       "6                         NaN               NaN                     NaN   \n",
       "7                         NaN               NaN                     NaN   \n",
       "8                         NaN               NaN                     NaN   \n",
       "9                         NaN               NaN                     NaN   \n",
       "10                        NaN               NaN                     NaN   \n",
       "11                        NaN               NaN                     NaN   \n",
       "12                        NaN               NaN                     NaN   \n",
       "13                        NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "1                         NaN               NaN                     NaN   \n",
       "2                         NaN               NaN                     NaN   \n",
       "3                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0                         NaN               NaN                     NaN   \n",
       "0               SRR8791701.bw  NeonGreen 2 – R2           SRR8791702.bw   \n",
       "1               SRR8791678.bw    mCherry 2 – R2           SRR8791679.bw   \n",
       "2               SRR8791697.bw    mCherry 2 – R2           SRR8791698.bw   \n",
       "0                         NaN               NaN                     NaN   \n",
       "\n",
       "       BWcond3_1_name               BWcond3_1  BWcond3_2_name  \\\n",
       "index                                                           \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1            ph5 – R1  salmone_ydcI_ph55_1.bw        ph5 – R2   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1             M3 – R1     StaphTCH_sigs_M3.bw         M3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "4                 NaN                     NaN             NaN   \n",
       "5                 NaN                     NaN             NaN   \n",
       "6                 NaN                     NaN             NaN   \n",
       "7                 NaN                     NaN             NaN   \n",
       "8                 NaN                     NaN             NaN   \n",
       "9                 NaN                     NaN             NaN   \n",
       "10                NaN                     NaN             NaN   \n",
       "11                NaN                     NaN             NaN   \n",
       "12                NaN                     NaN             NaN   \n",
       "13                NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "1                 NaN                     NaN             NaN   \n",
       "2                 NaN                     NaN             NaN   \n",
       "3                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "0        mCherry – R1           SRR8791680.bw    mCherry – R2   \n",
       "1      mCherry 3 – R1           SRR8791682.bw  mCherry 3 – R2   \n",
       "2                 NaN                     NaN             NaN   \n",
       "0                 NaN                     NaN             NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "4                         NaN            NaN                  NaN   \n",
       "5                         NaN            NaN                  NaN   \n",
       "6                         NaN            NaN                  NaN   \n",
       "7                         NaN            NaN                  NaN   \n",
       "8                         NaN            NaN                  NaN   \n",
       "9                         NaN            NaN                  NaN   \n",
       "10                        NaN            NaN                  NaN   \n",
       "11                        NaN            NaN                  NaN   \n",
       "12                        NaN            NaN                  NaN   \n",
       "13                        NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0               SRR8791681.bw            NaN                  NaN   \n",
       "1               SRR8791683.bw            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "4                NaN                  NaN  \n",
       "5                NaN                  NaN  \n",
       "6                NaN                  NaN  \n",
       "7                NaN                  NaN  \n",
       "8                NaN                  NaN  \n",
       "9                NaN                  NaN  \n",
       "10               NaN                  NaN  \n",
       "11               NaN                  NaN  \n",
       "12               NaN                  NaN  \n",
       "13               NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "TF_list['binding tables'] = TF_list['binding tables'].fillna('NA')\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fur\n",
      "fur\n",
      "sigma d\n",
      "Venn Error: p_putida sigma d\n",
      "fur\n",
      "ydci\n",
      "fur\n",
      "Venn Error: s_flexneri fur\n",
      "cody\n",
      "Venn Error: s_aureus cody\n",
      "sigs\n",
      "Venn Error: s_aureus sigs\n",
      "vrar\n",
      "fur\n",
      "Venn Error: y_pseudotuberculosis fur\n",
      "fleq\n",
      "gbdr\n",
      "pchr\n",
      "phob\n",
      "Venn Error: p_aeruginosa phob\n",
      "sphr\n",
      "Venn Error: p_aeruginosa sphr\n",
      "mvfr\n",
      "soxr\n",
      "rpon\n",
      "vfr\n",
      "algr\n",
      "exsa\n",
      "Venn Error: p_aeruginosa exsa\n",
      "gaca\n",
      "Venn Error: p_aeruginosa gaca\n",
      "mext\n",
      "wspr\n",
      "Venn Error: p_aeruginosa wspr\n",
      "hrdb\n",
      "Venn Error: s_coelicolor hrdb\n",
      "ndgr\n",
      "Venn Error: s_coelicolor ndgr\n",
      "scbr\n",
      "Venn Error: s_coelicolor scbr\n",
      "scbr2\n",
      "Venn Error: s_coelicolor scbr2\n",
      "amrz\n",
      "Curation Error: amrz\n",
      "lsr2\n",
      "Venn Error: s_venezuelae lsr2\n",
      "lexa1\n",
      "Venn Error: l_interrogans lexa1\n",
      "mksb\n",
      "Curation Error: mksb\n",
      "parb\n",
      "Curation Error: parb\n",
      "smc\n",
      "Curation Error: smc\n",
      "espr\n",
      "Curation Error: espr\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "    print(TF_name)\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc] \n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "#         break\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            other_venn(final_annot_DF, TRN, gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "\n",
    "        except:\n",
    "            print('Venn Error:',org_id, TF_name)\n",
    "            continue\n",
    "            \n",
    "    except:\n",
    "        print('Curation Error:',TF_name)\n",
    "        continue\n",
    "        \n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list.insert(12, \"curated information\", TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')\n",
    "\n",
    "#individual file updates\n",
    "for org in new_TF_list.organism_id.unique():\n",
    "    new_TF_list[new_TF_list.organism_id == org].to_csv(\"../data/\"+org+\"/TF_list.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - other strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>peak_intensity_measure</th>\n",
       "      <th>Method</th>\n",
       "      <th>TF Category</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Media_new</th>\n",
       "      <th>Supplement_new</th>\n",
       "      <th>Contents</th>\n",
       "      <th>Additional_Info</th>\n",
       "      <th>Motif_Available?</th>\n",
       "      <th>Motif_Parameters</th>\n",
       "      <th>Note</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>33172971.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>--</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>33172971.0</td>\n",
       "      <td>LB (Luria-Bertani) broth</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>{LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}</td>\n",
       "      <td>--</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>GSE65643</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky069</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>29394395.0</td>\n",
       "      <td>M9</td>\n",
       "      <td>{Carbon Source: {0.2% Fructose, Galactose, Ace...</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>--</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>GSE54901</td>\n",
       "      <td>https://doi.org/10.1038/ncomms5910</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>25222563.0</td>\n",
       "      <td>M9</td>\n",
       "      <td>{Carbon Source: {0.2% Glucose}, Iron-replete c...</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>--</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>GSE66482</td>\n",
       "      <td>https://doi.org/10.1038/ncomms8970</td>\n",
       "      <td>MACE S/N</td>\n",
       "      <td>ChIP-exo</td>\n",
       "      <td>1 – Single TF</td>\n",
       "      <td>26258987.0</td>\n",
       "      <td>M9</td>\n",
       "      <td>{Carbon Source: {0.2% Glucose}, Acidic stress ...</td>\n",
       "      <td>{M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...</td>\n",
       "      <td>--</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         TF          Organism       Strain Media  \\\n",
       "index                                              \n",
       "0      BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1      CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2       Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3       Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4      GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond   project  \\\n",
       "index                                                       \n",
       "0                    NaN            4         1       TCS   \n",
       "1                    NaN            2         1       TCS   \n",
       "2                    NaN            8         4  known_TF   \n",
       "3                    NaN            4         2  known_TF   \n",
       "4                    NaN            2         1  known_TF   \n",
       "\n",
       "                                          binding tables  accession  \\\n",
       "index                                                                 \n",
       "0                      EtOH|baer_EtOH_binding_table.json  GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json  GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...   GSE65643   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...   GSE54901   \n",
       "4                          M9|gade_M9_binding_table.json   GSE66482   \n",
       "\n",
       "                                             doi peak_intensity_measure  \\\n",
       "index                                                                     \n",
       "0      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "1      https://doi.org/10.1128/mSystems.00980-20               MACE S/N   \n",
       "2             https://doi.org/10.1093/nar/gky069               MACE S/N   \n",
       "3             https://doi.org/10.1038/ncomms5910               MACE S/N   \n",
       "4             https://doi.org/10.1038/ncomms8970               MACE S/N   \n",
       "\n",
       "         Method    TF Category        PMID                 Media_new  \\\n",
       "index                                                                  \n",
       "0      ChIP-exo  1 – Single TF  33172971.0  LB (Luria-Bertani) broth   \n",
       "1      ChIP-exo  1 – Single TF  33172971.0  LB (Luria-Bertani) broth   \n",
       "2      ChIP-exo  1 – Single TF  29394395.0                        M9   \n",
       "3      ChIP-exo  1 – Single TF  25222563.0                        M9   \n",
       "4      ChIP-exo  1 – Single TF  26258987.0                        M9   \n",
       "\n",
       "                                          Supplement_new  \\\n",
       "index                                                      \n",
       "0                                                5% EtOH   \n",
       "1                                                5% EtOH   \n",
       "2      {Carbon Source: {0.2% Fructose, Galactose, Ace...   \n",
       "3      {Carbon Source: {0.2% Glucose}, Iron-replete c...   \n",
       "4      {Carbon Source: {0.2% Glucose}, Acidic stress ...   \n",
       "\n",
       "                                                Contents Additional_Info  \\\n",
       "index                                                                      \n",
       "0       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}              --   \n",
       "1       {LB: {1% NaCl, 1% tryptone, 0.5% yeast extract}}              --   \n",
       "2      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...              --   \n",
       "3      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...              --   \n",
       "4      {M9: {M9 salts (5X) 20 mL/100 mL, glucose (20%...              --   \n",
       "\n",
       "       Motif_Available? Motif_Parameters Note BWcond1_1_name  \\\n",
       "index                                                          \n",
       "0                  True              NaN  NaN  bio-rep1 – R1   \n",
       "1                  True              NaN  NaN             R1   \n",
       "2                  True              NaN  NaN       Glu – R1   \n",
       "3                  True              NaN  NaN        Fe – R1   \n",
       "4                  True              NaN  NaN             R1   \n",
       "\n",
       "               BWcond1_1 BWcond1_2_name          BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                                       \n",
       "0      BaeR_R1_S31_R1.bw  bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1   \n",
       "1        CpxRR1_S1_R1.bw             R2    CpxRR2_S2_R2.bw            NaN   \n",
       "2           cra_glu_1.bw       Glu – R2       cra_glu_2.bw       Fru – R1   \n",
       "3            fur_fe_1.bw        Fe – R2        fur_fe_2.bw       DPD – R1   \n",
       "4              gade_1.bw             R2          gade_2.bw            NaN   \n",
       "\n",
       "                BWcond2_1 BWcond2_2_name           BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                                         \n",
       "0      BaerR_R2_S32_R1.bw  bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN   \n",
       "1                     NaN            NaN                 NaN            NaN   \n",
       "2            cra_fru_1.bw       Fru – R2        cra_fru_2.bw       Gal – R1   \n",
       "3            fur_dpd_1.bw       DPD – R2        fur_dpd_2.bw            NaN   \n",
       "4                     NaN            NaN                 NaN            NaN   \n",
       "\n",
       "          BWcond3_1 BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                                           \n",
       "0               NaN            NaN           NaN            NaN           NaN   \n",
       "1               NaN            NaN           NaN            NaN           NaN   \n",
       "2      cra_gal_1.bw       Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3               NaN            NaN           NaN            NaN           NaN   \n",
       "4               NaN            NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_folder = \"../data/e_coli/\"\n",
    "\n",
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "TF_list['binding tables'] = TF_list['binding tables'].fillna('NA')\n",
    "\n",
    "TF_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "    \n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "    \n",
    "    if genome_id == 'NC_000913.3':\n",
    "            continue\n",
    "            \n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    TF_list.Supplement.fillna(\"--\", inplace=True)\n",
    "    \n",
    " # write plot data files to folders\n",
    "    try:\n",
    "        df_list = []\n",
    "        final_annot_DF_list =[]\n",
    "        for file_index in range(len(curated_files)):\n",
    "            file = curated_files[file_index]\n",
    "            df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                     delimiter='\\t', header=None, \n",
    "                                     names = ['ref','condition', 'condition_name', \n",
    "                                              \"binding_peak_start\",'binding_peak_end', \n",
    "                                              'binding_peak_strength', 'direction', '.','ID'])\n",
    "            df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "            df_list.append(df)\n",
    "            final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "            final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "            final_annot_DF_list.append(final_annot_DF)\n",
    "        # merge conditions then do plots\n",
    "        df = pd.concat(df_list, axis=0); df.index = np.arange(1, len(df) + 1)\n",
    "        final_annot_DF = pd.concat(final_annot_DF_list, axis =0); final_annot_DF.index = np.arange(1, len(final_annot_DF.index) + 1)\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        if any(final_annot_DF.binding_peak_strength > 0):\n",
    "            other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "        try:\n",
    "            TRN = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/annotation/trn.csv\", index_col=0)\n",
    "            Venn_data_gen2(final_annot_DF, TRN, DF_gene_info).to_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/venn/\"+TF_list.TF.iloc[TF_index]+'_venn.csv')\n",
    "            \n",
    "        except:\n",
    "            continue\n",
    "    except:\n",
    "        continue\n",
    "\n",
    "    #compress bw files\n",
    "    bw_files = TF_list.iloc[TF_index,-15:-1:2].dropna().to_list()\n",
    "    zipObj = ZipFile(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/compressed/\"+TF_list.TF.iloc[TF_index]+\"_bw.zip\", 'w')\n",
    "    for bw in bw_files:\n",
    "        zipObj.write(\"../data/\"+org_id+\"/\"+genome_id+\"/bw/\"+bw, bw)\n",
    "    zipObj.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Go through all binding sequences and perform new run of MEME-ChIP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# db_dir = '../data/'\n",
    "\n",
    "# all_peaks = glob.glob(db_dir+\"*/*/curated_input/*.gff\")\n",
    "\n",
    "# for peak_file in all_peaks:\n",
    "#     print(peak_file)\n",
    "#     name= peak_file.split('/')[5][0:-12]\n",
    "#     tfbs = pd.read_csv(peak_file, sep='\\t', header=None,\n",
    "#                       names=['genome', 'program','filter', 'Start Position', 'End Position', '1', '2', '3', '4'])\n",
    "#     tfbs['mid'] = 0.5*(tfbs['End Position'] + tfbs['Start Position'])\n",
    "#     tfbs['length'] = tfbs['End Position'] - tfbs['Start Position']\n",
    "    \n",
    "#     strain_folder = \"/\".join(peak_file.split('/')[0:4])\n",
    "#     fasta = strain_folder+\"/sequence/sequence.fasta\"\n",
    "#     for seq_record in SeqIO.parse(fasta, \"fasta\"):\n",
    "#         sequence = seq_record.seq\n",
    "\n",
    "#     seq_list = []\n",
    "#     for i,row in tfbs.iterrows():\n",
    "#         start = int(row['Start Position']-20)\n",
    "#         stop = int(row['End Position']+20)\n",
    "#         seq = str(sequence[start:stop])\n",
    "#         seq_list.append(seq)\n",
    "        \n",
    "\n",
    "#     !{'mkdir '+strain_folder+'/meme-chip'}\n",
    "\n",
    "#     filename = strain_folder+'/meme-chip/'+name+'.txt'\n",
    "#     textfile = open(filename, \"w\")\n",
    "#     for i,seq in enumerate(seq_list):\n",
    "#         textfile.write(\">peak\"+str(i)+ \"\\n\")\n",
    "#         textfile.write(seq + \"\\n\")\n",
    "#     textfile.close()\n",
    "\n",
    "#     out_folder= strain_folder+'/meme-chip/'+name\n",
    "#     !{\"meme-chip -oc \"+out_folder+\" -time 240 -ccut 100 -dna -order 2 -meme-minw 5 -meme-maxw 45 \\\n",
    "#       -filter-thresh 0.001 \\\n",
    "#       -meme-mod zoops -meme-nmotifs 4 \"+filename}\n",
    "    \n",
    "#     # parse XML output, if available\n",
    "#     if exists(out_folder+'/meme_out/meme.xml'):\n",
    "#         tree = ET.parse(out_folder+'/meme_out/meme.xml')\n",
    "#         root = tree.getroot()\n",
    "\n",
    "#         e_thres = 0.001\n",
    "#         for motif in root.findall('./motifs/motif'):\n",
    "#             probs = motif.findall('probabilities/alphabet_matrix/alphabet_array/value')\n",
    "#             e_value = float(motif.get('e_value'))\n",
    "\n",
    "#             if e_value < e_thres:\n",
    "#                 final_outputs = strain_folder+'/meme-chip/outputs/'\n",
    "#                 !{'mkdir '+final_outputs}\n",
    "#                 # Get probabilities\n",
    "#                 prob_list = []\n",
    "#                 for prob in probs:\n",
    "#                     prob_list.append(float(prob.text))\n",
    "#                 # make PWM\n",
    "#                 pwm = pd.DataFrame(index=['A', 'C', 'G', 'T'])\n",
    "#                 for i,four_start in enumerate(range(0, len(prob_list),4)):\n",
    "#                     pwm['position-'+str(i+1)] = prob_list[four_start:four_start+4]\n",
    "#                 pwm.to_csv(final_outputs+name+\".csv\")\n",
    "#                 # reformat image and save\n",
    "#                 motif_num = motif.get('name')\n",
    "#                 img = Image.open(out_folder+\"/meme_out/logo\"+motif_num+\".png\")\n",
    "#                 img = ImageOps.expand(img, border=30)\n",
    "#                 draw = ImageDraw.Draw(img)\n",
    "#                 font = ImageFont.truetype(\"FONTS/ariali.ttf\", 36)\n",
    "#                 font_bold = ImageFont.truetype(\"FONTS/arial.ttf\", 36)\n",
    "#                 draw.text((30,310),\"E-value = \"+\"{:.1e}\".format(e_value),'#000000',font=font)\n",
    "#                 draw.text((30,0),\"Condition: \"+name.split(\"_\")[1],'#000000',font=font_bold)\n",
    "#                 img.save(final_outputs+name+\".png\")\n",
    "#                 #reflect change in e-value\n",
    "#                 e_thres = e_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Metadata for manuscript table S1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-4-aafd1252a3f5>:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  table_s1['v1.0.0 TF dashboard link'] = 'https://prochipdb.com/tf_dashboard.html?organism='+meta_combined.organism_id+'&tf='+meta_combined.TF+'&genome='+meta_combined.genome_id+'&i='+[str(i) for i in meta_combined.index.to_list()]\n",
      "<ipython-input-4-aafd1252a3f5>:10: FutureWarning: As the xlwt package is no longer maintained, the xlwt engine will be removed in a future version of pandas. This is the only engine in pandas that supports writing in the xls format. Install openpyxl and write to an xlsx file instead. You can set the option io.excel.xls.writer to 'xlwt' to silence this warning. While this option is deprecated and will also raise a warning, it can be globally set and the warning suppressed.\n",
      "  table_s1.to_excel('../../Supplementary Table 1.xls')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>TF</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>v1.0.0 TF dashboard link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Organism</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>MksB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.com/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>ParB</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.com/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium glutamicum</th>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>Smc</td>\n",
       "      <td>PRJNA529385</td>\n",
       "      <td>https://doi.org/10.1038/s41467-020-15238-4</td>\n",
       "      <td>https://prochipdb.com/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArcA-1</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://prochipdb.com/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia coli</th>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>ArcA-2</td>\n",
       "      <td>GSE182077</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://prochipdb.com/tf_dashboard.html?organi...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Strain      TF    accession  \\\n",
       "Organism                                                       \n",
       "Corynebacterium glutamicum   ATCC 13032    MksB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032    ParB  PRJNA529385   \n",
       "Corynebacterium glutamicum   ATCC 13032     Smc  PRJNA529385   \n",
       "Escherichia coli            K-12 MG1655  ArcA-1    GSE182077   \n",
       "Escherichia coli            K-12 MG1655  ArcA-2    GSE182077   \n",
       "\n",
       "                                                                   doi  \\\n",
       "Organism                                                                 \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Corynebacterium glutamicum  https://doi.org/10.1038/s41467-020-15238-4   \n",
       "Escherichia coli                                                   NaN   \n",
       "Escherichia coli                                                   NaN   \n",
       "\n",
       "                                                     v1.0.0 TF dashboard link  \n",
       "Organism                                                                       \n",
       "Corynebacterium glutamicum  https://prochipdb.com/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.com/tf_dashboard.html?organi...  \n",
       "Corynebacterium glutamicum  https://prochipdb.com/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.com/tf_dashboard.html?organi...  \n",
       "Escherichia coli            https://prochipdb.com/tf_dashboard.html?organi...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_ecoli = pd.read_csv('../data/e_coli/TF_list.csv', index_col=0)\n",
    "meta_other = pd.read_csv('../data/all_other/TF_list.csv', index_col=0)\n",
    "\n",
    "meta_combined = pd.concat([meta_ecoli, meta_other], axis=0)\n",
    "table_s1 = meta_combined[['Organism', 'Strain', 'TF', 'accession', 'doi']]\n",
    "table_s1['v1.0.0 TF dashboard link'] = 'https://prochipdb.com/tf_dashboard.html?organism='+meta_combined.organism_id+'&tf='+meta_combined.TF+'&genome='+meta_combined.genome_id+'&i='+[str(i) for i in meta_combined.index.to_list()]\n",
    "table_s1.sort_values(by=['Organism','Strain','TF'], inplace=True)\n",
    "table_s1.set_index('Organism', inplace=True)\n",
    "\n",
    "table_s1.to_excel('../../Supplementary Table 1.xls')\n",
    "table_s1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "e coli dataset\n",
      "num org 1\n",
      "num strain 4\n",
      "num TF 65\n",
      "num ChIP-exo 184 92.9%\n",
      "num ChIP-exo 14 7.1%\n",
      "num sample 198\n",
      "percent curated 92.3 60\n"
     ]
    }
   ],
   "source": [
    "print('e coli dataset')\n",
    "print('num org', len(meta_ecoli.Organism.unique()))\n",
    "print('num strain', len(meta_ecoli.Strain.unique()))\n",
    "print('num TF', len(meta_ecoli))\n",
    "num_samples = meta_ecoli.num_samples.sum()\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_ecoli[meta_ecoli.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_ecoli[meta_ecoli.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_ecoli['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_ecoli['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 13\n",
      "num strain 14\n",
      "num TF 35\n",
      "num ChIP-exo 28 38.4%\n",
      "num ChIP-exo 45 61.6%\n",
      "num sample 73\n",
      "percent curated 85.7 30\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_other.Organism.unique()))\n",
    "print('num strain', len(meta_other.Strain.unique()))\n",
    "print('num TF', len(meta_other))\n",
    "num_samples = meta_other.num_samples.sum()\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_other[meta_other.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_other[meta_other.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_other['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_other['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num org 14\n",
      "num strain 18\n",
      "num TF 100\n",
      "num ChIP-exo 212 78.2%\n",
      "num ChIP-exo 59 21.8%\n",
      "num sample 271\n",
      "percent curated 90.0 90\n"
     ]
    }
   ],
   "source": [
    "# print('e coli dataset')\n",
    "print('num org', len(meta_combined.Organism.unique()))\n",
    "print('num strain', len(meta_combined.Strain.unique()))\n",
    "print('num TF', len(meta_combined))\n",
    "num_samples = meta_combined.num_samples.sum()\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum(), \n",
    "      str(round(meta_combined[meta_combined.Method =='ChIP-exo'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num ChIP-exo', meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum(),\n",
    "       str(round(meta_combined[meta_combined.Method =='ChIP-seq'].num_samples.sum()/num_samples*100,1))+\"%\")\n",
    "print('num sample', num_samples)\n",
    "print('percent curated', round(100*len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3])/len([i for i in meta_combined['binding tables'].tolist()]),1),\n",
    "     len([i  for i in meta_combined['binding tables'].tolist() if len(str(i)) > 3]))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py38",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.20"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
