{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Search Output\n",
    "\n",
    "This notebook generates json files for searching through proChIPdb's genes and TFs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=CodY&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=SigS&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=VraR&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AmrZ</td>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>F113</td>\n",
       "      <td>SA (low Fe)</td>\n",
       "      <td>--</td>\n",
       "      <td>tf_dashboard.html?organism=p_fluorescens&amp;tf=Am...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>tf_dashboard.html?organism=s_flexneri&amp;tf=Fur&amp;g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Smc</td>\n",
       "      <td>Corynebacterium glutamicum</td>\n",
       "      <td>ATCC 13032</td>\n",
       "      <td>BHI</td>\n",
       "      <td>--</td>\n",
       "      <td>tf_dashboard.html?organism=c_glutamicum&amp;tf=Smc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>EspR</td>\n",
       "      <td>Mycobacterium tuberculosis</td>\n",
       "      <td>H37Rv</td>\n",
       "      <td>7H9 Broth (Difco)</td>\n",
       "      <td>0.2% gly, ADC, 0.05% Tween 80</td>\n",
       "      <td>tf_dashboard.html?organism=m_tuberculosis&amp;tf=E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>BHI</td>\n",
       "      <td>Fe</td>\n",
       "      <td>tf_dashboard.html?organism=y_pseudotuberculosi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>tf_dashboard.html?organism=s_enterica&amp;tf=Fur&amp;g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH5, MES</td>\n",
       "      <td>tf_dashboard.html?organism=s_enterica&amp;tf=YdcI&amp;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      TF                     Organism           Strain              Media  \\\n",
       "0   CodY        Staphylococcus aureus   LAC chromosome               RPMI   \n",
       "1   SigS        Staphylococcus aureus   USA300_TCH1516               RPMI   \n",
       "2   VraR        Staphylococcus aureus   USA300_TCH1516               RPMI   \n",
       "3   AmrZ      Pseudomonas fluorescens             F113        SA (low Fe)   \n",
       "4    Fur            Shigella flexneri        2a, 2457T                 M9   \n",
       "..   ...                          ...              ...                ...   \n",
       "95   Smc   Corynebacterium glutamicum       ATCC 13032                BHI   \n",
       "96  EspR   Mycobacterium tuberculosis            H37Rv  7H9 Broth (Difco)   \n",
       "97   Fur  Yersinia pseudotuberculosis         IP 32953                BHI   \n",
       "98   Fur          Salmonella enterica  Typhimurium LT2                 M9   \n",
       "99  YdcI          Salmonella enterica  Typhimurium LT2                 M9   \n",
       "\n",
       "                       Supplement  \\\n",
       "0                          10% LB   \n",
       "1                          10% LB   \n",
       "2                          10% LB   \n",
       "3                              --   \n",
       "4                              Fe   \n",
       "..                            ...   \n",
       "95                             --   \n",
       "96  0.2% gly, ADC, 0.05% Tween 80   \n",
       "97                             Fe   \n",
       "98                             Fe   \n",
       "99                       pH5, MES   \n",
       "\n",
       "                                                 link  \n",
       "0   tf_dashboard.html?organism=s_aureus&tf=CodY&ge...  \n",
       "1   tf_dashboard.html?organism=s_aureus&tf=SigS&ge...  \n",
       "2   tf_dashboard.html?organism=s_aureus&tf=VraR&ge...  \n",
       "3   tf_dashboard.html?organism=p_fluorescens&tf=Am...  \n",
       "4   tf_dashboard.html?organism=s_flexneri&tf=Fur&g...  \n",
       "..                                                ...  \n",
       "95  tf_dashboard.html?organism=c_glutamicum&tf=Smc...  \n",
       "96  tf_dashboard.html?organism=m_tuberculosis&tf=E...  \n",
       "97  tf_dashboard.html?organism=y_pseudotuberculosi...  \n",
       "98  tf_dashboard.html?organism=s_enterica&tf=Fur&g...  \n",
       "99  tf_dashboard.html?organism=s_enterica&tf=YdcI&...  \n",
       "\n",
       "[100 rows x 6 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prefix = '../data/'\n",
    "suffix = '/TF_list.csv'\n",
    "\n",
    "organisms = [f.name for f in os.scandir(prefix)]\n",
    "organisms.remove('all_other')\n",
    "\n",
    "cols_of_interest = ['TF', 'Organism', 'Strain', 'Media', 'Supplement']\n",
    "\n",
    "all_tfs = pd.DataFrame(columns = cols_of_interest + ['link'])\n",
    "\n",
    "for o in organisms:\n",
    "    try:\n",
    "        tfs = pd.read_csv(prefix + o + suffix, index_col = 0)\n",
    "    except NotADirectoryError:\n",
    "        continue\n",
    "        \n",
    "    # build link\n",
    "    for i in tfs.index:\n",
    "        l = 'tf_dashboard.html?organism=' + o\n",
    "        l += '&tf=' + tfs.TF[i]\n",
    "        l += '&genome=' + tfs.genome_id[i]\n",
    "        l += '&i=' + str(i)\n",
    "        tfs.loc[i, 'link'] = l\n",
    "    \n",
    "    # take info of interest\n",
    "    tfs = tfs[cols_of_interest + ['link']]\n",
    "    \n",
    "    # save\n",
    "    all_tfs = all_tfs.append(tfs, ignore_index = True)\n",
    "    \n",
    "all_tfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_tfs.to_json('../data/tfs_search.json', orient='records')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "prefix = '../data/'\n",
    "suffix = '_binding_table.json'\n",
    "\n",
    "all_genes = pd.DataFrame(columns = ['Gene', 'Locus', \n",
    "        'Organism', 'Strain', 'Binding_site_id',\n",
    "        'Condition', 'Peak_start', 'Peak_end',\n",
    "        'Peak_strength', 'link'])\n",
    "\n",
    "for j, tf_row in all_tfs.iterrows():\n",
    "    \n",
    "    # get folder info from the link & read in file\n",
    "    file = tf_row.link.replace('&', '=').split('=')\n",
    "    path = prefix + file[1] + '/' + file[5] + '/table/' + file[3].lower() + \"_*binding_table.json\"\n",
    "    all_matches = glob.glob(path)\n",
    "    \n",
    "    for f in all_matches:\n",
    "        # read in\n",
    "        try:\n",
    "            bt = pd.read_json(f)\n",
    "        except (FileNotFoundError, ValueError) as e:\n",
    "            continue\n",
    "\n",
    "        for i, row in bt.iterrows():\n",
    "\n",
    "            # get genes & loci\n",
    "            genes = row.target_genes.split(',')\n",
    "            loci = row.target_locus.split(',')\n",
    "\n",
    "            # rearrange data for new table\n",
    "            for g, l in zip(genes, loci):\n",
    "                if g == '':\n",
    "                    continue\n",
    "                all_genes = all_genes.append({\n",
    "                    'Gene':g,\n",
    "                    'Locus':l,\n",
    "                    'Organism':tf_row.Organism,\n",
    "                    'Strain':tf_row.Strain,\n",
    "                    'Binding_site_id':row['index'],\n",
    "                    'Condition':row.condition,\n",
    "                    'Peak_start':row.binding_peak_start,\n",
    "                    'Peak_end':row.binding_peak_end,\n",
    "                    'Peak_strength':row.binding_peak_strength,\n",
    "                    'link':tf_row.link}, ignore_index = True)\n",
    "            \n",
    "all_genes = all_genes.sort_values('Peak_start')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Binding_site_id</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Peak_start</th>\n",
       "      <th>Peak_end</th>\n",
       "      <th>Peak_strength</th>\n",
       "      <th>link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>87764</th>\n",
       "      <td>PP_0010</td>\n",
       "      <td>PP_0010</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>Sigma D-M9-1</td>\n",
       "      <td>sigma d + M9</td>\n",
       "      <td>43</td>\n",
       "      <td>82</td>\n",
       "      <td>8.09000</td>\n",
       "      <td>tf_dashboard.html?organism=p_putida&amp;tf=Sigma D...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88877</th>\n",
       "      <td>dnaA</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>SphR-LB-1</td>\n",
       "      <td>sphr + LB</td>\n",
       "      <td>66</td>\n",
       "      <td>149</td>\n",
       "      <td>1.98077</td>\n",
       "      <td>tf_dashboard.html?organism=p_aeruginosa&amp;tf=Sph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7159</th>\n",
       "      <td>thrC</td>\n",
       "      <td>b0004</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>RpoB-dpd-1</td>\n",
       "      <td>rpob + dpd</td>\n",
       "      <td>100</td>\n",
       "      <td>150</td>\n",
       "      <td>3.68000</td>\n",
       "      <td>tf_dashboard.html?organism=e_coli&amp;tf=RpoB&amp;geno...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7156</th>\n",
       "      <td>thrL</td>\n",
       "      <td>b0001</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>RpoB-dpd-1</td>\n",
       "      <td>rpob + dpd</td>\n",
       "      <td>100</td>\n",
       "      <td>150</td>\n",
       "      <td>3.68000</td>\n",
       "      <td>tf_dashboard.html?organism=e_coli&amp;tf=RpoB&amp;geno...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7157</th>\n",
       "      <td>thrA</td>\n",
       "      <td>b0002</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>RpoB-dpd-1</td>\n",
       "      <td>rpob + dpd</td>\n",
       "      <td>100</td>\n",
       "      <td>150</td>\n",
       "      <td>3.68000</td>\n",
       "      <td>tf_dashboard.html?organism=e_coli&amp;tf=RpoB&amp;geno...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87697</th>\n",
       "      <td>SCO7825</td>\n",
       "      <td>SCO7825</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>ScbR2-SMM-444</td>\n",
       "      <td>scbr2 + SMM</td>\n",
       "      <td>8641432</td>\n",
       "      <td>8645750</td>\n",
       "      <td>2.34000</td>\n",
       "      <td>tf_dashboard.html?organism=s_coelicolor&amp;tf=Scb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87090</th>\n",
       "      <td>SCO7828</td>\n",
       "      <td>SCO7828</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>HrdB-MS-1541</td>\n",
       "      <td>hrdb + MS</td>\n",
       "      <td>8645864</td>\n",
       "      <td>8646219</td>\n",
       "      <td>2.65645</td>\n",
       "      <td>tf_dashboard.html?organism=s_coelicolor&amp;tf=Hrd...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87722</th>\n",
       "      <td>SCO7833</td>\n",
       "      <td>SCO7833</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>ScbR2-SMM-469</td>\n",
       "      <td>scbr2 + SMM</td>\n",
       "      <td>8651620</td>\n",
       "      <td>8652701</td>\n",
       "      <td>4.41000</td>\n",
       "      <td>tf_dashboard.html?organism=s_coelicolor&amp;tf=Scb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87687</th>\n",
       "      <td>SCO7844</td>\n",
       "      <td>SCO7844</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>ScbR2-SMM-434</td>\n",
       "      <td>scbr2 + SMM</td>\n",
       "      <td>8660800</td>\n",
       "      <td>8662596</td>\n",
       "      <td>2.48000</td>\n",
       "      <td>tf_dashboard.html?organism=s_coelicolor&amp;tf=Scb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87691</th>\n",
       "      <td>SCO7845</td>\n",
       "      <td>SCO7845</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2) M145</td>\n",
       "      <td>ScbR2-SMM-438</td>\n",
       "      <td>scbr2 + SMM</td>\n",
       "      <td>8663839</td>\n",
       "      <td>8664941</td>\n",
       "      <td>2.07000</td>\n",
       "      <td>tf_dashboard.html?organism=s_coelicolor&amp;tf=Scb...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92032 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene    Locus                 Organism       Strain Binding_site_id  \\\n",
       "87764  PP_0010  PP_0010       Pseudomonas putida       KT2440    Sigma D-M9-1   \n",
       "88877     dnaA     dnaA   Pseudomonas aeruginosa         PAO1       SphR-LB-1   \n",
       "7159      thrC    b0004         Escherichia coli  K-12 MG1655      RpoB-dpd-1   \n",
       "7156      thrL    b0001         Escherichia coli  K-12 MG1655      RpoB-dpd-1   \n",
       "7157      thrA    b0002         Escherichia coli  K-12 MG1655      RpoB-dpd-1   \n",
       "...        ...      ...                      ...          ...             ...   \n",
       "87697  SCO7825  SCO7825  Streptomyces coelicolor   A3(2) M145   ScbR2-SMM-444   \n",
       "87090  SCO7828  SCO7828  Streptomyces coelicolor   A3(2) M145    HrdB-MS-1541   \n",
       "87722  SCO7833  SCO7833  Streptomyces coelicolor   A3(2) M145   ScbR2-SMM-469   \n",
       "87687  SCO7844  SCO7844  Streptomyces coelicolor   A3(2) M145   ScbR2-SMM-434   \n",
       "87691  SCO7845  SCO7845  Streptomyces coelicolor   A3(2) M145   ScbR2-SMM-438   \n",
       "\n",
       "          Condition Peak_start Peak_end  Peak_strength  \\\n",
       "87764  sigma d + M9         43       82        8.09000   \n",
       "88877     sphr + LB         66      149        1.98077   \n",
       "7159     rpob + dpd        100      150        3.68000   \n",
       "7156     rpob + dpd        100      150        3.68000   \n",
       "7157     rpob + dpd        100      150        3.68000   \n",
       "...             ...        ...      ...            ...   \n",
       "87697   scbr2 + SMM    8641432  8645750        2.34000   \n",
       "87090     hrdb + MS    8645864  8646219        2.65645   \n",
       "87722   scbr2 + SMM    8651620  8652701        4.41000   \n",
       "87687   scbr2 + SMM    8660800  8662596        2.48000   \n",
       "87691   scbr2 + SMM    8663839  8664941        2.07000   \n",
       "\n",
       "                                                    link  \n",
       "87764  tf_dashboard.html?organism=p_putida&tf=Sigma D...  \n",
       "88877  tf_dashboard.html?organism=p_aeruginosa&tf=Sph...  \n",
       "7159   tf_dashboard.html?organism=e_coli&tf=RpoB&geno...  \n",
       "7156   tf_dashboard.html?organism=e_coli&tf=RpoB&geno...  \n",
       "7157   tf_dashboard.html?organism=e_coli&tf=RpoB&geno...  \n",
       "...                                                  ...  \n",
       "87697  tf_dashboard.html?organism=s_coelicolor&tf=Scb...  \n",
       "87090  tf_dashboard.html?organism=s_coelicolor&tf=Hrd...  \n",
       "87722  tf_dashboard.html?organism=s_coelicolor&tf=Scb...  \n",
       "87687  tf_dashboard.html?organism=s_coelicolor&tf=Scb...  \n",
       "87691  tf_dashboard.html?organism=s_coelicolor&tf=Scb...  \n",
       "\n",
       "[92032 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes.to_json('../data/genes_search.json', orient='records')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
