{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for ChipDB. But before you run the notebook, make sure of the following: \n",
    "\n",
    "1. Concatnate files that contain multiple conditions for the same TF by using the following command on your terminal: `cat /path/prefix* >> new_file`.\n",
    "2. Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#basics:\n",
    "import numpy as np     \n",
    "import pandas as pd\n",
    "import urllib                      \n",
    "import gzip\n",
    "import os\n",
    "from os import path\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Peak Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate a data objects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prerequisit\n",
    "Ensure the following paths reflect your local **input** and **output** directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_dir = './'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "in_data = data_dir+ 'input_data/'\n",
    "in_gff = in_data+ 'gff_data/'\n",
    "out_data_venn = data_dir+ 'output_data/'+'venn_data/'\n",
    "out_data_table = data_dir+ 'output_data/'+'table_data/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "database = defaultdict(list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['yafc_glc_curated.csv',\n",
       " 'ybih_glc_curated.csv',\n",
       " 'yddm_glc_curated.csv',\n",
       " 'yagi_-fe_curated.csv',\n",
       " 'yeie_glc_curated.csv',\n",
       " 'ybaq_glc_curated.csv',\n",
       " 'ydci_cat.csv',\n",
       " 'yiaj_glc_curated.csv',\n",
       " 'yheo_glc_curated.csv',\n",
       " 'yiep_glc_curated.csv',\n",
       " 'ybao_glc_curated.csv']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filelist = [file for file in os.listdir(in_gff)]\n",
    "filelist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "filelist = [file for file in os.listdir(in_gff)]\n",
    "\n",
    "for i in range(0,len(filelist)):\n",
    "    df = pd.read_csv(path.join(in_gff,filelist[i]),index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "\n",
    "    df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "    database[i] = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_list =[]\n",
    "TF_gene = []\n",
    "for i in range (0,len(database)): \n",
    "    TF_list.append(database[i].condition_name[1][:4])\n",
    "    TF_gene.append(database[i].condition_name[2][:4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['YafC',\n",
       " 'YbiH',\n",
       " 'YddM',\n",
       " 'Yagi',\n",
       " 'YeiE',\n",
       " 'YbaQ',\n",
       " 'YdcI',\n",
       " 'YiaJ',\n",
       " 'YheO',\n",
       " 'YieP',\n",
       " 'YbaO']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load gene info + Biocyc TRN "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_EcoCyc_TF = pd.read_csv(path.join(data_dir,'input_data','trn_EcoCyc.csv'),index_col=0)\n",
    "DF_gene_info = pd.read_csv(path.join(data_dir,'input_data','gene_info.csv'),index_col=0)\n",
    "precise2_TRN = pd.read_csv(path.join(data_dir,'input_data','trn_precise2.csv'),index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       start  stop strand gene_name  length   operon  \\\n",
       "b0001    189   255      +      thrL      66  thrLABC   \n",
       "b0002    336  2799      +      thrA    2463  thrLABC   \n",
       "b0003   2800  3733      +      thrB     933  thrLABC   \n",
       "b0004   3733  5020      +      thrC    1287  thrLABC   \n",
       "b0005   5233  5530      +      yaaX     297     yaaX   \n",
       "\n",
       "                                       cog  start_codon_pos  \n",
       "b0001                    No COG Annotation              189  \n",
       "b0002                    No COG Annotation              336  \n",
       "b0003  Amino acid transport and metabolism             2800  \n",
       "b0004  Amino acid transport and metabolism             3733  \n",
       "b0005                     Function unknown             5233  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define start codon position (to account for +/- strands)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "DF_gene_info.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "    TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF +'-' + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '') & (genes[1] == '')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start > BP) & (stop > BP) & (strand == '+')) | ((start < BP) & (stop < BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start < BP) & (stop > BP)) | ((start > BP) & (stop < BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "                    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "TRN_data = defaultdict(list)\n",
    "TF_name = [\n",
    "    'YgeV',\n",
    " 'YheO',\n",
    " 'YbiH',\n",
    " 'YbaQ',\n",
    " 'YbaO',\n",
    " 'YieP',\n",
    " 'YagI',\n",
    " 'YiaJ',\n",
    " 'YddM',\n",
    " 'YdcI',\n",
    " 'YafC',\n",
    " 'YeiE']\n",
    "\n",
    "for i in TF_name: \n",
    "    TRN_data[i] = [x for x in  list(precise2_TRN.gene_name[precise2_TRN.index == i])  if  str(x) != 'nan']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen(Peak_DF): \n",
    "    TF_name = Peak_DF['index'][1][:4]\n",
    "    gene_list = [i for i in Peak_DF.target_genes if i != '']\n",
    "    chip_data= ','.join(list(gene_list)).split(',')\n",
    "    reg_data = TRN_data[TF_name]\n",
    "    all_genes = [i  for i in chip_data if i not in reg_data] + reg_data\n",
    "    \n",
    "    \n",
    "    TF = TF_name\n",
    "    reg_genes=reg_data\n",
    "    reg_only = []\n",
    "    chip_genes=chip_data\n",
    "    chip_only = []\n",
    "    shared_genes=[]\n",
    "    for i in all_genes: \n",
    "        if (i in reg_data) & (i not in chip_data):\n",
    "            reg_only.append(i)\n",
    "        elif (i in chip_data) & (i not in reg_data):\n",
    "            chip_only.append(i)\n",
    "        elif (i in chip_data) & (i in reg_data):\n",
    "            shared_genes.append(i)\n",
    "            \n",
    "    values = [TF,\n",
    "          len(reg_genes),\n",
    "          len(reg_only),\n",
    "          len(chip_genes),\n",
    "          len(chip_only),\n",
    "          len(shared_genes),\n",
    "          len(all_genes)]\n",
    "    \n",
    "    index_name = ['TF',\n",
    "              'reg_genes',\n",
    "              'reg_only',\n",
    "              'chip_genes',\n",
    "              'chip_only',\n",
    "              'shared_genes',\n",
    "              'all_genes']\n",
    "    \n",
    "    genes = ['',\n",
    "         reg_genes,\n",
    "         reg_only,\n",
    "         chip_genes,\n",
    "         chip_only,\n",
    "         shared_genes,\n",
    "         all_genes]\n",
    "    \n",
    "    same1 = defaultdict(list)\n",
    "    for i in range(0,len(index_name)):\n",
    "        same1[index_name[i]].append(values[i])\n",
    "\n",
    "    finall = pd.DataFrame.from_dict(same1, orient='index', columns = ['value'])\n",
    "    finall['list'] = genes\n",
    "    for i, row in finall.iterrows(): \n",
    "        if row.value == 0:\n",
    "            finall.list[i] = '; '.join(precise2_TRN.source[precise2_TRN.index == TF_name].unique())\n",
    "    return finall"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Gao2021; Ecocyc'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'; '.join(precise2_TRN.source[precise2_TRN.index == 'YdcI'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['iraP',\n",
       " 'gltP',\n",
       " 'yicG',\n",
       " 'lldP',\n",
       " 'yhdV',\n",
       " 'gutM',\n",
       " 'yojI',\n",
       " 'yobA',\n",
       " 'yobF',\n",
       " 'dtpA',\n",
       " 'trkG',\n",
       " 'wrbA',\n",
       " 'ybhI',\n",
       " 'sdhC',\n",
       " 'gltA',\n",
       " 'tomB',\n",
       " 'ddlA',\n",
       " 'nhaA',\n",
       " 'isrC',\n",
       " 'flu',\n",
       " 'smrB']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TRN_data['YdcI']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value</th>\n",
       "      <th>list</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <td>YdcI</td>\n",
       "      <td></td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reg_genes</th>\n",
       "      <td>21</td>\n",
       "      <td>[iraP, gltP, yicG, lldP, yhdV, gutM, yojI, yob...</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reg_only</th>\n",
       "      <td>19</td>\n",
       "      <td>[iraP, gltP, yicG, lldP, yhdV, gutM, yojI, yob...</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chip_genes</th>\n",
       "      <td>68</td>\n",
       "      <td>[nhaA, nhaR, mokC, hokC, maa, hha, gltA, ybhH,...</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chip_only</th>\n",
       "      <td>64</td>\n",
       "      <td>[nhaR, mokC, hokC, maa, hha, ybhH, yccJ, ydaU,...</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>shared_genes</th>\n",
       "      <td>2</td>\n",
       "      <td>[gltA, nhaA]</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all_genes</th>\n",
       "      <td>85</td>\n",
       "      <td>[nhaR, mokC, hokC, maa, hha, ybhH, yccJ, ydaU,...</td>\n",
       "      <td>Gao2021; Ecocyc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             value                                               list  \\\n",
       "TF            YdcI                                                      \n",
       "reg_genes       21  [iraP, gltP, yicG, lldP, yhdV, gutM, yojI, yob...   \n",
       "reg_only        19  [iraP, gltP, yicG, lldP, yhdV, gutM, yojI, yob...   \n",
       "chip_genes      68  [nhaA, nhaR, mokC, hokC, maa, hha, gltA, ybhH,...   \n",
       "chip_only       64  [nhaR, mokC, hokC, maa, hha, ybhH, yccJ, ydaU,...   \n",
       "shared_genes     2                                       [gltA, nhaA]   \n",
       "all_genes       85  [nhaR, mokC, hokC, maa, hha, ybhH, yccJ, ydaU,...   \n",
       "\n",
       "                       source  \n",
       "TF            Gao2021; Ecocyc  \n",
       "reg_genes     Gao2021; Ecocyc  \n",
       "reg_only      Gao2021; Ecocyc  \n",
       "chip_genes    Gao2021; Ecocyc  \n",
       "chip_only     Gao2021; Ecocyc  \n",
       "shared_genes  Gao2021; Ecocyc  \n",
       "all_genes     Gao2021; Ecocyc  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i = 6\n",
    "\n",
    "peak_annot_DF = annotate_peaks(database[i],500,DF_gene_info).fillna('')\n",
    "final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "venn_files = Venn_data_gen(final_annot_DF)\n",
    "\n",
    "venn_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>condition</th>\n",
       "      <th>binding_peak_start</th>\n",
       "      <th>binding_peak_end</th>\n",
       "      <th>binding_peak_strength</th>\n",
       "      <th>target_locus</th>\n",
       "      <th>target_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI-1</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>17249</td>\n",
       "      <td>17276</td>\n",
       "      <td>160.551500</td>\n",
       "      <td>b0019,b0020,b0018,b4412</td>\n",
       "      <td>nhaA,nhaR,mokC,hokC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>YdcI-2</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>400362</td>\n",
       "      <td>400391</td>\n",
       "      <td>755.438800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YdcI-3</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>400396</td>\n",
       "      <td>400422</td>\n",
       "      <td>661.472800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>YdcI-4</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>480298</td>\n",
       "      <td>480326</td>\n",
       "      <td>13.553280</td>\n",
       "      <td>b0459,b0460</td>\n",
       "      <td>maa,hha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>YdcI-5</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>753952</td>\n",
       "      <td>753980</td>\n",
       "      <td>11.534270</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>YdcI-6</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>754249</td>\n",
       "      <td>754277</td>\n",
       "      <td>17.167660</td>\n",
       "      <td>b0720</td>\n",
       "      <td>gltA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>YdcI-7</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>801106</td>\n",
       "      <td>801134</td>\n",
       "      <td>6.904169</td>\n",
       "      <td>b0769</td>\n",
       "      <td>ybhH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>YdcI-8</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1067103</td>\n",
       "      <td>1067132</td>\n",
       "      <td>14.061660</td>\n",
       "      <td>b1003</td>\n",
       "      <td>yccJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>YdcI-9</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1421701</td>\n",
       "      <td>1421720</td>\n",
       "      <td>10.287660</td>\n",
       "      <td>b1359,b1360,b1361,b1362</td>\n",
       "      <td>ydaU,ydaV,ydaW,rzpR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YdcI-10</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1710207</td>\n",
       "      <td>1710235</td>\n",
       "      <td>145.471900</td>\n",
       "      <td>b1631,b1632,b1633</td>\n",
       "      <td>rsxG,rsxE,nth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>YdcI-11</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1710716</td>\n",
       "      <td>1710748</td>\n",
       "      <td>734.173400</td>\n",
       "      <td>b1631,b1632,b1633</td>\n",
       "      <td>rsxG,rsxE,nth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>YdcI-12</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1905919</td>\n",
       "      <td>1905947</td>\n",
       "      <td>57.594680</td>\n",
       "      <td>b1821</td>\n",
       "      <td>mntP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>YdcI-13</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1906003</td>\n",
       "      <td>1906030</td>\n",
       "      <td>107.186900</td>\n",
       "      <td>b1821</td>\n",
       "      <td>mntP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YdcI-14</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1923026</td>\n",
       "      <td>1923053</td>\n",
       "      <td>22.718800</td>\n",
       "      <td>b1838</td>\n",
       "      <td>pphA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>YdcI-15</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>2305311</td>\n",
       "      <td>2305337</td>\n",
       "      <td>43.293770</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>YdcI-16</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>2826624</td>\n",
       "      <td>2826650</td>\n",
       "      <td>7.296546</td>\n",
       "      <td>b2703,b2704,b2705,b2706,b2707,b2708</td>\n",
       "      <td>srlE,srlB,srlD,srlM,srlR,srlQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>YdcI-17</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3416289</td>\n",
       "      <td>3416317</td>\n",
       "      <td>50.592800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>YdcI-18</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3775238</td>\n",
       "      <td>3775267</td>\n",
       "      <td>62.955920</td>\n",
       "      <td>b3600,b3601</td>\n",
       "      <td>mtlD,mtlR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>YdcI-19</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3816713</td>\n",
       "      <td>3816744</td>\n",
       "      <td>25.631220</td>\n",
       "      <td>b3644,b3642,b3643_2,b3643_1</td>\n",
       "      <td>yicC,pyrE,rph_2,rph_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>YdcI-20</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>4292270</td>\n",
       "      <td>4292299</td>\n",
       "      <td>26.685470</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>YdcI-21</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>17249</td>\n",
       "      <td>17276</td>\n",
       "      <td>160.551500</td>\n",
       "      <td>b0019,b0020,b0018,b4412</td>\n",
       "      <td>nhaA,nhaR,mokC,hokC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>YdcI-22</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>400362</td>\n",
       "      <td>400391</td>\n",
       "      <td>755.438800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>YdcI-23</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>400396</td>\n",
       "      <td>400422</td>\n",
       "      <td>661.472800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>YdcI-24</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>480298</td>\n",
       "      <td>480326</td>\n",
       "      <td>13.553280</td>\n",
       "      <td>b0459,b0460</td>\n",
       "      <td>maa,hha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>YdcI-25</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>753952</td>\n",
       "      <td>753980</td>\n",
       "      <td>11.534270</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>YdcI-26</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>754249</td>\n",
       "      <td>754277</td>\n",
       "      <td>17.167660</td>\n",
       "      <td>b0720</td>\n",
       "      <td>gltA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>YdcI-27</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>801106</td>\n",
       "      <td>801134</td>\n",
       "      <td>6.904169</td>\n",
       "      <td>b0769</td>\n",
       "      <td>ybhH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>YdcI-28</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1067103</td>\n",
       "      <td>1067132</td>\n",
       "      <td>14.061660</td>\n",
       "      <td>b1003</td>\n",
       "      <td>yccJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>YdcI-29</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1421701</td>\n",
       "      <td>1421720</td>\n",
       "      <td>10.287660</td>\n",
       "      <td>b1359,b1360,b1361,b1362</td>\n",
       "      <td>ydaU,ydaV,ydaW,rzpR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>YdcI-30</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1710207</td>\n",
       "      <td>1710235</td>\n",
       "      <td>145.471900</td>\n",
       "      <td>b1631,b1632,b1633</td>\n",
       "      <td>rsxG,rsxE,nth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>YdcI-31</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1710716</td>\n",
       "      <td>1710748</td>\n",
       "      <td>734.173400</td>\n",
       "      <td>b1631,b1632,b1633</td>\n",
       "      <td>rsxG,rsxE,nth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>YdcI-32</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1905919</td>\n",
       "      <td>1905947</td>\n",
       "      <td>57.594680</td>\n",
       "      <td>b1821</td>\n",
       "      <td>mntP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>YdcI-33</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1906003</td>\n",
       "      <td>1906030</td>\n",
       "      <td>107.186900</td>\n",
       "      <td>b1821</td>\n",
       "      <td>mntP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>YdcI-34</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>1923026</td>\n",
       "      <td>1923053</td>\n",
       "      <td>22.718800</td>\n",
       "      <td>b1838</td>\n",
       "      <td>pphA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>YdcI-35</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>2305311</td>\n",
       "      <td>2305337</td>\n",
       "      <td>43.293770</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>YdcI-36</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>2826624</td>\n",
       "      <td>2826650</td>\n",
       "      <td>7.296546</td>\n",
       "      <td>b2703,b2704,b2705,b2706,b2707,b2708</td>\n",
       "      <td>srlE,srlB,srlD,srlM,srlR,srlQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>YdcI-37</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3416289</td>\n",
       "      <td>3416317</td>\n",
       "      <td>50.592800</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>YdcI-38</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3775238</td>\n",
       "      <td>3775267</td>\n",
       "      <td>62.955920</td>\n",
       "      <td>b3600,b3601</td>\n",
       "      <td>mtlD,mtlR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>YdcI-39</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>3816713</td>\n",
       "      <td>3816744</td>\n",
       "      <td>25.631220</td>\n",
       "      <td>b3644,b3642,b3643_2,b3643_1</td>\n",
       "      <td>yicC,pyrE,rph_2,rph_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>YdcI-40</td>\n",
       "      <td>ydci + ace</td>\n",
       "      <td>4292270</td>\n",
       "      <td>4292299</td>\n",
       "      <td>26.685470</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      index   condition  binding_peak_start  binding_peak_end  \\\n",
       "1    YdcI-1  ydci + ace               17249             17276   \n",
       "2    YdcI-2  ydci + ace              400362            400391   \n",
       "3    YdcI-3  ydci + ace              400396            400422   \n",
       "4    YdcI-4  ydci + ace              480298            480326   \n",
       "5    YdcI-5  ydci + ace              753952            753980   \n",
       "6    YdcI-6  ydci + ace              754249            754277   \n",
       "7    YdcI-7  ydci + ace              801106            801134   \n",
       "8    YdcI-8  ydci + ace             1067103           1067132   \n",
       "9    YdcI-9  ydci + ace             1421701           1421720   \n",
       "10  YdcI-10  ydci + ace             1710207           1710235   \n",
       "11  YdcI-11  ydci + ace             1710716           1710748   \n",
       "12  YdcI-12  ydci + ace             1905919           1905947   \n",
       "13  YdcI-13  ydci + ace             1906003           1906030   \n",
       "14  YdcI-14  ydci + ace             1923026           1923053   \n",
       "15  YdcI-15  ydci + ace             2305311           2305337   \n",
       "16  YdcI-16  ydci + ace             2826624           2826650   \n",
       "17  YdcI-17  ydci + ace             3416289           3416317   \n",
       "18  YdcI-18  ydci + ace             3775238           3775267   \n",
       "19  YdcI-19  ydci + ace             3816713           3816744   \n",
       "20  YdcI-20  ydci + ace             4292270           4292299   \n",
       "21  YdcI-21  ydci + ace               17249             17276   \n",
       "22  YdcI-22  ydci + ace              400362            400391   \n",
       "23  YdcI-23  ydci + ace              400396            400422   \n",
       "24  YdcI-24  ydci + ace              480298            480326   \n",
       "25  YdcI-25  ydci + ace              753952            753980   \n",
       "26  YdcI-26  ydci + ace              754249            754277   \n",
       "27  YdcI-27  ydci + ace              801106            801134   \n",
       "28  YdcI-28  ydci + ace             1067103           1067132   \n",
       "29  YdcI-29  ydci + ace             1421701           1421720   \n",
       "30  YdcI-30  ydci + ace             1710207           1710235   \n",
       "31  YdcI-31  ydci + ace             1710716           1710748   \n",
       "32  YdcI-32  ydci + ace             1905919           1905947   \n",
       "33  YdcI-33  ydci + ace             1906003           1906030   \n",
       "34  YdcI-34  ydci + ace             1923026           1923053   \n",
       "35  YdcI-35  ydci + ace             2305311           2305337   \n",
       "36  YdcI-36  ydci + ace             2826624           2826650   \n",
       "37  YdcI-37  ydci + ace             3416289           3416317   \n",
       "38  YdcI-38  ydci + ace             3775238           3775267   \n",
       "39  YdcI-39  ydci + ace             3816713           3816744   \n",
       "40  YdcI-40  ydci + ace             4292270           4292299   \n",
       "\n",
       "    binding_peak_strength                         target_locus  \\\n",
       "1              160.551500              b0019,b0020,b0018,b4412   \n",
       "2              755.438800                                        \n",
       "3              661.472800                                        \n",
       "4               13.553280                          b0459,b0460   \n",
       "5               11.534270                                        \n",
       "6               17.167660                                b0720   \n",
       "7                6.904169                                b0769   \n",
       "8               14.061660                                b1003   \n",
       "9               10.287660              b1359,b1360,b1361,b1362   \n",
       "10             145.471900                    b1631,b1632,b1633   \n",
       "11             734.173400                    b1631,b1632,b1633   \n",
       "12              57.594680                                b1821   \n",
       "13             107.186900                                b1821   \n",
       "14              22.718800                                b1838   \n",
       "15              43.293770                                        \n",
       "16               7.296546  b2703,b2704,b2705,b2706,b2707,b2708   \n",
       "17              50.592800                                        \n",
       "18              62.955920                          b3600,b3601   \n",
       "19              25.631220          b3644,b3642,b3643_2,b3643_1   \n",
       "20              26.685470                                        \n",
       "21             160.551500              b0019,b0020,b0018,b4412   \n",
       "22             755.438800                                        \n",
       "23             661.472800                                        \n",
       "24              13.553280                          b0459,b0460   \n",
       "25              11.534270                                        \n",
       "26              17.167660                                b0720   \n",
       "27               6.904169                                b0769   \n",
       "28              14.061660                                b1003   \n",
       "29              10.287660              b1359,b1360,b1361,b1362   \n",
       "30             145.471900                    b1631,b1632,b1633   \n",
       "31             734.173400                    b1631,b1632,b1633   \n",
       "32              57.594680                                b1821   \n",
       "33             107.186900                                b1821   \n",
       "34              22.718800                                b1838   \n",
       "35              43.293770                                        \n",
       "36               7.296546  b2703,b2704,b2705,b2706,b2707,b2708   \n",
       "37              50.592800                                        \n",
       "38              62.955920                          b3600,b3601   \n",
       "39              25.631220          b3644,b3642,b3643_2,b3643_1   \n",
       "40              26.685470                                        \n",
       "\n",
       "                     target_genes  \n",
       "1             nhaA,nhaR,mokC,hokC  \n",
       "2                                  \n",
       "3                                  \n",
       "4                         maa,hha  \n",
       "5                                  \n",
       "6                            gltA  \n",
       "7                            ybhH  \n",
       "8                            yccJ  \n",
       "9             ydaU,ydaV,ydaW,rzpR  \n",
       "10                  rsxG,rsxE,nth  \n",
       "11                  rsxG,rsxE,nth  \n",
       "12                           mntP  \n",
       "13                           mntP  \n",
       "14                           pphA  \n",
       "15                                 \n",
       "16  srlE,srlB,srlD,srlM,srlR,srlQ  \n",
       "17                                 \n",
       "18                      mtlD,mtlR  \n",
       "19          yicC,pyrE,rph_2,rph_1  \n",
       "20                                 \n",
       "21            nhaA,nhaR,mokC,hokC  \n",
       "22                                 \n",
       "23                                 \n",
       "24                        maa,hha  \n",
       "25                                 \n",
       "26                           gltA  \n",
       "27                           ybhH  \n",
       "28                           yccJ  \n",
       "29            ydaU,ydaV,ydaW,rzpR  \n",
       "30                  rsxG,rsxE,nth  \n",
       "31                  rsxG,rsxE,nth  \n",
       "32                           mntP  \n",
       "33                           mntP  \n",
       "34                           pphA  \n",
       "35                                 \n",
       "36  srlE,srlB,srlD,srlM,srlR,srlQ  \n",
       "37                                 \n",
       "38                      mtlD,mtlR  \n",
       "39          yicC,pyrE,rph_2,rph_1  \n",
       "40                                 "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_annot_DF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate the data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run this block if you are ready to generate your data for ChIPdb."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range (0, len(database)): \n",
    "    peak_annot_DF = annotate_peaks(database[i],500,DF_gene_info).fillna('')\n",
    "    final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "    venn_files = Venn_data_gen(final_annot_DF)\n",
    "    final_annot_DF.to_json(out_data_table+TF_list[i]+'.json',orient='records')\n",
    "    venn_files.to_json(path.join(out_data_venn, TF_list[i]+'.json'),orient='records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
