{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Generation for Plots Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook generates data used to contruct `binding peak` tables and `venn diagrams` for ChIP-pro, as well as other  visuals. But before you run the notebook, make sure of the following: \n",
    "\n",
    "> Make sure to rename TF in the gff file to match the TF convention and not gene convension."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np \n",
    "import pandas as pd\n",
    "import glob\n",
    "import urllib                      \n",
    "import gzip\n",
    "from collections import defaultdict\n",
    "\n",
    "pd.set_option('display.max_columns', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binding peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function assigns bindings peaks for each TF and fins the corresponsing gene targets using the `gene_info.csv` table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def annotate_peaks(TF_name, TF_condition, peak_df,margin,gene_info):\n",
    "    \n",
    "    res_df = peak_df.copy()\n",
    "#     TF = peak_df.condition_name[1][:4]\n",
    "    for i,row in res_df.iterrows():\n",
    "        pos = row['binding_peak_start']\n",
    "        # Identify genes within MARGIN nt of binding peak\n",
    "        close_genes = gene_info[(gene_info.start_codon_pos > pos-margin) \n",
    "                                & (gene_info.start_codon_pos < pos+margin)]\n",
    "        for strand,group in close_genes.groupby('strand'):\n",
    "            #Remove genes that are completely transcribed before binding peak\n",
    "            if strand == '+':\n",
    "                group = group[group.stop > pos]\n",
    "            else:\n",
    "                group = group[group.start < pos]\n",
    "\n",
    "            operon = group.operon.unique()\n",
    "            # Ensure that we're only identifying one operon on either side of binding peak\n",
    "#             if len(operon) > 1:\n",
    "#                 print (operon)\n",
    "\n",
    "            # Get all genes in operon\n",
    "            bnums = gene_info[gene_info.operon.isin(operon)].index\n",
    "            \n",
    "            ## Add gene information to dataframe\n",
    "            if strand == '+':\n",
    "                res_df.loc[i,'TU_p'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_p'] = ','.join(bnums)\n",
    "            else:\n",
    "                res_df.loc[i,'TU_m'] = ','.join(operon)\n",
    "                res_df.loc[i,'genes_m'] = ','.join(bnums)\n",
    "    res_df['index'] = [TF_name +'-' + str(i) for i in range(1,peak_df.shape[0]+1)]\n",
    "    res_df['condition'] = [ TF_name.lower() + \" + \" + TF_condition for i in peak_df.condition_name]\n",
    "#     [peak_df.condition_name[2][:4]+' + '+peak_df.condition_name[1][5:8]]*peak_df.shape[0]\n",
    "    cols = ['index','condition','binding_peak_start','binding_peak_end',\n",
    "            'binding_peak_strength','TU_p','genes_p','TU_m','genes_m']\n",
    "    return res_df.reindex(columns = cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validate Binding Peaks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this function validates the accuracy of every gene target from the already indetified gene list for every binding site, identified from the previous function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper for closest gene\n",
    "\n",
    "def closest_locus_nc000913(peak_start, peak_stop, gene_info):\n",
    "    df_pos = gene_info[gene_info.strand==\"+\"]\n",
    "    df_pos = df_pos[df_pos.stop >= peak_start]\n",
    "    try:\n",
    "        gene_pos = df_pos.iloc[0,:]['gene_name']\n",
    "        dist_pos = abs(peak_stop - df_pos.iloc[0,:]['start'])\n",
    "    except:\n",
    "        dist_pos = np.inf\n",
    "    \n",
    "    df_neg = gene_info[gene_info.strand==\"-\"]\n",
    "    df_neg = df_neg[df_neg.start <= peak_stop]\n",
    "    try:\n",
    "        gene_neg = df_neg.iloc[-1,:]['gene_name']\n",
    "        dist_neg = abs(peak_start - df_neg.iloc[-1,:]['stop'])\n",
    "    except:\n",
    "        dist_neg = np.inf\n",
    "    \n",
    "    if dist_pos <= dist_neg:\n",
    "        return(gene_pos)\n",
    "    else:\n",
    "        return(gene_neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "def validate_peak_info(df,gene_info):\n",
    "    locusTag = defaultdict(list)\n",
    "    geneName = defaultdict(list)\n",
    "    for i,row in df.iterrows():\n",
    "        BP = row['binding_peak_start']\n",
    "        idx_name = i\n",
    "        genes = [row['genes_p'] , row['genes_m']]\n",
    "        if ((genes[0] == '' or str(genes[0]) == 'nan') & (genes[1] == '' or str(genes[1]) == 'nan')):\n",
    "            locusTag[idx_name].append('')\n",
    "            geneName[idx_name].append('')\n",
    "        for gene in genes:\n",
    "            if gene == '' or str(gene) == 'nan':\n",
    "                continue \n",
    "            gene_list = gene.split(',')\n",
    "            for g in gene_list: \n",
    "                name = DF_gene_info.loc[g].gene_name \n",
    "                strand = DF_gene_info.loc[g].strand \n",
    "                start = DF_gene_info.loc[g].start\n",
    "                stop = DF_gene_info.loc[g].stop\n",
    "                if ((start >= BP) & (stop >= BP) & (strand == '+')) | ((start <= BP) & (stop <= BP) & (strand == '-')):\n",
    "                        locusTag[idx_name].append(g)\n",
    "                        geneName[idx_name].append(name)\n",
    "                elif ((start <= BP) & (stop >= BP)) | ((start >= BP) & (stop <= BP)):\n",
    "                    locusTag[idx_name].append(g)\n",
    "                    geneName[idx_name].append(name)\n",
    "    \n",
    "    for k, v in locusTag.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            locusTag[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(locusTag[k])\n",
    "        locusTag[k] = genes\n",
    "\n",
    "    for k, v in geneName.items():\n",
    "        if ((len(v) == 1) & (v[0] == '')): \n",
    "            geneName[k] = ''\n",
    "            continue\n",
    "        genes = ','.join(geneName[k])\n",
    "        geneName[k] = genes\n",
    "\n",
    "    df_complete = df.loc[:,['index','condition','binding_peak_start',\n",
    "                            'binding_peak_end','binding_peak_strength']]\n",
    "    df_complete['target_locus'] = locusTag.values()\n",
    "    df_complete['target_genes'] = geneName.values()\n",
    "    df_complete['closest_gene'] = [closest_locus_nc000913(df_complete.binding_peak_start.iloc[i], \n",
    "                                           df_complete.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(df_complete))]\n",
    "    return df_complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn Diagrams"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure to update the list of `TF_names` as Ye adds more gff files into the dropbox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Venn_data_gen(Peak_DF, TRN_data): \n",
    "    TF_name = Peak_DF['index'][1].split(\"-\")[0].split(\" \")[0]\n",
    "    gene_list = [i for i in Peak_DF.target_genes if i != '']\n",
    "    chip_data= ','.join(list(gene_list)).split(',')\n",
    "    reg_data = TRN_data.loc[TF_name].gene_name.to_list()\n",
    "    all_genes = [i  for i in chip_data if i not in reg_data] + reg_data\n",
    "    \n",
    "    \n",
    "    TF = TF_name\n",
    "    reg_genes=reg_data\n",
    "    reg_only = []\n",
    "    chip_genes=chip_data\n",
    "    chip_only = []\n",
    "    shared_genes=[]\n",
    "    for i in all_genes: \n",
    "        if (i in reg_data) & (i not in chip_data):\n",
    "            reg_only.append(i)\n",
    "        elif (i in chip_data) & (i not in reg_data):\n",
    "            chip_only.append(i)\n",
    "        elif (i in chip_data) & (i in reg_data):\n",
    "            shared_genes.append(i)\n",
    "            \n",
    "    values = [TF,\n",
    "          len(reg_genes),\n",
    "          len(reg_only),\n",
    "          len(chip_genes),\n",
    "          len(chip_only),\n",
    "          len(shared_genes),\n",
    "          len(all_genes)]\n",
    "    \n",
    "    index_name = ['TF',\n",
    "              'reg_genes',\n",
    "              'reg_only',\n",
    "              'chip_genes',\n",
    "              'chip_only',\n",
    "              'shared_genes',\n",
    "              'all_genes']\n",
    "    \n",
    "    genes = ['; '.join(TRN_data.source[TRN_data.index == TF_name].unique()),\n",
    "         reg_genes,\n",
    "         reg_only,\n",
    "         chip_genes,\n",
    "         chip_only,\n",
    "         shared_genes,\n",
    "         all_genes]\n",
    "    \n",
    "    same1 = defaultdict(list)\n",
    "    for i in range(0,len(index_name)):\n",
    "        same1[index_name[i]].append(values[i])\n",
    "\n",
    "    finall = pd.DataFrame.from_dict(same1, orient='index', columns = ['value'])\n",
    "    finall['list'] = genes\n",
    "    for i, row in finall.iterrows(): \n",
    "        if row.value == 0:\n",
    "            finall.list[i] = ''\n",
    "    return finall"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Binding Width Histograms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "def binding_width_gen(TF, peak_df,out_dir): \n",
    "    widths = pd.DataFrame(peak_df.binding_peak_end - peak_df.binding_peak_start + 1, columns=[\"binding_width\"])\n",
    "    widths.to_csv(out_dir+TF+'_widths.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peak Position Scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "def peak_position_gen(TF, final_annot_DF,out_dir): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.loc[i+1,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        gene = peak.closest_gene\n",
    "        gene_info = DF_gene_info[DF_gene_info.gene_name == gene]\n",
    "        try:\n",
    "            if gene_info.strand[0] == \"-\":\n",
    "                dist = float((gene_info.stop - peak_center)/gene_info.length)\n",
    "            else:\n",
    "                dist = float((peak_center - gene_info.start)/gene_info.length)\n",
    "            peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "            counter+=1\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - NC_000913.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/e_coli/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GadW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadw_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GadX</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gadx_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>gadx_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gadx_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KdpE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>Tris-maleic acid (TMA)</td>\n",
       "      <td>0.1 mM KCl</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>KCl|kdpe_KCl_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>R1</td>\n",
       "      <td>KdpeE_R2_S34_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>KdpeE_R2_S34_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9P</td>\n",
       "      <td>Low phosphate concentration (10% of the M9 med...</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>M9P|phob_M9P_binding_table.json</td>\n",
       "      <td>GSE143856</td>\n",
       "      <td>https://doi.org/10.1128/mSystems.00980-20</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>PhoB_1_S16_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>PhoB_1_S16_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>PhoB_2_S6_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>PhoB_2_S6_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>RpoB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|rpob_dpd_binding_table.json;fe|rpob_fe_bin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>rpob_fe1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>rpob_fe2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>rpob_dpd1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>rpob_dpd2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YbaO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybao_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybao_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybao_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>YbaQ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>41.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|ybaq_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ybaq_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ybaq_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5, pH 7.0, or pH 8.5</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>40.0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>ph5|ydci_ph5_binding_table.json;ph7|ydci_ph7_b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH5 – R1</td>\n",
       "      <td>ydci_ph5_1.bw</td>\n",
       "      <td>pH5 – R2</td>\n",
       "      <td>ydci_ph5_2.bw</td>\n",
       "      <td>pH7 – R1</td>\n",
       "      <td>ydci_ph7_1.bw</td>\n",
       "      <td>pH7 – R2</td>\n",
       "      <td>ydci_ph7_2.bw</td>\n",
       "      <td>pH8 – R1</td>\n",
       "      <td>ydci_ph8_1.bw</td>\n",
       "      <td>pH8 – R2</td>\n",
       "      <td>ydci_ph8_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>YddM</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yddm_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yddm_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yddm_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YheO</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yheo_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yheo_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yheo_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>YiaG</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiag_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiag_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiag_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>YieP</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>100 mM 3-hydroxypropionic acid (3-HP)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yiep_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yiep_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yiep_2.bw</td>\n",
       "      <td>3HP – R1</td>\n",
       "      <td>yiep_3hp_1.bw</td>\n",
       "      <td>3HP – R2</td>\n",
       "      <td>yiep_3hp_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>YihL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|yihl_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>YihW</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|yihw_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yihw_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yihw_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>YjdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF1</td>\n",
       "      <td>M9|yjdc_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>YjhI</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|yjhi_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yjhi_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yjhi_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>YneJ</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|ynej_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynej_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynej_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>YnfL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|ynfl_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ynfl_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ynfl_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>YpdC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>glc|ypdc_glc_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>ypdc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ypdc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>YqhC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>yTF2</td>\n",
       "      <td>M9|yqhc_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yqhc_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yqhc_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ZraR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>1 mM ZnCl2</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>ZnCl2|zrar_ZnCl2_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>ZraR_1_S3_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>ZraR_1_S3_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>ZraR_2_S4_R1.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>O157:H7 EDL933</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>AE005174.2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>EDL933</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>edl933furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>edl933furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_010498.1</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SMS-3-5</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>smsfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>smsfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>CU928163.2</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>UMN026</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>umn026furfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>umn026furfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Dps</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|dps_expo_binding_table.json;sta|dps_sta_b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>dps_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>dps_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>dps_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>dps_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Fis</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|fis_expo_binding_table.json;sta|fis_sta_b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>fis_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>fis_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>fis_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>fis_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GyrA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|gyra_expo_binding_table.json;sta|gyra_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyra_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyra_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyra_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyra_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GyrB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|gyrb_expo_binding_table.json;sta|gyrb_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>gyrb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>gyrb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>gyrb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>gyrb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>H-NS</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|h-ns_expo_binding_table.json;sta|h-ns_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hns_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hns_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hns_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hns_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>HupA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|hupa_expo_binding_table.json;sta|hupa_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>HupB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|hupb_expo_binding_table.json;sta|hupb_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>hupb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>hupb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>hupb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>hupb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>InfA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|infa_expo_binding_table.json;sta|infa_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>infa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>infa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>infa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>infa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MukB</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|mukb_expo_binding_table.json;sta|mukb_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>mukb_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>mukb_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>mukb_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>mukb_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>RdgC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|rdgc_expo_binding_table.json;sta|rdgc_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rdgc_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rdgc_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rdgc_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rdgc_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Rob</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|rob_expo_binding_table.json;sta|rob_sta_b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>rob_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>rob_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>rob_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>rob_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>StpA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|stpa_expo_binding_table.json;sta|stpa_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>stpa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>stpa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>stpa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>stpa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TopA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|topa_expo_binding_table.json;sta|topa_sta...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>topa_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>topa_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>topa_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>topa_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Tus</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>conformation</td>\n",
       "      <td>expo|tus_expo_binding_table.json;sta|tus_sta_b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exponential growth – R1</td>\n",
       "      <td>tus_expo_1.bw</td>\n",
       "      <td>exponential growth – R2</td>\n",
       "      <td>tus_expo_2.bw</td>\n",
       "      <td>stationary growth – R1</td>\n",
       "      <td>tus_sta_1.bw</td>\n",
       "      <td>stationary growth – R2</td>\n",
       "      <td>tus_sta_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NarL</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>100 mM Nitrate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>nitrate|narl_nitrate_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>narl_nitrate_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>narl_nitrate_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PuuR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>Putrescine</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>putrescine|puur_putrescine_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>puur_putrescine_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>puur_putrescine_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>RcsA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>250 uM Paraquat (PQ) treatment for 20 min with...</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Misc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>rcsa_oxida_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>rcsa_oxida_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>RpoB (Cra and Crp KO exps)</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Acetate</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>Misc</td>\n",
       "      <td>MG1655 ace|rpob (cra and crp ko exps)_MG1655 a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>control (WT) – R1</td>\n",
       "      <td>mg1655_ace_rpob1.bw</td>\n",
       "      <td>control (WT) – R2</td>\n",
       "      <td>mg1655_ace_rpob2.bw</td>\n",
       "      <td>cra-KO – R1</td>\n",
       "      <td>cra_ko_ace_rpob1.bw</td>\n",
       "      <td>cra-KO – R2</td>\n",
       "      <td>cra_ko_ace_rpob2.bw</td>\n",
       "      <td>crp-KO – R1</td>\n",
       "      <td>crp_ko_ace_rpob1.bw</td>\n",
       "      <td>crp-KO – R2</td>\n",
       "      <td>crp_ko_ace_rpob2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>FlhC</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E-MTAB-2457</td>\n",
       "      <td>https://doi.org/10.1371/journal.pgen.1004649</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR475432.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR475433.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>FlhD</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E-MTAB-2457</td>\n",
       "      <td>https://doi.org/10.1371/journal.pgen.1004649</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR475429.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR475430.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>FliA</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E-MTAB-2457</td>\n",
       "      <td>https://doi.org/10.1371/journal.pgen.1004649</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR475428.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR475431.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Fnr</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>MOPS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE41195</td>\n",
       "      <td>https://doi.org/10.1371/journal.pgen.1003565</td>\n",
       "      <td>anaerobic – R1</td>\n",
       "      <td>GSM1010219_FNR_IP_ChIP-seq_Anaerobic_A.bw</td>\n",
       "      <td>anaerobic – R2</td>\n",
       "      <td>GSM1010220_FNR_IP_ChIP-seq_Anaerobic_B.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>RpoD</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>MOPS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>public</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE41195</td>\n",
       "      <td>https://doi.org/10.1371/journal.pgen.1003565</td>\n",
       "      <td>aerobic – R1</td>\n",
       "      <td>GSM1010221_Sigma70_IP_ChIP-seq_Aerobic_A_SET.bw</td>\n",
       "      <td>aerobic – R2</td>\n",
       "      <td>GSM1072326_Sigma70_IP_ChIP-seq_Aerobic_B_SET.bw</td>\n",
       "      <td>anaerobic – R1</td>\n",
       "      <td>GSM1010222_Sigma70_IP_ChIP-seq_Anaerobic_A_SET.bw</td>\n",
       "      <td>anaerobic – R2</td>\n",
       "      <td>GSM1072327_Sigma70_IP_ChIP-seq_Anaerobic_B_SET.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               TF          Organism          Strain  \\\n",
       "index                                                                 \n",
       "0                            BaeR  Escherichia coli     K-12 MG1655   \n",
       "1                            CpxR  Escherichia coli     K-12 MG1655   \n",
       "2                             Cra  Escherichia coli     K-12 MG1655   \n",
       "3                             Fur  Escherichia coli     K-12 MG1655   \n",
       "4                            GadE  Escherichia coli     K-12 MG1655   \n",
       "5                            GadW  Escherichia coli     K-12 MG1655   \n",
       "6                            GadX  Escherichia coli     K-12 MG1655   \n",
       "7                            KdpE  Escherichia coli     K-12 MG1655   \n",
       "8                            PhoB  Escherichia coli     K-12 MG1655   \n",
       "9                            RpoB  Escherichia coli     K-12 MG1655   \n",
       "10                           YbaO  Escherichia coli     K-12 MG1655   \n",
       "11                           YbaQ  Escherichia coli     K-12 MG1655   \n",
       "12                           YdcI  Escherichia coli     K-12 MG1655   \n",
       "13                           YddM  Escherichia coli     K-12 MG1655   \n",
       "14                           YheO  Escherichia coli     K-12 MG1655   \n",
       "15                           YiaG  Escherichia coli     K-12 MG1655   \n",
       "16                           YieP  Escherichia coli     K-12 MG1655   \n",
       "17                           YihL  Escherichia coli     K-12 MG1655   \n",
       "18                           YihW  Escherichia coli     K-12 MG1655   \n",
       "19                           YjdC  Escherichia coli     K-12 MG1655   \n",
       "20                           YjhI  Escherichia coli     K-12 MG1655   \n",
       "21                           YneJ  Escherichia coli     K-12 MG1655   \n",
       "22                           YnfL  Escherichia coli     K-12 MG1655   \n",
       "23                           YpdC  Escherichia coli     K-12 MG1655   \n",
       "24                           YqhC  Escherichia coli     K-12 MG1655   \n",
       "25                           ZraR  Escherichia coli     K-12 MG1655   \n",
       "26                            Fur  Escherichia coli  O157:H7 EDL933   \n",
       "27                            Fur  Escherichia coli         SMS-3-5   \n",
       "28                            Fur  Escherichia coli          UMN026   \n",
       "29                            Dps  Escherichia coli     K-12 MG1655   \n",
       "30                            Fis  Escherichia coli     K-12 MG1655   \n",
       "31                           GyrA  Escherichia coli     K-12 MG1655   \n",
       "32                           GyrB  Escherichia coli     K-12 MG1655   \n",
       "33                           H-NS  Escherichia coli     K-12 MG1655   \n",
       "34                           HupA  Escherichia coli     K-12 MG1655   \n",
       "35                           HupB  Escherichia coli     K-12 MG1655   \n",
       "36                           InfA  Escherichia coli     K-12 MG1655   \n",
       "37                           MukB  Escherichia coli     K-12 MG1655   \n",
       "38                           RdgC  Escherichia coli     K-12 MG1655   \n",
       "39                            Rob  Escherichia coli     K-12 MG1655   \n",
       "40                           StpA  Escherichia coli     K-12 MG1655   \n",
       "41                           TopA  Escherichia coli     K-12 MG1655   \n",
       "42                            Tus  Escherichia coli     K-12 MG1655   \n",
       "43                           NarL  Escherichia coli     K-12 MG1655   \n",
       "44                           PuuR  Escherichia coli     K-12 MG1655   \n",
       "45                           RcsA  Escherichia coli     K-12 MG1655   \n",
       "46     RpoB (Cra and Crp KO exps)  Escherichia coli     K-12 MG1655   \n",
       "47                           FlhC  Escherichia coli     K-12 MG1655   \n",
       "48                           FlhD  Escherichia coli     K-12 MG1655   \n",
       "49                           FliA  Escherichia coli     K-12 MG1655   \n",
       "50                            Fnr  Escherichia coli     K-12 MG1655   \n",
       "51                           RpoD  Escherichia coli     K-12 MG1655   \n",
       "\n",
       "                        Media  \\\n",
       "index                           \n",
       "0                          LB   \n",
       "1                          LB   \n",
       "2                          M9   \n",
       "3                          M9   \n",
       "4                          M9   \n",
       "5                          M9   \n",
       "6                          M9   \n",
       "7      Tris-maleic acid (TMA)   \n",
       "8                         M9P   \n",
       "9                          M9   \n",
       "10                         M9   \n",
       "11                         M9   \n",
       "12                         M9   \n",
       "13                         M9   \n",
       "14                         M9   \n",
       "15                         M9   \n",
       "16                         M9   \n",
       "17                         M9   \n",
       "18                         M9   \n",
       "19                         M9   \n",
       "20                         M9   \n",
       "21                         M9   \n",
       "22                         M9   \n",
       "23                         M9   \n",
       "24                         M9   \n",
       "25                         LB   \n",
       "26                         M9   \n",
       "27                         M9   \n",
       "28                         M9   \n",
       "29                         M9   \n",
       "30                         M9   \n",
       "31                         M9   \n",
       "32                         M9   \n",
       "33                         M9   \n",
       "34                         M9   \n",
       "35                         M9   \n",
       "36                         M9   \n",
       "37                         M9   \n",
       "38                         M9   \n",
       "39                         M9   \n",
       "40                         M9   \n",
       "41                         M9   \n",
       "42                         M9   \n",
       "43                         M9   \n",
       "44                         M9   \n",
       "45                         M9   \n",
       "46                         M9   \n",
       "47                         LB   \n",
       "48                         LB   \n",
       "49                         LB   \n",
       "50                       MOPS   \n",
       "51                       MOPS   \n",
       "\n",
       "                                              Supplement    genome_id  \\\n",
       "index                                                                   \n",
       "0                                                5% EtOH  NC_000913.3   \n",
       "1                                                5% EtOH  NC_000913.3   \n",
       "2          0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3   \n",
       "3             0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3   \n",
       "4                             pH 5.5 (adjusted with HCl)  NC_000913.3   \n",
       "5                             pH 5.5 (adjusted with HCl)  NC_000913.3   \n",
       "6                             pH 5.5 (adjusted with HCl)  NC_000913.3   \n",
       "7                                             0.1 mM KCl  NC_000913.3   \n",
       "8      Low phosphate concentration (10% of the M9 med...  NC_000913.3   \n",
       "9             0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3   \n",
       "10                                                   NaN  NC_000913.3   \n",
       "11                                                   NaN  NC_000913.3   \n",
       "12                             pH 5.5, pH 7.0, or pH 8.5  NC_000913.3   \n",
       "13                                                   NaN  NC_000913.3   \n",
       "14                                                   NaN  NC_000913.3   \n",
       "15                                                   NaN  NC_000913.3   \n",
       "16                 100 mM 3-hydroxypropionic acid (3-HP)  NC_000913.3   \n",
       "17                                                   NaN  NC_000913.3   \n",
       "18                                                   NaN  NC_000913.3   \n",
       "19                                                   NaN  NC_000913.3   \n",
       "20                                                   NaN  NC_000913.3   \n",
       "21                                                   NaN  NC_000913.3   \n",
       "22                                                   NaN  NC_000913.3   \n",
       "23                                                   NaN  NC_000913.3   \n",
       "24                                                   NaN  NC_000913.3   \n",
       "25                                            1 mM ZnCl2  NC_000913.3   \n",
       "26            0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)   AE005174.2   \n",
       "27            0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_010498.1   \n",
       "28            0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)   CU928163.2   \n",
       "29                                                   NaN  NC_000913.3   \n",
       "30                                                   NaN  NC_000913.3   \n",
       "31                                                   NaN  NC_000913.3   \n",
       "32                                                   NaN  NC_000913.3   \n",
       "33                                                   NaN  NC_000913.3   \n",
       "34                                                   NaN  NC_000913.3   \n",
       "35                                                   NaN  NC_000913.3   \n",
       "36                                                   NaN  NC_000913.3   \n",
       "37                                                   NaN  NC_000913.3   \n",
       "38                                                   NaN  NC_000913.3   \n",
       "39                                                   NaN  NC_000913.3   \n",
       "40                                                   NaN  NC_000913.3   \n",
       "41                                                   NaN  NC_000913.3   \n",
       "42                                                   NaN  NC_000913.3   \n",
       "43                                        100 mM Nitrate  NC_000913.3   \n",
       "44                                            Putrescine  NC_000913.3   \n",
       "45     250 uM Paraquat (PQ) treatment for 20 min with...  NC_000913.3   \n",
       "46                                          0.2% Acetate  NC_000913.3   \n",
       "47                                                   NaN  NC_000913.3   \n",
       "48                                                   NaN  NC_000913.3   \n",
       "49                                                   NaN  NC_000913.3   \n",
       "50                                                   NaN  NC_000913.3   \n",
       "51                                                   NaN  NC_000913.3   \n",
       "\n",
       "      organism_id  num_binding_sites  num_samples  num_cond       project  \\\n",
       "index                                                                       \n",
       "0          e_coli                NaN            4         1           TCS   \n",
       "1          e_coli                NaN            2         1           TCS   \n",
       "2          e_coli                NaN            8         4      known_TF   \n",
       "3          e_coli                NaN            4         2      known_TF   \n",
       "4          e_coli                NaN            2         1      known_TF   \n",
       "5          e_coli                NaN            2         1      known_TF   \n",
       "6          e_coli                NaN            2         1      known_TF   \n",
       "7          e_coli                NaN            2         1           TCS   \n",
       "8          e_coli                NaN            4         1           TCS   \n",
       "9          e_coli                NaN            4         2      known_TF   \n",
       "10         e_coli                4.0            2         1          yTF1   \n",
       "11         e_coli               41.0            2         1          yTF1   \n",
       "12         e_coli               40.0            6         3          yTF1   \n",
       "13         e_coli               18.0            2         1          yTF1   \n",
       "14         e_coli                4.0            2         1          yTF1   \n",
       "15         e_coli                NaN            2         1          yTF1   \n",
       "16         e_coli               23.0            4         1          yTF1   \n",
       "17         e_coli                NaN            2         1          yTF2   \n",
       "18         e_coli                NaN            2         1          yTF2   \n",
       "19         e_coli                NaN            2         1          yTF1   \n",
       "20         e_coli                NaN            2         1          yTF2   \n",
       "21         e_coli                NaN            2         1          yTF2   \n",
       "22         e_coli                NaN            2         1          yTF2   \n",
       "23         e_coli                NaN            2         1          yTF2   \n",
       "24         e_coli                NaN            2         1          yTF2   \n",
       "25         e_coli                NaN            3         1           TCS   \n",
       "26         e_coli                NaN            2         1        EDL933   \n",
       "27         e_coli                NaN            2         1       SMS-3-5   \n",
       "28         e_coli                NaN            2         1        UMN026   \n",
       "29         e_coli                NaN            4         2  conformation   \n",
       "30         e_coli                NaN            4         2  conformation   \n",
       "31         e_coli                NaN            4         2  conformation   \n",
       "32         e_coli                NaN            4         2  conformation   \n",
       "33         e_coli                NaN            4         2  conformation   \n",
       "34         e_coli                NaN            4         2  conformation   \n",
       "35         e_coli                NaN            4         2  conformation   \n",
       "36         e_coli                NaN            4         2  conformation   \n",
       "37         e_coli                NaN            4         2  conformation   \n",
       "38         e_coli                NaN            4         2  conformation   \n",
       "39         e_coli                NaN            4         2  conformation   \n",
       "40         e_coli                NaN            4         2  conformation   \n",
       "41         e_coli                NaN            4         2  conformation   \n",
       "42         e_coli                NaN            4         2  conformation   \n",
       "43         e_coli                NaN            2         1          Misc   \n",
       "44         e_coli                NaN            2         1          Misc   \n",
       "45         e_coli                NaN            2         1          Misc   \n",
       "46         e_coli                NaN            6         3          Misc   \n",
       "47         e_coli                NaN            2         1        public   \n",
       "48         e_coli                NaN            2         1        public   \n",
       "49         e_coli                NaN            2         1        public   \n",
       "50         e_coli                NaN            2         1        public   \n",
       "51         e_coli                NaN            4         1        public   \n",
       "\n",
       "                                          binding tables    accession  \\\n",
       "index                                                                   \n",
       "0                      EtOH|baer_EtOH_binding_table.json    GSE143856   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json    GSE143856   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...          NaN   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...          NaN   \n",
       "4                          M9|gade_M9_binding_table.json          NaN   \n",
       "5                          M9|gadw_M9_binding_table.json          NaN   \n",
       "6                          M9|gadx_M9_binding_table.json          NaN   \n",
       "7                        KCl|kdpe_KCl_binding_table.json    GSE143856   \n",
       "8                        M9P|phob_M9P_binding_table.json    GSE143856   \n",
       "9      dpd|rpob_dpd_binding_table.json;fe|rpob_fe_bin...          NaN   \n",
       "10                         M9|ybao_M9_binding_table.json          NaN   \n",
       "11                         M9|ybaq_M9_binding_table.json          NaN   \n",
       "12     ph5|ydci_ph5_binding_table.json;ph7|ydci_ph7_b...          NaN   \n",
       "13                         M9|yddm_M9_binding_table.json          NaN   \n",
       "14                         M9|yheo_M9_binding_table.json          NaN   \n",
       "15                         M9|yiag_M9_binding_table.json          NaN   \n",
       "16                         M9|yiep_M9_binding_table.json          NaN   \n",
       "17                       glc|yihl_glc_binding_table.json          NaN   \n",
       "18                       glc|yihw_glc_binding_table.json          NaN   \n",
       "19                         M9|yjdc_M9_binding_table.json          NaN   \n",
       "20                       glc|yjhi_glc_binding_table.json          NaN   \n",
       "21                       glc|ynej_glc_binding_table.json          NaN   \n",
       "22                       glc|ynfl_glc_binding_table.json          NaN   \n",
       "23                       glc|ypdc_glc_binding_table.json          NaN   \n",
       "24                         M9|yqhc_M9_binding_table.json          NaN   \n",
       "25                   ZnCl2|zrar_ZnCl2_binding_table.json          NaN   \n",
       "26                          Fe|fur_Fe_binding_table.json          NaN   \n",
       "27                          Fe|fur_Fe_binding_table.json          NaN   \n",
       "28                          Fe|fur_Fe_binding_table.json          NaN   \n",
       "29     expo|dps_expo_binding_table.json;sta|dps_sta_b...          NaN   \n",
       "30     expo|fis_expo_binding_table.json;sta|fis_sta_b...          NaN   \n",
       "31     expo|gyra_expo_binding_table.json;sta|gyra_sta...          NaN   \n",
       "32     expo|gyrb_expo_binding_table.json;sta|gyrb_sta...          NaN   \n",
       "33     expo|h-ns_expo_binding_table.json;sta|h-ns_sta...          NaN   \n",
       "34     expo|hupa_expo_binding_table.json;sta|hupa_sta...          NaN   \n",
       "35     expo|hupb_expo_binding_table.json;sta|hupb_sta...          NaN   \n",
       "36     expo|infa_expo_binding_table.json;sta|infa_sta...          NaN   \n",
       "37     expo|mukb_expo_binding_table.json;sta|mukb_sta...          NaN   \n",
       "38     expo|rdgc_expo_binding_table.json;sta|rdgc_sta...          NaN   \n",
       "39     expo|rob_expo_binding_table.json;sta|rob_sta_b...          NaN   \n",
       "40     expo|stpa_expo_binding_table.json;sta|stpa_sta...          NaN   \n",
       "41     expo|topa_expo_binding_table.json;sta|topa_sta...          NaN   \n",
       "42     expo|tus_expo_binding_table.json;sta|tus_sta_b...          NaN   \n",
       "43               nitrate|narl_nitrate_binding_table.json          NaN   \n",
       "44         putrescine|puur_putrescine_binding_table.json          NaN   \n",
       "45                                                   NaN          NaN   \n",
       "46     MG1655 ace|rpob (cra and crp ko exps)_MG1655 a...          NaN   \n",
       "47                                                   NaN  E-MTAB-2457   \n",
       "48                                                   NaN  E-MTAB-2457   \n",
       "49                                                   NaN  E-MTAB-2457   \n",
       "50                                                   NaN     GSE41195   \n",
       "51                                                   NaN     GSE41195   \n",
       "\n",
       "                                                doi           BWcond1_1_name  \\\n",
       "index                                                                          \n",
       "0         https://doi.org/10.1128/mSystems.00980-20            bio-rep1 – R1   \n",
       "1         https://doi.org/10.1128/mSystems.00980-20                       R1   \n",
       "2                                               NaN                 Glu – R1   \n",
       "3                                               NaN                  Fe – R1   \n",
       "4                                               NaN                       R1   \n",
       "5                                               NaN                       R1   \n",
       "6                                               NaN                       R1   \n",
       "7         https://doi.org/10.1128/mSystems.00980-20                       R1   \n",
       "8         https://doi.org/10.1128/mSystems.00980-20            bio-rep1 – R1   \n",
       "9                                               NaN                  Fe – R1   \n",
       "10                                              NaN                       R1   \n",
       "11                                              NaN                       R1   \n",
       "12                                              NaN                 pH5 – R1   \n",
       "13                                              NaN                       R1   \n",
       "14                                              NaN                       R1   \n",
       "15                                              NaN                       R1   \n",
       "16                                              NaN                       R1   \n",
       "17                                              NaN                       R1   \n",
       "18                                              NaN                       R1   \n",
       "19                                              NaN                       R1   \n",
       "20                                              NaN                       R1   \n",
       "21                                              NaN                       R1   \n",
       "22                                              NaN                       R1   \n",
       "23                                              NaN                       R1   \n",
       "24                                              NaN                       R1   \n",
       "25                                              NaN            bio-rep1 – R1   \n",
       "26                                              NaN                       R1   \n",
       "27                                              NaN                       R1   \n",
       "28                                              NaN                       R1   \n",
       "29                                              NaN  exponential growth – R1   \n",
       "30                                              NaN  exponential growth – R1   \n",
       "31                                              NaN  exponential growth – R1   \n",
       "32                                              NaN  exponential growth – R1   \n",
       "33                                              NaN  exponential growth – R1   \n",
       "34                                              NaN  exponential growth – R1   \n",
       "35                                              NaN  exponential growth – R1   \n",
       "36                                              NaN  exponential growth – R1   \n",
       "37                                              NaN  exponential growth – R1   \n",
       "38                                              NaN  exponential growth – R1   \n",
       "39                                              NaN  exponential growth – R1   \n",
       "40                                              NaN  exponential growth – R1   \n",
       "41                                              NaN  exponential growth – R1   \n",
       "42                                              NaN  exponential growth – R1   \n",
       "43                                              NaN                       R1   \n",
       "44                                              NaN                       R1   \n",
       "45                                              NaN                       R1   \n",
       "46                                              NaN        control (WT) – R1   \n",
       "47     https://doi.org/10.1371/journal.pgen.1004649                       R1   \n",
       "48     https://doi.org/10.1371/journal.pgen.1004649                       R1   \n",
       "49     https://doi.org/10.1371/journal.pgen.1004649                       R1   \n",
       "50     https://doi.org/10.1371/journal.pgen.1003565           anaerobic – R1   \n",
       "51     https://doi.org/10.1371/journal.pgen.1003565             aerobic – R1   \n",
       "\n",
       "                                             BWcond1_1  \\\n",
       "index                                                    \n",
       "0                                    BaeR_R1_S31_R1.bw   \n",
       "1                                      CpxRR1_S1_R1.bw   \n",
       "2                                         cra_glu_1.bw   \n",
       "3                                          fur_fe_1.bw   \n",
       "4                                            gade_1.bw   \n",
       "5                                            gadw_1.bw   \n",
       "6                                            gadx_1.bw   \n",
       "7                                   KdpeE_R2_S34_R1.bw   \n",
       "8                                     PhoB_1_S16_R1.bw   \n",
       "9                                          rpob_fe1.bw   \n",
       "10                                           ybao_1.bw   \n",
       "11                                           ybaq_1.bw   \n",
       "12                                       ydci_ph5_1.bw   \n",
       "13                                           yddm_1.bw   \n",
       "14                                           yheo_1.bw   \n",
       "15                                           yiag_1.bw   \n",
       "16                                           yiep_1.bw   \n",
       "17                                           yihl_1.bw   \n",
       "18                                           yihw_1.bw   \n",
       "19                                           yjdc_1.bw   \n",
       "20                                           yjhi_1.bw   \n",
       "21                                           ynej_1.bw   \n",
       "22                                           ynfl_1.bw   \n",
       "23                                           ypdc_1.bw   \n",
       "24                                           yqhc_1.bw   \n",
       "25                                     ZraR_1_S3_R1.bw   \n",
       "26                                  edl933furfinal1.bw   \n",
       "27                                     smsfurfinal1.bw   \n",
       "28                                  umn026furfinal1.bw   \n",
       "29                                       dps_expo_1.bw   \n",
       "30                                       fis_expo_1.bw   \n",
       "31                                      gyra_expo_1.bw   \n",
       "32                                      gyrb_expo_1.bw   \n",
       "33                                       hns_expo_1.bw   \n",
       "34                                      hupa_expo_1.bw   \n",
       "35                                      hupb_expo_1.bw   \n",
       "36                                      infa_expo_1.bw   \n",
       "37                                      mukb_expo_1.bw   \n",
       "38                                      rdgc_expo_1.bw   \n",
       "39                                       rob_expo_1.bw   \n",
       "40                                      stpa_expo_1.bw   \n",
       "41                                      topa_expo_1.bw   \n",
       "42                                       tus_expo_1.bw   \n",
       "43                                   narl_nitrate_1.bw   \n",
       "44                                puur_putrescine_1.bw   \n",
       "45                                     rcsa_oxida_1.bw   \n",
       "46                                 mg1655_ace_rpob1.bw   \n",
       "47                                        ERR475432.bw   \n",
       "48                                        ERR475429.bw   \n",
       "49                                        ERR475428.bw   \n",
       "50           GSM1010219_FNR_IP_ChIP-seq_Anaerobic_A.bw   \n",
       "51     GSM1010221_Sigma70_IP_ChIP-seq_Aerobic_A_SET.bw   \n",
       "\n",
       "                BWcond1_2_name  \\\n",
       "index                            \n",
       "0                bio-rep1 – R2   \n",
       "1                           R2   \n",
       "2                     Glu – R2   \n",
       "3                      Fe – R2   \n",
       "4                           R2   \n",
       "5                           R2   \n",
       "6                           R2   \n",
       "7                           R2   \n",
       "8                bio-rep1 – R2   \n",
       "9                      Fe – R2   \n",
       "10                          R2   \n",
       "11                          R2   \n",
       "12                    pH5 – R2   \n",
       "13                          R2   \n",
       "14                          R2   \n",
       "15                          R2   \n",
       "16                          R2   \n",
       "17                          R2   \n",
       "18                          R2   \n",
       "19                          R2   \n",
       "20                          R2   \n",
       "21                          R2   \n",
       "22                          R2   \n",
       "23                          R2   \n",
       "24                          R2   \n",
       "25               bio-rep1 – R2   \n",
       "26                          R2   \n",
       "27                          R2   \n",
       "28                          R2   \n",
       "29     exponential growth – R2   \n",
       "30     exponential growth – R2   \n",
       "31     exponential growth – R2   \n",
       "32     exponential growth – R2   \n",
       "33     exponential growth – R2   \n",
       "34     exponential growth – R2   \n",
       "35     exponential growth – R2   \n",
       "36     exponential growth – R2   \n",
       "37     exponential growth – R2   \n",
       "38     exponential growth – R2   \n",
       "39     exponential growth – R2   \n",
       "40     exponential growth – R2   \n",
       "41     exponential growth – R2   \n",
       "42     exponential growth – R2   \n",
       "43                          R2   \n",
       "44                          R2   \n",
       "45                          R2   \n",
       "46           control (WT) – R2   \n",
       "47                          R2   \n",
       "48                          R2   \n",
       "49                          R2   \n",
       "50              anaerobic – R2   \n",
       "51                aerobic – R2   \n",
       "\n",
       "                                             BWcond1_2  \\\n",
       "index                                                    \n",
       "0                                    BaeR_R1_S31_R2.bw   \n",
       "1                                      CpxRR2_S2_R2.bw   \n",
       "2                                         cra_glu_2.bw   \n",
       "3                                          fur_fe_2.bw   \n",
       "4                                            gade_2.bw   \n",
       "5                                            gadw_2.bw   \n",
       "6                                            gadx_2.bw   \n",
       "7                                   KdpeE_R2_S34_R2.bw   \n",
       "8                                     PhoB_1_S16_R2.bw   \n",
       "9                                          rpob_fe2.bw   \n",
       "10                                           ybao_2.bw   \n",
       "11                                           ybaq_2.bw   \n",
       "12                                       ydci_ph5_2.bw   \n",
       "13                                           yddm_2.bw   \n",
       "14                                           yheo_2.bw   \n",
       "15                                           yiag_2.bw   \n",
       "16                                           yiep_2.bw   \n",
       "17                                           yihl_2.bw   \n",
       "18                                           yihw_2.bw   \n",
       "19                                           yjdc_2.bw   \n",
       "20                                           yjhi_2.bw   \n",
       "21                                           ynej_2.bw   \n",
       "22                                           ynfl_2.bw   \n",
       "23                                           ypdc_2.bw   \n",
       "24                                           yqhc_2.bw   \n",
       "25                                     ZraR_1_S3_R2.bw   \n",
       "26                                  edl933furfinal2.bw   \n",
       "27                                     smsfurfinal2.bw   \n",
       "28                                  umn026furfinal2.bw   \n",
       "29                                       dps_expo_2.bw   \n",
       "30                                       fis_expo_2.bw   \n",
       "31                                      gyra_expo_2.bw   \n",
       "32                                      gyrb_expo_2.bw   \n",
       "33                                       hns_expo_2.bw   \n",
       "34                                      hupa_expo_2.bw   \n",
       "35                                      hupb_expo_2.bw   \n",
       "36                                      infa_expo_2.bw   \n",
       "37                                      mukb_expo_2.bw   \n",
       "38                                      rdgc_expo_2.bw   \n",
       "39                                       rob_expo_2.bw   \n",
       "40                                      stpa_expo_2.bw   \n",
       "41                                      topa_expo_2.bw   \n",
       "42                                       tus_expo_2.bw   \n",
       "43                                   narl_nitrate_2.bw   \n",
       "44                                puur_putrescine_2.bw   \n",
       "45                                     rcsa_oxida_2.bw   \n",
       "46                                 mg1655_ace_rpob2.bw   \n",
       "47                                        ERR475433.bw   \n",
       "48                                        ERR475430.bw   \n",
       "49                                        ERR475431.bw   \n",
       "50           GSM1010220_FNR_IP_ChIP-seq_Anaerobic_B.bw   \n",
       "51     GSM1072326_Sigma70_IP_ChIP-seq_Aerobic_B_SET.bw   \n",
       "\n",
       "               BWcond2_1_name  \\\n",
       "index                           \n",
       "0               bio-rep2 – R1   \n",
       "1                         NaN   \n",
       "2                    Fru – R1   \n",
       "3                    DPD – R1   \n",
       "4                         NaN   \n",
       "5                         NaN   \n",
       "6                         NaN   \n",
       "7                         NaN   \n",
       "8               bio-rep2 – R1   \n",
       "9                    DPD – R1   \n",
       "10                        NaN   \n",
       "11                        NaN   \n",
       "12                   pH7 – R1   \n",
       "13                        NaN   \n",
       "14                        NaN   \n",
       "15                        NaN   \n",
       "16                   3HP – R1   \n",
       "17                        NaN   \n",
       "18                        NaN   \n",
       "19                        NaN   \n",
       "20                        NaN   \n",
       "21                        NaN   \n",
       "22                        NaN   \n",
       "23                        NaN   \n",
       "24                        NaN   \n",
       "25              bio-rep2 – R1   \n",
       "26                        NaN   \n",
       "27                        NaN   \n",
       "28                        NaN   \n",
       "29     stationary growth – R1   \n",
       "30     stationary growth – R1   \n",
       "31     stationary growth – R1   \n",
       "32     stationary growth – R1   \n",
       "33     stationary growth – R1   \n",
       "34     stationary growth – R1   \n",
       "35     stationary growth – R1   \n",
       "36     stationary growth – R1   \n",
       "37     stationary growth – R1   \n",
       "38     stationary growth – R1   \n",
       "39     stationary growth – R1   \n",
       "40     stationary growth – R1   \n",
       "41     stationary growth – R1   \n",
       "42     stationary growth – R1   \n",
       "43                        NaN   \n",
       "44                        NaN   \n",
       "45                        NaN   \n",
       "46                cra-KO – R1   \n",
       "47                        NaN   \n",
       "48                        NaN   \n",
       "49                        NaN   \n",
       "50                        NaN   \n",
       "51             anaerobic – R1   \n",
       "\n",
       "                                               BWcond2_1  \\\n",
       "index                                                      \n",
       "0                                     BaerR_R2_S32_R1.bw   \n",
       "1                                                    NaN   \n",
       "2                                           cra_fru_1.bw   \n",
       "3                                           fur_dpd_1.bw   \n",
       "4                                                    NaN   \n",
       "5                                                    NaN   \n",
       "6                                                    NaN   \n",
       "7                                                    NaN   \n",
       "8                                        PhoB_2_S6_R1.bw   \n",
       "9                                           rpob_dpd1.bw   \n",
       "10                                                   NaN   \n",
       "11                                                   NaN   \n",
       "12                                         ydci_ph7_1.bw   \n",
       "13                                                   NaN   \n",
       "14                                                   NaN   \n",
       "15                                                   NaN   \n",
       "16                                         yiep_3hp_1.bw   \n",
       "17                                                   NaN   \n",
       "18                                                   NaN   \n",
       "19                                                   NaN   \n",
       "20                                                   NaN   \n",
       "21                                                   NaN   \n",
       "22                                                   NaN   \n",
       "23                                                   NaN   \n",
       "24                                                   NaN   \n",
       "25                                       ZraR_2_S4_R1.bw   \n",
       "26                                                   NaN   \n",
       "27                                                   NaN   \n",
       "28                                                   NaN   \n",
       "29                                          dps_sta_1.bw   \n",
       "30                                          fis_sta_1.bw   \n",
       "31                                         gyra_sta_1.bw   \n",
       "32                                         gyrb_sta_1.bw   \n",
       "33                                          hns_sta_1.bw   \n",
       "34                                         hupa_sta_1.bw   \n",
       "35                                         hupb_sta_1.bw   \n",
       "36                                         infa_sta_1.bw   \n",
       "37                                         mukb_sta_1.bw   \n",
       "38                                         rdgc_sta_1.bw   \n",
       "39                                          rob_sta_1.bw   \n",
       "40                                         stpa_sta_1.bw   \n",
       "41                                         topa_sta_1.bw   \n",
       "42                                          tus_sta_1.bw   \n",
       "43                                                   NaN   \n",
       "44                                                   NaN   \n",
       "45                                                   NaN   \n",
       "46                                   cra_ko_ace_rpob1.bw   \n",
       "47                                                   NaN   \n",
       "48                                                   NaN   \n",
       "49                                                   NaN   \n",
       "50                                                   NaN   \n",
       "51     GSM1010222_Sigma70_IP_ChIP-seq_Anaerobic_A_SET.bw   \n",
       "\n",
       "               BWcond2_2_name  \\\n",
       "index                           \n",
       "0               bio-rep2 – R2   \n",
       "1                         NaN   \n",
       "2                    Fru – R2   \n",
       "3                    DPD – R2   \n",
       "4                         NaN   \n",
       "5                         NaN   \n",
       "6                         NaN   \n",
       "7                         NaN   \n",
       "8               bio-rep2 – R2   \n",
       "9                    DPD – R2   \n",
       "10                        NaN   \n",
       "11                        NaN   \n",
       "12                   pH7 – R2   \n",
       "13                        NaN   \n",
       "14                        NaN   \n",
       "15                        NaN   \n",
       "16                   3HP – R2   \n",
       "17                        NaN   \n",
       "18                        NaN   \n",
       "19                        NaN   \n",
       "20                        NaN   \n",
       "21                        NaN   \n",
       "22                        NaN   \n",
       "23                        NaN   \n",
       "24                        NaN   \n",
       "25                        NaN   \n",
       "26                        NaN   \n",
       "27                        NaN   \n",
       "28                        NaN   \n",
       "29     stationary growth – R2   \n",
       "30     stationary growth – R2   \n",
       "31     stationary growth – R2   \n",
       "32     stationary growth – R2   \n",
       "33     stationary growth – R2   \n",
       "34     stationary growth – R2   \n",
       "35     stationary growth – R2   \n",
       "36     stationary growth – R2   \n",
       "37     stationary growth – R2   \n",
       "38     stationary growth – R2   \n",
       "39     stationary growth – R2   \n",
       "40     stationary growth – R2   \n",
       "41     stationary growth – R2   \n",
       "42     stationary growth – R2   \n",
       "43                        NaN   \n",
       "44                        NaN   \n",
       "45                        NaN   \n",
       "46                cra-KO – R2   \n",
       "47                        NaN   \n",
       "48                        NaN   \n",
       "49                        NaN   \n",
       "50                        NaN   \n",
       "51             anaerobic – R2   \n",
       "\n",
       "                                               BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                                     \n",
       "0                                     BaerR_R2_S32_R2.bw            NaN   \n",
       "1                                                    NaN            NaN   \n",
       "2                                           cra_fru_2.bw       Gal – R1   \n",
       "3                                           fur_dpd_2.bw            NaN   \n",
       "4                                                    NaN            NaN   \n",
       "5                                                    NaN            NaN   \n",
       "6                                                    NaN            NaN   \n",
       "7                                                    NaN            NaN   \n",
       "8                                        PhoB_2_S6_R2.bw            NaN   \n",
       "9                                           rpob_dpd2.bw            NaN   \n",
       "10                                                   NaN            NaN   \n",
       "11                                                   NaN            NaN   \n",
       "12                                         ydci_ph7_2.bw       pH8 – R1   \n",
       "13                                                   NaN            NaN   \n",
       "14                                                   NaN            NaN   \n",
       "15                                                   NaN            NaN   \n",
       "16                                         yiep_3hp_2.bw            NaN   \n",
       "17                                                   NaN            NaN   \n",
       "18                                                   NaN            NaN   \n",
       "19                                                   NaN            NaN   \n",
       "20                                                   NaN            NaN   \n",
       "21                                                   NaN            NaN   \n",
       "22                                                   NaN            NaN   \n",
       "23                                                   NaN            NaN   \n",
       "24                                                   NaN            NaN   \n",
       "25                                                   NaN            NaN   \n",
       "26                                                   NaN            NaN   \n",
       "27                                                   NaN            NaN   \n",
       "28                                                   NaN            NaN   \n",
       "29                                          dps_sta_2.bw            NaN   \n",
       "30                                          fis_sta_2.bw            NaN   \n",
       "31                                         gyra_sta_2.bw            NaN   \n",
       "32                                         gyrb_sta_2.bw            NaN   \n",
       "33                                          hns_sta_2.bw            NaN   \n",
       "34                                         hupa_sta_2.bw            NaN   \n",
       "35                                         hupb_sta_2.bw            NaN   \n",
       "36                                         infa_sta_2.bw            NaN   \n",
       "37                                         mukb_sta_2.bw            NaN   \n",
       "38                                         rdgc_sta_2.bw            NaN   \n",
       "39                                          rob_sta_2.bw            NaN   \n",
       "40                                         stpa_sta_2.bw            NaN   \n",
       "41                                         topa_sta_2.bw            NaN   \n",
       "42                                          tus_sta_2.bw            NaN   \n",
       "43                                                   NaN            NaN   \n",
       "44                                                   NaN            NaN   \n",
       "45                                                   NaN            NaN   \n",
       "46                                   cra_ko_ace_rpob2.bw    crp-KO – R1   \n",
       "47                                                   NaN            NaN   \n",
       "48                                                   NaN            NaN   \n",
       "49                                                   NaN            NaN   \n",
       "50                                                   NaN            NaN   \n",
       "51     GSM1072327_Sigma70_IP_ChIP-seq_Anaerobic_B_SET.bw            NaN   \n",
       "\n",
       "                 BWcond3_1 BWcond3_2_name            BWcond3_2 BWcond4_1_name  \\\n",
       "index                                                                           \n",
       "0                      NaN            NaN                  NaN            NaN   \n",
       "1                      NaN            NaN                  NaN            NaN   \n",
       "2             cra_gal_1.bw       Gal – R2         cra_gal_2.bw       Ace – R1   \n",
       "3                      NaN            NaN                  NaN            NaN   \n",
       "4                      NaN            NaN                  NaN            NaN   \n",
       "5                      NaN            NaN                  NaN            NaN   \n",
       "6                      NaN            NaN                  NaN            NaN   \n",
       "7                      NaN            NaN                  NaN            NaN   \n",
       "8                      NaN            NaN                  NaN            NaN   \n",
       "9                      NaN            NaN                  NaN            NaN   \n",
       "10                     NaN            NaN                  NaN            NaN   \n",
       "11                     NaN            NaN                  NaN            NaN   \n",
       "12           ydci_ph8_1.bw       pH8 – R2        ydci_ph8_2.bw            NaN   \n",
       "13                     NaN            NaN                  NaN            NaN   \n",
       "14                     NaN            NaN                  NaN            NaN   \n",
       "15                     NaN            NaN                  NaN            NaN   \n",
       "16                     NaN            NaN                  NaN            NaN   \n",
       "17                     NaN            NaN                  NaN            NaN   \n",
       "18                     NaN            NaN                  NaN            NaN   \n",
       "19                     NaN            NaN                  NaN            NaN   \n",
       "20                     NaN            NaN                  NaN            NaN   \n",
       "21                     NaN            NaN                  NaN            NaN   \n",
       "22                     NaN            NaN                  NaN            NaN   \n",
       "23                     NaN            NaN                  NaN            NaN   \n",
       "24                     NaN            NaN                  NaN            NaN   \n",
       "25                     NaN            NaN                  NaN            NaN   \n",
       "26                     NaN            NaN                  NaN            NaN   \n",
       "27                     NaN            NaN                  NaN            NaN   \n",
       "28                     NaN            NaN                  NaN            NaN   \n",
       "29                     NaN            NaN                  NaN            NaN   \n",
       "30                     NaN            NaN                  NaN            NaN   \n",
       "31                     NaN            NaN                  NaN            NaN   \n",
       "32                     NaN            NaN                  NaN            NaN   \n",
       "33                     NaN            NaN                  NaN            NaN   \n",
       "34                     NaN            NaN                  NaN            NaN   \n",
       "35                     NaN            NaN                  NaN            NaN   \n",
       "36                     NaN            NaN                  NaN            NaN   \n",
       "37                     NaN            NaN                  NaN            NaN   \n",
       "38                     NaN            NaN                  NaN            NaN   \n",
       "39                     NaN            NaN                  NaN            NaN   \n",
       "40                     NaN            NaN                  NaN            NaN   \n",
       "41                     NaN            NaN                  NaN            NaN   \n",
       "42                     NaN            NaN                  NaN            NaN   \n",
       "43                     NaN            NaN                  NaN            NaN   \n",
       "44                     NaN            NaN                  NaN            NaN   \n",
       "45                     NaN            NaN                  NaN            NaN   \n",
       "46     crp_ko_ace_rpob1.bw    crp-KO – R2  crp_ko_ace_rpob2.bw            NaN   \n",
       "47                     NaN            NaN                  NaN            NaN   \n",
       "48                     NaN            NaN                  NaN            NaN   \n",
       "49                     NaN            NaN                  NaN            NaN   \n",
       "50                     NaN            NaN                  NaN            NaN   \n",
       "51                     NaN            NaN                  NaN            NaN   \n",
       "\n",
       "          BWcond4_1 BWcond4_2_name     BWcond4_2  \n",
       "index                                             \n",
       "0               NaN            NaN           NaN  \n",
       "1               NaN            NaN           NaN  \n",
       "2      cra_ace_1.bw       Ace – R2  cra_ace_2.bw  \n",
       "3               NaN            NaN           NaN  \n",
       "4               NaN            NaN           NaN  \n",
       "5               NaN            NaN           NaN  \n",
       "6               NaN            NaN           NaN  \n",
       "7               NaN            NaN           NaN  \n",
       "8               NaN            NaN           NaN  \n",
       "9               NaN            NaN           NaN  \n",
       "10              NaN            NaN           NaN  \n",
       "11              NaN            NaN           NaN  \n",
       "12              NaN            NaN           NaN  \n",
       "13              NaN            NaN           NaN  \n",
       "14              NaN            NaN           NaN  \n",
       "15              NaN            NaN           NaN  \n",
       "16              NaN            NaN           NaN  \n",
       "17              NaN            NaN           NaN  \n",
       "18              NaN            NaN           NaN  \n",
       "19              NaN            NaN           NaN  \n",
       "20              NaN            NaN           NaN  \n",
       "21              NaN            NaN           NaN  \n",
       "22              NaN            NaN           NaN  \n",
       "23              NaN            NaN           NaN  \n",
       "24              NaN            NaN           NaN  \n",
       "25              NaN            NaN           NaN  \n",
       "26              NaN            NaN           NaN  \n",
       "27              NaN            NaN           NaN  \n",
       "28              NaN            NaN           NaN  \n",
       "29              NaN            NaN           NaN  \n",
       "30              NaN            NaN           NaN  \n",
       "31              NaN            NaN           NaN  \n",
       "32              NaN            NaN           NaN  \n",
       "33              NaN            NaN           NaN  \n",
       "34              NaN            NaN           NaN  \n",
       "35              NaN            NaN           NaN  \n",
       "36              NaN            NaN           NaN  \n",
       "37              NaN            NaN           NaN  \n",
       "38              NaN            NaN           NaN  \n",
       "39              NaN            NaN           NaN  \n",
       "40              NaN            NaN           NaN  \n",
       "41              NaN            NaN           NaN  \n",
       "42              NaN            NaN           NaN  \n",
       "43              NaN            NaN           NaN  \n",
       "44              NaN            NaN           NaN  \n",
       "45              NaN            NaN           NaN  \n",
       "46              NaN            NaN           NaN  \n",
       "47              NaN            NaN           NaN  \n",
       "48              NaN            NaN           NaN  \n",
       "49              NaN            NaN           NaN  \n",
       "50              NaN            NaN           NaN  \n",
       "51              NaN            NaN           NaN  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length</th>\n",
       "      <th>operon</th>\n",
       "      <th>cog</th>\n",
       "      <th>start_codon_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>b0001</th>\n",
       "      <td>189</td>\n",
       "      <td>255</td>\n",
       "      <td>+</td>\n",
       "      <td>thrL</td>\n",
       "      <td>66</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0002</th>\n",
       "      <td>336</td>\n",
       "      <td>2799</td>\n",
       "      <td>+</td>\n",
       "      <td>thrA</td>\n",
       "      <td>2463</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>No COG Annotation</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0003</th>\n",
       "      <td>2800</td>\n",
       "      <td>3733</td>\n",
       "      <td>+</td>\n",
       "      <td>thrB</td>\n",
       "      <td>933</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>2800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0004</th>\n",
       "      <td>3733</td>\n",
       "      <td>5020</td>\n",
       "      <td>+</td>\n",
       "      <td>thrC</td>\n",
       "      <td>1287</td>\n",
       "      <td>thrLABC</td>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>3733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b0005</th>\n",
       "      <td>5233</td>\n",
       "      <td>5530</td>\n",
       "      <td>+</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>297</td>\n",
       "      <td>yaaX</td>\n",
       "      <td>Function unknown</td>\n",
       "      <td>5233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       start  stop strand gene_name  length   operon  \\\n",
       "b0001    189   255      +      thrL      66  thrLABC   \n",
       "b0002    336  2799      +      thrA    2463  thrLABC   \n",
       "b0003   2800  3733      +      thrB     933  thrLABC   \n",
       "b0004   3733  5020      +      thrC    1287  thrLABC   \n",
       "b0005   5233  5530      +      yaaX     297     yaaX   \n",
       "\n",
       "                                       cog  start_codon_pos  \n",
       "b0001                    No COG Annotation              189  \n",
       "b0002                    No COG Annotation              336  \n",
       "b0003  Amino acid transport and metabolism             2800  \n",
       "b0004  Amino acid transport and metabolism             3733  \n",
       "b0005                     Function unknown             5233  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain = 'NC_000913.3'\n",
    "DF_gene_info = pd.read_csv(org_folder+strain+'/annotation/gene_info.csv',index_col=0)\n",
    "DF_gene_info['start_codon_pos'] = [row.start if row.strand == '+' else row.stop for idx,row in DF_gene_info.iterrows()]\n",
    "TRN = pd.read_csv(org_folder+strain+'/annotation/trn.csv',index_col=0)\n",
    "\n",
    "DF_gene_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "baer\n",
      "cpxr\n",
      "cra\n",
      "fur\n",
      "gade\n",
      "gadw\n",
      "gadx\n",
      "kdpe\n",
      "phob\n",
      "rpob\n",
      "ybao\n",
      "ybaq\n",
      "ydci\n",
      "yddm\n",
      "yheo\n",
      "yiag\n",
      "yiep\n",
      "yihl\n",
      "yihw\n",
      "yjdc\n",
      "yjhi\n",
      "ynej\n",
      "ynfl\n",
      "ypdc\n",
      "yqhc\n",
      "zrar\n",
      "fur\n",
      "fur\n",
      "fur\n",
      "dps\n",
      "fis\n",
      "gyra\n",
      "gyrb\n",
      "h-ns\n",
      "hupa\n",
      "hupb\n",
      "infa\n",
      "mukb\n",
      "rdgc\n",
      "rob\n",
      "stpa\n",
      "topa\n",
      "tus\n",
      "narl\n",
      "puur\n",
      "rcsa\n",
      "rpob (cra and crp ko exps)\n",
      "flhc\n",
      "flhd\n",
      "flia\n",
      "fnr\n",
      "rpod\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF[TF_index].lower()\n",
    "    print(TF_name)\n",
    "    strain = TF_list.genome_id[TF_index]\n",
    "    \n",
    "    if strain != 'NC_000913.3':\n",
    "        continue\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(org_folder+strain+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"\\\\\")[-1] for i in curated_loc] # change split to \"/\" on linux\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list.loc[TF_index, 'binding tables'] = table_string\n",
    "    else:\n",
    "        TF_list.loc[TF_index, 'binding tables'] = \"NA\"\n",
    "    TF_list.loc[TF_index, 'num_cond'] = max(1,len(curated_loc))\n",
    "    \n",
    "    # write plot data files to folders\n",
    "    for file_index in range(len(curated_files)):\n",
    "        file = curated_files[file_index]\n",
    "        df = pd.read_csv(org_folder+strain+\"/curated_input/\"+file,index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "        df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "        peak_annot_DF = annotate_peaks(TF_list.TF[TF_index], curated_cond[file_index], df, 500, DF_gene_info)\n",
    "        final_annot_DF = validate_peak_info(peak_annot_DF,DF_gene_info)\n",
    "        final_annot_DF.to_json(org_folder+strain+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "        binding_width_gen(TF_list.TF[TF_index], df, org_folder+strain+\"/binding_widths/\")\n",
    "        peak_position_gen(TF_list.TF[TF_index], final_annot_DF, org_folder+strain+\"/positions/\")\n",
    "        \n",
    "        #venn diagrams\n",
    "        try:\n",
    "            if strain == 'NC_000913.3' and TF_list.TF[TF_index].split(\" \")[0] in TRN.index:\n",
    "                venn_files = Venn_data_gen(final_annot_DF, TRN)\n",
    "                venn_files.to_json(org_folder+strain+\"/venn/\"+TF_list.TF[TF_index]+'_venn.json',orient='records')\n",
    "        except:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Other Organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_gff_to_geneinfo(org, strain):\n",
    "    genes = pd.read_csv('../data'+'/'+org+'/'+strain+'/annotation/genes.gff',index_col=0, delimiter='\\t', header=None, \n",
    "                        names = ['genome', 'ref', 'type', 'start','stop','.', 'strand', '-','details'])\n",
    "    genes[\"locus_tag\"] = [gene_detail[0].split(\" \")[1] for gene_detail in genes.details.str.split(\"; \")]\n",
    "    genes.set_index('locus_tag', inplace=True)\n",
    "    genes = genes.drop(['.', '-','ref','type'], axis=1)\n",
    "\n",
    "    return(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "def closest_locus(peak_start, peak_end, gene_info):\n",
    "    min_name = ''\n",
    "    min_value = float('inf')\n",
    "    for i in range(len(gene_info)):\n",
    "        if abs(peak_start - gene_info.stop[i]) < min_value:\n",
    "            min_name = gene_info.index[i]\n",
    "            min_value = abs(peak_start - gene_info.stop[i])\n",
    "        if abs(peak_end - gene_info.start[i]) < min_value:\n",
    "            min_name = gene_info.index[i]\n",
    "            min_value = abs(peak_end - gene_info.start[i])\n",
    "\n",
    "    return(min_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_table_gen(TF_name, TF_condition, peak_df, gene_info):\n",
    "    \n",
    "    table = peak_df.copy()\n",
    "    table[\"index\"] = [TF_name+'-'+str(i+1) for i in range(len(table))]\n",
    "    table[\"condition\"] = [TF_name.lower() + \" + \" + TF_condition for i in range(len(table))]\n",
    "    table.drop([\"condition_name\",\".\", \"ID\"], axis=1, inplace=True)\n",
    "    table[\"target_locus\"] = [closest_locus(table.binding_peak_start.iloc[i], \n",
    "                                           table.binding_peak_end.iloc[i],\n",
    "                                           gene_info) for i in range(len(table))]\n",
    "    table[\"target_genes\"] = table[\"target_locus\"]\n",
    "    table[\"closest_gene\"] = table[\"target_locus\"]\n",
    "\n",
    "    return(table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "def other_peak_position_gen(TF, final_annot_DF,out_dir, DF_gene_info): \n",
    "    peak_scatter_df = pd.DataFrame(columns=final_annot_DF.columns.to_list()+['gene'] +['normalized_dist'])\n",
    "    counter=0\n",
    "\n",
    "    for i in range(len(final_annot_DF)):\n",
    "        peak = final_annot_DF.iloc[i,:]\n",
    "        peak_center = np.mean([peak.binding_peak_start, peak.binding_peak_end])\n",
    "\n",
    "        for gene in peak.target_genes.split(\",\"):\n",
    "            gene_info = DF_gene_info[DF_gene_info.index == gene]\n",
    "            try:\n",
    "                if gene_info.strand.iloc[0] == \"-\":\n",
    "                    dist = float((gene_info.stop - peak_center)/(gene_info.stop-gene_info.start+1))\n",
    "                else:\n",
    "                    dist = float((peak_center - gene_info.start)/(gene_info.stop-gene_info.start+1))\n",
    "                peak_scatter_df.loc[counter] = peak.to_list()+[gene, dist]\n",
    "                counter+=1\n",
    "            except:\n",
    "                continue\n",
    "    peak_scatter_df['binding_peak_center'] = peak_scatter_df.binding_peak_start/2 + peak_scatter_df.binding_peak_end/2\n",
    "    peak_scatter_df.drop(['direction'], axis=1, inplace=True)\n",
    "    cols = ['index', 'condition', 'binding_peak_start', 'binding_peak_end', 'binding_peak_strength', 'target_locus', \n",
    "            'target_genes', 'gene', 'normalized_dist', 'binding_peak_center']\n",
    "    peak_scatter_df = peak_scatter_df[cols]\n",
    "    peak_scatter_df.to_csv(out_dir+TF+'_positions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_folder = \"../data/all_other/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>accession</th>\n",
       "      <th>doi</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>CP000647.1</td>\n",
       "      <td>k_pneumoniae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MGH78578</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>klebfurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>klebfurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>fe|fur_fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidafurfinal1_202003.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>M9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002947.4</td>\n",
       "      <td>p_putida</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>M9|sigma d_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>putidaSigmaD_202003.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>putidaSigmaD_202005.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>LT2</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2</td>\n",
       "      <td>Salmonella_Fur_Fe_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH5, MES</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>s_enterica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>LT2</td>\n",
       "      <td>MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MES5 – R1</td>\n",
       "      <td>salmone_ydcI_MES5_1.bw</td>\n",
       "      <td>MES5 – R2</td>\n",
       "      <td>salmone_ydcI_MES5_2.bw</td>\n",
       "      <td>MES5 – R3</td>\n",
       "      <td>salmone_ydcI_MES5_3.bw</td>\n",
       "      <td>MES5 – R4</td>\n",
       "      <td>salmone_ydcI_MES5_4.bw</td>\n",
       "      <td>ph5 – R1</td>\n",
       "      <td>salmone_ydcI_ph55_1.bw</td>\n",
       "      <td>ph5 – R2</td>\n",
       "      <td>salmone_ydcI_ph55_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>M9</td>\n",
       "      <td>Fe</td>\n",
       "      <td>AE014073.1</td>\n",
       "      <td>s_flexneri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>shigella_fur_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>shigella_fur_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>CP035369.1</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LAC</td>\n",
       "      <td>M9|cody_M9_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphLAC_new_cody1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphLAC_new_cody2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|sigs_RPMI_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M1 – R1</td>\n",
       "      <td>StaphTCH_sigs_M1.bw</td>\n",
       "      <td>M1 – R2</td>\n",
       "      <td>Staph_SigS_M1_R2.bw</td>\n",
       "      <td>M2 – R1</td>\n",
       "      <td>StaphTCH_sigs_M2.bw</td>\n",
       "      <td>M2 – R2</td>\n",
       "      <td>Staph_SigS_M2_R2.bw</td>\n",
       "      <td>M3 – R1</td>\n",
       "      <td>StaphTCH_sigs_M3.bw</td>\n",
       "      <td>M3 – R2</td>\n",
       "      <td>Staph_SigS_M3_R2.bw</td>\n",
       "      <td>M4 – R1</td>\n",
       "      <td>StaphTCH_sigs_M4.bw</td>\n",
       "      <td>M4 – R2</td>\n",
       "      <td>Staph_SigS_M4_R2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI</td>\n",
       "      <td>10% LB</td>\n",
       "      <td>NC_010079</td>\n",
       "      <td>s_aureus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>RPMI|vrar_RPMI_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>StaphTCH_vraR_7H7C11purified.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>StaphTCH_vraR_13F12F6purified.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>BHI</td>\n",
       "      <td>Fe</td>\n",
       "      <td>NZ_CP009712.1</td>\n",
       "      <td>y_pseudotuberculosis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2457T</td>\n",
       "      <td>Fe|fur_Fe_binding_table.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1</td>\n",
       "      <td>yersifurfinal1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>yersifurfinal2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FleQ</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742210.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742211.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GbdR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742212.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742213.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PchR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742214.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742215.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PhoB</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742216.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742217.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SphR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE128430</td>\n",
       "      <td>https://doi.org/10.1038/s41467-019-10778-w</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR8742218.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR8742219.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MvfR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776542.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776543.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SoxR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776544.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776545.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RpoN</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776546.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776547.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Vfr</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776548.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776549.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AlgR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776550.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776551.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ExsA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776552.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776553.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GacA</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776554.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776555.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MexT</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776558.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776559.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>WspR</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>PAO1</td>\n",
       "      <td>LB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_002516.1</td>\n",
       "      <td>p_aeruginosa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE65356</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv747</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1776560.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR1776561.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HrdB</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2)</td>\n",
       "      <td>MS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E-MTAB-6926</td>\n",
       "      <td>https://doi.org/10.1093/nar/gky1018</td>\n",
       "      <td>R1</td>\n",
       "      <td>ERR2639621.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>ERR2639622.bw</td>\n",
       "      <td>R3</td>\n",
       "      <td>ERR2639623.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NdgR</td>\n",
       "      <td>Streptomyces coelicolor</td>\n",
       "      <td>A3(2)</td>\n",
       "      <td>minimal media</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_003888.3</td>\n",
       "      <td>s_coelicolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE59010</td>\n",
       "      <td>https://doi.org/10.1186/s12864-015-1311-0</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1503874.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AmrZ</td>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>F113</td>\n",
       "      <td>SA (low Fe)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_016830.1</td>\n",
       "      <td>p_fluorescens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRP039494</td>\n",
       "      <td>https://doi.org/10.1186/1471-2164-15-237</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR1184483.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Lsr2</td>\n",
       "      <td>Streptomyces venezuelae</td>\n",
       "      <td>ATCC 10712</td>\n",
       "      <td>MS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_018750.1</td>\n",
       "      <td>s_venezuelae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE115439</td>\n",
       "      <td>https://doi.org/10.7554/eLife.47691</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR7282569.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LexA1</td>\n",
       "      <td>Leptospira interrogans</td>\n",
       "      <td>Copenhageni Fiocruz L1-130</td>\n",
       "      <td>EMJH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_005823.1</td>\n",
       "      <td>l_interrogans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>public_data</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSE73688</td>\n",
       "      <td>https://doi.org/10.1093/nar/gkv1536</td>\n",
       "      <td>R1</td>\n",
       "      <td>SRR2549977.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>SRR2549979.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF                     Organism                      Strain  \\\n",
       "index                                                                     \n",
       "0          Fur        Klebsiella pneumoniae                   MGH 78578   \n",
       "0          Fur           Pseudomonas putida                      KT2440   \n",
       "1      Sigma D           Pseudomonas putida                      KT2440   \n",
       "0          Fur          Salmonella enterica             Typhimurium LT2   \n",
       "1         YdcI          Salmonella enterica             Typhimurium LT2   \n",
       "0          Fur            Shigella flexneri                   2a, 2457T   \n",
       "0         CodY        Staphylococcus aureus              LAC chromosome   \n",
       "1         SigS        Staphylococcus aureus              USA300_TCH1516   \n",
       "2         VraR        Staphylococcus aureus              USA300_TCH1516   \n",
       "0          Fur  Yersinia pseudotuberculosis                    IP 32953   \n",
       "0         FleQ       Pseudomonas aeruginosa                        PAO1   \n",
       "1         GbdR       Pseudomonas aeruginosa                        PAO1   \n",
       "2         PchR       Pseudomonas aeruginosa                        PAO1   \n",
       "3         PhoB       Pseudomonas aeruginosa                        PAO1   \n",
       "4         SphR       Pseudomonas aeruginosa                        PAO1   \n",
       "5         MvfR       Pseudomonas aeruginosa                        PAO1   \n",
       "6         SoxR       Pseudomonas aeruginosa                        PAO1   \n",
       "7         RpoN       Pseudomonas aeruginosa                        PAO1   \n",
       "8          Vfr       Pseudomonas aeruginosa                        PAO1   \n",
       "9         AlgR       Pseudomonas aeruginosa                        PAO1   \n",
       "10        ExsA       Pseudomonas aeruginosa                        PAO1   \n",
       "11        GacA       Pseudomonas aeruginosa                        PAO1   \n",
       "12        MexT       Pseudomonas aeruginosa                        PAO1   \n",
       "13        WspR       Pseudomonas aeruginosa                        PAO1   \n",
       "0         HrdB      Streptomyces coelicolor                       A3(2)   \n",
       "1         NdgR      Streptomyces coelicolor                       A3(2)   \n",
       "0         AmrZ      Pseudomonas fluorescens                        F113   \n",
       "0         Lsr2      Streptomyces venezuelae                  ATCC 10712   \n",
       "0        LexA1       Leptospira interrogans  Copenhageni Fiocruz L1-130   \n",
       "\n",
       "               Media Supplement      genome_id           organism_id  \\\n",
       "index                                                                  \n",
       "0                 M9         Fe     CP000647.1          k_pneumoniae   \n",
       "0                 M9         Fe    NC_002947.4              p_putida   \n",
       "1                 M9        NaN    NC_002947.4              p_putida   \n",
       "0                 M9         Fe    NC_003197.2            s_enterica   \n",
       "1                 M9   pH5, MES    NC_003197.2            s_enterica   \n",
       "0                 M9         Fe     AE014073.1            s_flexneri   \n",
       "0               RPMI     10% LB     CP035369.1              s_aureus   \n",
       "1               RPMI     10% LB      NC_010079              s_aureus   \n",
       "2               RPMI     10% LB      NC_010079              s_aureus   \n",
       "0                BHI         Fe  NZ_CP009712.1  y_pseudotuberculosis   \n",
       "0                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "1                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "2                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "3                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "4                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "5                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "6                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "7                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "8                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "9                 LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "10                LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "11                LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "12                LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "13                LB        NaN    NC_002516.1          p_aeruginosa   \n",
       "0                 MS        NaN    NC_003888.3          s_coelicolor   \n",
       "1      minimal media        NaN    NC_003888.3          s_coelicolor   \n",
       "0        SA (low Fe)        NaN    NC_016830.1         p_fluorescens   \n",
       "0                 MS        NaN    NC_018750.1          s_venezuelae   \n",
       "0               EMJH        NaN    NC_005823.1         l_interrogans   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond         project  \\\n",
       "index                                                             \n",
       "0                    NaN            2         1        MGH78578   \n",
       "0                    NaN            1         1           2457T   \n",
       "1                    NaN            2         1           2457T   \n",
       "0                    NaN            1         1             LT2   \n",
       "1                    NaN            6         3             LT2   \n",
       "0                    NaN            2         1           2457T   \n",
       "0                    NaN            2         1             LAC   \n",
       "1                    NaN            8         1  USA300_TCH1516   \n",
       "2                    NaN            2         1  USA300_TCH1516   \n",
       "0                    NaN            2         1           2457T   \n",
       "0                    NaN            2         1     public_data   \n",
       "1                    NaN            2         1     public_data   \n",
       "2                    NaN            2         1     public_data   \n",
       "3                    NaN            2         1     public_data   \n",
       "4                    NaN            2         1     public_data   \n",
       "5                    NaN            2         1     public_data   \n",
       "6                    NaN            2         1     public_data   \n",
       "7                    NaN            2         1     public_data   \n",
       "8                    NaN            2         1     public_data   \n",
       "9                    NaN            2         1     public_data   \n",
       "10                   NaN            2         1     public_data   \n",
       "11                   NaN            2         1     public_data   \n",
       "12                   NaN            2         1     public_data   \n",
       "13                   NaN            2         1     public_data   \n",
       "0                    NaN            3         1     public_data   \n",
       "1                    NaN            1         1     public_data   \n",
       "0                    NaN            1         1     public_data   \n",
       "0                    NaN            2         1     public_data   \n",
       "0                    NaN            2         1     public_data   \n",
       "\n",
       "                                          binding tables    accession  \\\n",
       "index                                                                   \n",
       "0                           Fe|fur_Fe_binding_table.json          NaN   \n",
       "0                           fe|fur_fe_binding_table.json          NaN   \n",
       "1                       M9|sigma d_M9_binding_table.json          NaN   \n",
       "0                           Fe|fur_Fe_binding_table.json          NaN   \n",
       "1      MES2|ydci_MES2_binding_table.json;MES5|ydci_ME...          NaN   \n",
       "0                           Fe|fur_Fe_binding_table.json          NaN   \n",
       "0                          M9|cody_M9_binding_table.json          NaN   \n",
       "1                      RPMI|sigs_RPMI_binding_table.json          NaN   \n",
       "2                      RPMI|vrar_RPMI_binding_table.json          NaN   \n",
       "0                           Fe|fur_Fe_binding_table.json          NaN   \n",
       "0                                                    NaN    GSE128430   \n",
       "1                                                    NaN    GSE128430   \n",
       "2                                                    NaN    GSE128430   \n",
       "3                                                    NaN    GSE128430   \n",
       "4                                                    NaN    GSE128430   \n",
       "5                                                    NaN     GSE65356   \n",
       "6                                                    NaN     GSE65356   \n",
       "7                                                    NaN     GSE65356   \n",
       "8                                                    NaN     GSE65356   \n",
       "9                                                    NaN     GSE65356   \n",
       "10                                                   NaN     GSE65356   \n",
       "11                                                   NaN     GSE65356   \n",
       "12                                                   NaN     GSE65356   \n",
       "13                                                   NaN     GSE65356   \n",
       "0                                                    NaN  E-MTAB-6926   \n",
       "1                                                    NaN     GSE59010   \n",
       "0                                                    NaN    SRP039494   \n",
       "0                                                    NaN    GSE115439   \n",
       "0                                                    NaN     GSE73688   \n",
       "\n",
       "                                              doi BWcond1_1_name  \\\n",
       "index                                                              \n",
       "0                                             NaN             R1   \n",
       "0                                             NaN             R1   \n",
       "1                                             NaN             R1   \n",
       "0                                             NaN            NaN   \n",
       "1                                             NaN      MES5 – R1   \n",
       "0                                             NaN             R1   \n",
       "0                                             NaN             R1   \n",
       "1                                             NaN        M1 – R1   \n",
       "2                                             NaN             R1   \n",
       "0                                             NaN             R1   \n",
       "0      https://doi.org/10.1038/s41467-019-10778-w             R1   \n",
       "1      https://doi.org/10.1038/s41467-019-10778-w             R1   \n",
       "2      https://doi.org/10.1038/s41467-019-10778-w             R1   \n",
       "3      https://doi.org/10.1038/s41467-019-10778-w             R1   \n",
       "4      https://doi.org/10.1038/s41467-019-10778-w             R1   \n",
       "5              https://doi.org/10.1093/nar/gkv747             R1   \n",
       "6              https://doi.org/10.1093/nar/gkv747             R1   \n",
       "7              https://doi.org/10.1093/nar/gkv747             R1   \n",
       "8              https://doi.org/10.1093/nar/gkv747             R1   \n",
       "9              https://doi.org/10.1093/nar/gkv747             R1   \n",
       "10             https://doi.org/10.1093/nar/gkv747             R1   \n",
       "11             https://doi.org/10.1093/nar/gkv747             R1   \n",
       "12             https://doi.org/10.1093/nar/gkv747             R1   \n",
       "13             https://doi.org/10.1093/nar/gkv747             R1   \n",
       "0             https://doi.org/10.1093/nar/gky1018             R1   \n",
       "1       https://doi.org/10.1186/s12864-015-1311-0             R1   \n",
       "0        https://doi.org/10.1186/1471-2164-15-237             R1   \n",
       "0             https://doi.org/10.7554/eLife.47691             R1   \n",
       "0             https://doi.org/10.1093/nar/gkv1536             R1   \n",
       "\n",
       "                             BWcond1_1 BWcond1_2_name  \\\n",
       "index                                                   \n",
       "0                     klebfurfinal1.bw             R2   \n",
       "0            putidafurfinal1_202003.bw            NaN   \n",
       "1               putidaSigmaD_202003.bw             R2   \n",
       "0                                  NaN             R2   \n",
       "1               salmone_ydcI_MES5_1.bw      MES5 – R2   \n",
       "0                    shigella_fur_1.bw             R2   \n",
       "0                StaphLAC_new_cody1.bw             R2   \n",
       "1                  StaphTCH_sigs_M1.bw        M1 – R2   \n",
       "2      StaphTCH_vraR_7H7C11purified.bw             R2   \n",
       "0                    yersifurfinal1.bw             R2   \n",
       "0                        SRR8742210.bw             R2   \n",
       "1                        SRR8742212.bw             R2   \n",
       "2                        SRR8742214.bw             R2   \n",
       "3                        SRR8742216.bw             R2   \n",
       "4                        SRR8742218.bw             R2   \n",
       "5                        SRR1776542.bw             R2   \n",
       "6                        SRR1776544.bw             R2   \n",
       "7                        SRR1776546.bw             R2   \n",
       "8                        SRR1776548.bw             R2   \n",
       "9                        SRR1776550.bw             R2   \n",
       "10                       SRR1776552.bw             R2   \n",
       "11                       SRR1776554.bw             R2   \n",
       "12                       SRR1776558.bw             R2   \n",
       "13                       SRR1776560.bw             R2   \n",
       "0                        ERR2639621.bw             R2   \n",
       "1                        SRR1503874.bw            NaN   \n",
       "0                        SRR1184483.bw            NaN   \n",
       "0                        SRR7282569.bw             R2   \n",
       "0                        SRR2549977.bw             R2   \n",
       "\n",
       "                              BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                    \n",
       "0                      klebfurfinal2.bw            NaN   \n",
       "0                                   NaN            NaN   \n",
       "1                putidaSigmaD_202005.bw            NaN   \n",
       "0                Salmonella_Fur_Fe_2.bw            NaN   \n",
       "1                salmone_ydcI_MES5_2.bw      MES5 – R3   \n",
       "0                     shigella_fur_2.bw            NaN   \n",
       "0                 StaphLAC_new_cody2.bw            NaN   \n",
       "1                   Staph_SigS_M1_R2.bw        M2 – R1   \n",
       "2      StaphTCH_vraR_13F12F6purified.bw            NaN   \n",
       "0                     yersifurfinal2.bw            NaN   \n",
       "0                         SRR8742211.bw            NaN   \n",
       "1                         SRR8742213.bw            NaN   \n",
       "2                         SRR8742215.bw            NaN   \n",
       "3                         SRR8742217.bw            NaN   \n",
       "4                         SRR8742219.bw            NaN   \n",
       "5                         SRR1776543.bw            NaN   \n",
       "6                         SRR1776545.bw            NaN   \n",
       "7                         SRR1776547.bw            NaN   \n",
       "8                         SRR1776549.bw            NaN   \n",
       "9                         SRR1776551.bw            NaN   \n",
       "10                        SRR1776553.bw            NaN   \n",
       "11                        SRR1776555.bw            NaN   \n",
       "12                        SRR1776559.bw            NaN   \n",
       "13                        SRR1776561.bw            NaN   \n",
       "0                         ERR2639622.bw             R3   \n",
       "1                                   NaN            NaN   \n",
       "0                                   NaN            NaN   \n",
       "0                         SRR7282569.bw            NaN   \n",
       "0                         SRR2549979.bw            NaN   \n",
       "\n",
       "                    BWcond2_1 BWcond2_2_name               BWcond2_2  \\\n",
       "index                                                                  \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1      salmone_ydcI_MES5_3.bw      MES5 – R4  salmone_ydcI_MES5_4.bw   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1         StaphTCH_sigs_M2.bw        M2 – R2     Staph_SigS_M2_R2.bw   \n",
       "2                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "1                         NaN            NaN                     NaN   \n",
       "2                         NaN            NaN                     NaN   \n",
       "3                         NaN            NaN                     NaN   \n",
       "4                         NaN            NaN                     NaN   \n",
       "5                         NaN            NaN                     NaN   \n",
       "6                         NaN            NaN                     NaN   \n",
       "7                         NaN            NaN                     NaN   \n",
       "8                         NaN            NaN                     NaN   \n",
       "9                         NaN            NaN                     NaN   \n",
       "10                        NaN            NaN                     NaN   \n",
       "11                        NaN            NaN                     NaN   \n",
       "12                        NaN            NaN                     NaN   \n",
       "13                        NaN            NaN                     NaN   \n",
       "0               ERR2639623.bw            NaN                     NaN   \n",
       "1                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "0                         NaN            NaN                     NaN   \n",
       "\n",
       "      BWcond3_1_name               BWcond3_1 BWcond3_2_name  \\\n",
       "index                                                         \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1           ph5 – R1  salmone_ydcI_ph55_1.bw       ph5 – R2   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1            M3 – R1     StaphTCH_sigs_M3.bw        M3 – R2   \n",
       "2                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1                NaN                     NaN            NaN   \n",
       "2                NaN                     NaN            NaN   \n",
       "3                NaN                     NaN            NaN   \n",
       "4                NaN                     NaN            NaN   \n",
       "5                NaN                     NaN            NaN   \n",
       "6                NaN                     NaN            NaN   \n",
       "7                NaN                     NaN            NaN   \n",
       "8                NaN                     NaN            NaN   \n",
       "9                NaN                     NaN            NaN   \n",
       "10               NaN                     NaN            NaN   \n",
       "11               NaN                     NaN            NaN   \n",
       "12               NaN                     NaN            NaN   \n",
       "13               NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "1                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "0                NaN                     NaN            NaN   \n",
       "\n",
       "                    BWcond3_2 BWcond4_1_name            BWcond4_1  \\\n",
       "index                                                               \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1      salmone_ydcI_ph55_2.bw            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1         Staph_SigS_M3_R2.bw        M4 – R1  StaphTCH_sigs_M4.bw   \n",
       "2                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "2                         NaN            NaN                  NaN   \n",
       "3                         NaN            NaN                  NaN   \n",
       "4                         NaN            NaN                  NaN   \n",
       "5                         NaN            NaN                  NaN   \n",
       "6                         NaN            NaN                  NaN   \n",
       "7                         NaN            NaN                  NaN   \n",
       "8                         NaN            NaN                  NaN   \n",
       "9                         NaN            NaN                  NaN   \n",
       "10                        NaN            NaN                  NaN   \n",
       "11                        NaN            NaN                  NaN   \n",
       "12                        NaN            NaN                  NaN   \n",
       "13                        NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "1                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "0                         NaN            NaN                  NaN   \n",
       "\n",
       "      BWcond4_2_name            BWcond4_2  \n",
       "index                                      \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1            M4 – R2  Staph_SigS_M4_R2.bw  \n",
       "2                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "4                NaN                  NaN  \n",
       "5                NaN                  NaN  \n",
       "6                NaN                  NaN  \n",
       "7                NaN                  NaN  \n",
       "8                NaN                  NaN  \n",
       "9                NaN                  NaN  \n",
       "10               NaN                  NaN  \n",
       "11               NaN                  NaN  \n",
       "12               NaN                  NaN  \n",
       "13               NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  \n",
       "0                NaN                  NaN  "
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-139-54e1928e7f9a>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     27\u001b[0m                                           'binding_peak_strength', 'direction', '.','ID'])\n\u001b[0;32m     28\u001b[0m         \u001b[0mdf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 29\u001b[1;33m         \u001b[0mfinal_annot_DF\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mother_table_gen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mTF_list\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTF\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mTF_index\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcurated_cond\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mfile_index\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     30\u001b[0m         \u001b[0mfinal_annot_DF\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_json\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"../data/\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0morg_id\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m\"/\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mgenome_id\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m\"/table/\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mTF_name\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m\"_\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mcurated_cond\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mfile_index\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_binding_table.json'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0morient\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'records'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     31\u001b[0m         \u001b[0mbinding_width_gen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mTF_list\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTF\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mTF_index\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"../data/\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0morg_id\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m\"/\"\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mgenome_id\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m\"/binding_widths/\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-132-b2350181469d>\u001b[0m in \u001b[0;36mother_table_gen\u001b[1;34m(TF_name, TF_condition, peak_df, gene_info)\u001b[0m\n\u001b[0;32m      7\u001b[0m     table[\"target_locus\"] = [closest_locus(table.binding_peak_start.iloc[i], \n\u001b[0;32m      8\u001b[0m                                            \u001b[0mtable\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbinding_peak_end\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 9\u001b[1;33m                                            gene_info) for i in range(len(table))]\n\u001b[0m\u001b[0;32m     10\u001b[0m     \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_genes\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_locus\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     11\u001b[0m     \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"closest_gene\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_locus\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-132-b2350181469d>\u001b[0m in \u001b[0;36m<listcomp>\u001b[1;34m(.0)\u001b[0m\n\u001b[0;32m      7\u001b[0m     table[\"target_locus\"] = [closest_locus(table.binding_peak_start.iloc[i], \n\u001b[0;32m      8\u001b[0m                                            \u001b[0mtable\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbinding_peak_end\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 9\u001b[1;33m                                            gene_info) for i in range(len(table))]\n\u001b[0m\u001b[0;32m     10\u001b[0m     \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_genes\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_locus\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     11\u001b[0m     \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"closest_gene\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtable\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"target_locus\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-131-6257ff788dbd>\u001b[0m in \u001b[0;36mclosest_locus\u001b[1;34m(peak_start, peak_end, gene_info)\u001b[0m\n\u001b[0;32m      6\u001b[0m             \u001b[0mmin_name\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m             \u001b[0mmin_value\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mabs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpeak_start\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstop\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 8\u001b[1;33m         \u001b[1;32mif\u001b[0m \u001b[0mabs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpeak_end\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mmin_value\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      9\u001b[0m             \u001b[0mmin_name\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m             \u001b[0mmin_value\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mabs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpeak_end\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mgene_info\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    869\u001b[0m         \u001b[0mkey\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    870\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 871\u001b[1;33m             \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    872\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    873\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_value\u001b[1;34m(self, series, key)\u001b[0m\n\u001b[0;32m   4403\u001b[0m         \u001b[0mk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_convert_scalar_indexer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"getitem\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4404\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4405\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtz\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mseries\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"tz\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4406\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me1\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4407\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m0\u001b[0m \u001b[1;32mand\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mholds_integer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_boolean\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine._get_loc_duplicates\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine._maybe_get_bool_indexer\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32m<__array_function__ internals>\u001b[0m in \u001b[0;36mwhere\u001b[1;34m(*args, **kwargs)\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "\n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "\n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"\\\\\")[-1] for i in curated_loc] # different for linux vs windows\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    \n",
    "#     write plot data files to folders\n",
    "    for file_index in range(len(curated_files)):\n",
    "        file = curated_files[file_index]\n",
    "        df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "        df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "        final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "        final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list.insert(12, \"curated information\", TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')\n",
    "\n",
    "#individual file updates\n",
    "for org in new_TF_list.organism_id.unique():\n",
    "    new_TF_list[new_TF_list.organism_id == org].to_csv(\"../data/\"+org+\"/TF_list.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# E. coli - other strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>organism_id</th>\n",
       "      <th>num_binding_sites</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>num_cond</th>\n",
       "      <th>project</th>\n",
       "      <th>binding tables</th>\n",
       "      <th>BWcond1_1_name</th>\n",
       "      <th>BWcond1_1</th>\n",
       "      <th>BWcond1_2_name</th>\n",
       "      <th>BWcond1_2</th>\n",
       "      <th>BWcond2_1_name</th>\n",
       "      <th>BWcond2_1</th>\n",
       "      <th>BWcond2_2_name</th>\n",
       "      <th>BWcond2_2</th>\n",
       "      <th>BWcond3_1_name</th>\n",
       "      <th>BWcond3_1</th>\n",
       "      <th>BWcond3_2_name</th>\n",
       "      <th>BWcond3_2</th>\n",
       "      <th>BWcond4_1_name</th>\n",
       "      <th>BWcond4_1</th>\n",
       "      <th>BWcond4_2_name</th>\n",
       "      <th>BWcond4_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BaeR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|baer_EtOH_binding_table.json</td>\n",
       "      <td>bio-rep1 – R1</td>\n",
       "      <td>BaeR_R1_S31_R1.bw</td>\n",
       "      <td>bio-rep1 – R2</td>\n",
       "      <td>BaeR_R1_S31_R2.bw</td>\n",
       "      <td>bio-rep2 – R1</td>\n",
       "      <td>BaerR_R2_S32_R1.bw</td>\n",
       "      <td>bio-rep2 – R2</td>\n",
       "      <td>BaerR_R2_S32_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CpxR</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>LB</td>\n",
       "      <td>5% EtOH</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>TCS</td>\n",
       "      <td>EtOH|cpxr_EtOH_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>CpxRR1_S1_R1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>CpxRR2_S2_R2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cra</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.2% Fructose, Galactose, Acetate, or Glucose</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|cra_M9_binding_table.json;acetate|cra_aceta...</td>\n",
       "      <td>Glu – R1</td>\n",
       "      <td>cra_glu_1.bw</td>\n",
       "      <td>Glu – R2</td>\n",
       "      <td>cra_glu_2.bw</td>\n",
       "      <td>Fru – R1</td>\n",
       "      <td>cra_fru_1.bw</td>\n",
       "      <td>Fru – R2</td>\n",
       "      <td>cra_fru_2.bw</td>\n",
       "      <td>Gal – R1</td>\n",
       "      <td>cra_gal_1.bw</td>\n",
       "      <td>Gal – R2</td>\n",
       "      <td>cra_gal_2.bw</td>\n",
       "      <td>Ace – R1</td>\n",
       "      <td>cra_ace_1.bw</td>\n",
       "      <td>Ace – R2</td>\n",
       "      <td>cra_ace_2.bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...</td>\n",
       "      <td>Fe – R1</td>\n",
       "      <td>fur_fe_1.bw</td>\n",
       "      <td>Fe – R2</td>\n",
       "      <td>fur_fe_2.bw</td>\n",
       "      <td>DPD – R1</td>\n",
       "      <td>fur_dpd_1.bw</td>\n",
       "      <td>DPD – R2</td>\n",
       "      <td>fur_dpd_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GadE</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>K-12 MG1655</td>\n",
       "      <td>M9</td>\n",
       "      <td>pH 5.5 (adjusted with HCl)</td>\n",
       "      <td>NC_000913.3</td>\n",
       "      <td>e_coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>known_TF</td>\n",
       "      <td>M9|gade_M9_binding_table.json</td>\n",
       "      <td>R1</td>\n",
       "      <td>gade_1.bw</td>\n",
       "      <td>R2</td>\n",
       "      <td>gade_2.bw</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         TF          Organism       Strain Media  \\\n",
       "index                                              \n",
       "0      BaeR  Escherichia coli  K-12 MG1655    LB   \n",
       "1      CpxR  Escherichia coli  K-12 MG1655    LB   \n",
       "2       Cra  Escherichia coli  K-12 MG1655    M9   \n",
       "3       Fur  Escherichia coli  K-12 MG1655    M9   \n",
       "4      GadE  Escherichia coli  K-12 MG1655    M9   \n",
       "\n",
       "                                          Supplement    genome_id organism_id  \\\n",
       "index                                                                           \n",
       "0                                            5% EtOH  NC_000913.3      e_coli   \n",
       "1                                            5% EtOH  NC_000913.3      e_coli   \n",
       "2      0.2% Fructose, Galactose, Acetate, or Glucose  NC_000913.3      e_coli   \n",
       "3         0.1 mM FeCl2 or 0.2 mM 2,2-dipyridyl (DPD)  NC_000913.3      e_coli   \n",
       "4                         pH 5.5 (adjusted with HCl)  NC_000913.3      e_coli   \n",
       "\n",
       "       num_binding_sites  num_samples  num_cond   project  \\\n",
       "index                                                       \n",
       "0                    NaN            4         1       TCS   \n",
       "1                    NaN            2         1       TCS   \n",
       "2                    NaN            8         4  known_TF   \n",
       "3                    NaN            4         2  known_TF   \n",
       "4                    NaN            2         1  known_TF   \n",
       "\n",
       "                                          binding tables BWcond1_1_name  \\\n",
       "index                                                                     \n",
       "0                      EtOH|baer_EtOH_binding_table.json  bio-rep1 – R1   \n",
       "1                      EtOH|cpxr_EtOH_binding_table.json             R1   \n",
       "2      M9|cra_M9_binding_table.json;acetate|cra_aceta...       Glu – R1   \n",
       "3      dpd|fur_dpd_binding_table.json;fe|fur_fe_bindi...        Fe – R1   \n",
       "4                          M9|gade_M9_binding_table.json             R1   \n",
       "\n",
       "               BWcond1_1 BWcond1_2_name          BWcond1_2 BWcond2_1_name  \\\n",
       "index                                                                       \n",
       "0      BaeR_R1_S31_R1.bw  bio-rep1 – R2  BaeR_R1_S31_R2.bw  bio-rep2 – R1   \n",
       "1        CpxRR1_S1_R1.bw             R2    CpxRR2_S2_R2.bw            NaN   \n",
       "2           cra_glu_1.bw       Glu – R2       cra_glu_2.bw       Fru – R1   \n",
       "3            fur_fe_1.bw        Fe – R2        fur_fe_2.bw       DPD – R1   \n",
       "4              gade_1.bw             R2          gade_2.bw            NaN   \n",
       "\n",
       "                BWcond2_1 BWcond2_2_name           BWcond2_2 BWcond3_1_name  \\\n",
       "index                                                                         \n",
       "0      BaerR_R2_S32_R1.bw  bio-rep2 – R2  BaerR_R2_S32_R2.bw            NaN   \n",
       "1                     NaN            NaN                 NaN            NaN   \n",
       "2            cra_fru_1.bw       Fru – R2        cra_fru_2.bw       Gal – R1   \n",
       "3            fur_dpd_1.bw       DPD – R2        fur_dpd_2.bw            NaN   \n",
       "4                     NaN            NaN                 NaN            NaN   \n",
       "\n",
       "          BWcond3_1 BWcond3_2_name     BWcond3_2 BWcond4_1_name     BWcond4_1  \\\n",
       "index                                                                           \n",
       "0               NaN            NaN           NaN            NaN           NaN   \n",
       "1               NaN            NaN           NaN            NaN           NaN   \n",
       "2      cra_gal_1.bw       Gal – R2  cra_gal_2.bw       Ace – R1  cra_ace_1.bw   \n",
       "3               NaN            NaN           NaN            NaN           NaN   \n",
       "4               NaN            NaN           NaN            NaN           NaN   \n",
       "\n",
       "      BWcond4_2_name     BWcond4_2  \n",
       "index                               \n",
       "0                NaN           NaN  \n",
       "1                NaN           NaN  \n",
       "2           Ace – R2  cra_ace_2.bw  \n",
       "3                NaN           NaN  \n",
       "4                NaN           NaN  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_folder = \"../data/e_coli/\"\n",
    "\n",
    "TF_list = pd.read_csv(org_folder+'TF_list.csv', index_col=0)\n",
    "TF_list.drop([\"curated information\"], axis=1, inplace=True)\n",
    "\n",
    "TF_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/katherine/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "for TF_index in range(len(TF_list)):\n",
    "    \n",
    "    TF_name= TF_list.TF.iloc[TF_index].lower()\n",
    "    org_id = TF_list.organism_id.iloc[TF_index]\n",
    "    genome_id = TF_list.genome_id.iloc[TF_index]\n",
    "    \n",
    "    if genome_id == 'NC_000913.3':\n",
    "            continue\n",
    "            \n",
    "    #find files and format\n",
    "    curated_loc = glob.glob(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+TF_name+\"_*.gff\")\n",
    "    curated_loc.sort()\n",
    "    curated_files = [i.split(\"/\")[-1] for i in curated_loc]\n",
    "    curated_cond = [i.split(\"_\")[1] for i in curated_files]\n",
    "    table_string =\";\".join([curated_cond[i]+\"|\"+TF_name+\"_\"+curated_cond[i]+'_binding_table.json' for i in range(len(curated_loc))])\n",
    "    if table_string!='':\n",
    "        TF_list['binding tables'].iloc[TF_index] = table_string\n",
    "    else:\n",
    "        TF_list['binding tables'].iloc[TF_index] = \"NA\"\n",
    "    TF_list['num_cond'].iloc[TF_index] = max(1,len(curated_loc))\n",
    "    gene_info = other_gff_to_geneinfo(org_id, genome_id)\n",
    "    \n",
    "#     write plot data files to folders\n",
    "    for file_index in range(len(curated_files)):\n",
    "#         try:\n",
    "        file = curated_files[file_index]\n",
    "        df = pd.read_csv(\"../data/\"+org_id+\"/\"+genome_id+\"/\"+\"/curated_input/\"+file,index_col=0, \n",
    "                                 delimiter='\\t', header=None, \n",
    "                                 names = ['ref','condition', 'condition_name', \n",
    "                                          \"binding_peak_start\",'binding_peak_end', \n",
    "                                          'binding_peak_strength', 'direction', '.','ID'])\n",
    "        df = df.set_index(pd.Series(range(1,len(df)+1)))\n",
    "        final_annot_DF = other_table_gen(TF_list.TF.iloc[TF_index], curated_cond[file_index], df, gene_info)\n",
    "        final_annot_DF.to_json(\"../data/\"+org_id+\"/\"+genome_id+\"/table/\"+TF_name+\"_\"+curated_cond[file_index]+'_binding_table.json',orient='records')\n",
    "        binding_width_gen(TF_list.TF.iloc[TF_index], df, \"../data/\"+org_id+\"/\"+genome_id+\"/binding_widths/\")\n",
    "        other_peak_position_gen(TF_list.TF.iloc[TF_index], final_annot_DF, \"../data/\"+org_id+\"/\"+genome_id+\"/positions/\", gene_info)\n",
    "#         except:\n",
    "#             print(TF_name, file)\n",
    "#             continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_TF_list = TF_list.copy()\n",
    "new_TF_list['binding tables'].fillna('NA',inplace=True)\n",
    "new_TF_list.insert(12, \"curated information\", new_TF_list['binding tables'] != 'NA')\n",
    "new_TF_list.to_csv(org_folder+'TF_list.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "205.4px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
