{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Search Output\n",
    "\n",
    "This notebook generates json files for searching through ChIP-pro's genes and TFs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Media</th>\n",
       "      <th>Supplement</th>\n",
       "      <th>link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CodY</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>LAC chromosome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=CodY&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SigS</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=SigS&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VraR</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>USA300_TCH1516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=s_aureus&amp;tf=VraR&amp;ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Shigella flexneri</td>\n",
       "      <td>2a, 2457T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=s_flexneri&amp;tf=Fur&amp;g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>MGH 78578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=k_pneumoniae&amp;tf=Fur...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=p_putida&amp;tf=Fur&amp;gen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Sigma D</td>\n",
       "      <td>Pseudomonas putida</td>\n",
       "      <td>KT2440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=p_putida&amp;tf=Sigma D...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Yersinia pseudotuberculosis</td>\n",
       "      <td>IP 32953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=y_pseudotuberculosi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Fur</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fe</td>\n",
       "      <td>tf_dashboard.html?organism=s_enterica&amp;tf=Fur&amp;g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>YdcI</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>Typhimurium LT2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tf_dashboard.html?organism=s_enterica&amp;tf=YdcI&amp;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         TF                     Organism           Strain Media Supplement  \\\n",
       "0      CodY        Staphylococcus aureus   LAC chromosome   NaN        NaN   \n",
       "1      SigS        Staphylococcus aureus   USA300_TCH1516   NaN        NaN   \n",
       "2      VraR        Staphylococcus aureus   USA300_TCH1516   NaN        NaN   \n",
       "3       Fur            Shigella flexneri        2a, 2457T   NaN        NaN   \n",
       "4       Fur        Klebsiella pneumoniae        MGH 78578   NaN        NaN   \n",
       "..      ...                          ...              ...   ...        ...   \n",
       "62      Fur           Pseudomonas putida           KT2440   NaN        NaN   \n",
       "63  Sigma D           Pseudomonas putida           KT2440   NaN        NaN   \n",
       "64      Fur  Yersinia pseudotuberculosis         IP 32953   NaN        NaN   \n",
       "65      Fur          Salmonella enterica  Typhimurium LT2   NaN         Fe   \n",
       "66     YdcI          Salmonella enterica  Typhimurium LT2   NaN        NaN   \n",
       "\n",
       "                                                 link  \n",
       "0   tf_dashboard.html?organism=s_aureus&tf=CodY&ge...  \n",
       "1   tf_dashboard.html?organism=s_aureus&tf=SigS&ge...  \n",
       "2   tf_dashboard.html?organism=s_aureus&tf=VraR&ge...  \n",
       "3   tf_dashboard.html?organism=s_flexneri&tf=Fur&g...  \n",
       "4   tf_dashboard.html?organism=k_pneumoniae&tf=Fur...  \n",
       "..                                                ...  \n",
       "62  tf_dashboard.html?organism=p_putida&tf=Fur&gen...  \n",
       "63  tf_dashboard.html?organism=p_putida&tf=Sigma D...  \n",
       "64  tf_dashboard.html?organism=y_pseudotuberculosi...  \n",
       "65  tf_dashboard.html?organism=s_enterica&tf=Fur&g...  \n",
       "66  tf_dashboard.html?organism=s_enterica&tf=YdcI&...  \n",
       "\n",
       "[67 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prefix = '../data/'\n",
    "suffix = '/TF_list.csv'\n",
    "\n",
    "organisms = [f.name for f in os.scandir(prefix)]\n",
    "cols_of_interest = ['TF', 'Organism', 'Strain', 'Media', 'Supplement']\n",
    "\n",
    "all_tfs = pd.DataFrame(columns = cols_of_interest + ['link'])\n",
    "\n",
    "for o in organisms:\n",
    "    try:\n",
    "        tfs = pd.read_csv(prefix + o + suffix, index_col = 0)\n",
    "    except NotADirectoryError:\n",
    "        continue\n",
    "        \n",
    "    # build link\n",
    "    for i in tfs.index:\n",
    "        l = 'tf_dashboard.html?organism=' + o\n",
    "        l += '&tf=' + tfs.TF[i]\n",
    "        l += '&genome=' + tfs.genome_id[i]\n",
    "        l += '&i=' + str(i)\n",
    "        tfs.loc[i, 'link'] = l\n",
    "    \n",
    "    # take info of interest\n",
    "    tfs = tfs[cols_of_interest + ['link']]\n",
    "    \n",
    "    # save\n",
    "    all_tfs = all_tfs.append(tfs, ignore_index = True)\n",
    "    \n",
    "all_tfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_tfs.to_json('../data/tfs_search.json', orient='records')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "prefix = '../data/'\n",
    "suffix = '_binding_table.json'\n",
    "\n",
    "all_genes = pd.DataFrame(columns = ['Gene', 'Locus', \n",
    "        'Organism', 'Strain', 'Binding_site_id',\n",
    "        'Condition', 'Peak_start', 'Peak_end',\n",
    "        'Peak_strength', 'link'])\n",
    "\n",
    "for j, tf_row in all_tfs.iterrows():\n",
    "    \n",
    "    # get folder info from the link & read in file\n",
    "    file = tf_row.link.replace('&', '=').split('=')\n",
    "    f = prefix + file[1] + '/' + file[5] + '/table/' + file[3] + suffix\n",
    "    \n",
    "    # read in\n",
    "    try:\n",
    "        bt = pd.read_json(f)\n",
    "    except (FileNotFoundError, ValueError) as e:\n",
    "        continue\n",
    "    \n",
    "    for i, row in bt.iterrows():\n",
    "        \n",
    "        # get genes & loci\n",
    "        genes = row.target_genes.split(',')\n",
    "        loci = row.target_locus.split(',')\n",
    "        \n",
    "        # rearrange data for new table\n",
    "        for g, l in zip(genes, loci):\n",
    "            if g == '':\n",
    "                continue\n",
    "            all_genes = all_genes.append({\n",
    "                'Gene':g,\n",
    "                'Locus':l,\n",
    "                'Organism':tf_row.Organism,\n",
    "                'Strain':tf_row.Strain,\n",
    "                'Binding_site_id':row['index'],\n",
    "                'Condition':row.condition,\n",
    "                'Peak_start':row.binding_peak_start,\n",
    "                'Peak_end':row.binding_peak_end,\n",
    "                'Peak_strength':row.binding_peak_strength,\n",
    "                'link':tf_row.link}, ignore_index = True)\n",
    "            \n",
    "all_genes = all_genes.sort_values('Peak_start')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Binding_site_id</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Peak_start</th>\n",
       "      <th>Peak_end</th>\n",
       "      <th>Peak_strength</th>\n",
       "      <th>link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene, Locus, Organism, Strain, Binding_site_id, Condition, Peak_start, Peak_end, Peak_strength, link]\n",
       "Index: []"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes.to_json('../ChiPdb/data/genes_search.json', orient='records')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
