<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html xmlns:cis="http://zlab.bu.edu/schema/cisml" xmlns:fimo="http://noble.gs.washington.edu/schema/cisml" xmlns:mem="http://noble.gs.washington.edu/meme">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="UTF-8">
<title>FIMO Results</title>
<style type="text/css">
td.left {text-align: left;}
td.right {text-align: right; padding-right: 1cm;}
</style>
</head>
<body bgcolor="#D5F0FF">
<a name="top_buttons"></a>
<hr>
<table summary="buttons" align="left" cellspacing="0">
<tr>
<td bgcolor="#00FFFF"><a href="#database_and_motifs"><b>Database and Motifs</b></a></td>
<td bgcolor="#DDFFDD"><a href="#sec_i"><b>High-scoring Motif Occurences</b></a></td>
<td bgcolor="#DDDDFF"><a href="#debugging_information"><b>Debugging Information</b></a></td>
</tr>
</table>
<br/>
<br/>
<hr/>
<center><big><b>FIMO - Motif search tool</b></big></center>
<hr>
<p>
FIMO version 4.11.2, (Release date: Thu May 05 14:58:55 2016 -0700)
</p>
<p>
For further information on how to interpret these results
or to get a copy of the FIMO software please access
<a href="http://meme.nbcr.net">http://meme.nbcr.net</a></p>
<p>If you use FIMO in your research, please cite the following paper:<br>
Charles E. Grant, Timothy L. Bailey, and William Stafford Noble,
"FIMO: Scanning for occurrences of a given motif",
<i>Bioinformatics</i>, <b>27</b>(7):1017-1018, 2011.
<a href="http://bioinformatics.oxfordjournals.org/content/27/7/1017">[full text]</a></p>
<hr>
<center><big><b><a name="database_and_motifs">DATABASE AND MOTIFS</a></b></big></center>
<hr>
<div style="padding-left: 0.75in; line-height: 1em; font-family: monospace;">
<p>
  DATABASE ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt
  <br />
  Database contains 0 sequences, 0 residues
</p>
<p>
  MOTIFS ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml (DNA)
  <table>
    <thead>
      <tr>
        <th style="border-bottom: 1px dashed;">MOTIF</th>
        <th style="border-bottom: 1px dashed; padding-left: 1em;">WIDTH</th>
        <th style="border-bottom: 1px dashed; padding-left: 1em;text-align:left;" >
         BEST POSSIBLE MATCH
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td style="text-align:right;">1</td>
        <td style="text-align:right;padding-left: 1em;">42</td>
        <td style="text-align:left;padding-left: 1em;">TCGCTGGTTCGATTCCTCTCAGGAACACCATATTTTCAAAAG</td>
       </tr>
      <tr>
        <td style="text-align:right;">2</td>
        <td style="text-align:right;padding-left: 1em;">41</td>
        <td style="text-align:left;padding-left: 1em;">AACGATGCGTCTATAGTTCAGTTGGTAGAACGGCTGCCTGG</td>
       </tr>
      <tr>
        <td style="text-align:right;">3</td>
        <td style="text-align:right;padding-left: 1em;">45</td>
        <td style="text-align:left;padding-left: 1em;">TATGTTTCATAGTGAAATATATGCTTTAATGAATGTTCAATATTG</td>
       </tr>
      <tr>
        <td style="text-align:right;">4</td>
        <td style="text-align:right;padding-left: 1em;">31</td>
        <td style="text-align:left;padding-left: 1em;">CGAATAACGGTATACTGTTCTGATCATTTAA</td>
       </tr>
    </tbody>
  </table>
</p>
<p>
Random model letter frequencies (../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background):
<br/>

A 0.264 C 0.236 G 0.236 T 0.264 </p>
</div>
<hr>
<center><big><b><a name="sec_i">SECTION I: HIGH-SCORING MOTIF OCCURENCES</a></b></big></center>
<hr>
<ul>
<li>
There were 9 motif occurences with a p-value less than 0.0001.

The full set of motif occurences can be seen in the
tab-delimited plain text output file
<a href="fimo.txt">fimo.txt</a>, 
the GFF file 
<a href="fimo.gff">fimo.gff</a> 
which may be suitable for uploading to the 
<a href="http://genome.ucsc.edu/cgi-bin/hgTables">UCSC Genome Table Browser</a>
(assuming the FASTA input sequences included genomic coordinates in UCSC or Galaxy format),
or the XML file 
<a href="fimo.xml">fimo.xml</a>.
</li>
<li>
The p-value of a motif occurrence is defined as the
probability of a random sequence of the same length as the motif
matching that position of the sequence with as good or better a score.
</li>
<li>
The score for the match of a position in a sequence to a motif
is computed by summing the appropriate entries from each column of
the position-dependent scoring matrix that represents the motif.
</li>
<li>
The q-value of a motif occurrence is defined as the
false discovery rate if the occurrence is accepted as significant.
</li>
<li>The table is sorted by increasing p-value.</li>
</ul>
<table border="1">
<thead>
<tr>
<th>Motif</th>
<th>Sequence Name</th>
<th>Strand</th>
<th>Start</th>
<th>End</th>
<th>p-value</th>
<th>q-value</th>
<th>Matched Sequence</th>
</tr>
</thead>
<tbody>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak20</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">5</td>
      <td style="text-align:left;">46</td>
      <td style="text-align:left;">8.97e-16</td>
      <td style="text-align:left;">2.18e-12</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCACTGGTTCGAGTCCAGTCAGAGGAGCCAAATTCAAAAAAG</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak12</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">29</td>
      <td style="text-align:left;">70</td>
      <td style="text-align:left;">1.09e-15</td>
      <td style="text-align:left;">2.18e-12</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCGGTGGTTCGAGTCCACTCGGACGCACCAGATTTTCTTAAT</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak50</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">7</td>
      <td style="text-align:left;">48</td>
      <td style="text-align:left;">3.33e-14</td>
      <td style="text-align:left;">3.32e-11</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCGCTGGCTTCTTGCCTCTCAGGAAATCACTTATGTCCAAAT</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak49</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">20</td>
      <td style="text-align:left;">61</td>
      <td style="text-align:left;">3.33e-14</td>
      <td style="text-align:left;">3.32e-11</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCGCTGGCTTCTTGCCTCTCAGGAAATCACTTATGTCCAAAT</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak6</td>
      <td style="text-align:center;">-</td>
      <td style="text-align:left;">3</td>
      <td style="text-align:left;">44</td>
      <td style="text-align:left;">1.12e-13</td>
      <td style="text-align:left;">8.95e-11</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCGGGGGTTCGAATCCCCCCCTCACCGCCATATTTAAAGAAG</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak23</td>
      <td style="text-align:center;">-</td>
      <td style="text-align:left;">34</td>
      <td style="text-align:left;">75</td>
      <td style="text-align:left;">4.51e-13</td>
      <td style="text-align:left;">2.68e-10</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TCAGCGGTTCGATCCCGCTTAGCTCCACCAAATTTCCAACCC</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak22</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">27</td>
      <td style="text-align:left;">68</td>
      <td style="text-align:left;">4.69e-13</td>
      <td style="text-align:left;">2.68e-10</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TTGCAGGTTCGATTCCTGCAGGGGACACCATTTATCAGTTCG</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak43</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">3</td>
      <td style="text-align:left;">44</td>
      <td style="text-align:left;">1.67e-11</td>
      <td style="text-align:left;">8.31e-09</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TTAACGCTTCAAAATCTGTAAAGCACGCCATATCGCCGAAAG</td>
   </tr>
    <tr>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">peak37</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">33</td>
      <td style="text-align:left;">74</td>
      <td style="text-align:left;">2.5e-05</td>
      <td style="text-align:left;">0.0111</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">TTGCTGCAGCATGATCAGACCAACAAGGCCAATTGCCACAAT</td>
   </tr>
</tbody>
</table>

<hr>
<center><big><b><a name="debugging_information">DEBUGGING INFORMATION</a></b></big></center>
<hr>
<p>
Command line:
</p>
<pre>
/home/katherine/anaconda3/bin/fimo --parse-genomic-coord --verbosity 1 --oc ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/fimo_out_1 --bgfile ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background --motif 1 ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt
</pre>
<p>
Settings:
</p>
<pre>
<table>
  <tr>
    <td style="padding-right: 2em">output_directory = ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/fimo_out_1</td>
    <td style="padding-left: 5em; padding-right: 2em">MEME file name = ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/meme_out/meme.xml</td>
    <td style="padding-left: 5em; padding-right: 2em">sequence file name = ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/cusr_CuSO4.txt</td>
  </tr>  <tr>
    <td style="padding-right: 2em">background file name = ../data/e_coli/NC_000913.3/meme-chip/cusr_CuSO4/background</td>
    <td style="padding-left: 5em; padding-right: 2em">alphabet = DNA</td>
    <td style="padding-left: 5em; padding-right: 2em">max stored scores = 100000</td>
  </tr>  <tr>
    <td style="padding-right: 2em">allow clobber = true</td>
    <td style="padding-left: 5em; padding-right: 2em">compute q-values = true</td>
    <td style="padding-left: 5em; padding-right: 2em">parse genomic coord. = true</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">text only = false</td>
    <td style="padding-left: 5em; padding-right: 2em">scan both strands = true</td>
    <td style="padding-left: 5em; padding-right: 2em">max strand = false</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">threshold type = p-value</td>
    <td style="padding-left: 5em; padding-right: 2em">output theshold = 0.0001</td>
    <td style="padding-left: 5em; padding-right: 2em">pseudocount = 0.1</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">alpha = 1</td>
    <td style="padding-left: 5em; padding-right: 2em">verbosity = 1</td>
    <td style="padding-left: 5em; padding-right: 2em"></td>
  </tr>

</table>
</pre>
<p>
This information can be useful in the event you wish to report a
problem with the FIMO software.
</p>
<hr>
<span style="background-color: #DDDDFF"><a href="#top_buttons"><b>Go to top</b></a></span>
</body>
</html>
